
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="A detailed description of all outputs produced by the main Bactopia pipeline">
      
      
        <meta name="author" content="Robert A. Petit III">
      
      
        <link rel="canonical" href="https://bactopia.github.io/v2.2.0/output-overview/">
      
      
        <link rel="prev" href="../workflow-overview/">
      
      
        <link rel="next" href="../usage-complete/">
      
      
        
      
      
      <link rel="icon" href="../assets/favicon.ico">
      <meta name="generator" content="mkdocs-1.4.2, mkdocs-material-9.1.6+insiders-4.32.5">
    
    
      
        <title>Ouput Overview - Bactopia</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.dc5b59cc.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.6932e648.min.css">
      
      

    
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Ubuntu:300,300i,400,400i,700,700i%7CUbuntu+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Ubuntu";--md-code-font:"Ubuntu Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../custom/magnific-popup.css">
    
      <link rel="stylesheet" href="../custom/extra.css">
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      
  


  
  


  <script id="__analytics">function __md_analytics(){window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)},ga.l=+new Date,ga("create","UA-104199698-2","auto"),ga("set","anonymizeIp",!0),ga("send","pageview"),document.addEventListener("DOMContentLoaded",function(){document.forms.search&&document.forms.search.query.addEventListener("blur",function(){var e;this.value&&(e=document.location.pathname,ga("send","pageview",e+"?q="+this.value))}),document$.subscribe(function(){var a=document.forms.feedback;if(void 0!==a)for(var e of a.querySelectorAll("[type=submit]"))e.addEventListener("click",function(e){e.preventDefault();var t=document.location.pathname,e=this.getAttribute("data-md-value");ga("send","event","feedback","click",t,e),a.firstElementChild.disabled=!0;e=a.querySelector(".md-feedback__note [data-md-value='"+e+"']");e&&(e.hidden=!1)}),a.hidden=!1}),location$.subscribe(function(e){ga("send","pageview",e.pathname)})});var e=document.createElement("script");e.async=!0,e.src="https://www.google-analytics.com/analytics.js",document.getElementById("__analytics").insertAdjacentElement("afterEnd",e)}</script>

  
    <script>"undefined"!=typeof __md_analytics&&__md_analytics()</script>
  

    
    
      
        <meta  property="og:type"  content="website" >
      
        <meta  property="og:title"  content="Ouput Overview - Bactopia" >
      
        <meta  property="og:description"  content="A detailed description of all outputs produced by the main Bactopia pipeline" >
      
        <meta  property="og:image"  content="https://bactopia.github.io/v2.2.0/assets/images/social/output-overview.png" >
      
        <meta  property="og:image:type"  content="image/png" >
      
        <meta  property="og:image:width"  content="1200" >
      
        <meta  property="og:image:height"  content="630" >
      
        <meta  property="og:url"  content="https://bactopia.github.io/v2.2.0/output-overview/" >
      
        <meta  name="twitter:card"  content="summary_large_image" >
      
        <meta  name="twitter:title"  content="Ouput Overview - Bactopia" >
      
        <meta  name="twitter:description"  content="A detailed description of all outputs produced by the main Bactopia pipeline" >
      
        <meta  name="twitter:image"  content="https://bactopia.github.io/v2.2.0/assets/images/social/output-overview.png" >
      
    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#overview-of-bactopia-output" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow md-header--lifted" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="Bactopia" class="md-header__button md-logo" aria-label="Bactopia" data-md-component="logo">
      
  <img src="../assets/bactopia-small-logo.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Bactopia
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Ouput Overview
            
          </span>
        </div>
      </div>
    </div>
    
      <form class="md-header__option" data-md-component="palette">
        
          
          <input class="md-option" data-md-color-media="" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_0">
          
            <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M17 6H7c-3.31 0-6 2.69-6 6s2.69 6 6 6h10c3.31 0 6-2.69 6-6s-2.69-6-6-6zm0 10H7c-2.21 0-4-1.79-4-4s1.79-4 4-4h10c2.21 0 4 1.79 4 4s-1.79 4-4 4zM7 9c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"/></svg>
            </label>
          
        
          
          <input class="md-option" data-md-color-media="" data-md-color-scheme="slate" data-md-color-primary="red" data-md-color-accent="red"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_1">
          
            <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_0" hidden>
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M17 7H7a5 5 0 0 0-5 5 5 5 0 0 0 5 5h10a5 5 0 0 0 5-5 5 5 0 0 0-5-5m0 8a3 3 0 0 1-3-3 3 3 0 0 1 3-3 3 3 0 0 1 3 3 3 3 0 0 1-3 3Z"/></svg>
            </label>
          
        
      </form>
    
    
      <script>var media,input,key,value,palette=__md_get("__palette");if(palette&&palette.color){"(prefers-color-scheme)"===palette.color.media&&(media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']"),palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent"));for([key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/bactopia/bactopia" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
    
      
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
    
  
  
    
    
      <li class="md-tabs__item">
        <a href=".." class="md-tabs__link md-tabs__link--active">
          
  
  Bactopia

        </a>
      </li>
    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../bactopia-tools/" class="md-tabs__link">
          
  
  Bactopia Tools

        </a>
      </li>
    
  

      
    </ul>
  </div>
</nav>
    
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="Bactopia" class="md-nav__button md-logo" aria-label="Bactopia" data-md-component="logo">
      
  <img src="../assets/bactopia-small-logo.png" alt="logo">

    </a>
    Bactopia
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/bactopia/bactopia" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
    
  
  
    
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1" checked>
        
          <label class="md-nav__link" for="__nav_1" id="__nav_1_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    
  
    Bactopia
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_1_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_1">
            <span class="md-nav__icon md-icon"></span>
            
  
    Bactopia
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    Introduction
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../quick-start/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    Quick Start
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../installation/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    Installation
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../tutorial/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    Tutorial
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../datasets/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    Build Datasets
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../usage-basic/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    Basic Usage
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../workflow-overview/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    Workflow Overview
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  <span class="md-ellipsis">
    
  
    Output Overview
  

    
  </span>
  
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    
  
    Output Overview
  

    
  </span>
  
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#directories" class="md-nav__link">
    <span class="md-ellipsis">
      
        Directories
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Directories">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#annotation" class="md-nav__link">
    <span class="md-ellipsis">
      
        annotation
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#antimicrobial_resistance" class="md-nav__link">
    <span class="md-ellipsis">
      
        antimicrobial_resistance
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ariba" class="md-nav__link">
    <span class="md-ellipsis">
      
        ariba
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#assembly" class="md-nav__link">
    <span class="md-ellipsis">
      
        assembly
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="assembly">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#standard" class="md-nav__link">
    <span class="md-ellipsis">
      
        standard
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hybrid" class="md-nav__link">
    <span class="md-ellipsis">
      
        hybrid
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#quality-reports" class="md-nav__link">
    <span class="md-ellipsis">
      
        quality reports
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#blast" class="md-nav__link">
    <span class="md-ellipsis">
      
        blast
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#genome-size" class="md-nav__link">
    <span class="md-ellipsis">
      
        genome-size
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#kmers" class="md-nav__link">
    <span class="md-ellipsis">
      
        kmers
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#logs" class="md-nav__link">
    <span class="md-ellipsis">
      
        logs
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="logs">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#example-versions" class="md-nav__link">
    <span class="md-ellipsis">
      
        Example versions
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#mapping" class="md-nav__link">
    <span class="md-ellipsis">
      
        mapping
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#minmers" class="md-nav__link">
    <span class="md-ellipsis">
      
        minmers
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#mlst" class="md-nav__link">
    <span class="md-ellipsis">
      
        mlst
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#quality-control" class="md-nav__link">
    <span class="md-ellipsis">
      
        quality-control
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#variants" class="md-nav__link">
    <span class="md-ellipsis">
      
        variants
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../usage-complete/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    Complete Usage
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../troubleshooting/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    Troubleshooting
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../presentations/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    Presentations
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../enhancements/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    Enhancements to OSS
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../acknowledgements/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    Acknowledgements
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../changelog/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    Changelog
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" >
        
          <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    
  
    Bactopia Tools
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            
  
    Bactopia Tools
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    Introduction
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2_2" >
        
          <label class="md-nav__link" for="__nav_2_2" id="__nav_2_2_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    
  
    Subworkflows
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_2_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_2_2">
            <span class="md-nav__icon md-icon"></span>
            
  
    Subworkflows
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/ariba/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    ariba
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/bakta/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    bakta
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/eggnog/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    eggnog
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/gtdb/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    gtdb
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/merlin/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    merlin
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/pangenome/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    pangenome
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/snippy/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    snippy
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/staphtyper/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    staphtyper
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2_3" >
        
          <label class="md-nav__link" for="__nav_2_3" id="__nav_2_3_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    
  
    Modules
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_2_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_2_3">
            <span class="md-nav__icon md-icon"></span>
            
  
    Modules
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/abricate/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    abricate
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/agrvate/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    agrvate
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/amrfinderplus/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    amrfinderplus
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/busco/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    busco
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/checkm/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    checkm
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/ectyper/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    ectyper
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/emmtyper/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    emmtyper
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/fastani/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    fastani
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/gamma/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    gamma
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/genotyphi/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    genotyphi
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/hicap/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    hicap
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/hpsuissero/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    hpsuissero
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/ismapper/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    ismapper
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/kleborate/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    kleborate
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/kraken2/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    kraken2
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/legsta/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    legsta
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/lissero/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    lissero
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/mashdist/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    mashdist
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/mashtree/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    mashtree
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/mcroni/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    mcroni
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/meningotype/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    meningotype
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/mlst/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    mlst
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/mobsuite/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    mobsuite
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/mykrobe/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    mykrobe
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/ngmaster/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    ngmaster
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/pasty/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    pasty
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/pbptyper/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    pbptyper
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/plasmidfinder/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    plasmidfinder
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/prokka/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    prokka
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/rgi/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    rgi
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/seqsero2/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    seqsero2
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/seroba/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    seroba
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/shigatyper/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    shigatyper
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/shigeifinder/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    shigeifinder
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/sistr/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    sistr
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/spatyper/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    spatyper
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/ssuissero/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    ssuissero
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/staphopiasccmec/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    staphopiasccmec
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/tbprofiler/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    tbprofiler
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#directories" class="md-nav__link">
    <span class="md-ellipsis">
      
        Directories
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Directories">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#annotation" class="md-nav__link">
    <span class="md-ellipsis">
      
        annotation
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#antimicrobial_resistance" class="md-nav__link">
    <span class="md-ellipsis">
      
        antimicrobial_resistance
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ariba" class="md-nav__link">
    <span class="md-ellipsis">
      
        ariba
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#assembly" class="md-nav__link">
    <span class="md-ellipsis">
      
        assembly
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="assembly">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#standard" class="md-nav__link">
    <span class="md-ellipsis">
      
        standard
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hybrid" class="md-nav__link">
    <span class="md-ellipsis">
      
        hybrid
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#quality-reports" class="md-nav__link">
    <span class="md-ellipsis">
      
        quality reports
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#blast" class="md-nav__link">
    <span class="md-ellipsis">
      
        blast
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#genome-size" class="md-nav__link">
    <span class="md-ellipsis">
      
        genome-size
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#kmers" class="md-nav__link">
    <span class="md-ellipsis">
      
        kmers
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#logs" class="md-nav__link">
    <span class="md-ellipsis">
      
        logs
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="logs">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#example-versions" class="md-nav__link">
    <span class="md-ellipsis">
      
        Example versions
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#mapping" class="md-nav__link">
    <span class="md-ellipsis">
      
        mapping
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#minmers" class="md-nav__link">
    <span class="md-ellipsis">
      
        minmers
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#mlst" class="md-nav__link">
    <span class="md-ellipsis">
      
        mlst
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#quality-control" class="md-nav__link">
    <span class="md-ellipsis">
      
        quality-control
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#variants" class="md-nav__link">
    <span class="md-ellipsis">
      
        variants
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              
              <article class="md-content__inner md-typeset">
                

<!-- Edit button -->

  <a
    href="https://github.com/bactopia/bactopia.github.io/edit/master/docs/output-overview.md"
    title="edit.link.title"
    class="md-content__button md-icon"
  >
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25Z"/></svg>
  </a>


<!--
  Hack: check whether the content contains a h1 headline. If it
  doesn't, the page title (or respectively site name) is used
  as the main headline.
-->


<!-- Markdown content -->
<h1 id="overview-of-bactopia-output">Overview of Bactopia Output<a class="headerlink" href="#overview-of-bactopia-output" title="Permanent link">&para;</a></h1>
<p>After a successful run, Bactopia will have produced numerous output files. Just how many output files depends on the input datasets used (e.g. none, general datasets, species specific datasets, user populated datasets).</p>
<p>Here is the complete directory structure that is possible (using all available dataset options) with Bactopia. </p>
<div class="highlight"><pre><span></span><code>${SAMPLE_NAME}/
├── annotation
├── antimicrobial_resistance
├── ariba
├── assembly
├── blast
├── kmers
├── logs
├── mapping
├── minmers
├── mlst
├── quality-control
├── variants
└── ${SAMPLE_NAME}-genome-size.txt
</code></pre></div>
<p>For each type of analysis in Bactopia, a separate directory is created to hold the results. All samples processed by Bactopia will have this directory structure. The only difference is the usage of <code>${SAMPLE_NAME}</code> as a prefix for naming some output files.</p>
<h2 id="directories">Directories<a class="headerlink" href="#directories" title="Permanent link">&para;</a></h2>
<p>The following sections include a list of expected outputs as well as a brief description of each output file.</p>
<p>There are instances where additional files (e.g. <code>--keep_all_files</code> and <code>--ariba_noclean</code>) may be encountered. These files aren't described below, just the defaults. Also, using <code>--compress</code> will add a <em>gz</em> extension, but the original extension is maintained and its description still applies. </p>
<div class="admonition info">
<p class="admonition-title">Developer Descriptions Take Priority</p>
<p>If a developer described their software's outputs, their description was used with a link back to the software's documentation (major thanks for taking the time to do that!). In some cases there may have been slight formatting modifications made. In any case, if descriptions are not original credit will be properly given to the source.</p>
</div>
<h3 id="annotation"><code>annotation</code><a class="headerlink" href="#annotation" title="Permanent link">&para;</a></h3>
<p>The <code>annotation</code> directory will contain the outputs from <a href="https://github.com/tseemann/prokka">Prokka</a> annotation. These outputs include FASTA (proteins and genes), GFF3, GenBank, and many more. By default the included Prokka databases are used for annotation. However, if a <a href="datasets/#species-specific">Species Specific Dataset</a> was a created the RefSeq clustered proteins are used first for annotation.</p>
<p>File descriptions were directly taken from <a href="https://github.com/tseemann/prokka#output-files">Prokka's Output Files</a> section and slight modifications were made to the order of rows.</p>
<div class="highlight"><pre><span></span><code>${SAMPLE_NAME}/
└── annotation
    ├── ${SAMPLE_NAME}.err
    ├── ${SAMPLE_NAME}.faa
    ├── ${SAMPLE_NAME}.ffn
    ├── ${SAMPLE_NAME}.fna
    ├── ${SAMPLE_NAME}.fsa
    ├── ${SAMPLE_NAME}.gbk
    ├── ${SAMPLE_NAME}.gff
    ├── ${SAMPLE_NAME}.log
    ├── ${SAMPLE_NAME}.sqn
    ├── ${SAMPLE_NAME}.tbl
    ├── ${SAMPLE_NAME}.tsv
    └── ${SAMPLE_NAME}.txt
</code></pre></div>
<table>
<thead>
<tr>
<th>Extension</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>.err</td>
<td>Unacceptable annotations - the NCBI discrepancy report.</td>
</tr>
<tr>
<td>.faa</td>
<td>Protein FASTA file of the translated CDS sequences.</td>
</tr>
<tr>
<td>.ffn</td>
<td>Nucleotide FASTA file of all the prediction transcripts (CDS, rRNA, tRNA, tmRNA, misc_RNA)</td>
</tr>
<tr>
<td>.fna</td>
<td>Nucleotide FASTA file of the input contig sequences.</td>
</tr>
<tr>
<td>.fsa</td>
<td>Nucleotide FASTA file of the input contig sequences, used by "tbl2asn" to create the .sqn file. It is mostly the same as the .fna file, but with extra Sequin tags in the sequence description lines.</td>
</tr>
<tr>
<td>.gbk</td>
<td>This is a standard GenBank file derived from the master .gff. If the input to prokka was a multi-FASTA, then this will be a multi-GenBank, with one record for each sequence.</td>
</tr>
<tr>
<td>.gff</td>
<td>This is the master annotation in GFF3 format, containing both sequences and annotations. It can be viewed directly in Artemis or IGV.</td>
</tr>
<tr>
<td>.log</td>
<td>Contains all the output that Prokka produced during its run. This is a record of what settings you used.</td>
</tr>
<tr>
<td>.sqn</td>
<td>An ASN1 format "Sequin" file for submission to GenBank. It needs to be edited to set the correct taxonomy, authors, related publication etc.</td>
</tr>
<tr>
<td>.tbl</td>
<td>Feature Table file, used by "tbl2asn" to create the .sqn file.</td>
</tr>
<tr>
<td>.tsv</td>
<td>Tab-separated file of all features: locus_tag,ftype,len_bp,gene,EC_number,COG,product</td>
</tr>
<tr>
<td>.txt</td>
<td>Statistics relating to the annotated features found.</td>
</tr>
</tbody>
</table>
<h3 id="antimicrobial_resistance"><code>antimicrobial_resistance</code><a class="headerlink" href="#antimicrobial_resistance" title="Permanent link">&para;</a></h3>
<p>The <code>antimicrobial_resistance</code> directory will contain the output from <a href="https://github.com/ncbi/amr">NCBI's AMRFinderPlus</a>. The results of AMRFinderPlus using genes as and input, and proteins as an input are available. More information about the output format is available from the <a href="https://github.com/ncbi/amr/wiki/Running-AMRFinderPlus#output-format">AMRFinderPlus Wiki</a>.</p>
<div class="highlight"><pre><span></span><code>${SAMPLE_NAME}/
└── antimicrobial_resistance/
    ├── ${SAMPLE_NAME}-gene-report.txt
    └── ${SAMPLE_NAME}-protein-report.txt
</code></pre></div>
<table>
<thead>
<tr>
<th>Extension</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>-gene-report.txt</td>
<td>Results of using gene sequences as an input</td>
</tr>
<tr>
<td>-protein-report.txt</td>
<td>Results of using protein sequences as an input</td>
</tr>
</tbody>
</table>
<h3 id="ariba"><code>ariba</code><a class="headerlink" href="#ariba" title="Permanent link">&para;</a></h3>
<p>The <code>ariba</code> directory will contain the results of any <a href="https://github.com/sanger-pathogens/ariba/">Ariba</a> analysis (excluding MLST). Only the <a href="datasets/#general">Ariba databases</a> created during the dataset setup are used for analysis. For each Ariba database (e.g. <code>card</code> or <code>vfdb</code>), a separate folder with the name of the database is included in the <code>ariba</code> folder.</p>
<p>The file descriptions below were modified from Ariba's wiki entries for <a href="https://github.com/sanger-pathogens/ariba/wiki/Task:-run"><code>run</code></a> and <a href="https://github.com/sanger-pathogens/ariba/wiki/Task:-summary"><code>summary</code></a>.</p>
<div class="highlight"><pre><span></span><code>${SAMPLE_NAME}/
└── ariba
    └── ARIBA_DATABASE_NAME
        ├── assembled_genes.fa.gz
        ├── assembled_seqs.fa.gz
        ├── assemblies.fa.gz
        ├── debug.report.tsv
        ├── log.clusters.gz
        ├── report.tsv
        ├── summary.csv
        └── version_info.txt
</code></pre></div>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>assembled_genes.fa.gz</td>
<td>A gzipped FASTA file of only assembled gene sequences (with extensions).</td>
</tr>
<tr>
<td>assembled_seqs.fa.gz</td>
<td>A gzipped FASTA of the assembled sequences (genes and non-coding).</td>
</tr>
<tr>
<td>assemblies.fa.gz</td>
<td>A gzipped FASTA file of the assemblies (complete, unedited, contigs).</td>
</tr>
<tr>
<td>debug.report.tsv</td>
<td>The complete list of clusters, including those that did not pass filtering.</td>
</tr>
<tr>
<td>log.clusters.gz</td>
<td>Detailed logging for the progress of each cluster.</td>
</tr>
<tr>
<td>report.tsv</td>
<td>A detailed <a href="https://github.com/sanger-pathogens/ariba/wiki/Task:-run#report-file">report file</a> of clusters which passed filtering.</td>
</tr>
<tr>
<td>summary.csv</td>
<td>A more condensed <a href="https://github.com/sanger-pathogens/ariba/wiki/Task:-summary">summary</a> of the report.tsv</td>
</tr>
<tr>
<td>version_info.txt</td>
<td>Information on the versions of ARIBA and its dependencies at runtime.</td>
</tr>
</tbody>
</table>
<h3 id="assembly"><code>assembly</code><a class="headerlink" href="#assembly" title="Permanent link">&para;</a></h3>
<p>The <code>assembly</code> folder contains the results of the sample's assembly. </p>
<h4 id="standard">standard<a class="headerlink" href="#standard" title="Permanent link">&para;</a></h4>
<p>The standard assembly is managed by <a href="https://github.com/tseemann/shovill">Shovill</a> and by default <a href="https://github.com/ncbi/SKESA">SKESA</a> is used for assembly. Alternative assemblers include <a href="https://github.com/ablab/spades">SPAdes</a>, <a href="https://github.com/voutcn/megahit">MEGAHIT</a>, and <a href="https://github.com/dzerbino/velvet">Velvet</a>. Depending on the choice of assembler, additional output files (e.g. assembly graphs) may be given.</p>
<p>Files descriptions with some modifications were directly taken from <a href="https://github.com/tseemann/shovill#output-files">Shovill's Output Files</a> section as well as the <a href="https://sourceforge.net/p/flashpage/code/ci/master/tree/flash.c#l114">FLASH usage</a>.
<div class="highlight"><pre><span></span><code>${SAMPLE_NAME}/
└── assembly
    ├── cointigs.fa
    ├── flash.hist
    ├── flash.histogram
    ├── shovill.corrections
    ├── shovill.log
    ├── ${SAMPLE_NAME}.fna
    └── ${SAMPLE_NAME}.fna.json
</code></pre></div></p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>contigs.fa</td>
<td>Final assembly without renamed headers.</td>
</tr>
<tr>
<td>flash.hist</td>
<td>Numeric histogram of merged read lengths.</td>
</tr>
<tr>
<td>flash.histogram</td>
<td>Visual histogram of merged read lengths</td>
</tr>
<tr>
<td>shovill.log</td>
<td>Full log file for bug reporting</td>
</tr>
<tr>
<td>shovill.corrections</td>
<td>List of post-assembly corrections</td>
</tr>
<tr>
<td>${SAMPLE_NAME}.fna</td>
<td>The final assembly, with renamed header to include sample name</td>
</tr>
<tr>
<td>${SAMPLE_NAME}.fna.json</td>
<td>Summary statistics of the assembly</td>
</tr>
</tbody>
</table>
<div class="admonition info">
<p class="admonition-title">FASTA inputs are not reassembled by default</p>
<p>In the case where an assembly is given as an input, the only files that will be available are <code>${SAMPLE_NAME}.fna</code> (the original unmodified assembly) and <code>${SAMPLE_NAME}.fna.json</code>. If <code>--reassemble</code> is also given, then all the files seen above will be available.</p>
</div>
<h4 id="hybrid">hybrid<a class="headerlink" href="#hybrid" title="Permanent link">&para;</a></h4>
<p>If long reads are available to supplement input paired-end Illumina reads, a hybrid assembly can be created using <a href="https://github.com/rrwick/Unicycler">Unicycler</a>.</p>
<p>Files descriptions with some modifications were directly taken from <a href="https://github.com/rrwick/Unicycler#output-files">Unicycler's Output Files</a>.
<div class="highlight"><pre><span></span><code>${SAMPLE_NAME}/
└── assembly
    ├── 001_best_spades_graph.gfa
    ├── 002_overlaps_removed.gfa
    ├── 003_long_read_assembly.gfa
    ├── 004_bridges_applied.gfa
    ├── 005_final_clean.gfa
    ├── 006_polished.gfa
    ├── 007_rotated.gfa
    ├── assembly.fasta
    ├── assembly.gfa
    ├── ${SAMPLE_NAME}.fna
    ├── ${SAMPLE_NAME}.fna.json
    └── unicycler.log
</code></pre></div></p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>001_best_spades_graph.gfa</td>
<td>The best SPAdes short-read assembly graph, with a bit of graph clean-up</td>
</tr>
<tr>
<td>002_overlaps_removed.gfa</td>
<td>Overlap-free version of the SPAdes graph, with some more graph clean-up</td>
</tr>
<tr>
<td>003_long_read_assembly.gfa</td>
<td>Assembly graph after long read assembly</td>
</tr>
<tr>
<td>004_bridges_applied.gfa</td>
<td>Bridges applied, before any cleaning or merging</td>
</tr>
<tr>
<td>005_final_clean.gfa</td>
<td>Assembly graph after more redundant contigs removed</td>
</tr>
<tr>
<td>006_polished.gfa</td>
<td>Assembly graph after a round of Pilon polishing</td>
</tr>
<tr>
<td>007_rotated.gfa</td>
<td>Assembly graph after ircular replicons rotated and/or flipped to a start position</td>
</tr>
<tr>
<td>assembly.fasta</td>
<td>The final assembly in FASTA format (same contigs names as in assembly.gfa)</td>
</tr>
<tr>
<td>assembly.gfa</td>
<td>The final assembly in <a href="https://github.com/GFA-spec/GFA-spec/blob/master/GFA1.md">GFA v1</a> graph format</td>
</tr>
<tr>
<td>${SAMPLE_NAME}.fna</td>
<td>The final assembly with renamed header to include sample name</td>
</tr>
<tr>
<td>${SAMPLE_NAME}.fna.json</td>
<td>Summary statistics of the assembly</td>
</tr>
<tr>
<td>unicycler.log</td>
<td>Unicycler log file (same info as stdout)</td>
</tr>
</tbody>
</table>
<h4 id="quality-reports">quality reports<a class="headerlink" href="#quality-reports" title="Permanent link">&para;</a></h4>
<p>Each assembly will have its biological and technical quality assessed with <a href="https://github.com/Ecogenomics/CheckM">CheckM</a> and <a href="http://quast.sourceforge.net/">QUAST</a>. This assessment is done no matter the input type (paired, single, hybrid, or assembly).</p>
<p>Files descriptions with some modifications were directly taken from <a href="https://github.com/Ecogenomics/CheckM/blob/master/bin/checkm">CheckM's Usage</a>  and <a href="http://quast.sourceforge.net/docs/manual.html#sec3">QUAST's Output Files</a>.
<div class="highlight"><pre><span></span><code>assembly/
├── checkm
│   ├── bins/
│   ├── checkm-genes.aln
│   ├── checkm.log
│   ├── checkm-results.txt
│   ├── lineage.ms
│   └── storage/
└── quast
    ├── basic_stats/
    ├── icarus.html
    ├── icarus_viewers
    │   └── contig_size_viewer.html
    ├── predicted_genes
    │   ├── GCF_003431365_glimmer_genes.gff.gz
    │   └── GCF_003431365_glimmer.stderr
    ├── quast.log
    ├── report.{html|pdf|tex|tsv|txt}
    ├── transposed_report.tex
    ├── transposed_report.tsv
    └── transposed_report.txt
</code></pre></div></p>
<p><em>CheckM Outputs</em>  </p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>bins/</td>
<td>A folder with inputs (e.g. proteins) for processing by CheckM</td>
</tr>
<tr>
<td>checkm-genes.aln</td>
<td>Alignment of multi-copy genes and their AAI identity</td>
</tr>
<tr>
<td>checkm.log</td>
<td>The output log of CheckM</td>
</tr>
<tr>
<td>checkm-results.txt</td>
<td>Final results of <a href="https://github.com/Ecogenomics/CheckM/wiki/Workflows#lineage-specific-workflow">CheckM's lineage_wf</a></td>
</tr>
<tr>
<td>lineage.ms</td>
<td>Output file describing marker set for each bin</td>
</tr>
<tr>
<td>storage/</td>
<td>A folder with intermediate results from CheckM processing</td>
</tr>
</tbody>
</table>
<p><em>QUAST Outputs</em>  </p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>basic_stats</td>
<td>A folder with plots of assembly metrics (e.g. GC content, NGx, Nx)</td>
</tr>
<tr>
<td>icarus.html</td>
<td>Icarus main menu with links to interactive viewers.</td>
</tr>
<tr>
<td>icarus_viewers/</td>
<td>Additional reports for Icarus</td>
</tr>
<tr>
<td>predicted_genes/</td>
<td>Predicted gene output</td>
</tr>
<tr>
<td>quast.log</td>
<td>Detailed information about the QUAST run</td>
</tr>
<tr>
<td>report.{html|pdf }</td>
<td>Assessement summary including all tables and plots</td>
</tr>
<tr>
<td>report.{tex|tsv|txt}</td>
<td>Assessment summary in multiple different formats</td>
</tr>
<tr>
<td>transposed_report.{tex|tsv|txt}</td>
<td>Transposed version of the assessment summary</td>
</tr>
</tbody>
</table>
<h3 id="blast"><code>blast</code><a class="headerlink" href="#blast" title="Permanent link">&para;</a></h3>
<p>The <code>blast</code> directory contains the <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi">BLAST</a> results and a BLAST database of the sample assembly.</p>
<p>Each of the <a href="datasets/#blast">User Populated BLAST Sequences</a> (gene, primer, or protein) is BLASTed against the sample assembly. Also if <a href="datasets/#general">setup</a>, annotated genes are BLASTed against the <a href="https://ccb-microbe.cs.uni-saarland.de/plsdb/plasmids/download/">PLSDB</a> BLAST database.</p>
<p>By default, results are returned in tabular format.</p>
<div class="highlight"><pre><span></span><code>${SAMPLE_NAME}/
└── blast
    ├── blastdb
    │   ├── ${SAMPLE_NAME}.nhr
    │   ├── ${SAMPLE_NAME}.nin
    │   └── ${SAMPLE_NAME}.nsq
    ├── genes
    │   └── ${INPUT_NAME}.txt
    ├── primers
    │   └── ${INPUT_NAME}.txt
    ├── proteins
    │   └── ${INPUT_NAME}.txt
    └── ${SAMPLE_NAME}-plsdb.txt
</code></pre></div>
<table>
<thead>
<tr>
<th>Extension</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>.nhr</td>
<td>Sample assembly BLAST database header file</td>
</tr>
<tr>
<td>.nin</td>
<td>Sample assembly BLAST database index file</td>
</tr>
<tr>
<td>.nsq</td>
<td>Sample assembly BLAST database sequence file</td>
</tr>
<tr>
<td>-plsdb.txt</td>
<td>The BLAST results against the PLSDB BALST database assembly.</td>
</tr>
<tr>
<td>.txt</td>
<td>The BLAST results of user input sequence(s) against the sample assembly.</td>
</tr>
</tbody>
</table>
<h3 id="genome-size"><code>genome-size</code><a class="headerlink" href="#genome-size" title="Permanent link">&para;</a></h3>
<p>For every sample <code>${SAMPLE_NAME}-genome-size.txt</code> file is created. This file contains the genome size that was used for analysis. Genome size is used throughout Bactopia for various tasks including error correction, subsampling, and assembly.</p>
<p>By default, the genome size is estimated with Mash, but users have the option to provide their own value or completely disable genome size related features. Learn more about changing the genome size at <a href="usage-basic/#-genome_size">Basic Usage - Genome Size</a></p>
<h3 id="kmers"><code>kmers</code><a class="headerlink" href="#kmers" title="Permanent link">&para;</a></h3>
<p>The <code>kmers</code> directory contains <a href="https://github.com/mcveanlab/mccortex">McCortex</a> 31-mer counts of the cleaned up FASTQ sequences. </p>
<div class="highlight"><pre><span></span><code>${SAMPLE_NAME}/
└── kmers
    └── ${SAMPLE_NAME}.ctx
</code></pre></div>
<table>
<thead>
<tr>
<th>Extension</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>.ctx</td>
<td>A Cortex graph file of the 31-mer counts</td>
</tr>
</tbody>
</table>
<h3 id="logs"><code>logs</code><a class="headerlink" href="#logs" title="Permanent link">&para;</a></h3>
<p>The <code>logs</code> folder will contain useful files for debugging or reviewing what was executed. For each process (e.g. annotation or assembly) the STDOUT and STDERR is log, as well as the time of execution and program versions. These outputs are completely optional, and can be disabled using <code>--skip_logs</code> at runtime.</p>
<div class="highlight"><pre><span></span><code>${SAMPLE_NAME}/
└── logs/
    ├── ${PROCESS_NAME}
    │   ├── ${PROCESS_NAME}.err
    │   ├── ${PROCESS_NAME}.out
    │   ├── ${PROCESS_NAME}.sh
    │   ├── ${PROCESS_NAME}.trace
    │   ├── ${PROCESS_NAME}.versions
    │   ├── ${PROGRAM}.err
    │   └── ${PROGRAM}.out
    └── bactopia.versions
</code></pre></div>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>${PROCESS_NAME}.err</td>
<td>Any STDERR captured by the process.</td>
</tr>
<tr>
<td>${PROCESS_NAME}.out</td>
<td>Any STDERR captured by the process.</td>
</tr>
<tr>
<td>${PROCESS_NAME}.sh</td>
<td>The shell script that process used.</td>
</tr>
<tr>
<td>${PROCESS_NAME}.trace</td>
<td>Compute resource usage by the process (this will not always be available)</td>
</tr>
<tr>
<td>${PROCESS_NAME}.versions</td>
<td>Date and program versions used by the process</td>
</tr>
<tr>
<td>${PROGRAM}.err</td>
<td>STDERR captured for a specific program.</td>
</tr>
<tr>
<td>${PROGRAM}.err</td>
<td>STDOUT captured for a specific program.</td>
</tr>
<tr>
<td>bactopia.versions</td>
<td>Date and Bactopia/Nextflow versions used</td>
</tr>
</tbody>
</table>
<h4 id="example-versions">Example <code>versions</code><a class="headerlink" href="#example-versions" title="Permanent link">&para;</a></h4>
<p>Here is an example of the <code>bactopia.versions</code> file. 
<div class="highlight"><pre><span></span><code># Timestamp
2020-11-11T11:12:31-05:00
# Bactopia Version
bactopia 1.4.11
# Nextflow Version
nextflow 20.07.1
</code></pre></div></p>
<p>All the <code>.versions</code> files will follow this format. The first line is always <code># Timestamp</code> followed by the output of <code>date</code>. Then each line beginning with <code>#</code> will represent a new program and its version. </p>
<h3 id="mapping"><code>mapping</code><a class="headerlink" href="#mapping" title="Permanent link">&para;</a></h3>
<p>The <code>mapping-sequences</code> directory contains <a href="https://github.com/lh3/bwa/">BWA</a> (bwa-mem) mapping results for each of the <a href="datasets/#mapping">User Populated Mapping Sequences</a>. </p>
<div class="highlight"><pre><span></span><code>${SAMPLE_NAME}/
└── mapping
    └── ${MAPPING_INPUT}
        ├── ${MAPPING_INPUT}.bam
        └── ${MAPPING_INPUT}.coverage.txt
</code></pre></div>
<table>
<thead>
<tr>
<th>Extension</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>.bam</td>
<td>The alignments in <a href="http://en.wikipedia.org/wiki/SAMtools">BAM</a> format.</td>
</tr>
<tr>
<td>.coverage.txt</td>
<td>The per-base coverage of the mapping results</td>
</tr>
</tbody>
</table>
<h3 id="minmers"><code>minmers</code><a class="headerlink" href="#minmers" title="Permanent link">&para;</a></h3>
<p>The <code>minmers</code> directory contains <a href="https://github.com/marbl/Mash">Mash</a> and <a href="https://github.com/dib-lab/sourmash">Sourmash</a> sketches of the cleaned FASTQs. If setup, it also contains the results of queries against <a href="datasets/#general">RefSeq, GenBank and PLSDB</a></p>
<div class="highlight"><pre><span></span><code>${SAMPLE_NAME}/
└── minmers
    ├── ${SAMPLE_NAME}-genbank-k21.txt
    ├── ${SAMPLE_NAME}-genbank-k31.txt
    ├── ${SAMPLE_NAME}-genbank-k51.txt
    ├── ${SAMPLE_NAME}-k21.msh
    ├── ${SAMPLE_NAME}-k31.msh
    ├── ${SAMPLE_NAME}-plsdb-k21.txt
    ├── ${SAMPLE_NAME}-refseq-k21.txt
    └── ${SAMPLE_NAME}.sig
</code></pre></div>
<table>
<thead>
<tr>
<th>Extension</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>-genbank-k(21|31|51).txt</td>
<td>Sourmash LCA Gather results against <a href="https://sourmash.readthedocs.io/en/latest/databases.html">Sourmash GenBank Signature</a> (k=21,31,51)</td>
</tr>
<tr>
<td>-k(21|31).msh</td>
<td>A Mash sketch (k=21,31) of the sample</td>
</tr>
<tr>
<td>-plsdb-k21.txt</td>
<td>Mash Screen results against <a href="https://ccb-microbe.cs.uni-saarland.de/plsdb/plasmids/download/">PLSDB Mash Sketch</a></td>
</tr>
<tr>
<td>-refseq-k21.txt</td>
<td>Mash Screen results against <a href="https://mash.readthedocs.io/en/latest/data.html">Mash Refseq Sketch</a></td>
</tr>
<tr>
<td>.sig</td>
<td>A Sourmash signature (k=21,31,51) of the sample</td>
</tr>
</tbody>
</table>
<h3 id="mlst"><code>mlst</code><a class="headerlink" href="#mlst" title="Permanent link">&para;</a></h3>
<p>If a <a href="datasets/#species-specific">Species Specific Dataset</a> has been set up, the <code>mlst</code> directory will contain <a href="https://github.com/sanger-pathogens/ariba">Ariba</a> and <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi">BLAST</a> results for a <a href="https://pubmlst.org/">PubMLST.org</a> schema. For most organisms there is only one MLST schema available, and it will be labeled as <code>default</code>. In cases where a organism has multiple schemas available they will be named following pubMLST's naming.</p>
<div class="highlight"><pre><span></span><code>${SAMPLE_NAME}/
└── mlst
    └── ${SCHEMA}
       ├── ariba
       │   ├── assembled_genes.fa.gz
       │   ├── assembled_seqs.fa.gz
       │   ├── assemblies.fa.gz
       │   ├── debug.report.tsv
       │   ├── log.clusters.gz
       │   ├── mlst_report.details.tsv
       │   ├── mlst_report.tsv
       │   ├── report.tsv
       │   └── version_info.txt
       └── blast
           └── ${SAMPLE_NAME}-blast.json
</code></pre></div>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>assembled_genes.fa.gz</td>
<td>A gzipped FASTA file of only assembled gene sequences (with extensions).</td>
</tr>
<tr>
<td>assembled_seqs.fa.gz</td>
<td>A gzipped FASTA of the assembled sequences (genes and non-coding).</td>
</tr>
<tr>
<td>assemblies.fa.gz</td>
<td>A gzipped FASTA file of the assemblies (complete, unedited, contigs).</td>
</tr>
<tr>
<td>debug.report.tsv</td>
<td>The complete list of clusters, including those that did not pass filtering.</td>
</tr>
<tr>
<td>log.clusters.gz</td>
<td>Detailed logging for the progress of each cluster.</td>
</tr>
<tr>
<td>mlst_report.details.tsv</td>
<td>A more detailed summary of the allele calls.</td>
</tr>
<tr>
<td>mlst_report.tsv</td>
<td>A summary of the allele calls and identified sequence type.</td>
</tr>
<tr>
<td>report.tsv</td>
<td>A detailed <a href="https://github.com/sanger-pathogens/ariba/wiki/Task:-run#report-file">report file</a> of clusters which passed filtering.</td>
</tr>
<tr>
<td>summary.csv</td>
<td>A more condensed <a href="https://github.com/sanger-pathogens/ariba/wiki/Task:-summary">summary</a> of the report.tsv</td>
</tr>
<tr>
<td>version_info.txt</td>
<td>Information on the versions of ARIBA and its dependencies at runtime.</td>
</tr>
<tr>
<td>-blast.json</td>
<td>Allele calls and identified sequence type based on BLAST</td>
</tr>
</tbody>
</table>
<h3 id="quality-control"><code>quality-control</code><a class="headerlink" href="#quality-control" title="Permanent link">&para;</a></h3>
<p>The <code>quality-control</code> directory contains the cleaned up FASTQs (<a href="https://jgi.doe.gov/data-and-tools/bbtools/">BBTools</a> and <a href="https://github.com/mourisl/Lighter">Lighter</a>) and summary statitics (<a href="https://github.com/s-andrews/FastQC">FastQC</a> and <a href="https://github.com/rpetit3/fastq-scan">Fastq-Scan</a>) before and after cleanup.</p>
<div class="highlight"><pre><span></span><code>${SAMPLE_NAME}/
└── quality-control
    ├── logs
    │   ├── bbduk-adapter.log
    │   └── bbduk-phix.log
    ├── ${SAMPLE_NAME}(|_R1|_R2).fastq.gz
    └── summary-(original|final)
        ├── ${SAMPLE_NAME}(|_R1|_R2)-(original|final)_fastqc.html
        ├── ${SAMPLE_NAME}(|_R1|_R2)-(original|final)_fastqc.zip
        └── ${SAMPLE_NAME}(|_R1|_R2)-(original|final).json
</code></pre></div>
<table>
<thead>
<tr>
<th>Extension</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>-adapter.log</td>
<td>A description of how many reads were filtered during the adapter removal step</td>
</tr>
<tr>
<td>-phix.log</td>
<td>A description of how many reads were filtered during the PhiX removal step</td>
</tr>
<tr>
<td>.fastq.gz</td>
<td>The cleaned up FASTQ(s), <code>_R1</code> and <code>_R2</code> for paired-end reads, and an empty string for single-end reads.</td>
</tr>
<tr>
<td>_fastqc.html</td>
<td>The FastQC html report of the original and final FASTQ(s)</td>
</tr>
<tr>
<td>_fastqc.zip</td>
<td>The zipped FastQC full report of the original and final FASTQ(s)</td>
</tr>
<tr>
<td>.json</td>
<td>Summary statistics (e.g. qualities and read lengths) of the original and final FASTQ(s)</td>
</tr>
</tbody>
</table>
<h3 id="variants"><code>variants</code><a class="headerlink" href="#variants" title="Permanent link">&para;</a></h3>
<p>The <code>variants</code> directory contains the results of <a href="https://github.com/tseemann/snippy">Snippy</a> variant calls against one or more reference genomes. There are two subdirectories <code>auto</code> and <code>user</code>. </p>
<p>The <code>auto</code> directory includes variants calls against automatically selected reference genome(s) based on nearest Mash distance to RefSeq completed genomes. This process only happens if a <a href="datasets/#species-specific">Species Specific Dataset</a> was a created. By default, only a single reference genome (nearest) is selected. This feature can be disabled (<code>--disable_auto_variants</code>) or the number of genomes changed (<code>--max_references INT</code>).</p>
<p>The <code>user</code> directory contains variant calls against for each of the <a href="datasets/#reference-genomes">User Populated Reference Genomes</a>.</p>
<p>The following description of files was directly taken from <a href="https://github.com/tseemann/snippy#output-files">Snippy's Output Files</a> section. Slight modifications were made to the order of rows.</p>
<div class="highlight"><pre><span></span><code>${SAMPLE_NAME}/
└── variants
    └── (auto|user)
        └── ${REFERENCE_NAME}
            ├── ${SAMPLE_NAME}.aligned.fa
            ├── ${SAMPLE_NAME}.annotated.vcf
            ├── ${SAMPLE_NAME}.bam
            ├── ${SAMPLE_NAME}.bam.bai
            ├── ${SAMPLE_NAME}.bed
            ├── ${SAMPLE_NAME}.consensus.fa
            ├── ${SAMPLE_NAME}.consensus.subs.fa
            ├── ${SAMPLE_NAME}.consensus.subs.masked.fa
            ├── ${SAMPLE_NAME}.coverage.txt
            ├── ${SAMPLE_NAME}.csv
            ├── ${SAMPLE_NAME}.filt.vcf
            ├── ${SAMPLE_NAME}.gff
            ├── ${SAMPLE_NAME}.html
            ├── ${SAMPLE_NAME}.log
            ├── ${SAMPLE_NAME}.raw.vcf
            ├── ${SAMPLE_NAME}.subs.vcf
            ├── ${SAMPLE_NAME}.tab
            ├── ${SAMPLE_NAME}.txt
            └── ${SAMPLE_NAME}.vcf
</code></pre></div>
<table>
<thead>
<tr>
<th>Extension</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>.aligned.fa</td>
<td>A version of the reference but with <code>-</code> at position with <code>depth=0</code> and <code>N</code> for <code>0 &lt; depth &lt; --mincov</code> (<strong>does not have variants</strong>)</td>
</tr>
<tr>
<td>.annotated.vcf</td>
<td>The final variant calls with additional annotations from Reference genome's GenBank file</td>
</tr>
<tr>
<td>.bam</td>
<td>The alignments in <a href="http://en.wikipedia.org/wiki/SAMtools">BAM</a> format. Includes unmapped, multimapping reads. Excludes duplicates.</td>
</tr>
<tr>
<td>.bam.bai</td>
<td>Index for the .bam file</td>
</tr>
<tr>
<td>.bed</td>
<td>The variants in <a href="http://genome.ucsc.edu/FAQ/FAQformat.html#format1">BED</a> format</td>
</tr>
<tr>
<td>.consensus.fa</td>
<td>A version of the reference genome with <em>all</em> variants instantiated</td>
</tr>
<tr>
<td>.consensus.subs.fa</td>
<td>A version of the reference genome with <em>only substitution</em> variants instantiated</td>
</tr>
<tr>
<td>.consensus.subs.masked.fa</td>
<td>A version of the reference genome with <em>only substitution</em> variants instantiated and low-coverage regions masked</td>
</tr>
<tr>
<td>.coverage.txt</td>
<td>The per-base coverage of each position in the reference genome</td>
</tr>
<tr>
<td>.csv</td>
<td>A <a href="http://en.wikipedia.org/wiki/Comma-separated_values">comma-separated</a> version of the .tab file</td>
</tr>
<tr>
<td>.filt.vcf</td>
<td>The filtered variant calls from Freebayes</td>
</tr>
<tr>
<td>.gff</td>
<td>The variants in <a href="http://www.sequenceontology.org/gff3.shtml">GFF3</a> format</td>
</tr>
<tr>
<td>.html</td>
<td>A <a href="http://en.wikipedia.org/wiki/HTML">HTML</a> version of the .tab file</td>
</tr>
<tr>
<td>.log</td>
<td>A log file with the commands run and their outputs</td>
</tr>
<tr>
<td>.raw.vcf</td>
<td>The unfiltered variant calls from Freebayes</td>
</tr>
<tr>
<td>.subs.vcf</td>
<td><em>Only substitution</em> variants from the final annotated variants</td>
</tr>
<tr>
<td>.tab</td>
<td>A simple <a href="http://en.wikipedia.org/wiki/Tab-separated_values">tab-separated</a> summary of all the variants</td>
</tr>
<tr>
<td>.txt</td>
<td>A summary of the Snippy run.</td>
</tr>
<tr>
<td>.vcf</td>
<td>The final annotated variants in <a href="http://en.wikipedia.org/wiki/Variant_Call_Format">VCF</a> format</td>
</tr>
</tbody>
</table>

<!-- Last update of source file -->

  


              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright © 2019-2021 Robert A. Petit III. All rights reserved.
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs Insiders
    </a>
  
</div>
      
        <div class="md-social">
  
    
    
    
    
      
      
    
    <a href="https://github.com/bactopia/bactopia" target="_blank" rel="noopener" title="github.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
    </a>
  
    
    
    
    
      
      
    
    <a href="https://gitter.im/bactopia/Lobby" target="_blank" rel="noopener" title="gitter.im" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 384 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M66.4 322.5H16V0h50.4v322.5zM166.9 76.1h-50.4V512h50.4V76.1zm100.6 0h-50.4V512h50.4V76.1zM368 76h-50.4v247H368V76z"/></svg>
    </a>
  
    
    
    
    
      
      
    
    <a href="https://hub.docker.com/r/bactopia/bactopia" target="_blank" rel="noopener" title="hub.docker.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 640 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M349.9 236.3h-66.1v-59.4h66.1v59.4zm0-204.3h-66.1v60.7h66.1V32zm78.2 144.8H362v59.4h66.1v-59.4zm-156.3-72.1h-66.1v60.1h66.1v-60.1zm78.1 0h-66.1v60.1h66.1v-60.1zm276.8 100c-14.4-9.7-47.6-13.2-73.1-8.4-3.3-24-16.7-44.9-41.1-63.7l-14-9.3-9.3 14c-18.4 27.8-23.4 73.6-3.7 103.8-8.7 4.7-25.8 11.1-48.4 10.7H2.4c-8.7 50.8 5.8 116.8 44 162.1 37.1 43.9 92.7 66.2 165.4 66.2 157.4 0 273.9-72.5 328.4-204.2 21.4.4 67.6.1 91.3-45.2 1.5-2.5 6.6-13.2 8.5-17.1l-13.3-8.9zm-511.1-27.9h-66v59.4h66.1v-59.4zm78.1 0h-66.1v59.4h66.1v-59.4zm78.1 0h-66.1v59.4h66.1v-59.4zm-78.1-72.1h-66.1v60.1h66.1v-60.1z"/></svg>
    </a>
  
    
    
    
    
      
      
    
    <a href="https://twitter.com/rpetit3" target="_blank" rel="noopener" title="twitter.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"/></svg>
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    <script id="__config" type="application/json">{"base": "..", "features": ["navigation.tabs", "navigation.tabs.sticky"], "search": "../assets/javascripts/workers/search.6c7302c4.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../assets/javascripts/bundle.803ed251.min.js"></script>
      
        <script src="../custom/jquery.min.js"></script>
      
        <script src="../custom/jquery.magnific-popup.min.js"></script>
      
        <script src="../custom/extra.js"></script>
      
    
  </body>
</html>