
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="An extensive workflow to process Nanopore and Illumina sequencing for bacterial genomes.">
      
      
        <meta name="author" content="Robert A. Petit III">
      
      
        <link rel="canonical" href="https://bactopia.github.io/v2.1.1/bactopia-tools/merlin/">
      
      
        
      
      
      <link rel="icon" href="../../assets/favicon.ico">
      <meta name="generator" content="mkdocs-1.3.1, mkdocs-material-8.5.3+insiders-4.23.5">
    
    
      
        <title>merlin - Bactopia</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.fb3608a7.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.cbb835fc.min.css">
        
      
      

    
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Ubuntu:300,300i,400,400i,700,700i%7CUbuntu+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Ubuntu";--md-code-font:"Ubuntu Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../custom/magnific-popup.css">
    
      <link rel="stylesheet" href="../../custom/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      
  


  
  


  <script id="__analytics">function __md_analytics(){window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)},ga.l=+new Date,ga("create","UA-104199698-2","auto"),ga("set","anonymizeIp",!0),ga("send","pageview"),document.addEventListener("DOMContentLoaded",function(){document.forms.search&&document.forms.search.query.addEventListener("blur",function(){var e;this.value&&(e=document.location.pathname,ga("send","pageview",e+"?q="+this.value))}),document$.subscribe(function(){var a=document.forms.feedback;if(void 0!==a)for(var e of a.querySelectorAll("[type=submit]"))e.addEventListener("click",function(e){e.preventDefault();var t=document.location.pathname,e=this.getAttribute("data-md-value");ga("send","event","feedback","click",t,e),a.firstElementChild.disabled=!0;e=a.querySelector(".md-feedback__note [data-md-value='"+e+"']");e&&(e.hidden=!1)}),a.hidden=!1}),location$.subscribe(function(e){ga("send","pageview",e.pathname)})});var e=document.createElement("script");e.async=!0,e.src="https://www.google-analytics.com/analytics.js",document.getElementById("__analytics").insertAdjacentElement("afterEnd",e)}</script>

  
    <script>"undefined"!=typeof __md_analytics&&__md_analytics()</script>
  

    
    
      
        <meta  property="og:type"  content="website" >
      
        <meta  property="og:title"  content="merlin - Bactopia" >
      
        <meta  property="og:description"  content="An extensive workflow to process Nanopore and Illumina sequencing for bacterial genomes." >
      
        <meta  property="og:image"  content="https://bactopia.github.io/v2.1.1/assets/images/social/bactopia-tools/merlin.png" >
      
        <meta  property="og:image:type"  content="image/png" >
      
        <meta  property="og:image:width"  content="1200" >
      
        <meta  property="og:image:height"  content="630" >
      
        <meta  property="og:url"  content="https://bactopia.github.io/v2.1.1/bactopia-tools/merlin/" >
      
        <meta  name="twitter:card"  content="summary_large_image" >
      
        <meta  name="twitter:title"  content="merlin - Bactopia" >
      
        <meta  name="twitter:description"  content="An extensive workflow to process Nanopore and Illumina sequencing for bacterial genomes." >
      
        <meta  name="twitter:image"  content="https://bactopia.github.io/v2.1.1/assets/images/social/bactopia-tools/merlin.png" >
      
    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#bactopia-tool-merlin" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--lifted" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Bactopia" class="md-header__button md-logo" aria-label="Bactopia" data-md-component="logo">
      
  <img src="../../assets/bactopia-small-logo.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Bactopia
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              merlin
            
          </span>
        </div>
      </div>
    </div>
    
      <form class="md-header__option" data-md-component="palette">
        
          
          
          <input class="md-option" data-md-color-media="" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_0">
          
            <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M17 6H7c-3.31 0-6 2.69-6 6s2.69 6 6 6h10c3.31 0 6-2.69 6-6s-2.69-6-6-6zm0 10H7c-2.21 0-4-1.79-4-4s1.79-4 4-4h10c2.21 0 4 1.79 4 4s-1.79 4-4 4zM7 9c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"/></svg>
            </label>
          
        
          
          
          <input class="md-option" data-md-color-media="" data-md-color-scheme="slate" data-md-color-primary="red" data-md-color-accent="red"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_1">
          
            <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_0" hidden>
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M17 7H7a5 5 0 0 0-5 5 5 5 0 0 0 5 5h10a5 5 0 0 0 5-5 5 5 0 0 0-5-5m0 8a3 3 0 0 1-3-3 3 3 0 0 1 3-3 3 3 0 0 1 3 3 3 3 0 0 1-3 3Z"/></svg>
            </label>
          
        
      </form>
    
    
      <script>var media,input,key,value,palette=__md_get("__palette");if(palette&&palette.color){"(prefers-color-scheme)"===palette.color.media&&(media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']"),palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent"));for([key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/bactopia/bactopia" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.2.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
    
      <nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        




  
    
    
    
      <li class="md-tabs__item">
        <a href="../.." class="md-tabs__link">
          
  
    
  
  Bactopia

        </a>
      </li>
    
  

      
        

  




  
    
    
    
      <li class="md-tabs__item">
        <a href="../" class="md-tabs__link md-tabs__link--active">
          
  
    
  
  Bactopia Tools

        </a>
      </li>
    
  

      
        




  
    
    
    
      <li class="md-tabs__item">
        <a href="../../blog/" class="md-tabs__link">
          
  
    
  
  Blog

        </a>
      </li>
    
  

      
    </ul>
  </div>
</nav>
    
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Bactopia" class="md-nav__button md-logo" aria-label="Bactopia" data-md-component="logo">
      
  <img src="../../assets/bactopia-small-logo.png" alt="logo">

    </a>
    Bactopia
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/bactopia/bactopia" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.2.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_1" type="checkbox" id="__nav_1" >
        
        
          <label class="md-nav__link" for="__nav_1">
            
  
  <span class="md-ellipsis">
    Bactopia
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" aria-label="Bactopia" data-md-level="1">
          <label class="md-nav__title" for="__nav_1">
            <span class="md-nav__icon md-icon"></span>
            Bactopia
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Introduction
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../../quick-start/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Quick Start
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../../installation/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Installation
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorial/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Tutorial
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../../datasets/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Build Datasets
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../../usage-basic/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Basic Usage
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../../workflow-overview/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Workflow Overview
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../../output-overview/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Output Overview
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../../usage-complete/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Complete Usage
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../../troubleshooting/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Troubleshooting
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../../presentations/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Presentations
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../../enhancements/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Enhancements to OSS
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../../acknowledgements/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Acknowledgements
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../../changelog/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Changelog
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
    
  
  
    
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
          <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_2" type="checkbox" id="__nav_2" checked>
        
        
          <label class="md-nav__link" for="__nav_2">
            
  
  <span class="md-ellipsis">
    Bactopia Tools
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" aria-label="Bactopia Tools" data-md-level="1">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            Bactopia Tools
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Introduction
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
    
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
          <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_2_2" type="checkbox" id="__nav_2_2" checked>
        
        
          <label class="md-nav__link" for="__nav_2_2">
            
  
  <span class="md-ellipsis">
    Subworkflows
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" aria-label="Subworkflows" data-md-level="2">
          <label class="md-nav__title" for="__nav_2_2">
            <span class="md-nav__icon md-icon"></span>
            Subworkflows
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../ariba/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    ariba
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bakta/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    bakta
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../eggnog/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    eggnog
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../gtdb/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    gtdb
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  <span class="md-ellipsis">
    merlin
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    merlin
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#example-usage" class="md-nav__link">
    <span class="md-ellipsis">
      Example Usage
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#output-overview" class="md-nav__link">
    <span class="md-ellipsis">
      Output Overview
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Output Overview">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#results" class="md-nav__link">
    <span class="md-ellipsis">
      Results
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Results">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#top-level" class="md-nav__link">
    <span class="md-ellipsis">
      Top Level
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#agrvate" class="md-nav__link">
    <span class="md-ellipsis">
      AgrVATE
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ectyper" class="md-nav__link">
    <span class="md-ellipsis">
      ECTyper
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#emmtyper" class="md-nav__link">
    <span class="md-ellipsis">
      emmtyper
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hicap" class="md-nav__link">
    <span class="md-ellipsis">
      hicap
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hpsuissero" class="md-nav__link">
    <span class="md-ellipsis">
      HpsuisSero
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#kleborate" class="md-nav__link">
    <span class="md-ellipsis">
      Kleborate
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#legsta" class="md-nav__link">
    <span class="md-ellipsis">
      legsta
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#lissero" class="md-nav__link">
    <span class="md-ellipsis">
      LisSero
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#mash" class="md-nav__link">
    <span class="md-ellipsis">
      Mash
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#meningotype" class="md-nav__link">
    <span class="md-ellipsis">
      meningotype
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ngmaster" class="md-nav__link">
    <span class="md-ellipsis">
      ngmaster
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#seqsero2" class="md-nav__link">
    <span class="md-ellipsis">
      SeqSero2
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sistr" class="md-nav__link">
    <span class="md-ellipsis">
      SISTR
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#spatyper" class="md-nav__link">
    <span class="md-ellipsis">
      spaTyper
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ssuissero" class="md-nav__link">
    <span class="md-ellipsis">
      SsuisSero
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#staphopia-sccmec" class="md-nav__link">
    <span class="md-ellipsis">
      staphopia-sccmec
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#tbprofiler" class="md-nav__link">
    <span class="md-ellipsis">
      TBProfiler
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#audit-trail" class="md-nav__link">
    <span class="md-ellipsis">
      Audit Trail
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Audit Trail">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#logs" class="md-nav__link">
    <span class="md-ellipsis">
      Logs
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#nextflow-reports" class="md-nav__link">
    <span class="md-ellipsis">
      Nextflow Reports
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#program-versions" class="md-nav__link">
    <span class="md-ellipsis">
      Program Versions
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#parameters" class="md-nav__link">
    <span class="md-ellipsis">
      Parameters
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Parameters">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#required-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      Required Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#filtering-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      Filtering Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#mashdist-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      mashdist Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#agrvate-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      AgrVATE Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ectyper-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      ECTyper Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#emmtyper-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      emmtyper Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hicap-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      hicap Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hpsuissero-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      HpsuisSero Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#kleborate-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      Kleborate Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#legsta-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      legsta Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#lissero-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      LisSero Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#meningotype-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      meningotype Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ngmaster-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      ngmaster Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#seqsero2-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      SeqSero2 Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sistr-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      SISTR Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#spatyper-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      spaTyper Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#staphopia-sccmec-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      staphopia-sccmec Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ssuissero-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      SsuisSero Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#tbprofiler-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      TBProfiler Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#optional-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      Optional Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#max-job-request-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      Max Job Request Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#nextflow-configuration-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      Nextflow Configuration Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#nextflow-profile-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      Nextflow Profile Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#helpful-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      Helpful Parameters
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#citations" class="md-nav__link">
    <span class="md-ellipsis">
      Citations
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../pangenome/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    pangenome
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../snippy/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    snippy
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../staphtyper/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    staphtyper
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_2_3" type="checkbox" id="__nav_2_3" >
        
        
          <label class="md-nav__link" for="__nav_2_3">
            
  
  <span class="md-ellipsis">
    Modules
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" aria-label="Modules" data-md-level="2">
          <label class="md-nav__title" for="__nav_2_3">
            <span class="md-nav__icon md-icon"></span>
            Modules
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../abricate/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    abricate
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../agrvate/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    agrvate
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../amrfinderplus/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    amrfinderplus
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../busco/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    busco
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../checkm/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    checkm
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../ectyper/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    ectyper
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../emmtyper/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    emmtyper
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../fastani/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    fastani
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../gamma/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    gamma
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../genotyphi/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    genotyphi
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../hicap/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    hicap
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../hpsuissero/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    hpsuissero
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../ismapper/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    ismapper
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../kleborate/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    kleborate
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../kraken2/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    kraken2
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../legsta/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    legsta
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../lissero/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    lissero
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../mashdist/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    mashdist
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../mashtree/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    mashtree
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../mcroni/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    mcroni
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../meningotype/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    meningotype
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../mlst/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    mlst
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../mobsuite/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    mobsuite
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../mykrobe/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    mykrobe
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../ngmaster/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    ngmaster
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../plasmidfinder/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    plasmidfinder
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../prokka/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    prokka
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../rgi/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    rgi
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../seqsero2/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    seqsero2
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../seroba/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    seroba
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../shigatyper/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    shigatyper
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../sistr/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    sistr
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../spatyper/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    spatyper
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../ssuissero/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    ssuissero
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../staphopiasccmec/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    staphopiasccmec
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../tbprofiler/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    tbprofiler
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_3" type="checkbox" id="__nav_3" >
        
        
          <label class="md-nav__link" for="__nav_3">
            
  
  <span class="md-ellipsis">
    Blog
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" aria-label="Blog" data-md-level="1">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            Blog
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../../blog/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Bactopia Blog
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#example-usage" class="md-nav__link">
    <span class="md-ellipsis">
      Example Usage
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#output-overview" class="md-nav__link">
    <span class="md-ellipsis">
      Output Overview
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Output Overview">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#results" class="md-nav__link">
    <span class="md-ellipsis">
      Results
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Results">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#top-level" class="md-nav__link">
    <span class="md-ellipsis">
      Top Level
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#agrvate" class="md-nav__link">
    <span class="md-ellipsis">
      AgrVATE
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ectyper" class="md-nav__link">
    <span class="md-ellipsis">
      ECTyper
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#emmtyper" class="md-nav__link">
    <span class="md-ellipsis">
      emmtyper
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hicap" class="md-nav__link">
    <span class="md-ellipsis">
      hicap
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hpsuissero" class="md-nav__link">
    <span class="md-ellipsis">
      HpsuisSero
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#kleborate" class="md-nav__link">
    <span class="md-ellipsis">
      Kleborate
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#legsta" class="md-nav__link">
    <span class="md-ellipsis">
      legsta
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#lissero" class="md-nav__link">
    <span class="md-ellipsis">
      LisSero
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#mash" class="md-nav__link">
    <span class="md-ellipsis">
      Mash
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#meningotype" class="md-nav__link">
    <span class="md-ellipsis">
      meningotype
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ngmaster" class="md-nav__link">
    <span class="md-ellipsis">
      ngmaster
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#seqsero2" class="md-nav__link">
    <span class="md-ellipsis">
      SeqSero2
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sistr" class="md-nav__link">
    <span class="md-ellipsis">
      SISTR
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#spatyper" class="md-nav__link">
    <span class="md-ellipsis">
      spaTyper
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ssuissero" class="md-nav__link">
    <span class="md-ellipsis">
      SsuisSero
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#staphopia-sccmec" class="md-nav__link">
    <span class="md-ellipsis">
      staphopia-sccmec
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#tbprofiler" class="md-nav__link">
    <span class="md-ellipsis">
      TBProfiler
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#audit-trail" class="md-nav__link">
    <span class="md-ellipsis">
      Audit Trail
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Audit Trail">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#logs" class="md-nav__link">
    <span class="md-ellipsis">
      Logs
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#nextflow-reports" class="md-nav__link">
    <span class="md-ellipsis">
      Nextflow Reports
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#program-versions" class="md-nav__link">
    <span class="md-ellipsis">
      Program Versions
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#parameters" class="md-nav__link">
    <span class="md-ellipsis">
      Parameters
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Parameters">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#required-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      Required Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#filtering-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      Filtering Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#mashdist-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      mashdist Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#agrvate-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      AgrVATE Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ectyper-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      ECTyper Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#emmtyper-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      emmtyper Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hicap-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      hicap Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hpsuissero-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      HpsuisSero Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#kleborate-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      Kleborate Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#legsta-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      legsta Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#lissero-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      LisSero Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#meningotype-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      meningotype Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ngmaster-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      ngmaster Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#seqsero2-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      SeqSero2 Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sistr-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      SISTR Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#spatyper-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      spaTyper Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#staphopia-sccmec-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      staphopia-sccmec Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ssuissero-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      SsuisSero Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#tbprofiler-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      TBProfiler Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#optional-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      Optional Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#max-job-request-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      Max Job Request Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#nextflow-configuration-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      Nextflow Configuration Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#nextflow-profile-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      Nextflow Profile Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#helpful-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      Helpful Parameters
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#citations" class="md-nav__link">
    <span class="md-ellipsis">
      Citations
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                

<!-- Edit button -->

  <a
    href="https://github.com/bactopia/bactopia.github.io/edit/master/docs/bactopia-tools/merlin.md"
    title="Edit this page"
    class="md-content__button md-icon"
  >
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25Z"/></svg>
  </a>


<!--
  Hack: check whether the content contains a h1 headline. If it
  doesn't, the page title (or respectively site name) is used
  as the main headline.
-->


<!-- Markdown content -->
<h1 id="bactopia-tool-merlin">Bactopia Tool - <code>merlin</code><a class="headerlink" href="#bactopia-tool-merlin" title="Permanent link">&para;</a></h1>
<p><em>MinmER assisted species-specific bactopia tool seLectIoN</em>, or Merlin, uses distances based
on the RefSeq sketch downloaded by <code>bactopia datasets</code> to automatically run species-specific tools.</p>
<p>Currently Merlin knows 16 spells for which cover the following:</p>
<table>
<thead>
<tr>
<th>Genus/Species</th>
<th>Tools</th>
</tr>
</thead>
<tbody>
<tr>
<td>Escherichia</td>
<td><a href="/bactopia-tools/ectyper/">ECTyper</a></td>
</tr>
<tr>
<td>Haemophilus</td>
<td><a href="/bactopia-tools/hicap/">hicap</a>, <a href="/bactopia-tools/ssuissero/">HpsuisSero</a></td>
</tr>
<tr>
<td>Klebsiella</td>
<td><a href="/bactopia-tools/kleborate/">Kleborate</a></td>
</tr>
<tr>
<td>Legionella</td>
<td><a href="/bactopia-tools/legsta/">legsta</a></td>
</tr>
<tr>
<td>Listeria</td>
<td><a href="/bactopia-tools/lissero/">LisSero</a></td>
</tr>
<tr>
<td>Mycobacterium</td>
<td><a href="/bactopia-tools/tbprofiler/">TBProfiler</a></td>
</tr>
<tr>
<td>Neisseria</td>
<td><a href="/bactopia-tools/meningotype/">meningotype</a>, <a href="/bactopia-tools/ngmaster/">ngmaster</a></td>
</tr>
<tr>
<td>Salmonella</td>
<td><a href="/bactopia-tools/seqsero2/">SeqSero2</a>, <a href="/bactopia-tools/sistr/">SISTR</a></td>
</tr>
<tr>
<td>Staphylococcus</td>
<td><a href="/bactopia-tools/agrvate/">AgrVATE</a>, <a href="/bactopia-tools/spatyper/">spaTyper</a>, <a href="/bactopia-tools/staphopiasccmec/">staphopia-sccmec</a></td>
</tr>
<tr>
<td>Streptococcus</td>
<td><a href="/bactopia-tools/emmtyper/">emmtyper</a>, <a href="/bactopia-tools/ssuissero/">SsuisSero</a></td>
</tr>
</tbody>
</table>
<p>Merlin is avialable as an independent Bactopia Tool, or in the Bactopia with the <code>--ask_merlin</code> parameter. Even better,
if you want to force Merlin to execute all species-specific tools (no matter the distance), you can use <code>--full_merlin</code>.
Then all the spells will be unleashed!</p>
<h2 id="example-usage">Example Usage<a class="headerlink" href="#example-usage" title="Permanent link">&para;</a></h2>
<div class="highlight"><pre><span></span><code>bactopia --wf merlin \
  --bactopia /path/to/your/bactopia/results \ 
  --include includes.txt  
</code></pre></div>
<h2 id="output-overview">Output Overview<a class="headerlink" href="#output-overview" title="Permanent link">&para;</a></h2>
<p>Below is the default output structure for the <code>merlin</code> tool. Where possible the 
file descriptions below were modified from a tools description.</p>
<div class="highlight"><pre><span></span><code>merlin/
 &lt;SAMPLE_NAME&gt;
    agrvate
       &lt;SAMPLE_NAME&gt;-agr_gp.tab
       &lt;SAMPLE_NAME&gt;-blastn_log.txt
       &lt;SAMPLE_NAME&gt;-summary.tab
    ectyper
       &lt;SAMPLE_NAME&gt;.tsv
       blast_output_alleles.txt
    emmtyper
       &lt;SAMPLE_NAME&gt;.tsv
    hicap
       &lt;SAMPLE_NAME&gt;.tsv
    hpsuissero
       &lt;SAMPLE_NAME&gt;_serotyping_res.tsv
    kleborate
       &lt;SAMPLE_NAME&gt;.results.txt
    legsta
       &lt;SAMPLE_NAME&gt;.tsv
    lissero
       &lt;SAMPLE_NAME&gt;.tsv
    logs
       &lt;BACTOPIA_TOOL&gt;
           nf-&lt;BACTOPIA_TOOL&gt;.{begin,err,log,out,run,sh,trace}
           versions.yml
    mashdist
       &lt;SAMPLE_NAME&gt;-dist.txt
    meningotype
       &lt;SAMPLE_NAME&gt;.tsv
    ngmaster
       &lt;SAMPLE_NAME&gt;.tsv
    seqsero2
       &lt;SAMPLE_NAME&gt;_log.txt
       &lt;SAMPLE_NAME&gt;_result.tsv
       &lt;SAMPLE_NAME&gt;_result.txt
    sistr
       &lt;SAMPLE_NAME&gt;-allele.fasta.gz
       &lt;SAMPLE_NAME&gt;-allele.json.gz
       &lt;SAMPLE_NAME&gt;-cgmlst.csv
       &lt;SAMPLE_NAME&gt;.tsv
    spatyper
       &lt;SAMPLE_NAME&gt;.tsv
    ssuissero
       &lt;SAMPLE_NAME&gt;_serotyping_res.tsv
    staphopiasccmec
       &lt;SAMPLE_NAME&gt;.tsv
    tbprofiler
        &lt;SAMPLE_NAME&gt;.results.csv
        &lt;SAMPLE_NAME&gt;.results.json
        &lt;SAMPLE_NAME&gt;.results.txt
        bam
           &lt;SAMPLE_NAME&gt;.bam
        vcf
            &lt;SAMPLE_NAME&gt;.targets.csq.vcf.gz
 logs
    csvtk_concat
       &lt;BACTOPIA_TOOL&gt;
           nf-csvtk_concat.{begin,err,log,out,run,sh,trace}
           versions.yml
    custom_dumpsoftwareversions
        nf-custom_dumpsoftwareversions.{begin,err,log,out,run,sh,trace}
        versions.yml
 nf-reports
    merlin-dag.dot
    merlin-report.html
    merlin-timeline.html
    merlin-trace.txt
 agrvate.tsv
 ectyper.tsv
 emmtyper.tsv
 hicap.tsv
 hpsuissero.tsv
 kleborate.tsv
 legsta.tsv
 lissero.tsv
 meningotype.tsv
 ngmaster.tsv
 seqsero2.tsv
 sistr.tsv
 software_versions.yml
 software_versions_mqc.yml
 spatyper.tsv
 ssuissero.tsv
 staphopiasccmec.tsv
</code></pre></div>
<div class="admonition info">
<p class="admonition-title">Directory structure might be different</p>
<p><code>merlin</code> is available as a standalone Bactopia Tool, as well as from
the main Bactopia workflow (e.g. through Staphopia or Merlin). If executed 
from Bactopia, the <code>merlin</code> directory structure might be different, but the
output descriptions below still apply.</p>
</div>
<h3 id="results">Results<a class="headerlink" href="#results" title="Permanent link">&para;</a></h3>
<h4 id="top-level">Top Level<a class="headerlink" href="#top-level" title="Permanent link">&para;</a></h4>
<p>Below are results that are in the base directory.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>agrvate.tsv</td>
<td>A merged TSV file with <code>AgrVATE</code> results from all samples</td>
</tr>
<tr>
<td>ectyper.tsv</td>
<td>A merged TSV file with <code>ECTyper</code> results from all samples</td>
</tr>
<tr>
<td>emmtyper.tsv</td>
<td>A merged TSV file with <code>emmtyper</code> results from all samples</td>
</tr>
<tr>
<td>hicap.tsv</td>
<td>A merged TSV file with <code>hicap</code> results from all samples</td>
</tr>
<tr>
<td>hpsuissero.tsv</td>
<td>A merged TSV file with <code>HpsuisSero</code> results from all samples</td>
</tr>
<tr>
<td>kleborate.tsv</td>
<td>A merged TSV file with <code>Kleborate</code> results from all samples</td>
</tr>
<tr>
<td>legsta.tsv</td>
<td>A merged TSV file with <code>legsta</code> results from all samples</td>
</tr>
<tr>
<td>lissero.tsv</td>
<td>A merged TSV file with <code>LisSero</code> results from all samples</td>
</tr>
<tr>
<td>meningotype.tsv</td>
<td>A merged TSV file with <code>meningotype</code> results from all samples</td>
</tr>
<tr>
<td>ngmaster.tsv</td>
<td>A merged TSV file with <code>ngmaster</code> results from all samples</td>
</tr>
<tr>
<td>seqsero2.tsv</td>
<td>A merged TSV file with <code>seqsero2</code> results from all samples</td>
</tr>
<tr>
<td>sistr.tsv</td>
<td>A merged TSV file with <code>SISTR</code> results from all samples</td>
</tr>
<tr>
<td>spatyper.tsv</td>
<td>A merged TSV file with <code>spaTyper</code> results from all samples</td>
</tr>
<tr>
<td>ssuissero.tsv</td>
<td>A merged TSV file with <code>SsuisSero</code> results from all samples</td>
</tr>
<tr>
<td>staphopiasccmec.tsv</td>
<td>A merged TSV file with <code>staphopia-sccmec</code> results from all samples</td>
</tr>
</tbody>
</table>
<h4 id="agrvate">AgrVATE<a class="headerlink" href="#agrvate" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results from <a href="https://github.com/VishnuRaghuram94/AgrVATE">AgrVATE</a>.</p>
<table>
<thead>
<tr>
<th>Extension</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>-agr_gp.tab</td>
<td>Detailed report for <em>agr</em> kmer matches</td>
</tr>
<tr>
<td>-blastn_log.txt</td>
<td>Log files from programs called by <code>AgrVATE</code></td>
</tr>
<tr>
<td>-summary.tab</td>
<td>A final summary report for <em>agr</em> typing</td>
</tr>
</tbody>
</table>
<h4 id="ectyper">ECTyper<a class="headerlink" href="#ectyper" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results from <a href="https://github.com/phac-nml/ecoli_serotyping">ECTyper</a>.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>&lt;SAMPLE_NAME&gt;.tsv</td>
<td>A tab-delimited file with <code>ECTyper</code> result, see <a href="https://github.com/phac-nml/ecoli_serotyping#report-format">ECTyper - Report format</a> for details</td>
</tr>
<tr>
<td>blast_output_alleles.txt</td>
<td>Allele report generated from BLAST results</td>
</tr>
</tbody>
</table>
<h4 id="emmtyper">emmtyper<a class="headerlink" href="#emmtyper" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results from <a href="https://github.com/MDU-PHL/emmtyper">emmtyper</a>.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>&lt;SAMPLE_NAME&gt;.tsv</td>
<td>A tab-delimited file with <code>emmtyper</code> result, see <a href="https://github.com/MDU-PHL/emmtyper#result-format">emmtyper - Result format</a> for details</td>
</tr>
</tbody>
</table>
<h4 id="hicap">hicap<a class="headerlink" href="#hicap" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results from <a href="https://github.com/scwatts/hicap">hicap</a>.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>&lt;SAMPLE_NAME&gt;.gbk</td>
<td>GenBank file and cap locus annotations</td>
</tr>
<tr>
<td>&lt;SAMPLE_NAME&gt;.svg</td>
<td>Visualization of annotated cap locus</td>
</tr>
<tr>
<td>&lt;SAMPLE_NAME&gt;.tsv</td>
<td>A tab-delimited file with <code>hicap</code> results</td>
</tr>
</tbody>
</table>
<h4 id="hpsuissero">HpsuisSero<a class="headerlink" href="#hpsuissero" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results from <a href="https://github.com/jimmyliu1326/HpsuisSero">HpsuisSero</a>.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>&lt;SAMPLE_NAME&gt;_serotyping_res.tsv</td>
<td>A tab-delimited file with <code>HpsuisSero</code> result</td>
</tr>
</tbody>
</table>
<h4 id="kleborate">Kleborate<a class="headerlink" href="#kleborate" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results from <a href="https://github.com/katholt/Kleborate">Kleborate</a>.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>&lt;SAMPLE_NAME&gt;.results.tsv</td>
<td>A tab-delimited file with <code>Kleborate</code> result, see <a href="https://github.com/katholt/Kleborate/wiki/Tests-and-example-outputs#example-output">Kleborate - Example output</a> for more details</td>
</tr>
</tbody>
</table>
<h4 id="legsta">legsta<a class="headerlink" href="#legsta" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results from <a href="https://github.com/tseemann/legsta">legsta</a>.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>&lt;SAMPLE_NAME&gt;.tsv</td>
<td>A tab-delimited file with <code>legsta</code> result, see <a href="https://github.com/tseemann/legsta#output">legsta - Output</a> for more details</td>
</tr>
</tbody>
</table>
<h4 id="lissero">LisSero<a class="headerlink" href="#lissero" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results from <a href="https://github.com/MDU-PHL/LisSero">LisSero</a>.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>&lt;SAMPLE_NAME&gt;.tsv</td>
<td>A tab-delimited file with <code>LisSero</code> results</td>
</tr>
</tbody>
</table>
<h4 id="mash">Mash<a class="headerlink" href="#mash" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results from <a href="https://github.com/marbl/Mash">Mash</a>.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>&lt;SAMPLE_NAME&gt;-dist.txt</td>
<td>A tab-delimited file with <code>mash dist</code> results</td>
</tr>
</tbody>
</table>
<h4 id="meningotype">meningotype<a class="headerlink" href="#meningotype" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results from <a href="https://github.com/MDU-PHL/meningotype">meningotype</a> .</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>&lt;SAMPLE_NAME&gt;.tsv</td>
<td>A tab-delimited file with <code>meningotype</code> result</td>
</tr>
</tbody>
</table>
<h4 id="ngmaster">ngmaster<a class="headerlink" href="#ngmaster" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results from <a href="https://github.com/MDU-PHL/ngmaster">ngmaster</a>.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>&lt;SAMPLE_NAME&gt;.tsv</td>
<td>A tab-delimited file with <code>ngmaster</code> results</td>
</tr>
</tbody>
</table>
<h4 id="seqsero2">SeqSero2<a class="headerlink" href="#seqsero2" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results from <a href="https://github.com/denglab/SeqSero2">SeqSero2</a>.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>&lt;SAMPLE_NAME&gt;.tsv</td>
<td>A tab-delimited file with <code>SeqSero2</code> results</td>
</tr>
<tr>
<td>&lt;SAMPLE_NAME&gt;.txt</td>
<td>A text file with key-value pairs of <code>SeqSero2</code> results</td>
</tr>
</tbody>
</table>
<h4 id="sistr">SISTR<a class="headerlink" href="#sistr" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results from <a href="https://github.com/phac-nml/sistr_cmd">SISTR</a>.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>&lt;SAMPLE_NAME&gt;-allele.fasta.gz</td>
<td>A FASTA file of the cgMLST allele search results</td>
</tr>
<tr>
<td>&lt;SAMPLE_NAME&gt;-allele.json.gz</td>
<td>JSON formated cgMLST allele search results, see <a href="https://github.com/phac-nml/sistr_cmd#cgmlst-allele-search-results">SISTR - cgMLST search results</a> for more details</td>
</tr>
<tr>
<td>&lt;SAMPLE_NAME&gt;-cgmlst.csv</td>
<td>A comma-delimited summary of the cgMLST allele search results</td>
</tr>
<tr>
<td>&lt;SAMPLE_NAME&gt;.tsv</td>
<td>A tab-delimited file with <code>SISTR</code> results, see <a href="https://github.com/phac-nml/sistr_cmd#primary-results-output--o-sistr-results">SISTR - Primary results</a> for more details</td>
</tr>
</tbody>
</table>
<h4 id="spatyper">spaTyper<a class="headerlink" href="#spatyper" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results from <a href="https://github.com/HCGB-IGTP/spaTyper">spaTyper</a>.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>&lt;SAMPLE_NAME&gt;.tsv</td>
<td>A tab-delimited file with <code>spaTyper</code> results</td>
</tr>
</tbody>
</table>
<h4 id="ssuissero">SsuisSero<a class="headerlink" href="#ssuissero" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results from <a href="https://github.com/jimmyliu1326/SsuisSero">SsuisSero</a>.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>&lt;SAMPLE_NAME&gt;_serotyping_res.tsv</td>
<td>A tab-delimited file with <code>SsuisSero</code> results</td>
</tr>
</tbody>
</table>
<h4 id="staphopia-sccmec">staphopia-sccmec<a class="headerlink" href="#staphopia-sccmec" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results from <a href="https://github.com/staphopia/staphopia-sccmec">staphopia-sccmec</a>.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>&lt;SAMPLE_NAME&gt;.tsv</td>
<td>A tab-delimited file with <code>staphopia-sccmec</code> results</td>
</tr>
</tbody>
</table>
<h4 id="tbprofiler">TBProfiler<a class="headerlink" href="#tbprofiler" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results from <a href="https://github.com/jodyphelan/TBProfiler">TBProfiler</a>.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>&lt;SAMPLE_NAME&gt;.results.csv</td>
<td>A CSV formated <code>TBProfiler</code> result file of resistance and strain type</td>
</tr>
<tr>
<td>&lt;SAMPLE_NAME&gt;.results.json</td>
<td>A JSON formated <code>TBProfiler</code> result file of resistance and strain type</td>
</tr>
<tr>
<td>&lt;SAMPLE_NAME&gt;.results.txt</td>
<td>A text file with <code>TBProfiler</code> results</td>
</tr>
<tr>
<td>&lt;SAMPLE_NAME&gt;.bam</td>
<td>BAM file with alignment details</td>
</tr>
<tr>
<td>&lt;SAMPLE_NAME&gt;.targets.csq.vcf.gz</td>
<td>VCF with variant info again refernce genomes</td>
</tr>
</tbody>
</table>
<h3 id="audit-trail">Audit Trail<a class="headerlink" href="#audit-trail" title="Permanent link">&para;</a></h3>
<p>Below are files that can assist you in understanding which parameters and program versions were used.</p>
<h4 id="logs">Logs<a class="headerlink" href="#logs" title="Permanent link">&para;</a></h4>
<p>Each process that is executed will have a <code>logs</code> folder containing helpful files for you to review
if the need ever arises.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>nf-&lt;PROCESS_NAME&gt;.begin</td>
<td>An empty file used to designate the process started</td>
</tr>
<tr>
<td>nf-&lt;PROCESS_NAME&gt;.err</td>
<td>Contains STDERR outputs from the process</td>
</tr>
<tr>
<td>nf-&lt;PROCESS_NAME&gt;.log</td>
<td>Contains both STDERR and STDOUT outputs from the process</td>
</tr>
<tr>
<td>nf-&lt;PROCESS_NAME&gt;.out</td>
<td>Contains STDOUT outputs from the process</td>
</tr>
<tr>
<td>nf-&lt;PROCESS_NAME&gt;.run</td>
<td>The script Nextflow uses to stage/unstage files and queue processes based on given profile</td>
</tr>
<tr>
<td>nf-&lt;PROCESS_NAME&gt;.sh</td>
<td>The script executed by bash for the process</td>
</tr>
<tr>
<td>nf-&lt;PROCESS_NAME&gt;.trace</td>
<td>The Nextflow <a href="https://www.nextflow.io/docs/latest/tracing.html#trace-report">Trace</a> report for the process</td>
</tr>
<tr>
<td>versions.yml</td>
<td>A YAML formatted file with program versions</td>
</tr>
</tbody>
</table>
<h4 id="nextflow-reports">Nextflow Reports<a class="headerlink" href="#nextflow-reports" title="Permanent link">&para;</a></h4>
<p>These Nextflow reports provide great a great summary of your run. These can be used to optimize
resource usage and estimate expected costs if using cloud platforms.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>merlin-dag.dot</td>
<td>The Nextflow <a href="https://www.nextflow.io/docs/latest/tracing.html#dag-visualisation">DAG visualisation</a></td>
</tr>
<tr>
<td>merlin-report.html</td>
<td>The Nextflow <a href="https://www.nextflow.io/docs/latest/tracing.html#execution-report">Execution Report</a></td>
</tr>
<tr>
<td>merlin-timeline.html</td>
<td>The Nextflow <a href="https://www.nextflow.io/docs/latest/tracing.html#timeline-report">Timeline Report</a></td>
</tr>
<tr>
<td>merlin-trace.txt</td>
<td>The Nextflow <a href="https://www.nextflow.io/docs/latest/tracing.html#trace-report">Trace</a> report</td>
</tr>
</tbody>
</table>
<h4 id="program-versions">Program Versions<a class="headerlink" href="#program-versions" title="Permanent link">&para;</a></h4>
<p>At the end of each run, each of the <code>versions.yml</code> files are merged into the files below.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>software_versions.yml</td>
<td>A complete list of programs and versions used by each process</td>
</tr>
<tr>
<td>software_versions_mqc.yml</td>
<td>A complete list of programs and versions formatted for <a href="https://multiqc.info/">MultiQC</a></td>
</tr>
</tbody>
</table>
<h2 id="parameters">Parameters<a class="headerlink" href="#parameters" title="Permanent link">&para;</a></h2>
<h3 id="required-parameters">Required Parameters<a class="headerlink" href="#required-parameters" title="Permanent link">&para;</a></h3>
<p>Define where the pipeline should find input data and save output data.</p>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Description</th>
<th>Default</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>--bactopia</code></td>
<td>The path to bactopia results to use as inputs</td>
<td></td>
</tr>
</tbody>
</table>
<h3 id="filtering-parameters">Filtering Parameters<a class="headerlink" href="#filtering-parameters" title="Permanent link">&para;</a></h3>
<p>Use these parameters to specify which samples to include or exclude.</p>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Description</th>
<th>Default</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>--include</code></td>
<td>A text file containing sample names (one per line) to include from the analysis</td>
<td></td>
</tr>
<tr>
<td><code>--exclude</code></td>
<td>A text file containing sample names (one per line) to exclude from the analysis</td>
<td></td>
</tr>
</tbody>
</table>
<h3 id="mashdist-parameters">mashdist Parameters<a class="headerlink" href="#mashdist-parameters" title="Permanent link">&para;</a></h3>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Description</th>
<th>Default</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>--mash_sketch</code></td>
<td>The reference sequence as a Mash Sketch (.msh file)</td>
<td></td>
</tr>
<tr>
<td><code>--mash_seed</code></td>
<td>Seed to provide to the hash function</td>
<td>42</td>
</tr>
<tr>
<td><code>--mash_table</code></td>
<td>Table output (fields will be blank if they do not meet the p-value threshold)</td>
<td>False</td>
</tr>
<tr>
<td><code>--mash_m</code></td>
<td>Minimum copies of each k-mer required to pass noise filter for reads</td>
<td>1</td>
</tr>
<tr>
<td><code>--mash_w</code></td>
<td>Probability threshold for warning about low k-mer size.</td>
<td>0.01</td>
</tr>
<tr>
<td><code>--max_p</code></td>
<td>Maximum p-value to report.</td>
<td>1.0</td>
</tr>
<tr>
<td><code>--max_dist</code></td>
<td>Maximum distance to report.</td>
<td>1.0</td>
</tr>
<tr>
<td><code>--merlin_dist</code></td>
<td>Maximum distance to report when using Merlin .</td>
<td>0.1</td>
</tr>
<tr>
<td><code>--full_merlin</code></td>
<td>Go full Merlin and run all species-specific tools, no matter the Mash distance</td>
<td>False</td>
</tr>
<tr>
<td><code>--use_fastqs</code></td>
<td>Query with FASTQs instead of the assemblies</td>
<td>False</td>
</tr>
</tbody>
</table>
<h3 id="agrvate-parameters">AgrVATE Parameters<a class="headerlink" href="#agrvate-parameters" title="Permanent link">&para;</a></h3>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Description</th>
<th>Default</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>--typing_only</code></td>
<td>agr typing only. Skips agr operon extraction and frameshift detection</td>
<td>False</td>
</tr>
</tbody>
</table>
<h3 id="ectyper-parameters">ECTyper Parameters<a class="headerlink" href="#ectyper-parameters" title="Permanent link">&para;</a></h3>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Description</th>
<th>Default</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>--opid</code></td>
<td>Percent identity required for an O antigen allele match</td>
<td>90</td>
</tr>
<tr>
<td><code>--opcov</code></td>
<td>Minumum percent coverage required for an O antigen allele match</td>
<td>90</td>
</tr>
<tr>
<td><code>--hpid</code></td>
<td>Percent identity required for an H antigen allele match</td>
<td>95</td>
</tr>
<tr>
<td><code>--hpcov</code></td>
<td>Minumum percent coverage required for an H antigen allele match</td>
<td>50</td>
</tr>
<tr>
<td><code>--verify</code></td>
<td>Enable E. coli species verification</td>
<td>False</td>
</tr>
<tr>
<td><code>--print_alleles</code></td>
<td>Prints the allele sequences if enabled as the final column</td>
<td>False</td>
</tr>
</tbody>
</table>
<h3 id="emmtyper-parameters">emmtyper Parameters<a class="headerlink" href="#emmtyper-parameters" title="Permanent link">&para;</a></h3>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Description</th>
<th>Default</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>--emmtyper_wf</code></td>
<td>Workflow for emmtyper to use.</td>
<td>blast</td>
</tr>
<tr>
<td><code>--cluster_distance</code></td>
<td>Distance between cluster of matches to consider as different clusters</td>
<td>500</td>
</tr>
<tr>
<td><code>--percid</code></td>
<td>Minimal percent identity of sequence</td>
<td>95</td>
</tr>
<tr>
<td><code>--culling_limit</code></td>
<td>Total hits to return in a position</td>
<td>5</td>
</tr>
<tr>
<td><code>--mismatch</code></td>
<td>Threshold for number of mismatch to allow in BLAST hit</td>
<td>5</td>
</tr>
<tr>
<td><code>--align_diff</code></td>
<td>Threshold for difference between alignment length and subject length in BLAST</td>
<td>5</td>
</tr>
<tr>
<td><code>--gap</code></td>
<td>Threshold gap to allow in BLAST hit</td>
<td>2</td>
</tr>
<tr>
<td><code>--min_perfect</code></td>
<td>Minimum size of perfect match at 3 primer end</td>
<td>15</td>
</tr>
<tr>
<td><code>--min_good</code></td>
<td>Minimum size where there must be 2 matches for each mismatch</td>
<td>15</td>
</tr>
<tr>
<td><code>--max_size</code></td>
<td>Maximum size of PCR product</td>
<td>2000</td>
</tr>
</tbody>
</table>
<h3 id="hicap-parameters">hicap Parameters<a class="headerlink" href="#hicap-parameters" title="Permanent link">&para;</a></h3>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Description</th>
<th>Default</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>--database_dir</code></td>
<td>Directory containing locus database</td>
<td></td>
</tr>
<tr>
<td><code>--model_fp</code></td>
<td>Path to prodigal model</td>
<td></td>
</tr>
<tr>
<td><code>--full_sequence</code></td>
<td>Write the full input sequence out to the genbank file rather than just the region surrounding and including the locus</td>
<td>False</td>
</tr>
<tr>
<td><code>--hicap_debug</code></td>
<td>hicap will print debug messages</td>
<td>False</td>
</tr>
<tr>
<td><code>--gene_coverage</code></td>
<td>Minimum percentage coverage to consider a single gene complete</td>
<td>0.8</td>
</tr>
<tr>
<td><code>--gene_identity</code></td>
<td>Minimum percentage identity to consider a single gene complete</td>
<td>0.7</td>
</tr>
<tr>
<td><code>--broken_gene_length</code></td>
<td>Minimum length to consider a broken gene</td>
<td>60</td>
</tr>
<tr>
<td><code>--broken_gene_identity</code></td>
<td>Minimum percentage identity to consider a broken gene</td>
<td>0.8</td>
</tr>
</tbody>
</table>
<h3 id="hpsuissero-parameters">HpsuisSero Parameters<a class="headerlink" href="#hpsuissero-parameters" title="Permanent link">&para;</a></h3>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Description</th>
<th>Default</th>
</tr>
</thead>
<tbody>
<tr>
<td></td>
<td></td>
<td></td>
</tr>
</tbody>
</table>
<h3 id="kleborate-parameters">Kleborate Parameters<a class="headerlink" href="#kleborate-parameters" title="Permanent link">&para;</a></h3>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Description</th>
<th>Default</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>--skip_resistance</code></td>
<td>Turn off resistance genes screening</td>
<td>False</td>
</tr>
<tr>
<td><code>--skip_kaptive</code></td>
<td>Turn off Kaptive screening of K and O loci</td>
<td>False</td>
</tr>
<tr>
<td><code>--min_identity</code></td>
<td>Minimum alignment percent identity for main results</td>
<td>90.0</td>
</tr>
<tr>
<td><code>--kleborate_min_coverage</code></td>
<td>Minimum alignment percent coverage for main results</td>
<td>80.0</td>
</tr>
<tr>
<td><code>--min_spurious_identity</code></td>
<td>Minimum alignment percent identity for spurious results</td>
<td>80.0</td>
</tr>
<tr>
<td><code>--min_spurious_coverage</code></td>
<td>Minimum alignment percent coverage for spurious results</td>
<td>40.0</td>
</tr>
<tr>
<td><code>--min_kaptive_confidence</code></td>
<td>Minimum Kaptive confidence to call K/O loci - confidence levels below this will be reported as unknown</td>
<td>Good</td>
</tr>
<tr>
<td><code>--force_index</code></td>
<td>Rebuild the BLAST index at the start of execution</td>
<td>False</td>
</tr>
</tbody>
</table>
<h3 id="legsta-parameters">legsta Parameters<a class="headerlink" href="#legsta-parameters" title="Permanent link">&para;</a></h3>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Description</th>
<th>Default</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>--noheader</code></td>
<td>Don't print header row</td>
<td>False</td>
</tr>
</tbody>
</table>
<h3 id="lissero-parameters">LisSero Parameters<a class="headerlink" href="#lissero-parameters" title="Permanent link">&para;</a></h3>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Description</th>
<th>Default</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>--min_id</code></td>
<td>Minimum percent identity to accept a match</td>
<td>95.0</td>
</tr>
<tr>
<td><code>--min_cov</code></td>
<td>Minimum coverage of the gene to accept a match</td>
<td>95.0</td>
</tr>
</tbody>
</table>
<h3 id="meningotype-parameters">meningotype Parameters<a class="headerlink" href="#meningotype-parameters" title="Permanent link">&para;</a></h3>
<p>You can use these parameters to fine-tune your meningotype analysis</p>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Description</th>
<th>Default</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>--finetype</code></td>
<td>perform porA and fetA fine typing</td>
<td>False</td>
</tr>
<tr>
<td><code>--porB</code></td>
<td>perform porB sequence typing (NEIS2020)</td>
<td>False</td>
</tr>
<tr>
<td><code>--bast</code></td>
<td>perform Bexsero antigen sequence typing (BAST)</td>
<td>False</td>
</tr>
<tr>
<td><code>--mlst</code></td>
<td>perform MLST</td>
<td>False</td>
</tr>
<tr>
<td><code>--all</code></td>
<td>perform MLST, porA, fetA, porB, BAST typing</td>
<td>False</td>
</tr>
</tbody>
</table>
<h3 id="ngmaster-parameters">ngmaster Parameters<a class="headerlink" href="#ngmaster-parameters" title="Permanent link">&para;</a></h3>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Description</th>
<th>Default</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>--csv</code></td>
<td>output comma-separated format (CSV) rather than tab-separated</td>
<td>False</td>
</tr>
</tbody>
</table>
<h3 id="seqsero2-parameters">SeqSero2 Parameters<a class="headerlink" href="#seqsero2-parameters" title="Permanent link">&para;</a></h3>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Description</th>
<th>Default</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>--run_mode</code></td>
<td>Workflow to run. 'a' allele mode, or 'k' k-mer mode</td>
<td>k</td>
</tr>
<tr>
<td><code>--input_type</code></td>
<td>Input format to analyze. 'assembly' or 'fastq'</td>
<td>assembly</td>
</tr>
<tr>
<td><code>--bwa_mode</code></td>
<td>Algorithms for bwa mapping for allele mode</td>
<td>mem</td>
</tr>
</tbody>
</table>
<h3 id="sistr-parameters">SISTR Parameters<a class="headerlink" href="#sistr-parameters" title="Permanent link">&para;</a></h3>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Description</th>
<th>Default</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>--full_cgmlst</code></td>
<td>Use the full set of cgMLST alleles which can include highly similar alleles</td>
<td>False</td>
</tr>
</tbody>
</table>
<h3 id="spatyper-parameters">spaTyper Parameters<a class="headerlink" href="#spatyper-parameters" title="Permanent link">&para;</a></h3>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Description</th>
<th>Default</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>--repeats</code></td>
<td>List of spa repeats</td>
<td></td>
</tr>
<tr>
<td><code>--repeat_order</code></td>
<td>List spa types and order of repeats</td>
<td></td>
</tr>
<tr>
<td><code>--do_enrich</code></td>
<td>Do PCR product enrichment</td>
<td>False</td>
</tr>
</tbody>
</table>
<h3 id="staphopia-sccmec-parameters">staphopia-sccmec Parameters<a class="headerlink" href="#staphopia-sccmec-parameters" title="Permanent link">&para;</a></h3>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Description</th>
<th>Default</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>--hamming</code></td>
<td>Report the results as hamming distances</td>
<td>False</td>
</tr>
</tbody>
</table>
<h3 id="ssuissero-parameters">SsuisSero Parameters<a class="headerlink" href="#ssuissero-parameters" title="Permanent link">&para;</a></h3>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Description</th>
<th>Default</th>
</tr>
</thead>
<tbody>
<tr>
<td></td>
<td></td>
<td></td>
</tr>
</tbody>
</table>
<h3 id="tbprofiler-parameters">TBProfiler Parameters<a class="headerlink" href="#tbprofiler-parameters" title="Permanent link">&para;</a></h3>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Description</th>
<th>Default</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>--call_whole_genome</code></td>
<td>Call whole genome</td>
<td>False</td>
</tr>
<tr>
<td><code>--mapper</code></td>
<td>Mapping tool to use. If you are using nanopore data it will default to minimap2</td>
<td>bwa</td>
</tr>
<tr>
<td><code>--caller</code></td>
<td>Variant calling tool to use</td>
<td>freebayes</td>
</tr>
<tr>
<td><code>--calling_params</code></td>
<td>Extra variant caller options in quotes</td>
<td></td>
</tr>
<tr>
<td><code>--tb_min_depth</code></td>
<td>Minimum depth required to call variant</td>
<td>10</td>
</tr>
<tr>
<td><code>--tb_af</code></td>
<td>Minimum allele frequency to call variants</td>
<td>0.1</td>
</tr>
<tr>
<td><code>--tb_reporting_af</code></td>
<td>Minimum allele frequency to use variants for prediction</td>
<td>0.1</td>
</tr>
<tr>
<td><code>--coverage_fraction_threshold</code></td>
<td>Cutoff used to calculate fraction of region covered by &lt;= this value</td>
<td>0</td>
</tr>
<tr>
<td><code>--suspect</code></td>
<td>Use the suspect suite of tools to add ML predictions</td>
<td>False</td>
</tr>
<tr>
<td><code>--no_flagstat</code></td>
<td>Don't collect flagstats</td>
<td>False</td>
</tr>
<tr>
<td><code>--no_delly</code></td>
<td>Don't run delly</td>
<td>False</td>
</tr>
</tbody>
</table>
<h3 id="optional-parameters">Optional Parameters<a class="headerlink" href="#optional-parameters" title="Permanent link">&para;</a></h3>
<p>These optional parameters can be useful in certain settings.</p>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Description</th>
<th>Default</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>--outdir</code></td>
<td>Base directory to write results to</td>
<td>./</td>
</tr>
<tr>
<td><code>--run_name</code></td>
<td>Name of the directory to hold results</td>
<td>bactopia</td>
</tr>
<tr>
<td><code>--skip_compression</code></td>
<td>Ouput files will not be compressed</td>
<td>False</td>
</tr>
<tr>
<td><code>--keep_all_files</code></td>
<td>Keeps all analysis files created</td>
<td>False</td>
</tr>
</tbody>
</table>
<h3 id="max-job-request-parameters">Max Job Request Parameters<a class="headerlink" href="#max-job-request-parameters" title="Permanent link">&para;</a></h3>
<p>Set the top limit for requested resources for any single job.</p>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Description</th>
<th>Default</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>--max_retry</code></td>
<td>Maximum times to retry a process before allowing it to fail.</td>
<td>3</td>
</tr>
<tr>
<td><code>--max_cpus</code></td>
<td>Maximum number of CPUs that can be requested for any single job.</td>
<td>4</td>
</tr>
<tr>
<td><code>--max_memory</code></td>
<td>Maximum amount of memory (in GB) that can be requested for any single job.</td>
<td>32</td>
</tr>
<tr>
<td><code>--max_time</code></td>
<td>Maximum amount of time (in minutes) that can be requested for any single job.</td>
<td>120</td>
</tr>
<tr>
<td><code>--max_downloads</code></td>
<td>Maximum number of samples to download at a time</td>
<td>3</td>
</tr>
</tbody>
</table>
<h3 id="nextflow-configuration-parameters">Nextflow Configuration Parameters<a class="headerlink" href="#nextflow-configuration-parameters" title="Permanent link">&para;</a></h3>
<p>Parameters to fine-tune your Nextflow setup.</p>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Description</th>
<th>Default</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>--nfconfig</code></td>
<td>A Nextflow compatible config file for custom profiles, loaded last and will overwrite existing variables if set.</td>
<td></td>
</tr>
<tr>
<td><code>--publish_dir_mode</code></td>
<td>Method used to save pipeline results to output directory.</td>
<td>copy</td>
</tr>
<tr>
<td><code>--infodir</code></td>
<td>Directory to keep pipeline Nextflow logs and reports.</td>
<td>${params.outdir}/pipeline_info</td>
</tr>
<tr>
<td><code>--force</code></td>
<td>Nextflow will overwrite existing output files.</td>
<td>False</td>
</tr>
<tr>
<td><code>--cleanup_workdir</code></td>
<td>After Bactopia is successfully executed, the <code>work</code> directory will be deleted.</td>
<td>False</td>
</tr>
</tbody>
</table>
<h3 id="nextflow-profile-parameters">Nextflow Profile Parameters<a class="headerlink" href="#nextflow-profile-parameters" title="Permanent link">&para;</a></h3>
<p>Parameters to fine-tune your Nextflow setup.</p>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Description</th>
<th>Default</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>--condadir</code></td>
<td>Directory to Nextflow should use for Conda environments</td>
<td></td>
</tr>
<tr>
<td><code>--registry</code></td>
<td>Docker registry to pull containers from.</td>
<td>dockerhub</td>
</tr>
<tr>
<td><code>--singularity_cache</code></td>
<td>Directory where remote Singularity images are stored.</td>
<td></td>
</tr>
<tr>
<td><code>--singularity_pull_docker_container</code></td>
<td>Instead of directly downloading Singularity images for use with Singularity, force the workflow to pull and convert Docker containers instead.</td>
<td></td>
</tr>
<tr>
<td><code>--force_rebuild</code></td>
<td>Force overwrite of existing pre-built environments.</td>
<td>False</td>
</tr>
<tr>
<td><code>--queue</code></td>
<td>Comma-separated name of the queue(s) to be used by a job scheduler (e.g. AWS Batch or SLURM)</td>
<td>general,high-memory</td>
</tr>
<tr>
<td><code>--cluster_opts</code></td>
<td>Additional options to pass to the executor. (e.g. SLURM: '--account=my_acct_name'</td>
<td></td>
</tr>
<tr>
<td><code>--disable_scratch</code></td>
<td>All intermediate files created on worker nodes of will be transferred to the head node.</td>
<td>False</td>
</tr>
</tbody>
</table>
<h3 id="helpful-parameters">Helpful Parameters<a class="headerlink" href="#helpful-parameters" title="Permanent link">&para;</a></h3>
<p>Uncommonly used parameters that might be useful.</p>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Description</th>
<th>Default</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>--monochrome_logs</code></td>
<td>Do not use coloured log outputs.</td>
<td></td>
</tr>
<tr>
<td><code>--nfdir</code></td>
<td>Print directory Nextflow has pulled Bactopia to</td>
<td></td>
</tr>
<tr>
<td><code>--sleep_time</code></td>
<td>The amount of time (seconds) Nextflow will wait after setting up datasets before execution.</td>
<td>5</td>
</tr>
<tr>
<td><code>--validate_params</code></td>
<td>Boolean whether to validate parameters against the schema at runtime</td>
<td>True</td>
</tr>
<tr>
<td><code>--help</code></td>
<td>Display help text.</td>
<td></td>
</tr>
<tr>
<td><code>--wf</code></td>
<td>Specify which workflow or Bactopia Tool to execute</td>
<td>bactopia</td>
</tr>
<tr>
<td><code>--list_wfs</code></td>
<td>List the available workflows and Bactopia Tools to use with '--wf'</td>
<td></td>
</tr>
<tr>
<td><code>--show_hidden_params</code></td>
<td>Show all params when using <code>--help</code></td>
<td></td>
</tr>
<tr>
<td><code>--help_all</code></td>
<td>An alias for --help --show_hidden_params</td>
<td></td>
</tr>
<tr>
<td><code>--version</code></td>
<td>Display version text.</td>
<td></td>
</tr>
</tbody>
</table>
<h2 id="citations">Citations<a class="headerlink" href="#citations" title="Permanent link">&para;</a></h2>
<p>If you use Bactopia and <code>merlin</code> in your analysis, please cite the following.</p>
<ul>
<li>
<p><a href="https://bactopia.github.io/">Bactopia</a><br />
    Petit III RA, Read TD <a href="https://doi.org/10.1128/mSystems.00190-20">Bactopia - a flexible pipeline for complete analysis of bacterial genomes.</a> <em>mSystems</em> 5 (2020)</p>
</li>
<li>
<p><a href="https://github.com/VishnuRaghuram94/AgrVATE">AgrVATE</a><br />
    Raghuram V. <a href="https://github.com/VishnuRaghuram94/AgrVATE">AgrVATE: Rapid identification of Staphylococcus aureus agr locus type and agr operon variants.</a> (GitHub)</p>
</li>
<li>
<p><a href="https://github.com/phac-nml/ecoli_serotyping">ECTyper</a><br />
    Laing C, Bessonov K, Sung S, La Rose C <a href="https://github.com/phac-nml/ecoli_serotyping">ECTyper - In silico prediction of <em>Escherichia coli</em> serotype</a> (GitHub)  </p>
</li>
<li>
<p><a href="https://github.com/MDU-PHL/emmtyper">emmtyper</a><br />
    Tan A, Seemann T, Lacey D, Davies M, Mcintyre L, Frost H, Williamson D, Gonalves da Silva A <a href="https://github.com/MDU-PHL/emmtyper">emmtyper - emm Automatic Isolate Labeller</a> (GitHub)</p>
</li>
<li>
<p><a href="https://github.com/scwatts/hicap">hicap</a><br />
    Watts SC, Holt KE <a href="https://doi.org/10.1128/JCM.00190-19">hicap: in silico serotyping of the Haemophilus influenzae capsule locus.</a> <em>Journal of Clinical Microbiology</em> JCM.00190-19 (2019)</p>
</li>
<li>
<p><a href="https://github.com/jimmyliu1326/HpsuisSero">HpsuisSero</a><br />
    Lui J <a href="https://github.com/jimmyliu1326/HpsuisSero">HpsuisSero: Rapid <em>Haemophilus parasuis</em> serotyping</a> (GitHub)</p>
</li>
<li>
<p><a href="https://github.com/katholt/Kleborate">Kleborate</a><br />
    Lam MMC, Wick RR, Watts, SC, Cerdeira LT, Wyres KL, Holt KE <a href="https://doi.org/10.1038/s41467-021-24448-3">A genomic surveillance framework and genotyping tool for Klebsiella pneumoniae and its related species complex.</a> <em>Nat Commun</em> 12, 4188 (2021)</p>
</li>
<li>
<p><a href="https://github.com/tseemann/legsta">legsta</a><br />
    Seemann T <a href="https://github.com/tseemann/legsta">legsta: In silico Legionella pneumophila Sequence Based Typing</a> (GitHub)</p>
</li>
<li>
<p><a href="https://github.com/MDU-PHL/LisSero">LisSero</a><br />
    Kwong J, Zhang J, Seeman T, Horan, K, Gonalves da Silva A <a href="https://github.com/MDU-PHL/LisSero">LisSero - <em>In silico</em> serotype prediction for <em>Listeria monocytogenes</em></a> (GitHub)</p>
</li>
<li>
<p><a href="https://github.com/marbl/Mash">Mash</a><br />
    Ondov BD, Treangen TJ, Melsted P, Mallonee AB, Bergman NH, Koren S, Phillippy AM <a href="http://dx.doi.org/10.1186/s13059-016-0997-x">Mash: fast genome and metagenome distance estimation using MinHash</a>. <em>Genome Biol</em> 17, 132 (2016)</p>
</li>
<li>
<p><a href="https://github.com/MDU-PHL/meningotype">meningotype</a><br />
    Kwong JC, Gonalves da Silva A, Stinear TP, Howden BP, &amp; Seemann T <a href="https://github.com/MDU-PHL/meningotype">meningotype: in silico typing for <em>Neisseria meningitidis</em>.</a> (GitHub)</p>
</li>
<li>
<p><a href="https://github.com/MDU-PHL/ngmaster">ngmaster</a><br />
    Kwong J, Gonalves da Silva A, Schultz M, Seeman T <a href="https://github.com/MDU-PHL/ngmaster">ngmaster - <em>In silico</em> multi-antigen sequence typing for <em>Neisseria gonorrhoeae</em> (NG-MAST)</a> (GitHub)</p>
</li>
<li>
<p><a href="https://github.com/denglab/SeqSero2">SeqSero2</a><br />
    Zhang S, Den-Bakker HC, Li S, Dinsmore BA, Lane C, Lauer AC, Fields PI, Deng X. <a href="https://doi.org/10.1128/AEM.01746-19">SeqSero2: rapid and improved Salmonella serotype determination using whole genome sequencing data.</a> <em>Appl Environ Microbiology</em> 85(23):e01746-19 (2019)</p>
</li>
<li>
<p><a href="https://github.com/phac-nml/sistr_cmd">SISTR</a><br />
    Yoshida CE, Kruczkiewicz P, Laing CR, Lingohr EJ, Gannon VPJ, Nash JHE, Taboada EN <a href="https://doi.org/10.1371/journal.pone.0147101">The Salmonella In Silico Typing Resource (SISTR): An Open Web-Accessible Tool for Rapidly Typing and Subtyping Draft Salmonella Genome Assemblies.</a> <em>PloS One</em>, 11(1), e0147101. (2016)</p>
</li>
<li>
<p><a href="https://github.com/HCGB-IGTP/spaTyper">spaTyper</a><br />
    Sanchez-Herrero JF, and Sullivan M <a href="http://doi.org/10.5281/zenodo.4063625">spaTyper: Staphylococcal protein A (spa) characterization pipeline</a>. Zenodo. (2020)</p>
</li>
<li>
<p><a href="https://github.com/jimmyliu1326/SsuisSero">SsuisSero</a><br />
    Lui J <a href="https://github.com/jimmyliu1326/SsuisSero">SsuisSero: Rapid <em>Streptococcus suis</em> serotyping</a> (GitHub)</p>
</li>
<li>
<p><a href="https://github.com/staphopia/staphopia-sccmec">staphopia-sccmec</a><br />
    Petit III RA, Read TD <a href="http://dx.doi.org/10.7717/peerj.5261"><em>Staphylococcus aureus</em> viewed from the perspective of 40,000+ genomes.</a> <em>PeerJ</em> 6, e5261 (2018)</p>
</li>
<li>
<p><a href="https://github.com/jodyphelan/TBProfiler">TBProfiler</a><br />
    Phelan JE, OSullivan DM, Machado D, Ramos J, Oppong YEA, Campino S, OGrady J, McNerney R, Hibberd ML, Viveiros M, Huggett JF, Clark TG <a href="https://doi.org/10.1186/s13073-019-0650-x">Integrating informatics tools and portable sequencing technology for rapid detection of resistance to anti-tuberculous drugs.</a> <em>Genome Med</em> 11, 41 (2019)</p>
</li>
</ul>

<!-- Last update of source file -->

  


              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
    
    <nav class="md-footer__inner md-grid" aria-label="Footer" >
      
        
        <a href="../gtdb/" class="md-footer__link md-footer__link--prev" aria-label="Previous: gtdb" rel="prev">
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
          </div>
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Previous
              </span>
              gtdb
            </div>
          </div>
        </a>
      
      
        
        <a href="../pangenome/" class="md-footer__link md-footer__link--next" aria-label="Next: pangenome" rel="next">
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Next
              </span>
              pangenome
            </div>
          </div>
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4Z"/></svg>
          </div>
        </a>
      
    </nav>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright  2019-2021 Robert A. Petit III. All rights reserved.
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs Insiders
    </a>
  
</div>
      
        <div class="md-social">
  
    
    
      
      
    
    <a href="https://github.com/bactopia/bactopia" target="_blank" rel="noopener" title="github.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.2.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
    </a>
  
    
    
      
      
    
    <a href="https://gitter.im/bactopia/Lobby" target="_blank" rel="noopener" title="gitter.im" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 384 512"><!--! Font Awesome Free 6.2.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M66.4 322.5H16V0h50.4v322.5zM166.9 76.1h-50.4V512h50.4V76.1zm100.6 0h-50.4V512h50.4V76.1zM368 76h-50.4v247H368V76z"/></svg>
    </a>
  
    
    
      
      
    
    <a href="https://hub.docker.com/r/bactopia/bactopia" target="_blank" rel="noopener" title="hub.docker.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 640 512"><!--! Font Awesome Free 6.2.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M349.9 236.3h-66.1v-59.4h66.1v59.4zm0-204.3h-66.1v60.7h66.1V32zm78.2 144.8H362v59.4h66.1v-59.4zm-156.3-72.1h-66.1v60.1h66.1v-60.1zm78.1 0h-66.1v60.1h66.1v-60.1zm276.8 100c-14.4-9.7-47.6-13.2-73.1-8.4-3.3-24-16.7-44.9-41.1-63.7l-14-9.3-9.3 14c-18.4 27.8-23.4 73.6-3.7 103.8-8.7 4.7-25.8 11.1-48.4 10.7H2.4c-8.7 50.8 5.8 116.8 44 162.1 37.1 43.9 92.7 66.2 165.4 66.2 157.4 0 273.9-72.5 328.4-204.2 21.4.4 67.6.1 91.3-45.2 1.5-2.5 6.6-13.2 8.5-17.1l-13.3-8.9zm-511.1-27.9h-66v59.4h66.1v-59.4zm78.1 0h-66.1v59.4h66.1v-59.4zm78.1 0h-66.1v59.4h66.1v-59.4zm-78.1-72.1h-66.1v60.1h66.1v-60.1z"/></svg>
    </a>
  
    
    
      
      
    
    <a href="https://twitter.com/rpetit3" target="_blank" rel="noopener" title="twitter.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.2.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"/></svg>
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    <script id="__config" type="application/json">{"base": "../..", "features": ["navigation.tabs", "navigation.tabs.sticky"], "search": "../../assets/javascripts/workers/search.720157f5.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.config.lang": "en", "search.config.pipeline": "stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version.title": "Select version"}}</script>
    
    
      <script src="../../assets/javascripts/bundle.0d3e07ec.min.js"></script>
      
        <script src="../../custom/jquery.min.js"></script>
      
        <script src="../../custom/jquery.magnific-popup.min.js"></script>
      
        <script src="../../custom/extra.js"></script>
      
    
  </body>
</html>