



<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
        <meta name="description" content="An extensive workflow to process Illumina sequencing for bacterial genomes.">
      
      
        <link rel="canonical" href="https://bactopia.github.io/acknowledgements/">
      
      
        <meta name="author" content="Robert A. Petit III">
      
      
        <meta name="lang:clipboard.copy" content="Copy to clipboard">
      
        <meta name="lang:clipboard.copied" content="Copied to clipboard">
      
        <meta name="lang:search.language" content="en">
      
        <meta name="lang:search.pipeline.stopwords" content="True">
      
        <meta name="lang:search.pipeline.trimmer" content="True">
      
        <meta name="lang:search.result.none" content="No matching documents">
      
        <meta name="lang:search.result.one" content="1 matching document">
      
        <meta name="lang:search.result.other" content="# matching documents">
      
        <meta name="lang:search.tokenizer" content="[\s\-]+">
      
      <link rel="shortcut icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.0.4, mkdocs-material-4.4.1">
    
    
      
        <title>Acknowledgements - Bactopia</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/application.30686662.css">
      
      
    
    
      <script src="../assets/javascripts/modernizr.74668098.js"></script>
    
    
      
        <link href="https://fonts.gstatic.com" rel="preconnect" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Ubuntu:300,400,400i,700|Ubuntu+Mono&display=fallback">
        <style>body,input{font-family:"Ubuntu","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Ubuntu Mono","Courier New",Courier,monospace}</style>
      
    
    <link rel="stylesheet" href="../assets/fonts/material-icons.css">
    
    
    
      
        
<script>
  window.ga = window.ga || function() {
    (ga.q = ga.q || []).push(arguments)
  }
  ga.l = +new Date
  /* Setup integration and send page view */
  ga("create", "UA-104199698-2", "bactopia.github.io")
  ga("set", "anonymizeIp", true)
  ga("send", "pageview")
  /* Register handler to log search on blur */
  document.addEventListener("DOMContentLoaded", () => {
    if (document.forms.search) {
      var query = document.forms.search.query
      query.addEventListener("blur", function() {
        if (this.value) {
          var path = document.location.pathname;
          ga("send", "pageview", path + "?q=" + this.value)
        }
      })
    }
  })
</script>
<script async src="https://www.google-analytics.com/analytics.js"></script>
      
    
    
  </head>
  
    <body dir="ltr">
  
    <svg class="md-svg">
      <defs>
        
        
          <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448" viewBox="0 0 416 448" id="__github"><path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19T128 352t-18.125-8.5-10.75-19T96 304t3.125-20.5 10.75-19T128 256t18.125 8.5 10.75 19T160 304zm160 0q0 10-3.125 20.5t-10.75 19T288 352t-18.125-8.5-10.75-19T256 304t3.125-20.5 10.75-19T288 256t18.125 8.5 10.75 19T320 304zm40 0q0-30-17.25-51T296 232q-10.25 0-48.75 5.25Q229.5 240 208 240t-39.25-2.75Q130.75 232 120 232q-29.5 0-46.75 21T56 304q0 22 8 38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0 37.25-1.75t35-7.375 30.5-15 20.25-25.75T360 304zm56-44q0 51.75-15.25 82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5T212 416q-19.5 0-35.5-.75t-36.875-3.125-38.125-7.5-34.25-12.875T37 371.5t-21.5-28.75Q0 312 0 260q0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25 30.875Q171.5 96 212 96q37 0 70 8 26.25-20.5 46.75-30.25T376 64q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34 99.5z"/></svg>
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" data-md-component="overlay" for="__drawer"></label>
    
      <a href="#acknowledgements" tabindex="1" class="md-skip">
        Skip to content
      </a>
    
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href="https://bactopia.github.io/" title="Bactopia" class="md-header-nav__button md-logo">
          
            <i class="md-icon">Ó†å</i>
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="__drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <div class="md-flex__ellipsis md-header-nav__title" data-md-component="title">
          
            <span class="md-header-nav__topic">
              Bactopia
            </span>
            <span class="md-header-nav__topic">
              
                Acknowledgements
              
            </span>
          
        </div>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <label class="md-icon md-icon--search md-header-nav__button" for="__search"></label>
          
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query" data-md-state="active">
      <label class="md-icon md-search__icon" for="__search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset" tabindex="-1">
        &#xE5CD;
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
        
      </div>
      
        <div class="md-flex__cell md-flex__cell--shrink">
          <div class="md-header-nav__source">
            


  

<a href="https://github.com/bactopia/bactopia/" title="Go to repository" class="md-source" data-md-source="github">
  
    <div class="md-source__icon">
      <svg viewBox="0 0 24 24" width="24" height="24">
        <use xlink:href="#__github" width="24" height="24"></use>
      </svg>
    </div>
  
  <div class="md-source__repository">
    GitHub
  </div>
</a>
          </div>
        </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
        
      
      
        

  

<nav class="md-tabs md-tabs--active" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  
  
    <li class="md-tabs__item">
      
        <a href=".." title="Bactopia" class="md-tabs__link md-tabs__link--active">
          Bactopia
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../bactopia-tools/" title="Bactopia Tools" class="md-tabs__link">
          Bactopia Tools
        </a>
      
    </li>
  

      
    </ul>
  </div>
</nav>
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="__drawer">
    <a href="https://bactopia.github.io/" title="Bactopia" class="md-nav__button md-logo">
      
        <i class="md-icon">Ó†å</i>
      
    </a>
    Bactopia
  </label>
  
    <div class="md-nav__source">
      


  

<a href="https://github.com/bactopia/bactopia/" title="Go to repository" class="md-source" data-md-source="github">
  
    <div class="md-source__icon">
      <svg viewBox="0 0 24 24" width="24" height="24">
        <use xlink:href="#__github" width="24" height="24"></use>
      </svg>
    </div>
  
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-1" type="checkbox" id="nav-1" checked>
    
    <label class="md-nav__link" for="nav-1">
      Bactopia
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-1">
        Bactopia
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href=".." title="Introduction" class="md-nav__link">
      Introduction
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../quick-start/" title="Quick Start" class="md-nav__link">
      Quick Start
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../installation/" title="Installation" class="md-nav__link">
      Installation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../tutorial/" title="Tutorial" class="md-nav__link">
      Tutorial
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../datasets/" title="Build Datasets" class="md-nav__link">
      Build Datasets
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../usage-basic/" title="Basic Usage" class="md-nav__link">
      Basic Usage
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../workflow-overview/" title="Workflow Overview" class="md-nav__link">
      Workflow Overview
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../output-overview/" title="Output Overview" class="md-nav__link">
      Output Overview
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../usage-complete/" title="Complete Usage" class="md-nav__link">
      Complete Usage
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../troubleshooting/" title="Troubleshooting" class="md-nav__link">
      Troubleshooting
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="__toc">
    
      
    
    
      <label class="md-nav__link md-nav__link--active" for="__toc">
        Acknowledgements
      </label>
    
    <a href="./" title="Acknowledgements" class="md-nav__link md-nav__link--active">
      Acknowledgements
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#public-datasets" title="Public Datasets" class="md-nav__link">
    Public Datasets
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#ariba-reference-datasets" title="Ariba Reference Datasets" class="md-nav__link">
    Ariba Reference Datasets
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#minmer-datasets" title="Minmer Datasets" class="md-nav__link">
    Minmer Datasets
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#everything-else" title="Everything Else" class="md-nav__link">
    Everything Else
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#software-included-in-bactopia" title="Software Included In Bactopia" class="md-nav__link">
    Software Included In Bactopia
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
    
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-2" type="checkbox" id="nav-2">
    
    <label class="md-nav__link" for="nav-2">
      Bactopia Tools
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-2">
        Bactopia Tools
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../bactopia-tools/" title="Introduction" class="md-nav__link">
      Introduction
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../bactopia-tools/eggnog/" title="eggnog" class="md-nav__link">
      eggnog
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../bactopia-tools/fastani/" title="fastani" class="md-nav__link">
      fastani
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../bactopia-tools/gtdb/" title="gtdb" class="md-nav__link">
      gtdb
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../bactopia-tools/ismapper/" title="ismapper" class="md-nav__link">
      ismapper
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../bactopia-tools/mashtree/" title="mashtree" class="md-nav__link">
      mashtree
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../bactopia-tools/phyloflash/" title="phyloflash" class="md-nav__link">
      phyloflash
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../bactopia-tools/pirate/" title="pirate" class="md-nav__link">
      pirate
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../bactopia-tools/roary/" title="roary" class="md-nav__link">
      roary
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../bactopia-tools/summary/" title="summary" class="md-nav__link">
      summary
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#public-datasets" title="Public Datasets" class="md-nav__link">
    Public Datasets
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#ariba-reference-datasets" title="Ariba Reference Datasets" class="md-nav__link">
    Ariba Reference Datasets
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#minmer-datasets" title="Minmer Datasets" class="md-nav__link">
    Minmer Datasets
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#everything-else" title="Everything Else" class="md-nav__link">
    Everything Else
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#software-included-in-bactopia" title="Software Included In Bactopia" class="md-nav__link">
    Software Included In Bactopia
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/bactopia/bactopia/edit/master/docs/acknowledgements.md" title="Edit this page" class="md-icon md-content__icon">&#xE3C9;</a>
                
                
                <h1 id="acknowledgements">Acknowledgements<a class="headerlink" href="#acknowledgements" title="Permanent link">&para;</a></h1>
<p>Bactopia is truly a case of <em>"standing upon the shoulders of giants"</em>. Nearly 
every component of Bactopia was created by others and made freely available to the public.</p>
<p>I would like to personally extend my many thanks and gratitude to the authors
of these software packages and public datasets. If you've made it this far, I 
owe you a beer üçª (or coffee ‚òï!) if we ever encounter one another in person. 
Really, thank you very much!</p>
<div class="admonition info">
<p class="admonition-title">Please Cite Datasets and Tools</p>
<p>If you have used Bactopia in your work, please be sure to cite any datasets or software you may have used. A citation link for each dataset/software has been made available.</p>
<p>A BibTeX file of each citation is also available at <a href="https://github.com/bactopia/bactopia/blob/master/docs/data/bactopia-datasets-software.bib">Bactopia Datasets and Software BibTeX</a></p>
</div>
<h2 id="public-datasets">Public Datasets<a class="headerlink" href="#public-datasets" title="Permanent link">&para;</a></h2>
<p>Below is a list of public datasets (alphabetical) that could have potentially 
been included during the <em><a href="../datasets/">Build Datasets</a></em> step.</p>
<h3 id="ariba-reference-datasets">Ariba Reference Datasets<a class="headerlink" href="#ariba-reference-datasets" title="Permanent link">&para;</a></h3>
<p>These datasets are available using Ariba's <code>getref</code> function. You can learn 
more about this function at <a href="https://github.com/sanger-pathogens/ariba/wiki/Task:-getref">Ariba's Wiki</a>.</p>
<ul>
<li>
<p><strong><a href="http://en.mediterranee-infection.com/article.php?laref=283%26titre=arg-annot">ARG-ANNOT</a></strong><br />
<em>Gupta, S. K. et al. <a href="http://www.ncbi.nlm.nih.gov/pubmed/24145532">ARG-ANNOT, a new bioinformatic tool to 
discover antibiotic resistance genes in bacterial genomes.</a> 
Antimicrob. Agents Chemother. 58, 212‚Äì220 (2014).</em>  </p>
</li>
<li>
<p><strong><a href="https://card.mcmaster.ca/">CARD</a></strong><br />
<em>Alcock, Brian P., et al. <a href="https://www.ncbi.nlm.nih.gov/pubmed/31665441">CARD 2020: antibiotic resistome surveillance with the comprehensive antibiotic resistance database.</a> Nucleic acids research 48.D1, D517-D525 (2020).</em></p>
</li>
<li>
<p><strong><a href="https://megares.meglab.org/">MEGARes</a></strong><br />
<em>Lakin, S. M. et al. <a href="http://www.ncbi.nlm.nih.gov/pubmed/27899569">MEGARes: an antimicrobial resistance database for high 
throughput sequencing</a>. 
Nucleic Acids Res. 45, D574‚ÄìD580 (2017).</em>  </p>
</li>
<li>
<p><strong><a href="https://www.ncbi.nlm.nih.gov/bioproject/?term=PRJNA313047">NCBI Reference Gene Catalog</a></strong><br />
<em>Feldgarden, M. et al. <a href="https://doi.org/10.1128/AAC.00483-19">Validating the NCBI AMRFinder Tool and Resistance Gene Database Using Antimicrobial Resistance Genotype-Phenotype Correlations in a Collection of NARMS Isolates</a>. Antimicrob. Agents Chemother. (2019)</em>  </p>
</li>
<li>
<p><strong><a href="https://cge.cbs.dtu.dk/services/PlasmidFinder/">plasmidfinder</a></strong><br />
<em>Carattoli, A. et al. <a href="http://www.ncbi.nlm.nih.gov/pubmed/24777092">In silico detection and typing of plasmids using 
PlasmidFinder and plasmid multilocus sequence typing.</a> 
Antimicrob. Agents Chemother. 58, 3895‚Äì3903 (2014).</em></p>
</li>
<li>
<p><strong><a href="https://cge.cbs.dtu.dk//services/ResFinder/">resfinder</a></strong><br />
<em>Zankari, E. et al. <a href="http://www.ncbi.nlm.nih.gov/pubmed/22782487">Identification of acquired antimicrobial resistance genes.</a> 
J. Antimicrob. Chemother. 67, 2640‚Äì2644 (2012).</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/katholt/srst2">SRST2</a></strong><br />
<em>Inouye, M. et al. <a href="http://www.ncbi.nlm.nih.gov/pubmed/25422674">SRST2: Rapid genomic surveillance for public health and 
hospital microbiology labs.</a> 
Genome Med. 6, 90 (2014).</em>  </p>
</li>
<li>
<p><strong><a href="http://www.mgc.ac.cn/VFs/">VFDB</a></strong><br />
<em>Chen, L., Zheng, D., Liu, B., Yang, J. &amp; Jin, Q. <a href="http://www.ncbi.nlm.nih.gov/pubmed/26578559">VFDB 2016: hierarchical 
and refined dataset for big data analysis--10 years on.</a> 
Nucleic Acids Res. 44, D694‚Äì7 (2016).</em>  </p>
</li>
<li>
<p><strong><a href="https://cge.cbs.dtu.dk/services/VirulenceFinder/">VirulenceFinder</a></strong><br />
<em>Joensen, K. G. et al. <a href="http://www.ncbi.nlm.nih.gov/pubmed/24574290">Real-time whole-genome sequencing for routine typing, 
surveillance, and outbreak detection of verotoxigenic Escherichia coli.</a> 
J. Clin. Microbiol. 52, 1501‚Äì1510 (2014).</em>  </p>
</li>
</ul>
<h3 id="minmer-datasets">Minmer Datasets<a class="headerlink" href="#minmer-datasets" title="Permanent link">&para;</a></h3>
<ul>
<li>
<p><strong><a href="https://mash.readthedocs.io/en/latest/data.html">Mash Refseq (release 88) Sketch</a></strong><br />
<em>Ondov, B. D. et al. <a href="https://doi.org/10.1101/557314">Mash Screen: High-throughput sequence containment 
estimation for genome discovery.</a> bioRxiv 557314 (2019).</em>  </p>
</li>
<li>
<p><strong><a href="https://sourmash.readthedocs.io/en/latest/databases.html">Sourmash Genbank LCA Signature</a></strong><br />
<em>Titus Brown, C. &amp; Irber, L. <a href="http://joss.theoj.org/papers/10.21105/joss.00027">sourmash: a library for MinHash sketching of DNA.</a> 
JOSS 1, 27 (2016).</em>  </p>
</li>
</ul>
<h3 id="everything-else">Everything Else<a class="headerlink" href="#everything-else" title="Permanent link">&para;</a></h3>
<ul>
<li>
<p><strong><a href="http://eggnog.embl.de/">eggNOG 5.0 Database</a></strong><br />
<em>J. Huerta-Cepas, D. Szklarczyk, D. Heller, A. Hern√°ndez-Plaza, S. K. Forslund, H. Cook, D. R. Mende, I. Letunic, T. Rattei, L. J. Jensen, C. von Mering, P. Bork, <a href="https://doi.org/10.1093/nar/gky1085">eggNOG 5.0: a hierarchical, functionally and phylogenetically annotated orthology resource based on 5090 organisms and 2502 viruses.</a> Nucleic Acids Res. 47, D309‚ÄìD314 (2019).</em>  </p>
</li>
<li>
<p><strong><a href="https://gtdb.ecogenomic.org/">Genome Taxonomy Database</a></strong><br />
<em>Parks, D. H. et al. 
<a href="https://doi.org/10.1038/nbt.4229">A standardized bacterial taxonomy based on genome phylogeny substantially revises the tree of life.</a>
 Nat. Biotechnol. 36, 996‚Äì1004 (2018)</em><br />
<em>Parks, D. H. et al. 
<a href="https://doi.org/10.1101/771964">Selection of representative genomes for 24,706 bacterial and archaeal species clusters provide a complete genome-based taxonomy.</a>
 bioRxiv 771964 (2019)</em>  </p>
</li>
<li>
<p><strong><a href="https://www.ncbi.nlm.nih.gov/refseq/">NCBI RefSeq Database</a></strong><br />
<em>O‚ÄôLeary, N. A. et al. <a href="http://dx.doi.org/10.1093/nar/gkv1189">Reference sequence (RefSeq) database at NCBI: current status, 
taxonomic expansion, and functional annotation</a>. 
Nucleic Acids Res. 44, D733‚Äì45 (2016).</em>  </p>
</li>
<li>
<p><strong><a href="https://ccb-microbe.cs.uni-saarland.de/plsdb/plasmids/download/">PLSDB - A plasmid database</a></strong><br />
<em>Galata, V., Fehlmann, T., Backes, C. &amp; Keller, A. <a href="http://dx.doi.org/10.1093/nar/gky1050">PLSDB: a resource of complete bacterial 
plasmids</a>. 
Nucleic Acids Res. 47, D195‚ÄìD202 (2019).</em>  </p>
</li>
<li>
<p><strong><a href="https://pubmlst.org/">PubMLST.org</a></strong><br />
<em>Jolley, K. A., Bray, J. E. &amp; Maiden, M. C. J. <a href="http://dx.doi.org/10.12688/wellcomeopenres.14826.1">Open-access bacterial population genomics: BIGSdb 
software, the PubMLST.org website and their applications</a>. 
Wellcome Open Res 3, 124 (2018).</em>  </p>
</li>
<li>
<p><strong><a href="https://www.arb-silva.de/">SILVA rRNA Database</a></strong><br />
<em>Quast, C. et al. 
<a href="https://doi.org/10.1093/nar/gks1219">The SILVA ribosomal RNA gene database project: improved data processing and web-based tools.</a> 
Nucleic Acids Res. 41, D590‚Äì6 (2013)</em></p>
</li>
</ul>
<h2 id="software-included-in-bactopia">Software Included In Bactopia<a class="headerlink" href="#software-included-in-bactopia" title="Permanent link">&para;</a></h2>
<p>Below is a list of software (alphabetical) used (directly and indirectly) by Bactopia. A link to the software page as well as the citation (if available) have been included.</p>
<ul>
<li>
<p><strong><a href="https://github.com/ncbi/amr">AMRFinderPlus</a></strong><br />
Find acquired antimicrobial resistance genes and some point mutations in protein or assembled nucleotide sequences.<br />
<em>Feldgarden, M. et al. <a href="https://doi.org/10.1128/AAC.00483-19">Validating the NCBI AMRFinder Tool and Resistance Gene Database Using Antimicrobial Resistance Genotype-Phenotype Correlations in a Collection of NARMS Isolates</a>. Antimicrob. Agents Chemother. (2019)</em>  </p>
</li>
<li>
<p><strong><a href="http://130.235.244.92/ARAGORN/Downloads/">Aragorn</a></strong><br />
Finds transfer RNA features (tRNA)<br />
<em>Laslett D. and B. Canback, <a href="https://doi.org/10.1093/nar/gkh152">ARAGORN, a program to detect tRNA genes and tmRNA genes in nucleotide sequences.</a> Nucleic Acids Res. 32(1):11-6. (2004)</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/sanger-pathogens/ariba">Ariba</a></strong><br />
Antimicrobial Resistance Identification By Assembly<br />
<em>Hunt, M. et al. <a href="http://dx.doi.org/10.1099/mgen.0.000131">ARIBA: rapid antimicrobial resistance genotyping directly from 
sequencing reads</a>. 
Microb Genom 3, e000131 (2017).</em>  </p>
</li>
<li>
<p><strong><a href="https://www.niehs.nih.gov/research/resources/software/biostatistics/art/index.cfm">ART</a></strong><br />
A set of simulation tools to generate synthetic next-generation sequencing reads<br />
<em>W. Huang, L. Li, J. R. Myers, G. T. Marth, <a href="http://dx.doi.org/10.1093/bioinformatics/btr708">ART: a next-generation sequencing read simulator.</a> Bioinformatics. 28, 593‚Äì594 (2012).</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/rpetit3/assembly-scan">assembly-scan</a></strong><br />
Generate basic stats for an assembly.<br />
<em>Petit III, R. A. <a href="https://github.com/rpetit3/assembly-scan">assembly-scan: generate basic stats for an 
assembly</a>.</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/tseemann/barrnap">Barrnap</a></strong><br />
Bacterial ribosomal RNA predictor<br />
<em>Seemann, T. <a href="https://github.com/tseemann/barrnap">Barrnap: Bacterial ribosomal RNA predictor</a></em>  </p>
</li>
<li>
<p><strong><a href="https://jgi.doe.gov/data-and-tools/bbtools/">BBTools</a></strong><br />
BBTools is a suite of fast, multithreaded bioinformatics tools designed for analysis of DNA and RNA sequence data.<br />
<em>Bushnell, B. <a href="http://sourceforge.net/projects/bbmap/">BBMap short read aligner, and other bioinformatic tools.</a></em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/samtools/bcftools">BCFtools</a></strong><br />
Utilities for variant calling and manipulating VCFs and BCFs.<br />
<em>Danecek, P. et al. <a href="http://github.com/samtools/bcftools">BCFtools - Utilities for variant calling and manipulating VCFs and BCFs.</a></em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/arq5x/bedtools2">Bedtools</a></strong><br />
A powerful toolset for genome arithmetic.<br />
<em>Quinlan, A. R. &amp; Hall, I. M. <a href="http://dx.doi.org/10.1093/bioinformatics/btq033">BEDTools: a flexible suite of utilities for 
comparing genomic features</a>. 
Bioinformatics 26, 841‚Äì842 (2010).</em>  </p>
</li>
<li>
<p><strong><a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi">BLAST</a></strong><br />
Basic Local Alignment Search Tool<br />
<em>Camacho, C. et al. <a href="http://dx.doi.org/10.1186/1471-2105-10-421">BLAST+: architecture and applications</a>. 
BMC Bioinformatics 10, 421 (2009).</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/BenLangmead/bowtie2">Bowtie2</a></strong><br />
A fast and sensitive gapped read aligner<br />
<em>B. Langmead, S. L. Salzberg, <a href="http://dx.doi.org/10.1038/nmeth.1923">Fast gapped-read alignment with Bowtie 2.</a> Nat. Methods. 9, 357‚Äì359 (2012).</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/lh3/bwa/">BWA</a></strong><br />
Burrow-Wheeler Aligner for short-read alignment<br />
<em>Li, H. <a href="http://arxiv.org/abs/1303.3997">Aligning sequence reads, clone sequences and assembly contigs with 
BWA-MEM</a>. arXiv [q-bio.GN] (2013).</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/weizhongli/cdhit">CD-Hit</a></strong><br />
Accelerated for clustering the next-generation sequencing data<br />
<em>Li, W. &amp; Godzik, A. <a href="http://dx.doi.org/10.1093/bioinformatics/btl158">Cd-hit: a fast program for clustering and comparing large sets of protein 
or nucleotide sequences</a>. 
Bioinformatics 22, 1658‚Äì1659 (2006).</em><br />
<em>Fu, L., Niu, B., Zhu, Z., Wu, S. &amp; Li, W. <a href="http://dx.doi.org/10.1093/bioinformatics/bts565">CD-HIT: accelerated for clustering the next-generation 
sequencing data</a>. 
Bioinformatics 28, 3150‚Äì3152 (2012).</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/Ecogenomics/CheckM">CheckM</a></strong><br />
Assess the quality of microbial genomes recovered from isolates, single cells, and metagenomes<br />
<em>D. H. Parks, M. Imelfort, C. T. Skennerton, P. Hugenholtz, G. W. Tyson, <a href="http://dx.doi.org/10.1101/gr.186072.114">CheckM: assessing the quality of microbial genomes recovered from isolates, single cells, and metagenomes.</a> Genome Res. 25, 1043‚Äì1055 (2015).</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/xavierdidelot/ClonalFrameML">ClonalFramML</a></strong><br />
Efficient Inference of Recombination in Whole Bacterial Genomes<br />
<em>Didelot, X. &amp; Wilson, D. J. 
<a href="http://dx.doi.org/10.1371/journal.pcbi.1004041">ClonalFrameML: efficient inference of recombination in whole bacterial genomes.</a>
 PLoS Comput. Biol. 11, e1004041 (2015)</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/bbuchfink/diamond">DIAMOND</a></strong><br />
Accelerated BLAST compatible local sequence aligner.<br />
<em>B. Buchfink, C. Xie, D. H. Huson, <a href="http://dx.doi.org/10.1038/nmeth.3176">Fast and sensitive protein alignment using DIAMOND.</a> Nat. Methods. 12, 59‚Äì60 (2015).</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/eggnogdb/eggnog-mapper">eggNOG-mapper</a></strong><br />
Fast genome-wide functional annotation through orthology assignment<br />
<em>J. Huerta-Cepas, K. Forslund, L. P. Coelho, D. Szklarczyk, L. J. Jensen, C. von Mering, P. Bork, <a href="http://dx.doi.org/10.1093/molbev/msx148">Fast Genome-Wide Functional Annotation through Orthology Assignment by eggNOG-Mapper.</a> Mol. Biol. Evol. 34, 2115‚Äì2122 (2017).</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/ParBLiSS/FastANI">FastANI</a></strong><br />
Fast Whole-Genome Similarity (ANI) Estimation<br />
<em>Jain, C., Rodriguez-R, L. M., Phillippy, A. M., Konstantinidis, K. T. &amp; Aluru, S. 
<a href="http://dx.doi.org/10.1038/s41467-018-07641-9">High throughput ANI analysis of 90K prokaryotic genomes reveals clear species boundaries.</a>
 Nat. Commun. 9, 5114 (2018)</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/s-andrews/FastQC">FastQC</a></strong><br />
A quality control analysis tool for high throughput sequencing data.<br />
<em>Andrews, S. <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc">FastQC: a quality control tool for high throughput sequence data.</a>.</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/rpetit3/fastq-dl">fastq-dl</a></strong><br />
Download FASTQ files from SRA or ENA repositories.<br />
<em>Petit III, R. A. <a href="https://github.com/rpetit3/fastq-dl">fastq-dl: Download FASTQ files from SRA or ENA repositories.</a></em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/rpetit3/fastq-scan">fastq-scan</a></strong><br />
Output FASTQ summary statistics in JSON format<br />
<em>Petit III, R. A. <a href="https://github.com/rpetit3/fastq-scan">fastq-scan: generate summary statistics of input FASTQ sequences.</a></em>  </p>
</li>
<li>
<p><strong><a href="http://www.microbesonline.org/fasttree">FastTree 2</a></strong><br />
Approximately-maximum-likelihood phylogenetic trees<br />
<em>Price, M. N., Dehal, P. S. &amp; Arkin, A. P. 
<a href="https://dx.doi.org/10.1371%2Fjournal.pone.0009490">FastTree 2 ‚Äì Approximately Maximum-Likelihood Trees for Large Alignments.</a>
 PLoS One 5, e9490 (2010)</em>  </p>
</li>
<li>
<p><strong><a href="https://ccb.jhu.edu/software/FLASH/">FLASH</a></strong><br />
A fast and accurate tool to merge paired-end reads.<br />
<em>Magoƒç, T., and S. L. Salzberg, <a href="https://doi.org/10.1093/bioinformatics/btr507">FLASH: fast length adjustment of short reads to improve genome assemblies.</a> Bioinformatics 27.21 (2011): 2957-2963.</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/ekg/freebayes">freebayes</a></strong><br />
Bayesian haplotype-based genetic polymorphism discovery and genotyping<br />
<em>Garrison E., and G. Marth, <a href="https://arxiv.org/abs/1207.3907">Haplotype-based variant detection from short-read sequencing.</a> arXiv preprint arXiv:1207.3907 [q-bio.GN] (2012)</em>  </p>
</li>
<li>
<p><strong><a href="https://www.gnu.org/software/parallel/">GNU Parallel</a></strong><br />
A shell tool for executing jobs in parallel<br />
<em>Tange, O. <a href="https://doi.org/10.5281/zenodo.1146014">GNU Parallel</a> 2018, March 2018</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/Ecogenomics/GTDBTk">GTDB-Tk</a></strong><br />
A toolkit for assigning objective taxonomic classifications to bacterial and archaeal genomes<br />
<em>Chaumeil, P.-A., Mussig, A. J., Hugenholtz, P. &amp; Parks, D. H. 
<a href="https://doi.org/10.1093/bioinformatics/btz848">GTDB-Tk: a toolkit to classify genomes with the Genome Taxonomy Database.</a>
 Bioinformatics (2019)</em>  </p>
</li>
<li>
<p><strong><a href="http://hmmer.org/">HMMER</a></strong><br />
Biosequence analysis using profile hidden Markov models<br />
<em>Eddy, S. R. 
<a href="https://doi.org/10.1371/journal.pcbi.1002195">Accelerated Profile HMM Searches.</a> 
PLoS Comput. Biol. 7, e1002195 (2011)</em>  </p>
</li>
<li>
<p><strong><a href="http://eddylab.org/infernal/">Infernal</a></strong><br />
Searches DNA sequence databases for RNA structure and sequence similarities<br />
<em>Nawrocki, E. P., and S. R. Eddy, <a href="https://doi.org/10.1093/bioinformatics/btt509">Infernal 1.1: 100-fold faster RNA homology searches.</a> Bioinformatics, 29(22), 2933-2935. (2013)</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/Cibiv/IQ-TREE">IQ-TREE</a></strong><br />
Efficient phylogenomic software by maximum likelihood<br />
<em>L.-T. Nguyen, H.A. Schmidt, A. von Haeseler, B.Q. Minh (2015) 
<a href="https://doi.org/10.1093/molbev/msu300">IQ-TREE: A fast and effective stochastic algorithm for estimating maximum likelihood phylogenies.</a>
 Mol. Biol. Evol., 32:268-274.</em><br />
<em>S. Kalyaanamoorthy, B.Q. Minh, T.K.F. Wong, A. von Haeseler, L.S. Jermiin (2017) 
<a href="https://doi.org/10.1038/nmeth.4285">ModelFinder: Fast model selection for accurate phylogenetic estimates.</a> 
Nat. Methods, 14:587-589.</em><br />
<em>D.T. Hoang, O. Chernomor, A. von Haeseler, B.Q. Minh, L.S. Vinh (2018) <a href="https://doi.org/10.1093/molbev/msx281">
UFBoot2: Improving the ultrafast bootstrap approximation.</a>
 Mol. Biol. Evol., 35:518‚Äì522.</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/jhawkey/IS_mapper">ISMapper</a></strong><br />
IS mapping software<br />
<em>Hawkey, J. et al. <a href="http://dx.doi.org/10.1186/s12864-015-1860-2">ISMapper: identifying transposase insertion sites in 
bacterial genomes from short read sequence data</a>. 
BMC Genomics 16, 667 (2015).</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/mourisl/Lighter">Lighter</a></strong><br />
Fast and memory-efficient sequencing error corrector<br />
<em>Song, L., Florea, L. and B. Langmead, <a href="http://genomebiology.com/2014/15/11/509/">Lighter: Fast and Memory-efficient Sequencing Error Correction without Counting</a>. Genome Biol. 2014 Nov 15;15(11):509.</em></p>
</li>
<li>
<p><strong><a href="https://mafft.cbrc.jp/alignment/software/">MAFFT</a></strong><br />
Multiple alignment program for amino acid or nucleotide sequences<br />
<em>Katoh, K. &amp; Standley, D. M. 
<a href="https://doi.org/10.1093/molbev/mst010">MAFFT multiple sequence alignment software version 7: improvements in performance and usability.</a> 
Mol. Biol. Evol. 30, 772‚Äì780 (2013)</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/marbl/Mash">Mash</a></strong><br />
Fast genome and metagenome distance estimation using MinHash<br />
<em>Ondov, B. D. et al. <a href="http://dx.doi.org/10.1186/s13059-016-0997-x">Mash: fast genome and metagenome distance 
estimation using MinHash</a>. 
Genome Biol. 17, 132 (2016).</em><br />
<em>Ondov, B. D. et al. <a href="http://dx.doi.org/10.1101/557314">Mash Screen: High-throughput sequence 
containment estimation for genome discovery</a>. 
bioRxiv 557314 (2019).</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/lskatz/mashtree">Mashtree</a></strong><br />
Create a tree using Mash distances<br />
<em>Katz, L. S., Griswold, T., Morrison, S., Caravas, J., Zhang, S., den Bakker, H.C., Deng, X., and Carleton, H. A. <a href="https://doi.org/10.21105/joss.01762">Mashtree: a rapid comparison of whole genome sequence files.</a> Journal of Open Source Software, 4(44), 1762, (2019)</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/kwongj/maskrc-svg">maskrc-svg</a></strong><br />
Masks recombination as detected by ClonalFrameML or Gubbins<br />
<em>Kwong, J. <a href="https://github.com/kwongj/maskrc-svg">maskrc-svg - Masks recombination as detected by ClonalFrameML or Gubbins and draws an SVG.</a></em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/mcveanlab/mccortex">McCortex</a></strong><br />
De novo genome assembly and multisample variant calling<br />
<em>Turner, I., Garimella, K. V., Iqbal, Z. and G. McVean, <a href="http://dx.doi.org/10.1093/bioinformatics/bty157">Integrating long-range 
connectivity information into de Bruijn graphs.</a> 
Bioinformatics 34, 2556‚Äì2565 (2018).</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/voutcn/megahit">MEGAHIT</a></strong><br />
Ultra-fast and memory-efficient (meta-)genome assembler<br />
<em>Li, D., et al. <a href="https://doi.org/10.1093/bioinformatics/btv033">MEGAHIT: an ultra-fast single-node solution for large and complex metagenomics assembly via succinct de Bruijn graph.</a> Bioinformatics 31.10 (2015): 1674-1676.</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/ctSkennerton/minced">MinCED</a></strong><br />
Mining CRISPRs in Environmental Datasets<br />
<em>Skennerton, C. <a href="https://github.com/ctSkennerton/minced">MinCED: Mining CRISPRs in Environmental Datasets</a></em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/lh3/minimap2">Minimap2</a></strong><br />
A versatile pairwise aligner for genomic and spliced nucleotide sequences<br />
<em>Li, H. <a href="https://doi.org/10.1093/bioinformatics/bty191">Minimap2: pairwise alignment for nucleotide sequences.</a> Bioinformatics, 34:3094-3100. (2018)</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/kblin/ncbi-genome-download">ncbi-genome-download</a></strong><br />
Scripts to download genomes from the NCBI FTP servers<br />
<em>Blin, K. <a href="https://github.com/kblin/ncbi-genome-download">ncbi-genome-download: Scripts to download genomes from the NCBI FTP 
servers</a></em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/nextflow-io/nextflow">Nextflow</a></strong><br />
A DSL for data-driven computational pipelines.<br />
<em>Di Tommaso, P., Chatzou, M., Floden, E. W., Barja, P.P., Palumbo, E., Notredame, C., 2017. <a href="https://www.nature.com/articles/nbt.3820.pdf?origin=ppub">Nextflow enables reproducible computational workflows.</a> Nat. Biotechnol. 35, 316‚Äì319.</em></p>
</li>
<li>
<p><strong><a href="http://hmmer.org/">nhmmer</a></strong><br />
DNA homology search with profile HMMs.<br />
<em>Wheeler, T. J. &amp; Eddy, S. R. 
<a href="https://doi.org/10.1093/bioinformatics/btt403">nhmmer: DNA homology search with profile HMMs.</a>
 Bioinformatics 29, 2487‚Äì2489 (2013)</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/HRGV/phyloFlash">phyloFlash</a></strong><br />
A pipeline to rapidly reconstruct the SSU rRNAs and explore phylogenetic composition of an illumina (meta)genomic dataset.<br />
<em>H. R. Gruber-Vodicka, B.KB. Seah, E. Pruesse. 
<a href="https://doi.org/10.1101/521922">phyloFlash ‚Äî Rapid SSU rRNA profiling and targeted assembly from metagenomes.</a> 
bioRxiv 521922</em>  </p>
</li>
<li>
<p><strong><a href="https://zlib.net/pigz/">Pigz</a></strong><br />
A parallel implementation of gzip for modern multi-processor, multi-core machines.<br />
<em>Adler, M. <a href="https://zlib.net/pigz/">pigz: A parallel implementation of gzip for modern multi-processor, multi-core machines.</a> Jet Propulsion Laboratory (2015).</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/broadinstitute/pilon/">Pilon</a></strong><br />
An automated genome assembly improvement and variant detection tool<br />
<em>Walker, B. J., et al. <a href="https://doi.org/10.1371/journal.pone.0112963">Pilon: an integrated tool for comprehensive microbial variant detection and genome assembly improvement.</a> PloS one 9.11 (2014): e112963.</em>  </p>
</li>
<li>
<p><strong><a href="http://github.com/SionBayliss/PIRATE">PIRATE</a></strong><br />
A toolbox for pangenome analysis and threshold evaluation.<br />
<em>S. C. Bayliss, H. A. Thorpe, N. M. Coyle, S. K. Sheppard, E. J. Feil 
<a href="https://doi.org/10.1093/gigascience/giz119">PIRATE: A fast and scalable pangenomics toolbox for clustering diverged orthologues in bacteria.</a> 
Gigascience. 8 (2019)</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/matsen/pplacer">pplacer</a></strong><br />
Phylogenetic placement and downstream analysis<br />
<em>Matsen, F. A., Kodner, R. B. &amp; Armbrust, E. V. 
<a href="https://doi.org/10.1186/1471-2105-11-538">pplacer: linear time maximum-likelihood and Bayesian phylogenetic placement of sequences onto a fixed reference tree.</a>
 BMC Bioinformatics 11, 538 (2010)</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/hyattpd/Prodigal">Prodigal</a></strong><br />
Fast, reliable protein-coding gene prediction for prokaryotic genomes.<br />
<em>Hyatt, D., et al. <a href="https://doi.org/10.1186/1471-2105-11-119">Prodigal: prokaryotic gene recognition and translation initiation site identification.</a> BMC Bioinformatics 11.1 (2010): 119.</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/tseemann/prokka">Prokka</a></strong><br />
Rapid prokaryotic genome annotation<br />
<em>Seemann, T. <a href="http://dx.doi.org/10.1093/bioinformatics/btu153">Prokka: rapid prokaryotic genome annotation</a>. 
Bioinformatics 30, 2068‚Äì2069 (2014).</em>  </p>
</li>
<li>
<p><strong><a href="http://quast.sourceforge.net/">QUAST</a></strong><br />
Quality Assessment Tool for Genome<br />
<em>A. Gurevich, V. Saveliev, N. Vyahhi, G. Tesler, <a href="http://dx.doi.org/10.1093/bioinformatics/btt086">QUAST: quality assessment tool for genome assemblies.</a> Bioinformatics. 29, 1072‚Äì1075 (2013).</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/lbcb-sci/racon">Racon</a></strong><br />
Ultrafast consensus module for raw de novo genome assembly of long uncorrected reads<br />
<em>R. Vaser, I. Soviƒá, N. Nagarajan, M. ≈†ikiƒá, <a href="http://dx.doi.org/10.1101/gr.214270.116">Fast and accurate de novo genome assembly from long uncorrected reads.</a> Genome Res. 27, 737‚Äì746 (2017).</em>  </p>
</li>
<li>
<p><strong><a href="http://www.cbs.dtu.dk/services/RNAmmer/">RNAmmer</a></strong><br />
Consistent and rapid annotation of ribosomal RNA genes<br />
<em>Lagesen, K., et al. <a href="https://dx.doi.org/10.1093%2Fnar%2Fgkm160">RNAmmer: consistent annotation of rRNA genes in genomic sequences.</a> Nucleic Acids Res 35.9: 3100-3108. (2007)</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/sanger-pathogens/Roary">Roary</a></strong><br />
Rapid large-scale prokaryote pan genome analysis<br />
<em>Page, A. J. et al. 
<a href="https://doi.org/10.1093/bioinformatics/btv421">Roary: rapid large-scale prokaryote pan genome analysis.</a>
 Bioinformatics 31, 3691‚Äì3693 (2015)</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/tseemann/samclip">samclip</a></strong><br />
Filter SAM file for soft and hard clipped alignments<br />
<em>Seemann, T. <a href="https://github.com/tseemann/samclip">Samclip: Filter SAM file for soft and hard clipped alignments</a></em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/samtools/samtools">Samtools</a></strong><br />
Tools for manipulating next-generation sequencing data<br />
<em>Li, H. et al. <a href="http://dx.doi.org/10.1093/bioinformatics/btp352">The Sequence Alignment/Map format and SAMtools</a>. 
Bioinformatics 25, 2078‚Äì2079 (2009).</em></p>
</li>
<li>
<p><strong><a href="https://github.com/lh3/seqtk">Seqtk</a></strong><br />
A fast and lightweight tool for processing sequences in the FASTA or FASTQ format.<br />
<em>Li, H. <a href="https://github.com/lh3/seqtk">Toolkit for processing sequences in FASTA/Q formats</a></em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/tseemann/shovill">Shovill</a></strong><br />
Faster assembly of Illumina reads<br />
<em>Seemann, T. <a href="https://github.com/tseemann/shovill">Shovill: De novo assembly pipeline for Illumina paired reads</a></em>  </p>
</li>
<li>
<p><strong><a href="http://www.cbs.dtu.dk/services/SignalP-4.0/">SignalP</a></strong><br />
Finds signal peptide features in CDS<br />
<em>Petersen, T. N., et al. <a href="https://doi.org/10.1038/nmeth.1701">SignalP 4.0: discriminating signal peptides from transmembrane regions.</a> Nature methods 8.10: 785.(2011)</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/ncbi/SKESA">SKESA</a></strong><br />
Strategic Kmer Extension for Scrupulous Assemblies<br />
<em>Souvorov, A., Agarwala, R. and D. J. Lipman. <a href="https://doi.org/10.1186/s13059-018-1540-z">SKESA: strategic k-mer extension for scrupulous assemblies.</a> Genome Biology 19:153 (2018)._</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/tseemann/snippy">Snippy</a></strong><br />
Rapid haploid variant calling and core genome alignment<br />
<em>Seemann, T. <a href="https://github.com/tseemann/snippy">Snippy: fast bacterial variant calling from NGS reads</a>
(2015)</em>  </p>
</li>
<li>
<p><strong><a href="http://snpeff.sourceforge.net/">SnpEff</a></strong><br />
Genomic variant annotations and functional effect prediction toolbox.<br />
<em>Cingolani, P., et al. <a href="https://doi.org/10.4161/fly.19695">A program for annotating and predicting the effects of single nucleotide polymorphisms, SnpEff: SNPs in the genome of Drosophila melanogaster strain w1118; iso-2; iso-3.</a> Fly, 6(2), 80-92 (2012)</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/tseemann/snp-dists">snp-dists</a></strong><br />
Pairwise SNP distance matrix from a FASTA sequence alignment<br />
<em>Seemann, T. <a href="https://github.com/tseemann/snp-dists">snp-dists - Pairwise SNP distance matrix from a FASTA sequence alignment.</a></em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/sanger-pathogens/snp-sites">SNP-sites</a></strong><br />
Rapidly extracts SNPs from a multi-FASTA alignment.<br />
<em>Page, A. J., et al. <a href="https://dx.doi.org/10.1099%2Fmgen.0.000056">SNP-sites: rapid efficient extraction of SNPs from multi-FASTA alignments.</a> Microbial Genomics 2.4 (2016).</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/dib-lab/sourmash">Sourmash</a></strong><br />
Compute and compare MinHash signatures for DNA data sets.<br />
<em>Titus Brown, C. and L. Irber <a href="http://dx.doi.org/10.21105/joss.00027">sourmash: a library for MinHash sketching 
of DNA</a>. JOSS 1, 27 (2016).</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/ablab/spades">SPAdes</a></strong><br />
An assembly toolkit containing various assembly pipelines.<br />
<em>Bankevich, A., et al. <a href="https://doi.org/10.1089/cmb.2012.0021">SPAdes: a new genome assembly algorithm and its applications to single-cell sequencing.</a> Journal of computational biology 19.5 (2012): 455-477.</em>  </p>
</li>
<li>
<p><strong><a href="http://www.usadellab.org/cms/index.php?page=trimmomatic">Trimmomatic</a></strong><br />
A flexible read trimming tool for Illumina NGS data<br />
<em>Bolger, A. M., Lohse, M., and B. Usadel. <a href="https://doi.org/10.1093/bioinformatics/btu170">Trimmomatic: a flexible trimmer for Illumina sequence data.</a> Bioinformatics 30.15 (2014): 2114-2120.</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/rrwick/Unicycler">Unicycler</a></strong><br />
Hybrid assembly pipeline for bacterial genomes<br />
<em>R. R. Wick, L. M. Judd, C. L. Gorrie, K. E. Holt, <a href="http://dx.doi.org/10.1371/journal.pcbi.1005595">Unicycler: Resolving bacterial genome assemblies from short and long sequencing reads.</a> PLoS Comput. Biol. 13, e1005595 (2017).</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/rpetit3/vcf-annotator">VCF-Annotator</a></strong><br />
Add biological annotations to variants in a VCF file.<br />
<em>Petit III, R. A. <a href="https://github.com/rpetit3/vcf-annotator">VCF-Annotator: Add biological annotations to variants 
in a VCF file.</a>.</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/vcflib/vcflib">Vcflib</a></strong><br />
a simple C++ library for parsing and manipulating VCF files<br />
<em>Garrison, E. <a href="https://github.com/vcflib/vcflib">Vcflib: A C++ library for parsing and manipulating VCF files</a></em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/dzerbino/velvet">Velvet</a></strong><br />
Short read de novo assembler using de Bruijn graphs<br />
<em>Zerbino, D. R., and E. Birney. <a href="http://www.genome.org/cgi/doi/10.1101/gr.074492.107">Velvet: algorithms for de novo short read assembly using de Bruijn graphs.</a> Genome research 18.5 (2008): 821-829.</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/torognes/vsearch">VSEARCH</a></strong><br />
Versatile open-source tool for metagenomics<br />
<em>Rognes, T., Flouri, T., Nichols, B., Quince, C. &amp; Mah√©, F. 
<a href="https://doi.org/10.7717/peerj.2584">VSEARCH: a versatile open source tool for metagenomics.</a>
 PeerJ 4, e2584 (2016)</em>  </p>
</li>
<li>
<p><strong><a href="https://github.com/atks/vt">vt</a></strong><br />
A tool set for short variant discovery in genetic sequence data.<br />
<em>Tan, A., Abecasis, G. R., and H. M. Kang, <a href="https://doi.org/10.1093/bioinformatics/btv112">Unified representation of genetic variants.</a> Bioinformatics, 31(13), 2202-2204. (2015)</em>  </p>
</li>
</ul>
                
                  
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../troubleshooting/" title="Troubleshooting" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                Troubleshooting
              </span>
            </div>
          </a>
        
        
          <a href="../bactopia-tools/" title="Introduction" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                Introduction
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            Copyright ¬© 2019 Robert A. Petit III. All rights reserved.
          </div>
        
        powered by
        <a href="https://www.mkdocs.org">MkDocs</a>
        and
        <a href="https://squidfunk.github.io/mkdocs-material/">
          Material for MkDocs</a>
      </div>
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../assets/javascripts/application.c648116f.js"></script>
      
      <script>app.initialize({version:"1.0.4",url:{base:".."}})</script>
      
    
  </body>
</html>