
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="An full guide for using Bactopia. Including a detailed description of each step in the pipeline.">
      
      
        <meta name="author" content="Robert A. Petit III">
      
      
        <link rel="canonical" href="https://bactopia.github.io/v3.1.0/full-guide/">
      
      
        <link rel="prev" href="../bactopia/merlin/">
      
      
        <link rel="next" href="../changelog/">
      
      
        
      
      
      <link rel="icon" href="../assets/favicon.ico">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.5.31+insiders-4.53.12">
    
    
      
        <title>Full Guide - Bactopia</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.dad623df.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.ab4e12ef.min.css">
      
      


    
    
      
    
    
      
        
        
        
        <link rel="stylesheet" href="../assets/external/fonts.googleapis.com/css.2196aa76.css">
        <style>:root{--md-text-font:"Ubuntu";--md-code-font:"Ubuntu Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../custom/magnific-popup.css">
    
      <link rel="stylesheet" href="../custom/extra.css">
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      
  


  
  

<script id="__analytics">function __md_analytics(){function n(){dataLayer.push(arguments)}window.dataLayer=window.dataLayer||[],n("js",new Date),n("config","G-QH76FN9N78"),document.addEventListener("DOMContentLoaded",function(){document.forms.search&&document.forms.search.query.addEventListener("blur",function(){this.value&&n("event","search",{search_term:this.value})}),document$.subscribe(function(){var a=document.forms.feedback;if(void 0!==a)for(var e of a.querySelectorAll("[type=submit]"))e.addEventListener("click",function(e){e.preventDefault();var t=document.location.pathname,e=this.getAttribute("data-md-value");n("event","feedback",{page:t,data:e}),a.firstElementChild.disabled=!0;e=a.querySelector(".md-feedback__note [data-md-value='"+e+"']");e&&(e.hidden=!1)}),a.hidden=!1}),location$.subscribe(function(e){n("config","G-QH76FN9N78",{page_path:e.pathname})})});var e=document.createElement("script");e.async=!0,e.src="https://www.googletagmanager.com/gtag/js?id=G-QH76FN9N78",document.getElementById("__analytics").insertAdjacentElement("afterEnd",e)}</script>
  
    <script>var consent;"undefined"==typeof __md_analytics||(consent=__md_get("__consent"))&&consent.analytics&&__md_analytics()</script>
  

    
    
  
<meta property="og:type" content="website" />
<meta property="og:title" content="Full Guide - Bactopia" />
<meta property="og:description" content="An full guide for using Bactopia. Including a detailed description of each step in the pipeline." />
<meta property="og:image" content="https://bactopia.github.io/v3.1.0/assets/images/social/full-guide.png" />
<meta property="og:image:type" content="image/png" />
<meta property="og:image:width" content="1200" />
<meta property="og:image:height" content="630" />
<meta property="og:url" content="https://bactopia.github.io/v3.1.0/full-guide/" />
<meta property="twitter:card" content="summary_large_image" />
<meta property="twitter:title" content="Full Guide - Bactopia" />
<meta property="twitter:description" content="An full guide for using Bactopia. Including a detailed description of each step in the pipeline." />
<meta property="twitter:image" content="https://bactopia.github.io/v3.1.0/assets/images/social/full-guide.png" />
</head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#step-1-gather" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
        <aside class="md-banner">
          <div class="md-banner__inner md-grid md-typeset">
            
              <button class="md-banner__button md-icon" aria-label="Don't show this again">
                
                <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
              </button>
            
            
  <!-- Add announcement here, including arbitrary HTML -->
  🎉 Please take a minutes to complete the <a href="https://forms.gle/BA2BDHRQvZtpVcD38" target="_blank" rel="noopener">2024 Bactopia Usage Survey</a> 🎉

          </div>
          
            <script>var content,el=document.querySelector("[data-md-component=announce]");el&&(content=el.querySelector(".md-typeset"),__md_hash(content.innerHTML)===__md_get("__announce")&&(el.hidden=!0))</script>
          
        </aside>
      
    </div>
    
      <div data-md-color-scheme="default" data-md-component="outdated" hidden>
        
          <aside class="md-banner md-banner--warning">
            <div class="md-banner__inner md-grid md-typeset">
              
  You're not viewing the latest version.
  <a href="../.."> 
    <strong>Click here to go to latest.</strong>
  </a>

            </div>
            <script>var el=document.querySelector("[data-md-component=outdated]"),outdated=__md_get("__outdated",sessionStorage);!0===outdated&&el&&(el.hidden=!1)</script>
          </aside>
        
      </div>
    
    
      

  

<header class="md-header md-header--shadow md-header--lifted" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="Bactopia" class="md-header__button md-logo" aria-label="Bactopia" data-md-component="logo">
      
  <img src="../assets/bactopia-small-logo.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Bactopia
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Full Guide
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme)" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m14.3 16-.7-2h-3.2l-.7 2H7.8L11 7h2l3.2 9h-1.9M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69m-9.15 3.96h2.3L12 9l-1.15 3.65Z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme="default" data-md-color-primary="teal" data-md-color-accent="teal"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_2" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme="slate" data-md-color-primary="deep-purple" data-md-color-accent="deep-purple"  aria-label="Switch to system preference"  type="radio" name="__palette" id="__palette_2">
    
      <label class="md-header__button md-icon" title="Switch to system preference" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12c0-2.42-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var media,input,key,value,palette=__md_get("__palette");if(palette&&palette.color){"(prefers-color-scheme)"===palette.color.media&&(media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']"),palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent"));for([key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/bactopia/bactopia" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.6.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
    
      
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
    
  
  
    
    
      <li class="md-tabs__item md-tabs__item--active">
        <a href=".." class="md-tabs__link">
          
  
  Bactopia

        </a>
      </li>
    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../bactopia-tools/" class="md-tabs__link">
          
  
  Bactopia Tools

        </a>
      </li>
    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../impact-and-outreach/" class="md-tabs__link">
          
  
  Impact & Outreach

        </a>
      </li>
    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../blog/" class="md-tabs__link">
          
  
  Blog

        </a>
      </li>
    
  

      
    </ul>
  </div>
</nav>
    
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="Bactopia" class="md-nav__button md-logo" aria-label="Bactopia" data-md-component="logo">
      
  <img src="../assets/bactopia-small-logo.png" alt="logo">

    </a>
    Bactopia
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/bactopia/bactopia" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.6.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
    
  
  
  
    
    
    
      
        
        
      
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1" checked>
        
          
          <label class="md-nav__link" for="__nav_1" id="__nav_1_label" tabindex="">
            
  
  <span class="md-ellipsis">
    
  
    Bactopia
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_1_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_1">
            <span class="md-nav__icon md-icon"></span>
            
  
    Bactopia
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    Introduction
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../quick-start/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    Quick Start
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../installation/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    Installation
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../beginners-guide/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    Beginner's Guide
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../tutorial/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    Tutorial
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_6" >
        
          
          <label class="md-nav__link" for="__nav_1_6" id="__nav_1_6_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    
  
    Workflow Steps
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_6">
            <span class="md-nav__icon md-icon"></span>
            
  
    Workflow Steps
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia/gather/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    Gather
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia/qc/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    QC
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia/assembler/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    Assembler
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_6_4" >
        
          
          <label class="md-nav__link" for="__nav_1_6_4" id="__nav_1_6_4_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    
  
    Annotator
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_1_6_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_6_4">
            <span class="md-nav__icon md-icon"></span>
            
  
    Annotator
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia/annotator/prokka/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    Prokka
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia/annotator/bakta/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    Bakta
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia/sketcher/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    Sketcher
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia/sequence-typing/mlst/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    Sequence Typing
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia/antimicrobial-resistance/amrfinderplus/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    Antimicrobial Resistance
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia/merlin/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    Merlin
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  <span class="md-ellipsis">
    
  
    Full Guide
  

    
  </span>
  
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    
  
    Full Guide
  

    
  </span>
  
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#step-1-gather" class="md-nav__link">
    <span class="md-ellipsis">
      
        Step 1 - Gather
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 1 - Gather">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#outputs" class="md-nav__link">
    <span class="md-ellipsis">
      
        Outputs
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Outputs">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#merged-results" class="md-nav__link">
    <span class="md-ellipsis">
      
        Merged Results
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gather" class="md-nav__link">
    <span class="md-ellipsis">
      
        gather
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#failed-quality-checks" class="md-nav__link">
    <span class="md-ellipsis">
      
        Failed Quality Checks
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#parameters" class="md-nav__link">
    <span class="md-ellipsis">
      
        Parameters
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Parameters">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#required" class="md-nav__link">
    <span class="md-ellipsis">
      
        Required
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#dataset" class="md-nav__link">
    <span class="md-ellipsis">
      
        Dataset
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#citations" class="md-nav__link">
    <span class="md-ellipsis">
      
        Citations
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-2-qc" class="md-nav__link">
    <span class="md-ellipsis">
      
        Step 2 - QC
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 2 - QC">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#outputs_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        Outputs
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Outputs">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#quality-control" class="md-nav__link">
    <span class="md-ellipsis">
      
        Quality Control
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#failed-quality-checks_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        Failed Quality Checks
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#qc-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      
        QC Parameters
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#citations_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        Citations
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-3-assembler" class="md-nav__link">
    <span class="md-ellipsis">
      
        Step 3 - Assembler
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 3 - Assembler">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#outputs_2" class="md-nav__link">
    <span class="md-ellipsis">
      
        Outputs
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Outputs">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#merged-results_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        Merged Results
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#dragonflye" class="md-nav__link">
    <span class="md-ellipsis">
      
        Dragonflye
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#shovill" class="md-nav__link">
    <span class="md-ellipsis">
      
        Shovill
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hybrid-assembly-unicycler" class="md-nav__link">
    <span class="md-ellipsis">
      
        Hybrid Assembly (Unicycler)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hybrid-assembly-short-read-polishing" class="md-nav__link">
    <span class="md-ellipsis">
      
        Hybrid Assembly (Short Read Polishing)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#failed-quality-checks_2" class="md-nav__link">
    <span class="md-ellipsis">
      
        Failed Quality Checks
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#assembler-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      
        Assembler Parameters
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#citations_2" class="md-nav__link">
    <span class="md-ellipsis">
      
        Citations
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-4-annotator" class="md-nav__link">
    <span class="md-ellipsis">
      
        Step 4 - Annotator
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 4 - Annotator">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#outputs_3" class="md-nav__link">
    <span class="md-ellipsis">
      
        Outputs
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Outputs">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#prokka" class="md-nav__link">
    <span class="md-ellipsis">
      
        Prokka
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#bakta" class="md-nav__link">
    <span class="md-ellipsis">
      
        Bakta
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#parameters_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        Parameters
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Parameters">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#prokka_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        Prokka
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#bakta-download" class="md-nav__link">
    <span class="md-ellipsis">
      
        Bakta Download
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#bakta_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        Bakta
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#citations_3" class="md-nav__link">
    <span class="md-ellipsis">
      
        Citations
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-5-sketcher" class="md-nav__link">
    <span class="md-ellipsis">
      
        Step 5 - Sketcher
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 5 - Sketcher">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#outputs_4" class="md-nav__link">
    <span class="md-ellipsis">
      
        Outputs
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Outputs">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#sketcher" class="md-nav__link">
    <span class="md-ellipsis">
      
        sketcher
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sketcher-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      
        Sketcher Parameters
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#citations_4" class="md-nav__link">
    <span class="md-ellipsis">
      
        Citations
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-6-sequence-typing" class="md-nav__link">
    <span class="md-ellipsis">
      
        Step 6 - Sequence Typing
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 6 - Sequence Typing">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#outputs_5" class="md-nav__link">
    <span class="md-ellipsis">
      
        Outputs
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Outputs">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#merged-results_2" class="md-nav__link">
    <span class="md-ellipsis">
      
        Merged Results
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#mlst" class="md-nav__link">
    <span class="md-ellipsis">
      
        mlst
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#parameters_2" class="md-nav__link">
    <span class="md-ellipsis">
      
        Parameters
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#citations_5" class="md-nav__link">
    <span class="md-ellipsis">
      
        Citations
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-7-antibiotic-resistance" class="md-nav__link">
    <span class="md-ellipsis">
      
        Step 7 - Antibiotic Resistance
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 7 - Antibiotic Resistance">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#outputs_6" class="md-nav__link">
    <span class="md-ellipsis">
      
        Outputs
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Outputs">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#merged-results_3" class="md-nav__link">
    <span class="md-ellipsis">
      
        Merged Results
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#amrfinder" class="md-nav__link">
    <span class="md-ellipsis">
      
        AMRFinder+
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#parameters_3" class="md-nav__link">
    <span class="md-ellipsis">
      
        Parameters
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#citations_6" class="md-nav__link">
    <span class="md-ellipsis">
      
        Citations
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-8-merlin" class="md-nav__link">
    <span class="md-ellipsis">
      
        Step 8 - Merlin
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 8 - Merlin">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#outputs_7" class="md-nav__link">
    <span class="md-ellipsis">
      
        Outputs
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Outputs">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#merged-results_4" class="md-nav__link">
    <span class="md-ellipsis">
      
        Merged Results
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#agrvate" class="md-nav__link">
    <span class="md-ellipsis">
      
        AgrVATE
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#clermontyping" class="md-nav__link">
    <span class="md-ellipsis">
      
        ClermonTyping
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ectyper" class="md-nav__link">
    <span class="md-ellipsis">
      
        ECTyper
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#emmtyper" class="md-nav__link">
    <span class="md-ellipsis">
      
        emmtyper
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hicap" class="md-nav__link">
    <span class="md-ellipsis">
      
        hicap
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hpsuissero" class="md-nav__link">
    <span class="md-ellipsis">
      
        HpsuisSero
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#genotyphi" class="md-nav__link">
    <span class="md-ellipsis">
      
        GenoTyphi
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#kleborate" class="md-nav__link">
    <span class="md-ellipsis">
      
        Kleborate
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#legsta" class="md-nav__link">
    <span class="md-ellipsis">
      
        legsta
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#lissero" class="md-nav__link">
    <span class="md-ellipsis">
      
        LisSero
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#mash" class="md-nav__link">
    <span class="md-ellipsis">
      
        Mash
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#meningotype" class="md-nav__link">
    <span class="md-ellipsis">
      
        meningotype
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ngmaster" class="md-nav__link">
    <span class="md-ellipsis">
      
        ngmaster
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#pasty" class="md-nav__link">
    <span class="md-ellipsis">
      
        pasty
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#pbptyper" class="md-nav__link">
    <span class="md-ellipsis">
      
        pbptyper
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#seqsero2" class="md-nav__link">
    <span class="md-ellipsis">
      
        SeqSero2
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#seroba" class="md-nav__link">
    <span class="md-ellipsis">
      
        Seroba
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#shigapass" class="md-nav__link">
    <span class="md-ellipsis">
      
        ShigaPass
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#shigatyper" class="md-nav__link">
    <span class="md-ellipsis">
      
        ShigaTyper
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#shigeifinder" class="md-nav__link">
    <span class="md-ellipsis">
      
        ShigEiFinder
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sistr" class="md-nav__link">
    <span class="md-ellipsis">
      
        SISTR
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#spatyper" class="md-nav__link">
    <span class="md-ellipsis">
      
        spaTyper
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ssuissero" class="md-nav__link">
    <span class="md-ellipsis">
      
        SsuisSero
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#staphopia-sccmec" class="md-nav__link">
    <span class="md-ellipsis">
      
        staphopia-sccmec
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#tbprofiler" class="md-nav__link">
    <span class="md-ellipsis">
      
        TBProfiler
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#mashdist" class="md-nav__link">
    <span class="md-ellipsis">
      
        mashdist
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#agrvate_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        AgrVATE
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ectyper_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        ECTyper
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#emmtyper_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        emmtyper
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hicap_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        hicap
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#genotyphi_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        GenoTyphi
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#kleborate_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        Kleborate
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#legsta_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        legsta
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#lissero_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        LisSero
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#meningotype_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        meningotype
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ngmaster_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        ngmaster
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#pasty_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        pasty
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#pbptyper_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        pbptyper
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#seqsero2_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        SeqSero2
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sistr_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        SISTR
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#spatyper_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        spaTyper
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#staphopia-sccmec_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        staphopia-sccmec
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#citations_7" class="md-nav__link">
    <span class="md-ellipsis">
      
        Citations
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#additional-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      
        Additional Parameters
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Additional Parameters">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#optional" class="md-nav__link">
    <span class="md-ellipsis">
      
        Optional
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#max-job-request" class="md-nav__link">
    <span class="md-ellipsis">
      
        Max Job Request
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#nextflow-configuration" class="md-nav__link">
    <span class="md-ellipsis">
      
        Nextflow Configuration
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#institutional-config-options" class="md-nav__link">
    <span class="md-ellipsis">
      
        Institutional config options
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#nextflow-profile" class="md-nav__link">
    <span class="md-ellipsis">
      
        Nextflow Profile
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#helpful" class="md-nav__link">
    <span class="md-ellipsis">
      
        Helpful
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../changelog/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    Changelog
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" >
        
          
          <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    
  
    Bactopia Tools
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            
  
    Bactopia Tools
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    Introduction
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2_2" >
        
          
          <label class="md-nav__link" for="__nav_2_2" id="__nav_2_2_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    
  
    Subworkflows
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_2_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_2_2">
            <span class="md-nav__icon md-icon"></span>
            
  
    Subworkflows
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/ariba/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    ariba
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/bakta/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    bakta
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/eggnog/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    eggnog
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/gtdb/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    gtdb
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/mashtree/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    mashtree
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/merlin/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    merlin
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/pangenome/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    pangenome
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/scrubber/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    scrubber
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/snippy/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    snippy
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/staphtyper/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    staphtyper
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/tbprofiler/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    tbprofiler
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2_3" >
        
          
          <label class="md-nav__link" for="__nav_2_3" id="__nav_2_3_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    
  
    Modules
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_2_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_2_3">
            <span class="md-nav__icon md-icon"></span>
            
  
    Modules
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/abricate/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    abricate
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/abritamr/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    abritamr
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/agrvate/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    agrvate
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/amrfinderplus/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    amrfinderplus
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/blastn/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    blastn
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/blastp/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    blastp
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/blastx/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    blastx
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/bracken/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    bracken
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/btyper3/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    btyper3
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/busco/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    busco
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/checkm/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    checkm
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/clermontyping.md" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    clermontyping
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/defensefinder/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    defensefinder
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/ectyper/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    ectyper
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/emmtyper/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    emmtyper
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/fastani/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    fastani
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/gamma/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    gamma
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/genotyphi/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    genotyphi
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/hicap/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    hicap
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/hpsuissero/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    hpsuissero
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/ismapper/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    ismapper
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/kleborate/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    kleborate
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/kraken2/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    kraken2
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/legsta/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    legsta
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/lissero/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    lissero
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/mashdist/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    mashdist
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/mcroni/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    mcroni
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/meningotype/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    meningotype
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/midas/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    midas
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/mlst/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    mlst
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/mobsuite/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    mobsuite
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/mykrobe/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    mykrobe
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/ngmaster/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    ngmaster
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/pasty/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    pasty
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/pbptyper/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    pbptyper
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/phispy/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    phispy
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/plasmidfinder/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    plasmidfinder
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/pneumocat/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    pneumocat
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/quast/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    quast
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/rgi/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    rgi
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/sccmec/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    sccmec
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/seqsero2/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    seqsero2
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/seroba/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    seroba
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/shigapass/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    shigapass
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/shigatyper/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    shigatyper
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/shigeifinder/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    shigeifinder
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/sistr/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    sistr
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/spatyper/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    spatyper
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/ssuissero/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    ssuissero
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/staphopiasccmec/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    staphopiasccmec
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/stecfinder/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    stecfinder
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/tblastn/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    tblastn
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/tblastx/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    tblastx
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" >
        
          
          <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    
  
    Impact & Outreach
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            
  
    Impact & Outreach
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../impact-and-outreach/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    Impact 
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../impact-and-outreach/acknowledgements/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    Acknowledgements
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../impact-and-outreach/citations/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    Citations
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../impact-and-outreach/enhancements/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    Enhancements to OSS
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../impact-and-outreach/presentations/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    Presentations
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" >
        
          
          <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    
  
    Blog
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            
  
    Blog
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
    
  
  
    <li class="md-nav__item">
      <a href="../blog/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    Bactopia Blog
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_2" >
        
          
          <label class="md-nav__link" for="__nav_4_2" id="__nav_4_2_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    
  
    Categories
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4_2">
            <span class="md-nav__icon md-icon"></span>
            
  
    Categories
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
    
  
  
    <li class="md-nav__item">
      <a href="../blog/category/community/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    Community
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    
  
  
    <li class="md-nav__item">
      <a href="../blog/category/tutorial/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    
  
    Tutorial
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#step-1-gather" class="md-nav__link">
    <span class="md-ellipsis">
      
        Step 1 - Gather
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 1 - Gather">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#outputs" class="md-nav__link">
    <span class="md-ellipsis">
      
        Outputs
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Outputs">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#merged-results" class="md-nav__link">
    <span class="md-ellipsis">
      
        Merged Results
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gather" class="md-nav__link">
    <span class="md-ellipsis">
      
        gather
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#failed-quality-checks" class="md-nav__link">
    <span class="md-ellipsis">
      
        Failed Quality Checks
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#parameters" class="md-nav__link">
    <span class="md-ellipsis">
      
        Parameters
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Parameters">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#required" class="md-nav__link">
    <span class="md-ellipsis">
      
        Required
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#dataset" class="md-nav__link">
    <span class="md-ellipsis">
      
        Dataset
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#citations" class="md-nav__link">
    <span class="md-ellipsis">
      
        Citations
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-2-qc" class="md-nav__link">
    <span class="md-ellipsis">
      
        Step 2 - QC
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 2 - QC">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#outputs_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        Outputs
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Outputs">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#quality-control" class="md-nav__link">
    <span class="md-ellipsis">
      
        Quality Control
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#failed-quality-checks_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        Failed Quality Checks
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#qc-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      
        QC Parameters
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#citations_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        Citations
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-3-assembler" class="md-nav__link">
    <span class="md-ellipsis">
      
        Step 3 - Assembler
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 3 - Assembler">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#outputs_2" class="md-nav__link">
    <span class="md-ellipsis">
      
        Outputs
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Outputs">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#merged-results_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        Merged Results
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#dragonflye" class="md-nav__link">
    <span class="md-ellipsis">
      
        Dragonflye
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#shovill" class="md-nav__link">
    <span class="md-ellipsis">
      
        Shovill
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hybrid-assembly-unicycler" class="md-nav__link">
    <span class="md-ellipsis">
      
        Hybrid Assembly (Unicycler)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hybrid-assembly-short-read-polishing" class="md-nav__link">
    <span class="md-ellipsis">
      
        Hybrid Assembly (Short Read Polishing)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#failed-quality-checks_2" class="md-nav__link">
    <span class="md-ellipsis">
      
        Failed Quality Checks
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#assembler-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      
        Assembler Parameters
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#citations_2" class="md-nav__link">
    <span class="md-ellipsis">
      
        Citations
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-4-annotator" class="md-nav__link">
    <span class="md-ellipsis">
      
        Step 4 - Annotator
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 4 - Annotator">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#outputs_3" class="md-nav__link">
    <span class="md-ellipsis">
      
        Outputs
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Outputs">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#prokka" class="md-nav__link">
    <span class="md-ellipsis">
      
        Prokka
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#bakta" class="md-nav__link">
    <span class="md-ellipsis">
      
        Bakta
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#parameters_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        Parameters
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Parameters">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#prokka_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        Prokka
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#bakta-download" class="md-nav__link">
    <span class="md-ellipsis">
      
        Bakta Download
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#bakta_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        Bakta
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#citations_3" class="md-nav__link">
    <span class="md-ellipsis">
      
        Citations
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-5-sketcher" class="md-nav__link">
    <span class="md-ellipsis">
      
        Step 5 - Sketcher
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 5 - Sketcher">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#outputs_4" class="md-nav__link">
    <span class="md-ellipsis">
      
        Outputs
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Outputs">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#sketcher" class="md-nav__link">
    <span class="md-ellipsis">
      
        sketcher
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sketcher-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      
        Sketcher Parameters
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#citations_4" class="md-nav__link">
    <span class="md-ellipsis">
      
        Citations
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-6-sequence-typing" class="md-nav__link">
    <span class="md-ellipsis">
      
        Step 6 - Sequence Typing
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 6 - Sequence Typing">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#outputs_5" class="md-nav__link">
    <span class="md-ellipsis">
      
        Outputs
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Outputs">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#merged-results_2" class="md-nav__link">
    <span class="md-ellipsis">
      
        Merged Results
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#mlst" class="md-nav__link">
    <span class="md-ellipsis">
      
        mlst
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#parameters_2" class="md-nav__link">
    <span class="md-ellipsis">
      
        Parameters
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#citations_5" class="md-nav__link">
    <span class="md-ellipsis">
      
        Citations
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-7-antibiotic-resistance" class="md-nav__link">
    <span class="md-ellipsis">
      
        Step 7 - Antibiotic Resistance
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 7 - Antibiotic Resistance">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#outputs_6" class="md-nav__link">
    <span class="md-ellipsis">
      
        Outputs
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Outputs">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#merged-results_3" class="md-nav__link">
    <span class="md-ellipsis">
      
        Merged Results
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#amrfinder" class="md-nav__link">
    <span class="md-ellipsis">
      
        AMRFinder+
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#parameters_3" class="md-nav__link">
    <span class="md-ellipsis">
      
        Parameters
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#citations_6" class="md-nav__link">
    <span class="md-ellipsis">
      
        Citations
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-8-merlin" class="md-nav__link">
    <span class="md-ellipsis">
      
        Step 8 - Merlin
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 8 - Merlin">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#outputs_7" class="md-nav__link">
    <span class="md-ellipsis">
      
        Outputs
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Outputs">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#merged-results_4" class="md-nav__link">
    <span class="md-ellipsis">
      
        Merged Results
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#agrvate" class="md-nav__link">
    <span class="md-ellipsis">
      
        AgrVATE
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#clermontyping" class="md-nav__link">
    <span class="md-ellipsis">
      
        ClermonTyping
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ectyper" class="md-nav__link">
    <span class="md-ellipsis">
      
        ECTyper
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#emmtyper" class="md-nav__link">
    <span class="md-ellipsis">
      
        emmtyper
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hicap" class="md-nav__link">
    <span class="md-ellipsis">
      
        hicap
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hpsuissero" class="md-nav__link">
    <span class="md-ellipsis">
      
        HpsuisSero
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#genotyphi" class="md-nav__link">
    <span class="md-ellipsis">
      
        GenoTyphi
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#kleborate" class="md-nav__link">
    <span class="md-ellipsis">
      
        Kleborate
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#legsta" class="md-nav__link">
    <span class="md-ellipsis">
      
        legsta
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#lissero" class="md-nav__link">
    <span class="md-ellipsis">
      
        LisSero
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#mash" class="md-nav__link">
    <span class="md-ellipsis">
      
        Mash
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#meningotype" class="md-nav__link">
    <span class="md-ellipsis">
      
        meningotype
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ngmaster" class="md-nav__link">
    <span class="md-ellipsis">
      
        ngmaster
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#pasty" class="md-nav__link">
    <span class="md-ellipsis">
      
        pasty
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#pbptyper" class="md-nav__link">
    <span class="md-ellipsis">
      
        pbptyper
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#seqsero2" class="md-nav__link">
    <span class="md-ellipsis">
      
        SeqSero2
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#seroba" class="md-nav__link">
    <span class="md-ellipsis">
      
        Seroba
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#shigapass" class="md-nav__link">
    <span class="md-ellipsis">
      
        ShigaPass
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#shigatyper" class="md-nav__link">
    <span class="md-ellipsis">
      
        ShigaTyper
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#shigeifinder" class="md-nav__link">
    <span class="md-ellipsis">
      
        ShigEiFinder
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sistr" class="md-nav__link">
    <span class="md-ellipsis">
      
        SISTR
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#spatyper" class="md-nav__link">
    <span class="md-ellipsis">
      
        spaTyper
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ssuissero" class="md-nav__link">
    <span class="md-ellipsis">
      
        SsuisSero
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#staphopia-sccmec" class="md-nav__link">
    <span class="md-ellipsis">
      
        staphopia-sccmec
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#tbprofiler" class="md-nav__link">
    <span class="md-ellipsis">
      
        TBProfiler
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#mashdist" class="md-nav__link">
    <span class="md-ellipsis">
      
        mashdist
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#agrvate_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        AgrVATE
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ectyper_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        ECTyper
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#emmtyper_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        emmtyper
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hicap_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        hicap
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#genotyphi_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        GenoTyphi
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#kleborate_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        Kleborate
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#legsta_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        legsta
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#lissero_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        LisSero
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#meningotype_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        meningotype
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ngmaster_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        ngmaster
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#pasty_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        pasty
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#pbptyper_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        pbptyper
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#seqsero2_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        SeqSero2
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sistr_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        SISTR
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#spatyper_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        spaTyper
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#staphopia-sccmec_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        staphopia-sccmec
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#citations_7" class="md-nav__link">
    <span class="md-ellipsis">
      
        Citations
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#additional-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      
        Additional Parameters
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Additional Parameters">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#optional" class="md-nav__link">
    <span class="md-ellipsis">
      
        Optional
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#max-job-request" class="md-nav__link">
    <span class="md-ellipsis">
      
        Max Job Request
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#nextflow-configuration" class="md-nav__link">
    <span class="md-ellipsis">
      
        Nextflow Configuration
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#institutional-config-options" class="md-nav__link">
    <span class="md-ellipsis">
      
        Institutional config options
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#nextflow-profile" class="md-nav__link">
    <span class="md-ellipsis">
      
        Nextflow Profile
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#helpful" class="md-nav__link">
    <span class="md-ellipsis">
      
        Helpful
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              
              <article class="md-content__inner md-typeset">
                
                  

<!--
  Copyright (c) 2016-2023 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
-->

<!-- Actions -->


  <!-- Edit button -->
  
    <a href="https://github.com/bactopia/bactopia.github.io/edit/master/docs/full-guide.md" title="Edit on GitHub" class="md-content__button md-icon">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M10 20H6V4h7v5h5v3.1l2-2V8l-6-6H6c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h4v-2m10.2-7c.1 0 .3.1.4.2l1.3 1.3c.2.2.2.6 0 .8l-1 1-2.1-2.1 1-1c.1-.1.2-.2.4-.2m0 3.9L14.1 23H12v-2.1l6.1-6.1 2.1 2.1Z"/></svg>
    </a>
  

  <!-- View button -->
  


  <h1>Full Guide</h1>

<p>Bactopia is a complete pipeline for the analysis of bacterial genomes, which includes
more than 150 bioinformatics tools. In this section, each step of the pipeline will
be described in detail. This includes the input data, output data, and the parameters
for each step.</p>
<p><a class="zoom" href="assets/bactopia-workflow.png">
<img alt="Bactopia Workflow" src="../assets/bactopia-workflow.png" />
</a></p>
<p>Looking at the workflow overview below, it might not look like much is happening, but I can
assure you that a lot is going on. The workflow is broken down into 8 steps, which are:</p>
<ol>
<li><strong>Gather</strong> - Collect all the data in one place</li>
<li><strong>QC</strong> - Quality control of the data</li>
<li><strong>Assembler</strong> - Assemble the reads into contigs</li>
<li><strong>Annotator</strong> - Annotate the contigs</li>
<li><strong>Sketcher</strong> - Create a sketch of the contigs, and query databases</li>
<li><strong>Sequence Typing</strong> - Determine the sequence type of the contigs</li>
<li><strong>Antibiotic Resistance</strong> - Determine the antibiotic resistance of the contigs and proteins</li>
<li><strong>Merlin</strong> - Automatically run species-specific tools based on distance</li>
</ol>
<p>This guide is meant to be extensive, so it will be very detailed. If you are looking for
a guide to get started quickly, please check out the <a href="../beginners-guide/">Beginner's Guide</a>.</p>
<p>Otherwise, let's get started!</p>
<h2 id="step-1-gather">Step 1 - Gather<a class="headerlink" href="#step-1-gather" title="Permanent link">&para;</a></h2>
<p>The main purpose of the <code>gather</code> step is to get all the samples into a single place. This
includes downloading samples from ENA/SRA or NCBI Assembly. The tools used are:</p>
<table>
<thead>
<tr>
<th>Tool</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="https://www.niehs.nih.gov/research/resources/software/biostatistics/art/index.cfm">art</a></td>
<td>For simulating error-free reads for an input assembly</td>
</tr>
<tr>
<td><a href="https://github.com/rpetit3/fastq-dl">fastq-dl</a></td>
<td>Downloading FASTQ files from ENA/SRA</td>
</tr>
<tr>
<td><a href="https://github.com/kblin/ncbi-genome-download">ncbi-genome-download</a></td>
<td>Downloading FASTA files from NCBI Assembly</td>
</tr>
</tbody>
</table>
<p>This <code>gather</code> step also does basic QC checks to help prevent downstream failures.</p>
<h3 id="outputs">Outputs<a class="headerlink" href="#outputs" title="Permanent link">&para;</a></h3>
<h4 id="merged-results">Merged Results<a class="headerlink" href="#merged-results" title="Permanent link">&para;</a></h4>
<p>Below are results that are concatenated into a single file.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>meta.tsv</td>
<td>A tab-delimited file with bactopia metadata for all samples</td>
</tr>
</tbody>
</table>
<h4 id="gather">gather<a class="headerlink" href="#gather" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results from the <code>gather</code> subworkflow.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>-meta.tsv</td>
<td>A tab-delimited file with bactopia metadata for each sample</td>
</tr>
</tbody>
</table>
<h4 id="failed-quality-checks">Failed Quality Checks<a class="headerlink" href="#failed-quality-checks" title="Permanent link">&para;</a></h4>
<p>Built into Bactopia are few basic quality checks to help prevent downstream failures.
If a sample fails one of these checks, it will be excluded from further analysis. By
excluding these samples, complete pipeline failures are prevented.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>-gzip-error.txt</td>
<td>Sample failed Gzip checks and excluded from further analysis</td>
</tr>
<tr>
<td>-low-basepair-proportion-error.txt</td>
<td>Sample failed basepair proportion checks and excluded from further analysis</td>
</tr>
<tr>
<td>-low-read-count-error.txt</td>
<td>Sample failed read count checks and excluded from further analysis</td>
</tr>
<tr>
<td>-low-sequence-depth-error.txt</td>
<td>Sample failed sequenced basepair checks and excluded from further analysis</td>
</tr>
</tbody>
</table>
<div class="admonition info">
<p class="admonition-title">Poor samples are excluded to prevent downstream failures</p>
<p>Samples that fail any of the QC checks will be excluded from further analysis.
Those samples will generate a <code>*-error.txt</code> file with the error message. Excluding
these samples prevents downstream failures that cause the whole workflow to fail.</p>
</div>
<details class="warning">
<summary>Example Error: Input FASTQ(s) failed Gzip checks</summary>
<p>If input FASTQ(s) fail to pass Gzip test, the sample will be excluded from
further analysis.</p>
<p><strong>Example Text from &lt;SAMPLE_NAME&gt;-gzip-error.txt</strong><br />
<em>&lt;SAMPLE_NAME&gt; FASTQs failed Gzip tests. Please check the input FASTQs. Further
analysis is discontinued.</em></p>
</details>
<details class="warning">
<summary>Example Error: Input FASTQs have disproportionate number of reads</summary>
<p>If input FASTQ(s) for a sample have disproportionately different number of reads
between the two pairs, the sample will be excluded from further analysis. You can
adjust this minimum read count using the <code>--min_proportion</code> parameter.</p>
<p><strong>Example Text from &lt;SAMPLE_NAME&gt;-low-basepair-proportion-error.txt</strong><br />
<em>&lt;SAMPLE_NAME&gt; FASTQs failed to meet the minimum shared basepairs (<code>X``). They
shared</code>Y<code>basepairs, with R1 having</code>A<code>bp and R2 having</code>B` bp. Further
analysis is discontinued.</em></p>
</details>
<details class="warning">
<summary>Example Error: Input FASTQ(s) has too few reads</summary>
<p>If input FASTQ(s) for a sample have less than the minimum required reads, the
sample will be excluded from further analysis. You can adjust this minimum read
count using the <code>--min_reads</code> parameter.</p>
<p><strong>Example Text from &lt;SAMPLE_NAME&gt;-low-read-count-error.txt</strong><br />
<em>&lt;SAMPLE_NAME&gt; FASTQ(s) contain <code>X</code> total reads. This does not exceed the required
minimum <code>Y</code> read count. Further analysis is discontinued.</em></p>
</details>
<details class="warning">
<summary>Example Error: Input FASTQ(s) has too little sequenced basepairs</summary>
<p>If input FASTQ(s) for a sample fails to meet the minimum number of sequenced
basepairs, the sample will be excluded from further analysis. You can
adjust this minimum read count using the <code>--min_basepairs</code> parameter.</p>
<p><strong>Example Text from &lt;SAMPLE_NAME&gt;-low-sequence-depth-error.txt</strong><br />
<em>&lt;SAMPLE_NAME&gt; FASTQ(s) contain <code>X</code> total basepairs. This does not exceed the
required minimum <code>Y</code> bp. Further analysis is discontinued.</em></p>
</details>
<h3 id="parameters">Parameters<a class="headerlink" href="#parameters" title="Permanent link">&para;</a></h3>
<h4 id="required"><i class="fa-xl fas fa-terminal"></i> Required<a class="headerlink" href="#required" title="Permanent link">&para;</a></h4>
<p>The following parameters are how you will provide either local or remote samples to be processed by Bactopia.</p>
<table>
<thead>
<tr>
<th style="text-align: left;">Parameter</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-file-alt"></i><code>--samples</code></td>
<td>A FOFN (via bactopia prepare) with sample names and paths to FASTQ/FASTAs to process <br/><strong>Type:</strong> <code>string</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-file-archive"></i><code>--r1</code></td>
<td>First set of compressed (gzip) Illumina paired-end FASTQ reads (requires --r2 and --sample) <br/><strong>Type:</strong> <code>string</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-file-archive"></i><code>--r2</code></td>
<td>Second set of compressed (gzip) Illumina paired-end FASTQ reads (requires --r1 and --sample) <br/><strong>Type:</strong> <code>string</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-file-archive"></i><code>--se</code></td>
<td>Compressed (gzip) Illumina single-end FASTQ reads  (requires --sample) <br/><strong>Type:</strong> <code>string</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-level-up"></i><code>--ont</code></td>
<td>Compressed (gzip) Oxford Nanopore FASTQ reads  (requires --sample) <br/><strong>Type:</strong> <code>string</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-level-up"></i><code>--hybrid</code></td>
<td>Create hybrid assembly using Unicycler.  (requires --r1, --r2, --ont and --sample) <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-level-up"></i><code>--short_polish</code></td>
<td>Create hybrid assembly from long-read assembly and short read polishing.  (requires --r1, --r2, --ont and --sample) <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-file"></i><code>--sample</code></td>
<td>Sample name to use for the input sequences <br/><strong>Type:</strong> <code>string</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-file-alt"></i><code>--accessions</code></td>
<td>A file containing ENA/SRA Experiment accessions or NCBI Assembly accessions to processed <br/><strong>Type:</strong> <code>string</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-font"></i><code>--accession</code></td>
<td>Sample name to use for the input sequences <br/><strong>Type:</strong> <code>string</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-file-archive"></i><code>--assembly</code></td>
<td>A assembled genome in compressed FASTA format. (requires --sample) <br/><strong>Type:</strong> <code>string</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-level-up"></i><code>--check_samples</code></td>
<td>Validate the input FOFN provided by --samples <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
</tbody>
</table>
<h4 id="dataset"><i class="fa-xl fas fa-exclamation-circle"></i> Dataset<a class="headerlink" href="#dataset" title="Permanent link">&para;</a></h4>
<p>Define where the pipeline should find input data and save output data.</p>
<table>
<thead>
<tr>
<th style="text-align: left;">Parameter</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-bacterium"></i><code>--species</code></td>
<td>Name of species for species-specific dataset to use <br/><strong>Type:</strong> <code>string</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-print"></i><code>--ask_merlin</code></td>
<td>Ask Merlin to execute species specific Bactopia tools based on Mash distances <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-angle-double-down"></i><code>--coverage</code></td>
<td>Reduce samples to a given coverage, requires a genome size <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>100</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-arrows-alt-h"></i><code>--genome_size</code></td>
<td>Expected genome size (bp) for all samples, required for read error correction and read subsampling <br/><strong>Type:</strong> <code>string</code>, <strong>Default:</strong> <code>0</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-print"></i><code>--use_bakta</code></td>
<td>Use Bakta for annotation, instead of Prokka <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
</tbody>
</table>
<h3 id="citations">Citations<a class="headerlink" href="#citations" title="Permanent link">&para;</a></h3>
<p>If you use Bactopia and <code>gather</code> results in your analysis, please cite the following.</p>
<ul>
<li>
<p><a href="https://bactopia.github.io/">Bactopia</a><br />
    Petit III RA, Read TD <a href="https://doi.org/10.1128/mSystems.00190-20">Bactopia - a flexible pipeline for complete analysis of bacterial genomes.</a> <em>mSystems</em> 5 (2020)</p>
</li>
<li>
<p><a href="https://www.niehs.nih.gov/research/resources/software/biostatistics/art/index.cfm">ART</a><br />
    Huang W, Li L, Myers JR, Marth GT <a href="http://dx.doi.org/10.1093/bioinformatics/btr708">ART: a next-generation sequencing read simulator.</a> <em>Bioinformatics</em> 28, 593–594 (2012)</p>
</li>
<li>
<p><a href="https://github.com/rpetit3/fastq-dl">fastq-dl</a><br />
    Petit III RA <a href="https://github.com/rpetit3/fastq-dl">fastq-dl: Download FASTQ files from SRA or ENA repositories.</a> (GitHub)</p>
</li>
<li>
<p><a href="https://github.com/rpetit3/fastq-scan">fastq-scan</a><br />
    Petit III RA <a href="https://github.com/rpetit3/fastq-scan">fastq-scan: generate summary statistics of input FASTQ sequences.</a> (GitHub)</p>
</li>
<li>
<p><a href="https://github.com/kblin/ncbi-genome-download">ncbi-genome-download</a><br />
    Blin K <a href="https://github.com/kblin/ncbi-genome-download">ncbi-genome-download: Scripts to download genomes from the NCBI FTP servers</a> (GitHub)</p>
</li>
<li>
<p><a href="https://zlib.net/pigz/">Pigz</a><br />
    Adler M. <a href="https://zlib.net/pigz/">pigz: A parallel implementation of gzip for modern multi-processor, multi-core machines.</a> <em>Jet Propulsion Laboratory</em> (2015)</p>
</li>
</ul>
<h2 id="step-2-qc">Step 2 - QC<a class="headerlink" href="#step-2-qc" title="Permanent link">&para;</a></h2>
<p>The <code>qc</code> module uses  a variety of tools to perform quality control on Illumina and
Oxford Nanopore reads. The tools used are:</p>
<table>
<thead>
<tr>
<th>Tool</th>
<th>Technology</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="https://jgi.doe.gov/data-and-tools/bbtools/">bbtools</a></td>
<td>Illumina</td>
<td>A suite of tools for manipulating reads</td>
</tr>
<tr>
<td><a href="https://github.com/OpenGene/fastp">fastp</a></td>
<td>Illumina</td>
<td>A tool designed to provide fast all-in-one preprocessing for FastQ files</td>
</tr>
<tr>
<td><a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">fastqc</a></td>
<td>Illumina</td>
<td>A quality control tool for high throughput sequence data</td>
</tr>
<tr>
<td><a href="https://github.com/rpetit3/fastq-scan">fastq_scan</a></td>
<td>Nanopore</td>
<td>A tool for quickly scanning FASTQ files</td>
</tr>
<tr>
<td><a href="https://github.com/mourisl/Lighter">lighter</a></td>
<td>Illumina</td>
<td>A tool for correcting sequencing errors in Illumina reads</td>
</tr>
<tr>
<td><a href="https://github.com/wdecoster/NanoPlot">NanoPlot</a></td>
<td>Nanopore</td>
<td>A tool for plotting long read sequencing data</td>
</tr>
<tr>
<td><a href="https://github.com/esteinig/nanoq">nanoq</a></td>
<td>Nanopore</td>
<td>A tool for calculating quality metrics for Oxford Nanopore reads</td>
</tr>
<tr>
<td><a href="https://github.com/rrwick/Porechop">porechop</a></td>
<td>Nanopore</td>
<td>A tool for removing adapters from Oxford Nanopore reads</td>
</tr>
<tr>
<td><a href="https://github.com/mbhall88/rasusa">rasusa</a></td>
<td>Nanopore</td>
<td>Randomly subsample sequencing reads to a specified coverage</td>
</tr>
</tbody>
</table>
<p>Similar to the <code>gather</code> step, the <code>qc</code> step will also stop samples that fail to meet
basic QC checks from continuing downstream.</p>
<h3 id="outputs_1">Outputs<a class="headerlink" href="#outputs_1" title="Permanent link">&para;</a></h3>
<h4 id="quality-control">Quality Control<a class="headerlink" href="#quality-control" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results from <code>qc</code> subworkflow.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>&lt;SAMPLE_NAME&gt;.fastq.gz</td>
<td>A gzipped FASTQ file containing the cleaned Illumina single-end, or Oxford Nanopore reads</td>
</tr>
<tr>
<td>&lt;SAMPLE_NAME&gt;_R{1|2}.fastq.gz</td>
<td>A gzipped FASTQ file containing the cleaned Illumina paired-end reads</td>
</tr>
<tr>
<td>&lt;SAMPLE_NAME&gt;-{final|original}.json</td>
<td>A JSON file containing the QC results generated by <a href="https://github.com/rpetit3/fastq-scan">fastq-scan</a></td>
</tr>
<tr>
<td>&lt;SAMPLE_NAME&gt;-{final|original}_fastqc.html</td>
<td>(Illumina Only) A HTML report of the QC results generated by <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">fastqc</a></td>
</tr>
<tr>
<td>&lt;SAMPLE_NAME&gt;-{final|original}_fastqc.zip</td>
<td>(Illumina Only) A zip file containing the complete set of <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">fastqc</a> results</td>
</tr>
<tr>
<td>&lt;SAMPLE_NAME&gt;-{final|original}_fastp.json</td>
<td>(Illumina Only) A JSON file containing the QC results generated by <a href="https://github.com/OpenGene/fastp">fastp</a></td>
</tr>
<tr>
<td>&lt;SAMPLE_NAME&gt;-{final|original}_fastp.html</td>
<td>(Illumina Only) A HTML report of the QC results generated by <a href="https://github.com/OpenGene/fastp">fastp</a></td>
</tr>
<tr>
<td>&lt;SAMPLE_NAME&gt;-{final|original}_NanoPlot-report.html</td>
<td>(ONT Only) A HTML report of the QC results generated by <a href="https://github.com/wdecoster/NanoPlot">NanoPlot</a></td>
</tr>
<tr>
<td>&lt;SAMPLE_NAME&gt;-{final|original}_NanoPlot.tar.gz</td>
<td>(ONT Only) A tarball containing the complete set of <a href="https://github.com/wdecoster/NanoPlot">NanoPlot</a> results</td>
</tr>
</tbody>
</table>
<h4 id="failed-quality-checks_1">Failed Quality Checks<a class="headerlink" href="#failed-quality-checks_1" title="Permanent link">&para;</a></h4>
<p>Built into Bactopia are few basic quality checks to help prevent downstream failures.
If a sample fails one of these checks, it will be excluded from further analysis. By
excluding these samples, complete pipeline failures are prevented.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>.error-fastq.gz</td>
<td>A gzipped FASTQ file of Illumina Single-End or Oxford Nanopore reads that failed QC</td>
</tr>
<tr>
<td>_R{1|2}.error-fastq.gz</td>
<td>A gzipped FASTQ file of Illumina Single-End or Oxford Nanopore reads that failed QC</td>
</tr>
<tr>
<td>-low-read-count-error.txt</td>
<td>Sample failed read count checks and excluded from further analysis</td>
</tr>
<tr>
<td>-low-sequence-coverage-error.txt</td>
<td>Sample failed sequenced coverage checks and excluded from further analysis</td>
</tr>
<tr>
<td>-low-sequence-depth-error.txt</td>
<td>Sample failed sequenced basepair checks and excluded from further analysis</td>
</tr>
</tbody>
</table>
<div class="admonition info">
<p class="admonition-title">Poor samples are excluded to prevent downstream failures</p>
<p>Samples that fail any of the QC checks will be excluded from further analysis.
Those samples will generate a <code>*-error.txt</code> file with the error message. Excluding
these samples prevents downstream failures that cause the whole workflow to fail.</p>
</div>
<details class="warning">
<summary>Example Error: After QC, too few reads remain</summary>
<p>If after cleaning reads, a sample has less than the minimum required reads, the
sample will be excluded from further analysis. You can adjust this minimum read
count using the <code>--min_reads</code> parameter.</p>
<p><strong>Example Text from &lt;SAMPLE_NAME&gt;-low-read-count-error.txt</strong><br />
<em>&lt;SAMPLE_NAME&gt; FASTQ(s) contain <code>X</code> total reads. This does not exceed the required
minimum <code>Y</code> read count. Further analysis is discontinued.</em></p>
</details>
<details class="warning">
<summary>Example Error: After QC, too little sequence coverage remains</summary>
<p>If after cleaning reads, a sample has failed to meet the minimum sequence 
coverage required, the sample will be excluded from further analysis. You can
adjust this minimum read count using the <code>--min_coverage</code> parameter.</p>
<p><strong>Note:</strong> This check is only performed when a genome size is available.</p>
<p><strong>Example Text from &lt;SAMPLE_NAME&gt;-low-sequence-coverage-error.txt</strong><br />
<em>After QC, &lt;SAMPLE_NAME&gt; FASTQ(s) contain <code>X</code> total basepairs. This does not
exceed the required minimum <code>Y</code> bp (<code>Z</code>x coverage). Further analysis is
discontinued.</em></p>
</details>
<details class="warning">
<summary>Example Error: After QC, too little sequenced basepairs remain</summary>
<p>If after cleaning reads, a sample has failed to meet the minimum number of
sequenced basepairs, the sample will be excluded from further analysis. You can
adjust this minimum read count using the <code>--min_basepairs</code> parameter.</p>
<p><strong>Example Text from &lt;SAMPLE_NAME&gt;-low-sequence-depth-error.txt</strong><br />
<em>&lt;SAMPLE_NAME&gt; FASTQ(s) contain <code>X</code> total basepairs. This does not exceed the
required minimum <code>Y</code> bp. Further analysis is discontinued.</em></p>
</details>
<h3 id="qc-parameters"><i class="fa-xl fas fa-exclamation-circle"></i> QC Parameters<a class="headerlink" href="#qc-parameters" title="Permanent link">&para;</a></h3>
<table>
<thead>
<tr>
<th style="text-align: left;">Parameter</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-fast-forward"></i><code>--use_bbmap</code></td>
<td>Illumina reads will be QC'd using BBMap <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-fast-forward"></i><code>--use_porechop</code></td>
<td>Use Porechop to remove adapters from ONT reads <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-fast-forward"></i><code>--skip_qc</code></td>
<td>The QC step will be skipped and it will be assumed the inputs sequences have already been QCed. <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-fast-forward"></i><code>--skip_qc_plots</code></td>
<td>QC Plot creation by FastQC or Nanoplot will be skipped <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-fast-forward"></i><code>--skip_error_correction</code></td>
<td>FLASH error correction of reads will be skipped. <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-file-alt"></i><code>--adapters</code></td>
<td>A FASTA file containing adapters to remove <br/><strong>Type:</strong> <code>string</code>, <strong>Default:</strong> <code>/home/robert_petit/bactopia/data/EMPTY_ADAPTERS</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-hashtag"></i><code>--adapter_k</code></td>
<td>Kmer length used for finding adapters. <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>23</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-file-alt"></i><code>--phix</code></td>
<td>phiX174 reference genome to remove <br/><strong>Type:</strong> <code>string</code>, <strong>Default:</strong> <code>/home/robert_petit/bactopia/data/EMPTY_PHIX</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-hashtag"></i><code>--phix_k</code></td>
<td>Kmer length used for finding phiX174. <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>31</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-boxes"></i><code>--ktrim</code></td>
<td>Trim reads to remove bases matching reference kmers <br/><strong>Type:</strong> <code>string</code>, <strong>Default:</strong> <code>r</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-angle-double-down"></i><code>--mink</code></td>
<td>Look for shorter kmers at read tips down to this length, when k-trimming or masking. <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>11</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-hashtag"></i><code>--hdist</code></td>
<td>Maximum Hamming distance for ref kmers (subs only) <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>1</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-boxes"></i><code>--tpe</code></td>
<td>When kmer right-trimming, trim both reads to the minimum length of either <br/><strong>Type:</strong> <code>string</code>, <strong>Default:</strong> <code>t</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-boxes"></i><code>--tbo</code></td>
<td>Trim adapters based on where paired reads overlap <br/><strong>Type:</strong> <code>string</code>, <strong>Default:</strong> <code>t</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-boxes"></i><code>--qtrim</code></td>
<td>Trim read ends to remove bases with quality below trimq. <br/><strong>Type:</strong> <code>string</code>, <strong>Default:</strong> <code>rl</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-hashtag"></i><code>--trimq</code></td>
<td>Regions with average quality BELOW this will be trimmed if qtrim is set to something other than f <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>6</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-angle-double-down"></i><code>--maq</code></td>
<td>Reads with average quality (after trimming) below this will be discarded <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>10</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-angle-double-down"></i><code>--minlength</code></td>
<td>Reads shorter than this after trimming will be discarded <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>35</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-hashtag"></i><code>--ftm</code></td>
<td>If positive, right-trim length to be equal to zero, modulo this number <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>5</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-boxes"></i><code>--tossjunk</code></td>
<td>Discard reads with invalid characters as bases <br/><strong>Type:</strong> <code>string</code>, <strong>Default:</strong> <code>t</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-boxes"></i><code>--ain</code></td>
<td>When detecting pair names, allow identical names <br/><strong>Type:</strong> <code>string</code>, <strong>Default:</strong> <code>f</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-boxes"></i><code>--qout</code></td>
<td>PHRED offset to use for output FASTQs <br/><strong>Type:</strong> <code>string</code>, <strong>Default:</strong> <code>33</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-hashtag"></i><code>--maxcor</code></td>
<td>Max number of corrections within a 20bp window <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>1</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-hashtag"></i><code>--sampleseed</code></td>
<td>Set to a positive number to use as the random number generator seed for sampling <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>42</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-angle-double-down"></i><code>--ont_minlength</code></td>
<td>ONT Reads shorter than this will be discarded <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>1000</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-angle-double-down"></i><code>--ont_minqual</code></td>
<td>Minimum average read quality filter of ONT reads <br/><strong>Type:</strong> <code>integer</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-italic"></i><code>--porechop_opts</code></td>
<td>Extra Porechop options in quotes <br/><strong>Type:</strong> <code>string</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-italic"></i><code>--nanoplot_opts</code></td>
<td>Extra NanoPlot options in quotes <br/><strong>Type:</strong> <code>string</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-italic"></i><code>--bbduk_opts</code></td>
<td>Extra BBDuk options in quotes <br/><strong>Type:</strong> <code>string</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-italic"></i><code>--fastp_opts</code></td>
<td>Extra fastp options in quotes <br/><strong>Type:</strong> <code>string</code></td>
</tr>
</tbody>
</table>
<h3 id="citations_1">Citations<a class="headerlink" href="#citations_1" title="Permanent link">&para;</a></h3>
<p>If you use Bactopia and <code>qc</code> results in your analysis, please cite the following.</p>
<ul>
<li>
<p><a href="https://bactopia.github.io/">Bactopia</a><br />
    Petit III RA, Read TD <a href="https://doi.org/10.1128/mSystems.00190-20">Bactopia - a flexible pipeline for complete analysis of bacterial genomes.</a> <em>mSystems</em> 5 (2020)</p>
</li>
<li>
<p><a href="https://jgi.doe.gov/data-and-tools/bbtools/">BBTools</a><br />
    Bushnell B <a href="http://sourceforge.net/projects/bbmap/">BBMap short read aligner, and other bioinformatic tools.</a> (Link)</p>
</li>
<li>
<p><a href="https://github.com/OpenGene/fastp">fastp</a><br />
    Chen S, Zhou Y, Chen Y, and Gu J <a href="https://doi.org/10.1093/bioinformatics/bty560">fastp: an ultra-fast all-in-one FASTQ preprocessor.</a> <em>Bioinformatics</em>, 34(17), i884–i890. (2018)</p>
</li>
<li>
<p><a href="https://github.com/rpetit3/fastq-scan">fastq-scan</a><br />
    Petit III RA <a href="https://github.com/rpetit3/fastq-scan">fastq-scan: generate summary statistics of input FASTQ sequences.</a> (GitHub)</p>
</li>
<li>
<p><a href="https://github.com/s-andrews/FastQC">FastQC</a><br />
    Andrews S <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc">FastQC: a quality control tool for high throughput sequence data.</a> (WebLink)</p>
</li>
<li>
<p><a href="https://github.com/mourisl/Lighter">Lighter</a><br />
    Song L, Florea L, Langmead B <a href="https://doi.org/10.1186/s13059-014-0509-9">Lighter: Fast and Memory-efficient Sequencing Error Correction without Counting</a>. <em>Genome Biol.</em> 15(11):509 (2014)</p>
</li>
<li>
<p><a href="https://github.com/wdecoster/NanoPlot">NanoPlot</a><br />
    De Coster W, D’Hert S, Schultz DT, Cruts M, Van Broeckhoven C <a href="https://doi.org/10.1093/bioinformatics/bty149">NanoPack: visualizing and processing long-read sequencing data</a> <em>Bioinformatics</em> Volume 34, Issue 15 (2018) </p>
</li>
<li>
<p><a href="https://github.com/esteinig/nanoq">Nanoq</a><br />
    Steinig E <a href="https://github.com/esteinig/nanoq">Nanoq: Minimal but speedy quality control for nanopore reads in Rust</a> (GitHub)</p>
</li>
<li>
<p><a href="https://zlib.net/pigz/">Pigz</a><br />
    Adler M. <a href="https://zlib.net/pigz/">pigz: A parallel implementation of gzip for modern multi-processor, multi-core machines.</a> <em>Jet Propulsion Laboratory</em> (2015)</p>
</li>
<li>
<p><a href="https://github.com/rrwick/Porechop">Porechop</a><br />
    Wick RR, Judd LM, Gorrie CL, Holt KE. <a href="https://doi.org/10.1099/mgen.0.000132">Completing bacterial genome assemblies with multiplex MinION sequencing.</a> <em>Microb Genom.</em> 3(10):e000132 (2017)</p>
</li>
<li>
<p><a href="https://github.com/mbhall88/rasusa">Rasusa</a><br />
    Hall MB <a href="https://doi.org/10.5281/zenodo.3731394">Rasusa: Randomly subsample sequencing reads to a specified coverage.</a> (2019).</p>
</li>
</ul>
<h2 id="step-3-assembler">Step 3 - Assembler<a class="headerlink" href="#step-3-assembler" title="Permanent link">&para;</a></h2>
<p>The <code>assembler</code> module uses a variety of assembly tools to create an assembly of
Illumina and Oxford Nanopore reads. The tools used are:</p>
<table>
<thead>
<tr>
<th>Tool</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="https://github.com/rpetit3/dragonflye">Dragonflye</a></td>
<td>Assembly of Oxford Nanopore reads, as well as hybrid assembly with short-read polishing</td>
</tr>
<tr>
<td><a href="https://github.com/tseemann/shovill">Shovill</a></td>
<td>Assembly of Illumina paired-end reads</td>
</tr>
<tr>
<td><a href="https://github.com/rpetit/shovill">Shovill-SE</a></td>
<td>Assembly of Illumina single-end reads</td>
</tr>
<tr>
<td><a href="https://github.com/rrwick/Unicycler">Unicycler</a></td>
<td>Hybrid assembly, using short-reads first then long-reads</td>
</tr>
</tbody>
</table>
<p>Summary statistics for each assembly are generated using <a href="https://github.com/rpetit3/assembly-scan">assembly-scan</a>.</p>
<h3 id="outputs_2">Outputs<a class="headerlink" href="#outputs_2" title="Permanent link">&para;</a></h3>
<h4 id="merged-results_1">Merged Results<a class="headerlink" href="#merged-results_1" title="Permanent link">&para;</a></h4>
<p>Below are results that are concatenated into a single file.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>assembly-scan.tsv</td>
<td>Assembly statistics for all samples</td>
</tr>
</tbody>
</table>
<h4 id="dragonflye">Dragonflye<a class="headerlink" href="#dragonflye" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results for Oxford Nanopore reads using
<a href="https://github.com/rpetit3/dragonflye">Dragonflye</a>.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>&lt;SAMPLE_NAME&gt;.fna.gz</td>
<td>The final assembly produced by Dragonflye</td>
</tr>
<tr>
<td>&lt;SAMPLE_NAME&gt;.tsv</td>
<td>A tab-delimited file containing assembly statistics</td>
</tr>
<tr>
<td>flye-info.txt</td>
<td>A log file containing information about the Flye assembly</td>
</tr>
<tr>
<td>{flye|miniasm|raven}-unpolished.fasta.gz</td>
<td>Raw unprocessed assembly produced by the used assembler</td>
</tr>
<tr>
<td>{flye|miniasm|raven}-unpolished.gfa.gz</td>
<td>Raw unprocessed assembly graph produced by the used assembler</td>
</tr>
</tbody>
</table>
<h4 id="shovill">Shovill<a class="headerlink" href="#shovill" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results for Illumina reads using
<a href="https://github.com/tseemann/shovill">Shovill</a> or <a href="https://github.com/rpetit3/shovill">Shovill-SE</a>.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>&lt;SAMPLE_NAME&gt;.fna.gz</td>
<td>The final assembly produced by Dragonflye</td>
</tr>
<tr>
<td>&lt;SAMPLE_NAME&gt;.tsv</td>
<td>A tab-delimited file containing assembly statistics</td>
</tr>
<tr>
<td>flash.hist</td>
<td>(Paired-End Only) Numeric histogram of merged read lengths.</td>
</tr>
<tr>
<td>flash.histogram</td>
<td>(Paired-End Only) Visual histogram of merged read lengths</td>
</tr>
<tr>
<td>{megahit|spades|velvet}-unpolished.gfa.gz</td>
<td>Raw unprocessed assembly graph produced by the used assembler</td>
</tr>
<tr>
<td>shovill.corrections</td>
<td>List of post-assembly corrections made by Shovill</td>
</tr>
</tbody>
</table>
<h4 id="hybrid-assembly-unicycler">Hybrid Assembly (Unicycler)<a class="headerlink" href="#hybrid-assembly-unicycler" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results for a hybrid assembly using
<a href="https://github.com/rrwick/Unicycler">Unicycler</a> (<code>--hybrid</code>). When using Unicycler,
the short-reads are assembled first, then the long-reads are used to polish the
assembly.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>&lt;SAMPLE_NAME&gt;.fna.gz</td>
<td>The final assembly produced by Dragonflye</td>
</tr>
<tr>
<td>&lt;SAMPLE_NAME&gt;.tsv</td>
<td>A tab-delimited file containing assembly statistics</td>
</tr>
<tr>
<td>unicycler-unpolished.fasta.gz</td>
<td>Raw unprocessed assembly produced by Unicycler</td>
</tr>
<tr>
<td>unicycler-unpolished.fasta.gz</td>
<td>Raw unprocessed assembly graph produced by Unicycler</td>
</tr>
</tbody>
</table>
<h4 id="hybrid-assembly-short-read-polishing">Hybrid Assembly (Short Read Polishing)<a class="headerlink" href="#hybrid-assembly-short-read-polishing" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results for a hybrid assembly using
<a href="https://github.com/rpetit3/dragonflye">Dragonflye</a> (<code>--short_polish</code>). When using
Dragonflye, the long-reads are assembled first, then the short-reads are used
to polish the assembly.</p>
<div class="admonition tip">
<p class="admonition-title">Prefer <code>--short_polish</code> over <code>--hybrid</code> with recent ONT sequencing</p>
<p>Using <a href="https://github.com/rrwick/Unicycler">Unicycler</a> (<code>--hybrid</code>) to create a hybrid
assembly works great when you have low-coverage noisy long-reads. However, if you are
using recent ONT sequencing, you likely have high-coverage and using the <code>--short_polish</code>
method is going to yeild better results (<em>and be faster!</em>) than <code>--hybrid</code>.</p>
</div>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>&lt;SAMPLE_NAME&gt;.fna.gz</td>
<td>The final assembly produced by Dragonflye</td>
</tr>
<tr>
<td>&lt;SAMPLE_NAME&gt;.tsv</td>
<td>A tab-delimited file containing assembly statistics</td>
</tr>
<tr>
<td>flye-info.txt</td>
<td>A log file containing information about the Flye assembly</td>
</tr>
<tr>
<td>{flye|miniasm|raven}-unpolished.fasta.gz</td>
<td>Raw unprocessed assembly produced by the used assembler</td>
</tr>
<tr>
<td>{flye|miniasm|raven}-unpolished.gfa.gz</td>
<td>Raw unprocessed assembly graph produced by the used assembler</td>
</tr>
</tbody>
</table>
<h4 id="failed-quality-checks_2">Failed Quality Checks<a class="headerlink" href="#failed-quality-checks_2" title="Permanent link">&para;</a></h4>
<p>Built into Bactopia are few basic quality checks to help prevent downstream failures.
If a sample fails one of these checks, it will be excluded from further analysis. By
excluding these samples, complete pipeline failures are prevented.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>-assembly-error.txt</td>
<td>Sample failed read count checks and excluded from further analysis</td>
</tr>
</tbody>
</table>
<div class="admonition info">
<p class="admonition-title">Poor samples are excluded to prevent downstream failures</p>
<p>Samples that fail any of the QC checks will be excluded from further analysis.
Those samples will generate a <code>*-error.txt</code> file with the error message. Excluding
these samples prevents downstream failures that cause the whole workflow to fail.</p>
</div>
<details class="warning">
<summary>Example Error: Assembled Successfully, but 0 Contigs</summary>
<p>If a sample assembles successfully, but 0 contigs are formed, the sample will be
excluded from further analysis.</p>
<p><strong>Example Text from &lt;SAMPLE_NAME&gt;-assembly-error.txt</strong><br />
<em>&lt;SAMPLE_NAME&gt; assembled successfully, but 0 contigs were formed. Please investigate
&lt;SAMPLE_NAME&gt; to determine a cause (e.g. metagenomic, contaminants, etc...) for this
outcome. Further assembly-based analysis of &lt;SAMPLE_NAME&gt; will be discontinued.</em></p>
</details>
<details class="warning">
<summary>Example Error: Assembled successfully, but poor assembly size</summary>
<p>If you sample assembles successfully, but the assembly size is less than the minimum
allowed genome size, the sample will be excluded from further analysis. You can
adjust this minimum size using the <code>--min_genome_size</code> parameter.</p>
<p><strong>Example Text from &lt;SAMPLE_NAME&gt;-assembly-error.txt</strong><br />
<em>&lt;SAMPLE_NAME&gt; assembled size (000 bp) is less than the minimum allowed genome
size (000 bp). If this is unexpected, please investigate &lt;SAMPLE_NAME&gt; to
determine a cause (e.g. metagenomic, contaminants, etc...) for the poor assembly.
Otherwise, adjust the <code>--min_genome_size</code> parameter to fit your need. Further
assembly based analysis of &lt;SAMPLE_NAME&gt; will be discontinued.</em></p>
</details>
<h3 id="assembler-parameters"><i class="fa-xl fas fa-exclamation-circle"></i> Assembler Parameters<a class="headerlink" href="#assembler-parameters" title="Permanent link">&para;</a></h3>
<table>
<thead>
<tr>
<th style="text-align: left;">Parameter</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-boxes"></i><code>--shovill_assembler</code></td>
<td>Assembler to be used by Shovill <br/><strong>Type:</strong> <code>string</code>, <strong>Default:</strong> <code>skesa</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-boxes"></i><code>--dragonflye_assembler</code></td>
<td>Assembler to be used by Dragonflye <br/><strong>Type:</strong> <code>string</code>, <strong>Default:</strong> <code>flye</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-cut"></i><code>--use_unicycler</code></td>
<td>Use unicycler for paired end assembly <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-angle-double-down"></i><code>--min_contig_len</code></td>
<td>Minimum contig length &lt;0=AUTO&gt; <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>500</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-angle-double-down"></i><code>--min_contig_cov</code></td>
<td>Minimum contig coverage &lt;0=AUTO&gt; <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>2</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-italic"></i><code>--contig_namefmt</code></td>
<td>Format of contig FASTA IDs in 'printf' style <br/><strong>Type:</strong> <code>string</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-italic"></i><code>--shovill_opts</code></td>
<td>Extra assembler options in quotes for Shovill <br/><strong>Type:</strong> <code>string</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-hashtag"></i><code>--shovill_kmers</code></td>
<td>K-mers to use <blank=AUTO> <br/><strong>Type:</strong> <code>string</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-italic"></i><code>--dragonflye_opts</code></td>
<td>Extra assembler options in quotes for Dragonflye <br/><strong>Type:</strong> <code>string</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-cut"></i><code>--trim</code></td>
<td>Enable adaptor trimming <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-fast-forward"></i><code>--no_stitch</code></td>
<td>Disable read stitching for paired-end reads <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-fast-forward"></i><code>--no_corr</code></td>
<td>Disable post-assembly correction <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-boxes"></i><code>--unicycler_mode</code></td>
<td>Bridging mode used by Unicycler <br/><strong>Type:</strong> <code>string</code>, <strong>Default:</strong> <code>normal</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-angle-double-down"></i><code>--min_polish_size</code></td>
<td>Contigs shorter than this value (bp) will not be polished using Pilon <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>10000</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-angle-double-down"></i><code>--min_component_size</code></td>
<td>Graph dead ends smaller than this size (bp) will be removed from the final graph <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>1000</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-angle-double-down"></i><code>--min_dead_end_size</code></td>
<td>Graph dead ends smaller than this size (bp) will be removed from the final graph <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>1000</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-redo"></i><code>--nanohq</code></td>
<td>For Flye, use '--nano-hq' instead of --nano-raw <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-font"></i><code>--medaka_model</code></td>
<td>The model to use for Medaka polishing <br/><strong>Type:</strong> <code>string</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-hashtag"></i><code>--medaka_rounds</code></td>
<td>The number of Medaka polishing rounds to conduct <br/><strong>Type:</strong> <code>integer</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-hashtag"></i><code>--racon_rounds</code></td>
<td>The number of Racon polishing rounds to conduct <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>1</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-fast-forward"></i><code>--no_polish</code></td>
<td>Skip the assembly polishing step <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-fast-forward"></i><code>--no_miniasm</code></td>
<td>Skip miniasm+Racon bridging <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-fast-forward"></i><code>--no_rotate</code></td>
<td>Do not rotate completed replicons to start at a standard gene <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-redo"></i><code>--reassemble</code></td>
<td>If reads were simulated, they will be used to create a new assembly. <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-angle-double-down"></i><code>--polypolish_rounds</code></td>
<td>Number of polishing rounds to conduct with Polypolish for short read polishing <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>1</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-angle-double-down"></i><code>--pilon_rounds</code></td>
<td>Number of polishing rounds to conduct with Pilon for short read polishing <br/><strong>Type:</strong> <code>integer</code></td>
</tr>
</tbody>
</table>
<h3 id="citations_2">Citations<a class="headerlink" href="#citations_2" title="Permanent link">&para;</a></h3>
<p>If you use Bactopia and <code>assembler</code> results in your analysis, please cite the following.</p>
<ul>
<li>
<p><a href="https://bactopia.github.io/">Bactopia</a><br />
    Petit III RA, Read TD <a href="https://doi.org/10.1128/mSystems.00190-20">Bactopia - a flexible pipeline for complete analysis of bacterial genomes.</a> <em>mSystems</em> 5 (2020)</p>
</li>
<li>
<p><a href="https://github.com/tseemann/any2fasta">any2fasta</a><br />
    Seemann T <a href="https://github.com/tseemann/any2fasta">any2fasta: Convert various sequence formats to FASTA</a> (GitHub)</p>
</li>
<li>
<p><a href="https://github.com/rpetit3/assembly-scan">assembly-scan</a><br />
    Petit III RA <a href="https://github.com/rpetit3/assembly-scan">assembly-scan: generate basic stats for an assembly</a> (GitHub)</p>
</li>
<li>
<p><a href="https://github.com/lh3/bwa/">BWA</a><br />
    Li H <a href="http://arxiv.org/abs/1303.3997">Aligning sequence reads, clone sequences and assembly contigs with BWA-MEM</a>. <em>arXiv</em> [q-bio.GN] (2013)</p>
</li>
<li>
<p><a href="https://bioinf.shenwei.me/csvtk/">csvtk</a><br />
    Shen, W <a href="https://github.com/shenwei356/csvtk/">csvtk: A cross-platform, efficient and practical CSV/TSV toolkit in Golang.</a> (GitHub)</p>
</li>
<li>
<p><a href="https://github.com/rpetit3/dragonflye">Dragonflye</a><br />
    Petit III RA <a href="https://github.com/rpetit3/dragonflye">Dragonflye: Assemble bacterial isolate genomes from Nanopore reads.</a> (GitHub)</p>
</li>
<li>
<p><a href="https://ccb.jhu.edu/software/FLASH/">FLASH</a><br />
    Magoč T, Salzberg SL <a href="https://doi.org/10.1093/bioinformatics/btr507">FLASH: fast length adjustment of short reads to improve genome assemblies.</a> <em>Bioinformatics</em> 27.21 2957-2963 (2011)</p>
</li>
<li>
<p><a href="https://github.com/fenderglass/Flye">Flye</a><br />
    Kolmogorov M, Yuan J, Lin Y, Pevzner P <a href="https://doi.org/10.1038/s41587-019-0072-8">Assembly of Long Error-Prone Reads Using Repeat Graphs</a> <em>Nature Biotechnology</em> (2019)</p>
</li>
<li>
<p><a href="https://github.com/nanoporetech/medaka">Medaka</a><br />
    ONT Research <a href="https://github.com/nanoporetech/medaka">Medaka: Sequence correction provided by ONT Research</a> (GitHub)</p>
</li>
<li>
<p><a href="https://github.com/voutcn/megahit">MEGAHIT</a><br />
    Li D, Liu C-M, Luo R, Sadakane K, Lam T-W <a href="https://doi.org/10.1093/bioinformatics/btv033">MEGAHIT: an ultra-fast single-node solution for large and complex metagenomics assembly via succinct de Bruijn graph.</a> <em>Bioinformatics</em> 31.10 1674-1676 (2015)</p>
</li>
<li>
<p><a href="https://github.com/lh3/miniasm">Miniasm</a><br />
    Li H <a href="https://github.com/lh3/miniasm">Miniasm: Ultrafast de novo assembly for long noisy reads</a> (GitHub)</p>
</li>
<li>
<p><a href="https://github.com/lh3/minimap2">Minimap2</a><br />
    Li H <a href="https://doi.org/10.1093/bioinformatics/bty191">Minimap2: pairwise alignment for nucleotide sequences.</a> <em>Bioinformatics</em> 34:3094-3100 (2018)</p>
</li>
<li>
<p><a href="https://github.com/esteinig/nanoq">Nanoq</a><br />
    Steinig E <a href="https://github.com/esteinig/nanoq">Nanoq: Minimal but speedy quality control for nanopore reads in Rust</a> (GitHub)</p>
</li>
<li>
<p><a href="https://zlib.net/pigz/">Pigz</a><br />
    Adler M. <a href="https://zlib.net/pigz/">pigz: A parallel implementation of gzip for modern multi-processor, multi-core machines.</a> <em>Jet Propulsion Laboratory</em> (2015)</p>
</li>
<li>
<p><a href="https://github.com/broadinstitute/pilon/">Pilon</a><br />
    Walker BJ, Abeel T,  Shea T, Priest M, Abouelliel A, Sakthikumar S, Cuomo CA, Zeng Q, Wortman J, Young SK, Earl AM <a href="https://doi.org/10.1371/journal.pone.0112963">Pilon: an integrated tool for comprehensive microbial variant detection and genome assembly improvement.</a> <em>PloS one</em> 9.11 e112963 (2014)</p>
</li>
<li>
<p><a href="https://github.com/lbcb-sci/racon">Racon</a><br />
    Vaser R, Sović I, Nagarajan N, Šikić M <a href="http://dx.doi.org/10.1101/gr.214270.116">Fast and accurate de novo genome assembly from long uncorrected reads.</a> <em>Genome Res</em> 27, 737–746 (2017)</p>
</li>
<li>
<p><a href="https://github.com/mbhall88/rasusa">Rasusa</a><br />
    Hall MB <a href="https://doi.org/10.5281/zenodo.3731394">Rasusa: Randomly subsample sequencing reads to a specified coverage.</a> (2019).</p>
</li>
<li>
<p><a href="https://github.com/lbcb-sci/raven">Raven</a><br />
    Vaser R, Šikić M <a href="https://doi.org/10.1038/s43588-021-00073-4">Time- and memory-efficient genome assembly with Raven.</a> <em>Nat Comput Sci</em> 1, 332–336 (2021)</p>
</li>
<li>
<p><a href="https://github.com/tseemann/samclip">samclip</a><br />
    Seemann T <a href="https://github.com/tseemann/samclip">Samclip: Filter SAM file for soft and hard clipped alignments</a> (GitHub)</p>
</li>
<li>
<p><a href="https://github.com/samtools/samtools">Samtools</a><br />
    Li H, Handsaker B, Wysoker A, Fennell T, Ruan J, Homer N, Marth G, Abecasis G, Durbin R <a href="http://dx.doi.org/10.1093/bioinformatics/btp352">The Sequence Alignment/Map format and SAMtools</a>. <em>Bioinformatics</em> 25, 2078–2079 (2009)</p>
</li>
<li>
<p><a href="https://github.com/tseemann/shovill">Shovill</a><br />
    Seemann T <a href="https://github.com/tseemann/shovill">Shovill: De novo assembly pipeline for Illumina paired reads</a> (GitHub)</p>
</li>
<li>
<p><a href="https://github.com/rpetit3/shovill">Shovill-SE</a><br />
    Petit III RA <a href="https://github.com/rpetit3/shovill">Shovill-SE: A fork of Shovill that includes support for single end reads.</a> (GitHub)</p>
</li>
<li>
<p><a href="https://github.com/ncbi/SKESA">SKESA</a><br />
    Souvorov A, Agarwala R, Lipman DJ <a href="https://doi.org/10.1186/s13059-018-1540-z">SKESA: strategic k-mer extension for scrupulous assemblies.</a> <em>Genome Biology</em> 19:153 (2018)</p>
</li>
<li>
<p><a href="https://github.com/ablab/spades">SPAdes</a><br />
    Bankevich A, Nurk S, Antipov D, Gurevich AA, Dvorkin M, Kulikov AS, Lesin VM, Nikolenko SI, Pham S, Prjibelski AD, Pyshkin AV, Sirotkin AV, Vyahhi N, Tesler G, Alekseyev MA, Pevzner PA <a href="https://doi.org/10.1089/cmb.2012.0021">SPAdes: a new genome assembly algorithm and its applications to single-cell sequencing.</a> <em>Journal of computational biology</em> 19.5 455-477 (2012)</p>
</li>
<li>
<p><a href="https://github.com/rrwick/Unicycler">Unicycler</a><br />
    Wick RR, Judd LM, Gorrie CL, Holt KE <a href="http://dx.doi.org/10.1371/journal.pcbi.1005595">Unicycler: Resolving bacterial genome assemblies from short and long sequencing reads.</a> <em>PLoS Comput. Biol.</em> 13, e1005595 (2017)</p>
</li>
<li>
<p><a href="https://github.com/dzerbino/velvet">Velvet</a><br />
    Zerbino DR, Birney E <a href="http://www.genome.org/cgi/doi/10.1101/gr.074492.107">Velvet: algorithms for de novo short read assembly using de Bruijn graphs.</a> <em>Genome research</em> 18.5 821-829 (2008)</p>
</li>
</ul>
<h2 id="step-4-annotator">Step 4 - Annotator<a class="headerlink" href="#step-4-annotator" title="Permanent link">&para;</a></h2>
<h3 id="outputs_3">Outputs<a class="headerlink" href="#outputs_3" title="Permanent link">&para;</a></h3>
<h4 id="prokka">Prokka<a class="headerlink" href="#prokka" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results from <a href="https://github.com/tseemann/prokka">Prokka</a>.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>.blastdb.tar.gz</td>
<td>A gzipped tar archive of BLAST+ database of the contigs, genes, and proteins</td>
</tr>
<tr>
<td>.faa.gz</td>
<td>Protein FASTA file of the translated CDS sequences.</td>
</tr>
<tr>
<td>.ffn.gz</td>
<td>Nucleotide FASTA file of all the prediction transcripts (CDS, rRNA, tRNA, tmRNA, misc_RNA)</td>
</tr>
<tr>
<td>.fna.gz</td>
<td>Nucleotide FASTA file of the input contig sequences.</td>
</tr>
<tr>
<td>.gbk.gz</td>
<td>This is a standard GenBank file derived from the master .gff. If the input to prokka was a multi-FASTA, then this will be a multi-GenBank, with one record for each sequence.</td>
</tr>
<tr>
<td>.gff.gz</td>
<td>This is the master annotation in GFF3 format, containing both sequences and annotations. It can be viewed directly in Artemis or IGV.</td>
</tr>
<tr>
<td>.sqn.gz</td>
<td>An ASN1 format "Sequin" file for submission to GenBank. It needs to be edited to set the correct taxonomy, authors, related publication etc.</td>
</tr>
<tr>
<td>.tbl.gz</td>
<td>Feature Table file, used by "tbl2asn" to create the .sqn file.</td>
</tr>
<tr>
<td>.tsv</td>
<td>Tab-separated file of all features (locus_tag,ftype,len_bp,gene,EC_number,COG,product)</td>
</tr>
<tr>
<td>.txt</td>
<td>Statistics relating to the annotated features found.</td>
</tr>
</tbody>
</table>
<h4 id="bakta">Bakta<a class="headerlink" href="#bakta" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results from <a href="https://github.com/oschwengers/bakta">Bakta</a>.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>.blastdb.tar.gz</td>
<td>A gzipped tar archive of BLAST+ database of the contigs, genes, and proteins</td>
</tr>
<tr>
<td>.embl.gz</td>
<td>Annotations &amp; sequences in (multi) EMBL format</td>
</tr>
<tr>
<td>.faa.gz</td>
<td>CDS/sORF amino acid sequences as FASTA</td>
</tr>
<tr>
<td>.ffn.gz</td>
<td>Feature nucleotide sequences as FASTA</td>
</tr>
<tr>
<td>.fna.gz</td>
<td>Replicon/contig DNA sequences as FASTA</td>
</tr>
<tr>
<td>.gbff.gz</td>
<td>Annotations &amp; sequences in (multi) GenBank format</td>
</tr>
<tr>
<td>.gff3.gz</td>
<td>Annotations &amp; sequences in GFF3 format</td>
</tr>
<tr>
<td>.hypotheticals.faa.gz</td>
<td>Hypothetical protein CDS amino acid sequences as FASTA</td>
</tr>
<tr>
<td>.hypotheticals.tsv</td>
<td>Further information on hypothetical protein CDS as simple human readable tab separated values</td>
</tr>
<tr>
<td>.tsv</td>
<td>Annotations as simple human readable tab separated values</td>
</tr>
<tr>
<td>.txt</td>
<td>Broad summary of <code>Bakta</code> annotations</td>
</tr>
</tbody>
</table>
<h3 id="parameters_1">Parameters<a class="headerlink" href="#parameters_1" title="Permanent link">&para;</a></h3>
<h4 id="prokka_1"><i class="fa-xl fas fa-exclamation-circle"></i> Prokka<a class="headerlink" href="#prokka_1" title="Permanent link">&para;</a></h4>
<table>
<thead>
<tr>
<th style="text-align: left;">Parameter</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-file-alt"></i><code>--proteins</code></td>
<td>FASTA file of trusted proteins to first annotate from <br/><strong>Type:</strong> <code>string</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-file-alt"></i><code>--prodigal_tf</code></td>
<td>Training file to use for Prodigal <br/><strong>Type:</strong> <code>string</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-check"></i><code>--compliant</code></td>
<td>Force Genbank/ENA/DDJB compliance <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-school"></i><code>--centre</code></td>
<td>Sequencing centre ID <br/><strong>Type:</strong> <code>string</code>, <strong>Default:</strong> <code>Bactopia</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-hashtag"></i><code>--prokka_coverage</code></td>
<td>Minimum coverage on query protein <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>80</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-italic"></i><code>--prokka_evalue</code></td>
<td>Similarity e-value cut-off <br/><strong>Type:</strong> <code>string</code>, <strong>Default:</strong> <code>1e-09</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-italic"></i><code>--prokka_opts</code></td>
<td>Extra Prokka options in quotes. <br/><strong>Type:</strong> <code>string</code></td>
</tr>
</tbody>
</table>
<h4 id="bakta-download"><i class="fa-xl fas fa-exclamation-circle"></i> Bakta Download<a class="headerlink" href="#bakta-download" title="Permanent link">&para;</a></h4>
<table>
<thead>
<tr>
<th style="text-align: left;">Parameter</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-file-alt"></i><code>--bakta_db</code></td>
<td>Tarball or path to the Bakta database <br/><strong>Type:</strong> <code>string</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-file-alt"></i><code>--bakta_db_type</code></td>
<td>Which Bakta DB to download 'full' (~30GB) or 'light' (~2GB) <br/><strong>Type:</strong> <code>string</code>, <strong>Default:</strong> <code>full</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-file-alt"></i><code>--bakta_save_as_tarball</code></td>
<td>Save the Bakta database as a tarball <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-italic"></i><code>--download_bakta</code></td>
<td>Download the Bakta database to the path given by --bakta_db <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
</tbody>
</table>
<h4 id="bakta_1"><i class="fa-xl fas fa-exclamation-circle"></i> Bakta<a class="headerlink" href="#bakta_1" title="Permanent link">&para;</a></h4>
<table>
<thead>
<tr>
<th style="text-align: left;">Parameter</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-file-alt"></i><code>--proteins</code></td>
<td>FASTA file of trusted proteins to first annotate from <br/><strong>Type:</strong> <code>string</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-file-alt"></i><code>--prodigal_tf</code></td>
<td>Training file to use for Prodigal <br/><strong>Type:</strong> <code>string</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-file-alt"></i><code>--replicons</code></td>
<td>Replicon information table (tsv/csv) <br/><strong>Type:</strong> <code>string</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-hashtag"></i><code>--min_contig_length</code></td>
<td>Minimum contig size to annotate <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>1</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-italic"></i><code>--keep_contig_headers</code></td>
<td>Keep original contig headers <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-italic"></i><code>--compliant</code></td>
<td>Force Genbank/ENA/DDJB compliance <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-italic"></i><code>--skip_trna</code></td>
<td>Skip tRNA detection &amp; annotation <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-italic"></i><code>--skip_tmrna</code></td>
<td>Skip tmRNA detection &amp; annotation <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-italic"></i><code>--skip_rrna</code></td>
<td>Skip rRNA detection &amp; annotation <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-italic"></i><code>--skip_ncrna</code></td>
<td>Skip ncRNA detection &amp; annotation <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-italic"></i><code>--skip_ncrna_region</code></td>
<td>Skip ncRNA region detection &amp; annotation <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-italic"></i><code>--skip_crispr</code></td>
<td>Skip CRISPR array detection &amp; annotation <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-italic"></i><code>--skip_cds</code></td>
<td>Skip CDS detection &amp; annotation <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-italic"></i><code>--skip_sorf</code></td>
<td>Skip sORF detection &amp; annotation <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-italic"></i><code>--skip_gap</code></td>
<td>Skip gap detection &amp; annotation <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-italic"></i><code>--skip_ori</code></td>
<td>Skip oriC/oriT detection &amp; annotation <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-italic"></i><code>--bakta_opts</code></td>
<td>Extra Backa options in quotes. Example: '--gram +' <br/><strong>Type:</strong> <code>string</code></td>
</tr>
</tbody>
</table>
<h3 id="citations_3">Citations<a class="headerlink" href="#citations_3" title="Permanent link">&para;</a></h3>
<p>If you use Bactopia and <code>annotator</code> results in your analysis, please cite the following.</p>
<ul>
<li>
<p><a href="https://bactopia.github.io/">Bactopia</a><br />
    Petit III RA, Read TD <a href="https://doi.org/10.1128/mSystems.00190-20">Bactopia - a flexible pipeline for complete analysis of bacterial genomes.</a> <em>mSystems</em> 5 (2020)</p>
</li>
<li>
<p><a href="https://github.com/tseemann/prokka">Prokka</a><br />
    Seemann T <a href="http://dx.doi.org/10.1093/bioinformatics/btu153">Prokka: rapid prokaryotic genome annotation</a> <em>Bioinformatics</em> 30, 2068–2069 (2014)</p>
</li>
<li>
<p><a href="https://github.com/oschwengers/bakta">Bakta</a><br />
    Schwengers O, Jelonek L, Dieckmann MA, Beyvers S, Blom J, Goesmann A <a href="https://doi.org/10.1099/mgen.0.000685">Bakta - rapid and standardized annotation of bacterial genomes via alignment-free sequence identification.</a> <em>Microbial Genomics</em> 7(11) (2021)</p>
</li>
</ul>
<h2 id="step-5-sketcher">Step 5 - Sketcher<a class="headerlink" href="#step-5-sketcher" title="Permanent link">&para;</a></h2>
<p>The <code>sketcher</code> module uses <a href="https://github.com/marbl/Mash">Mash</a> and
<a href="https://github.com/dib-lab/sourmash">Sourmash</a> to create sketches and query
<a href="https://www.ncbi.nlm.nih.gov/refseq/">RefSeq</a> and <a href="https://gtdb.ecogenomic.org/">GTDB</a>.</p>
<h3 id="outputs_4">Outputs<a class="headerlink" href="#outputs_4" title="Permanent link">&para;</a></h3>
<h4 id="sketcher">sketcher<a class="headerlink" href="#sketcher" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results from the <code>sketcher</code> subworkflow.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>&lt;SAMPLE_NAME&gt;-k{21|31}.msh</td>
<td>A Mash sketch of the input assembly for k=21 and k=31</td>
</tr>
<tr>
<td>&lt;SAMPLE_NAME&gt;-mash-refseq88-k21.txt</td>
<td>The results of querying the Mash sketch against RefSeq88</td>
</tr>
<tr>
<td>&lt;SAMPLE_NAME&gt;-sourmash-gtdb-rs207-k31.txt</td>
<td>The results of querying the Sourmash sketch against GTDB-rs207</td>
</tr>
<tr>
<td>&lt;SAMPLE_NAME&gt;.sig</td>
<td>A Sourmash sketch of the input assembly for k=21, k=31, and k=51</td>
</tr>
</tbody>
</table>
<h3 id="sketcher-parameters"><i class="fa-xl fas fa-exclamation-circle"></i> Sketcher Parameters<a class="headerlink" href="#sketcher-parameters" title="Permanent link">&para;</a></h3>
<table>
<thead>
<tr>
<th style="text-align: left;">Parameter</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-hashtag"></i><code>--sketch_size</code></td>
<td>Sketch size. Each sketch will have at most this many non-redundant min-hashes. <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>10000</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-hashtag"></i><code>--sourmash_scale</code></td>
<td>Choose number of hashes as 1 in FRACTION of input k-mers <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>10000</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-fast-forward"></i><code>--no_winner_take_all</code></td>
<td>Disable winner-takes-all strategy for identity estimates <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-hashtag"></i><code>--screen_i</code></td>
<td>Minimum identity to report. <br/><strong>Type:</strong> <code>number</code>, <strong>Default:</strong> <code>0.8</code></td>
</tr>
</tbody>
</table>
<h3 id="citations_4">Citations<a class="headerlink" href="#citations_4" title="Permanent link">&para;</a></h3>
<p>If you use Bactopia and <code>sketcher</code> results in your analysis, please cite the following.</p>
<ul>
<li>
<p><a href="https://bactopia.github.io/">Bactopia</a><br />
    Petit III RA, Read TD <a href="https://doi.org/10.1128/mSystems.00190-20">Bactopia - a flexible pipeline for complete analysis of bacterial genomes.</a> <em>mSystems</em> 5 (2020)</p>
</li>
<li>
<p><a href="https://gtdb.ecogenomic.org/">Genome Taxonomy Database</a><br />
    Parks DH, Chuvochina M, Rinke C, Mussig AJ, Chaumeil P-A, Hugenholtz P <a href="https://doi.org/10.1093/nar/gkab776">GTDB: an ongoing census of bacterial and archaeal diversity through a phylogenetically consistent, rank normalized and complete genome-based taxonomy</a> <em>Nucleic Acids Research</em> gkab776 (2021)</p>
</li>
<li>
<p><a href="https://github.com/marbl/Mash">Mash</a><br />
    Ondov BD, Treangen TJ, Melsted P, Mallonee AB, Bergman NH, Koren S, Phillippy AM <a href="http://dx.doi.org/10.1186/s13059-016-0997-x">Mash: fast genome and metagenome distance estimation using MinHash</a>. <em>Genome Biol</em> 17, 132 (2016)</p>
</li>
<li>
<p><a href="https://github.com/marbl/Mash">Mash</a><br />
    Ondov BD, Starrett GJ, Sappington A, Kostic A, Koren S, Buck CB, Phillippy AM <a href="https://doi.org/10.1186/s13059-019-1841-x">Mash Screen: high-throughput sequence containment estimation for genome discovery</a> <em>Genome Biol</em> 20, 232 (2019)</p>
</li>
<li>
<p><a href="https://www.ncbi.nlm.nih.gov/refseq/">NCBI RefSeq Database</a><br />
    O'Leary NA, Wright MW, Brister JR, Ciufo S, Haddad D, McVeigh R, Rajput B, Robbertse B, Smith-White B, Ako-Adjei D, Astashyn A, Badretdin A, Bao Y, Blinkova O0, Brover V, Chetvernin V, Choi J, Cox E, Ermolaeva O, Farrell CM, Goldfarb T, Gupta T, Haft D, Hatcher E, Hlavina W, Joardar VS, Kodali VK, Li W, Maglott D, Masterson P, McGarvey KM, Murphy MR, O'Neill K, Pujar S, Rangwala SH, Rausch D, Riddick LD, Schoch C, Shkeda A, Storz SS, Sun H, Thibaud-Nissen F, Tolstoy I, Tully RE, Vatsan AR, Wallin C, Webb D, Wu W, Landrum MJ, Kimchi A, Tatusova T, DiCuccio M, Kitts P, Murphy TD, Pruitt KD <a href="https://doi.org/10.1093/nar/gkv1189">Reference sequence (RefSeq) database at NCBI: current status, taxonomic expansion, and functional annotation.</a> <em>Nucleic Acids Res.</em> 44, D733–45 (2016)</p>
</li>
<li>
<p><a href="https://github.com/dib-lab/sourmash">Sourmash</a><br />
    Brown CT, Irber L <a href="http://dx.doi.org/10.21105/joss.00027">sourmash: a library for MinHash sketching of DNA</a>. <em>JOSS</em> 1, 27 (2016)</p>
</li>
</ul>
<h2 id="step-6-sequence-typing">Step 6 - Sequence Typing<a class="headerlink" href="#step-6-sequence-typing" title="Permanent link">&para;</a></h2>
<h3 id="outputs_5">Outputs<a class="headerlink" href="#outputs_5" title="Permanent link">&para;</a></h3>
<h4 id="merged-results_2">Merged Results<a class="headerlink" href="#merged-results_2" title="Permanent link">&para;</a></h4>
<p>Below are results that are concatenated into a single file.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>mlst.tsv</td>
<td>A merged TSV file with <code>mlst</code> results from all samples</td>
</tr>
</tbody>
</table>
<h4 id="mlst">mlst<a class="headerlink" href="#mlst" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results from <a href="https://github.com/tseemann/mlst">mlst</a>.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>&lt;SAMPLE_NAME&gt;.tsv</td>
<td>A tab-delimited file with <code>mlst</code> result, see <a href="https://github.com/tseemann/mlst#usage">mlst - Usage</a> for more details</td>
</tr>
</tbody>
</table>
<h3 id="parameters_2"><i class="fa-xl fas fa-exclamation-circle"></i> Parameters<a class="headerlink" href="#parameters_2" title="Permanent link">&para;</a></h3>
<table>
<thead>
<tr>
<th style="text-align: left;">Parameter</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--scheme</code></td>
<td>Don't autodetect, force this scheme on all inputs <br/><strong>Type:</strong> <code>string</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--minid</code></td>
<td>Minimum DNA percent identity of full allelle to consider 'similar' <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>95</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--mincov</code></td>
<td>Minimum DNA percent coverage to report partial allele at all <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>10</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--minscore</code></td>
<td>Minimum score out of 100 to match a scheme <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>50</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--nopath</code></td>
<td>Strip filename paths from FILE column <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--mlst_db</code></td>
<td>A custom MLST database to use, either a tarball or a directory <br/><strong>Type:</strong> <code>string</code></td>
</tr>
</tbody>
</table>
<h3 id="citations_5">Citations<a class="headerlink" href="#citations_5" title="Permanent link">&para;</a></h3>
<p>If you use Bactopia and <code>sequence typing</code> results in your analysis, please cite the following.</p>
<ul>
<li>
<p><a href="https://bactopia.github.io/">Bactopia</a><br />
    Petit III RA, Read TD <a href="https://doi.org/10.1128/mSystems.00190-20">Bactopia - a flexible pipeline for complete analysis of bacterial genomes.</a> <em>mSystems</em> 5 (2020)</p>
</li>
<li>
<p><a href="https://bioinf.shenwei.me/csvtk/">csvtk</a><br />
    Shen, W <a href="https://github.com/shenwei356/csvtk/">csvtk: A cross-platform, efficient and practical CSV/TSV toolkit in Golang.</a> (GitHub)</p>
</li>
<li>
<p><a href="https://github.com/tseemann/mlst">mlst</a><br />
    Seemann T <a href="https://github.com/tseemann/mlst">mlst: scan contig files against PubMLST typing schemes</a> (GitHub)</p>
</li>
<li>
<p><a href="https://pubmlst.org/">PubMLST.org</a><br />
    Jolley KA, Bray JE, Maiden MCJ <a href="http://dx.doi.org/10.12688/wellcomeopenres.14826.1">Open-access bacterial population genomics: BIGSdb software, the PubMLST.org website and their applications.</a> <em>Wellcome Open Res</em> 3, 124 (2018)</p>
</li>
</ul>
<h2 id="step-7-antibiotic-resistance">Step 7 - Antibiotic Resistance<a class="headerlink" href="#step-7-antibiotic-resistance" title="Permanent link">&para;</a></h2>
<h3 id="outputs_6">Outputs<a class="headerlink" href="#outputs_6" title="Permanent link">&para;</a></h3>
<h4 id="merged-results_3">Merged Results<a class="headerlink" href="#merged-results_3" title="Permanent link">&para;</a></h4>
<p>Below are results that are concatenated into a single file.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>amrfinderplus-genes.tsv</td>
<td>A merged TSV file with <code>AMRFinder+</code> results using nucleotide inputs</td>
</tr>
<tr>
<td>amrfinderplus-proteins.tsv</td>
<td>A merged TSV file with <code>AMRFinder+</code> results using protein inputs</td>
</tr>
</tbody>
</table>
<h4 id="amrfinder">AMRFinder+<a class="headerlink" href="#amrfinder" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results from <a href="https://github.com/ncbi/amr">AMRFinder+</a>.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>-genes.tsv</td>
<td>A TSV file with <code>AMRFinder+</code> results using nucleotide inputs</td>
</tr>
<tr>
<td>-proteins.tsv</td>
<td>A TSV file with <code>AMRFinder+</code> results using protein inputs</td>
</tr>
</tbody>
</table>
<h3 id="parameters_3"><i class="fa-xl fas fa-exclamation-circle"></i> Parameters<a class="headerlink" href="#parameters_3" title="Permanent link">&para;</a></h3>
<table>
<thead>
<tr>
<th style="text-align: left;">Parameter</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--ident_min</code></td>
<td>Minimum proportion of identical amino acids in alignment for hit (0..1) <br/><strong>Type:</strong> <code>number</code>, <strong>Default:</strong> <code>-1</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--coverage_min</code></td>
<td>Minimum coverage of the reference protein (0..1) <br/><strong>Type:</strong> <code>number</code>, <strong>Default:</strong> <code>0.5</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--organism</code></td>
<td>Taxonomy group to run additional screens against <br/><strong>Type:</strong> <code>string</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--translation_table</code></td>
<td>NCBI genetic code for translated BLAST <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>11</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--amrfinder_noplus</code></td>
<td>Disable running AMRFinder+ with the --plus option <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--report_common</code></td>
<td>Report proteins common to a taxonomy group <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--report_all_equal</code></td>
<td>Report all equally-scoring BLAST and HMM matches <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--amrfinder_opts</code></td>
<td>Extra AMRFinder+ options in quotes. <br/><strong>Type:</strong> <code>string</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--amrfinder_db</code></td>
<td>A custom AMRFinder+ database to use, either a tarball or a folder <br/><strong>Type:</strong> <code>string</code></td>
</tr>
</tbody>
</table>
<h3 id="citations_6">Citations<a class="headerlink" href="#citations_6" title="Permanent link">&para;</a></h3>
<p>If you use Bactopia and <code>antibiotic resistance</code> results in your analysis, please cite the following.</p>
<ul>
<li>
<p><a href="https://bactopia.github.io/">Bactopia</a><br />
    Petit III RA, Read TD <a href="https://doi.org/10.1128/mSystems.00190-20">Bactopia - a flexible pipeline for complete analysis of bacterial genomes.</a> <em>mSystems</em> 5 (2020)</p>
</li>
<li>
<p><a href="https://github.com/ncbi/amr">AMRFinderPlus</a><br />
    Feldgarden M, Brover V, Haft DH, Prasad AB, Slotta DJ, Tolstoy I, Tyson GH, Zhao S, Hsu C-H, McDermott PF, Tadesse DA, Morales C, Simmons M, Tillman G, Wasilenko J, Folster JP, Klimke W <a href="https://doi.org/10.1128/AAC.00483-19">Validating the NCBI AMRFinder Tool and Resistance Gene Database Using Antimicrobial Resistance Genotype-Phenotype Correlations in a Collection of NARMS Isolates</a>. <em>Antimicrob. Agents Chemother.</em> (2019)</p>
</li>
<li>
<p><a href="https://bioinf.shenwei.me/csvtk/">csvtk</a><br />
    Shen, W <a href="https://github.com/shenwei356/csvtk/">csvtk: A cross-platform, efficient and practical CSV/TSV toolkit in Golang.</a> (GitHub)</p>
</li>
</ul>
<h2 id="step-8-merlin">Step 8 - Merlin<a class="headerlink" href="#step-8-merlin" title="Permanent link">&para;</a></h2>
<h3 id="outputs_7">Outputs<a class="headerlink" href="#outputs_7" title="Permanent link">&para;</a></h3>
<h4 id="merged-results_4">Merged Results<a class="headerlink" href="#merged-results_4" title="Permanent link">&para;</a></h4>
<p>Below are results that are concatenated into a single file.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>agrvate.tsv</td>
<td>A merged TSV file with <code>AgrVATE</code> results from all samples</td>
</tr>
<tr>
<td>clermontyping.csv</td>
<td>A merged TSV file with <code>ClermonTyping</code> results from all samples</td>
</tr>
<tr>
<td>ectyper.tsv</td>
<td>A merged TSV file with <code>ECTyper</code> results from all samples</td>
</tr>
<tr>
<td>emmtyper.tsv</td>
<td>A merged TSV file with <code>emmtyper</code> results from all samples</td>
</tr>
<tr>
<td>genotyphi.tsv</td>
<td>A merged TSV file with <code>genotyphi</code> results from all samples</td>
</tr>
<tr>
<td>hicap.tsv</td>
<td>A merged TSV file with <code>hicap</code> results from all samples</td>
</tr>
<tr>
<td>hpsuissero.tsv</td>
<td>A merged TSV file with <code>HpsuisSero</code> results from all samples</td>
</tr>
<tr>
<td>kleborate.tsv</td>
<td>A merged TSV file with <code>Kleborate</code> results from all samples</td>
</tr>
<tr>
<td>legsta.tsv</td>
<td>A merged TSV file with <code>legsta</code> results from all samples</td>
</tr>
<tr>
<td>lissero.tsv</td>
<td>A merged TSV file with <code>LisSero</code> results from all samples</td>
</tr>
<tr>
<td>meningotype.tsv</td>
<td>A merged TSV file with <code>meningotype</code> results from all samples</td>
</tr>
<tr>
<td>ngmaster.tsv</td>
<td>A merged TSV file with <code>ngmaster</code> results from all samples</td>
</tr>
<tr>
<td>pasty.tsv</td>
<td>A merged TSV file with <code>pasty</code> results from all samples</td>
</tr>
<tr>
<td>pbptyper.tsv</td>
<td>A merged TSV file with <code>pbptyper</code> results from all samples</td>
</tr>
<tr>
<td>seqsero2.tsv</td>
<td>A merged TSV file with <code>seqsero2</code> results from all samples</td>
</tr>
<tr>
<td>seroba.tsv</td>
<td>A merged TSV file with <code>seroba</code> results from all samples</td>
</tr>
<tr>
<td>shigapass.csv</td>
<td>A merged CSV file with <code>ShigaPass</code> results from all samples</td>
</tr>
<tr>
<td>shigatyper.tsv</td>
<td>A merged TSV file with <code>ShigaTyper</code> results from all samples</td>
</tr>
<tr>
<td>shigeifinder.tsv</td>
<td>A merged TSV file with <code>ShigEiFinder</code> results from all samples</td>
</tr>
<tr>
<td>sistr.tsv</td>
<td>A merged TSV file with <code>SISTR</code> results from all samples</td>
</tr>
<tr>
<td>spatyper.tsv</td>
<td>A merged TSV file with <code>spaTyper</code> results from all samples</td>
</tr>
<tr>
<td>ssuissero.tsv</td>
<td>A merged TSV file with <code>SsuisSero</code> results from all samples</td>
</tr>
<tr>
<td>staphopiasccmec.tsv</td>
<td>A merged TSV file with <code>staphopia-sccmec</code> results from all samples</td>
</tr>
<tr>
<td>stecfinder.tsv</td>
<td>A merged TSV file with <code>stecfinder</code> results from all samples</td>
</tr>
</tbody>
</table>
<h4 id="agrvate">AgrVATE<a class="headerlink" href="#agrvate" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results from <a href="https://github.com/VishnuRaghuram94/AgrVATE">AgrVATE</a>.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>-agr_gp.tab</td>
<td>A detailed report for <em>agr</em> kmer matches</td>
</tr>
<tr>
<td>-blastn_log.txt</td>
<td>Log files from programs called by <code>AgrVATE</code></td>
</tr>
<tr>
<td>-summary.tab</td>
<td>A final summary report for <em>agr</em> typing</td>
</tr>
</tbody>
</table>
<h4 id="clermontyping">ClermonTyping<a class="headerlink" href="#clermontyping" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results from <a href="https://github.com/happykhan/ClermonTyping">ClermonTyping</a>.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>&lt;SAMPLE_NAME&gt;.blast.xml</td>
<td>A BLAST XML file with the results of the ClermonTyping analysis</td>
</tr>
<tr>
<td>&lt;SAMPLE_NAME&gt;.html</td>
<td>A HTML file with the results of the ClermonTyping analysis</td>
</tr>
<tr>
<td>&lt;SAMPLE_NAME&gt;.mash.tsv</td>
<td>A TSV file with the Mash distances</td>
</tr>
<tr>
<td>&lt;SAMPLE_NAME&gt;.phylogroups.txt</td>
<td>A TSV file with the final phylogroup assignments</td>
</tr>
</tbody>
</table>
<h4 id="ectyper">ECTyper<a class="headerlink" href="#ectyper" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results from <a href="https://github.com/phac-nml/ecoli_serotyping">ECTyper</a>.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>&lt;SAMPLE_NAME&gt;.tsv</td>
<td>A tab-delimited file with <code>ECTyper</code> result, see <a href="https://github.com/phac-nml/ecoli_serotyping#report-format">ECTyper - Report format</a> for details</td>
</tr>
<tr>
<td>blast_output_alleles.txt</td>
<td>Allele report generated from BLAST results</td>
</tr>
</tbody>
</table>
<h4 id="emmtyper">emmtyper<a class="headerlink" href="#emmtyper" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results from <a href="https://github.com/MDU-PHL/emmtyper">emmtyper</a>.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>&lt;SAMPLE_NAME&gt;.tsv</td>
<td>A tab-delimited file with <code>emmtyper</code> result, see <a href="https://github.com/MDU-PHL/emmtyper#result-format">emmtyper - Result format</a> for details</td>
</tr>
</tbody>
</table>
<h4 id="hicap">hicap<a class="headerlink" href="#hicap" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results from <a href="https://github.com/scwatts/hicap">hicap</a>.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>&lt;SAMPLE_NAME&gt;.gbk</td>
<td>GenBank file and cap locus annotations</td>
</tr>
<tr>
<td>&lt;SAMPLE_NAME&gt;.svg</td>
<td>Visualization of annotated cap locus</td>
</tr>
<tr>
<td>&lt;SAMPLE_NAME&gt;.tsv</td>
<td>A tab-delimited file with <code>hicap</code> results</td>
</tr>
</tbody>
</table>
<h4 id="hpsuissero">HpsuisSero<a class="headerlink" href="#hpsuissero" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results from <a href="https://github.com/jimmyliu1326/HpsuisSero">HpsuisSero</a>.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>&lt;SAMPLE_NAME&gt;_serotyping_res.tsv</td>
<td>A tab-delimited file with <code>HpsuisSero</code> result</td>
</tr>
</tbody>
</table>
<h4 id="genotyphi">GenoTyphi<a class="headerlink" href="#genotyphi" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results from <a href="https://github.com/katholt/genotyphi">GenoTyphi</a>. A
full description of the GenoTyphi output is available at <a href="https://github.com/katholt/genotyphi/blob/main/README.md#explanation-of-columns-in-the-output">GenoTyphi - Output</a></p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>&lt;SAMPLE_NAME&gt;_predictResults.tsv</td>
<td>A tab-delimited file with <code>GenoTyphi</code> results</td>
</tr>
<tr>
<td>&lt;SAMPLE_NAME&gt;.csv</td>
<td>The output of <code>mykrobe predict</code> in comma-separated format</td>
</tr>
<tr>
<td>&lt;SAMPLE_NAME&gt;.json</td>
<td>The output of <code>mykrobe predict</code> in JSON format</td>
</tr>
</tbody>
</table>
<h4 id="kleborate">Kleborate<a class="headerlink" href="#kleborate" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results from <a href="https://github.com/katholt/Kleborate">Kleborate</a>.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>&lt;SAMPLE_NAME&gt;.results.txt</td>
<td>A tab-delimited file with <code>Kleborate</code> result, see  <a href="https://github.com/katholt/Kleborate/wiki/Tests-and-example-outputs#example-output">Kleborate - Example output</a> for more details.</td>
</tr>
</tbody>
</table>
<h4 id="legsta">legsta<a class="headerlink" href="#legsta" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results from <a href="https://github.com/tseemann/legsta">legsta</a>.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>&lt;SAMPLE_NAME&gt;.tsv</td>
<td>A tab-delimited file with <code>legsta</code> result, see <a href="https://github.com/tseemann/legsta#output">legsta - Output</a> for more details</td>
</tr>
</tbody>
</table>
<h4 id="lissero">LisSero<a class="headerlink" href="#lissero" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results from <a href="https://github.com/MDU-PHL/LisSero">LisSero</a>.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>&lt;SAMPLE_NAME&gt;.tsv</td>
<td>A tab-delimited file with <code>LisSero</code> results</td>
</tr>
</tbody>
</table>
<h4 id="mash">Mash<a class="headerlink" href="#mash" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results from <a href="https://github.com/marbl/Mash">Mash</a>.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>&lt;SAMPLE_NAME&gt;-dist.txt</td>
<td>A tab-delimited file with <code>mash dist</code> results</td>
</tr>
</tbody>
</table>
<h4 id="meningotype">meningotype<a class="headerlink" href="#meningotype" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results from <a href="https://github.com/MDU-PHL/meningotype">meningotype</a> .</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>&lt;SAMPLE_NAME&gt;.tsv</td>
<td>A tab-delimited file with <code>meningotype</code> result</td>
</tr>
</tbody>
</table>
<h4 id="ngmaster">ngmaster<a class="headerlink" href="#ngmaster" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results from <a href="https://github.com/MDU-PHL/ngmaster">ngmaster</a>.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>&lt;SAMPLE_NAME&gt;.tsv</td>
<td>A tab-delimited file with <code>ngmaster</code> results</td>
</tr>
</tbody>
</table>
<h4 id="pasty">pasty<a class="headerlink" href="#pasty" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results from <a href="https://github.com/rpetit3/pasty">pasty</a>.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>.blastn.tsv</td>
<td>A tab-delimited file of all blast hits</td>
</tr>
<tr>
<td>.details.tsv</td>
<td>A tab-delimited file with details for each serogroup</td>
</tr>
<tr>
<td>.tsv</td>
<td>A tab-delimited file with the predicted serogroup</td>
</tr>
</tbody>
</table>
<h4 id="pbptyper">pbptyper<a class="headerlink" href="#pbptyper" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results from <a href="https://github.com/rpetit3/pbptyper">pbptyper</a>.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>.tblastn.tsv</td>
<td>A tab-delimited file of all blast hits</td>
</tr>
<tr>
<td>.tsv</td>
<td>A tab-delimited file with the predicted PBP type</td>
</tr>
</tbody>
</table>
<h4 id="seqsero2">SeqSero2<a class="headerlink" href="#seqsero2" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results from <a href="https://github.com/denglab/SeqSero2">SeqSero2</a>.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>&lt;SAMPLE_NAME&gt;_result.tsv</td>
<td>A tab-delimited file with <code>SeqSero2</code> results</td>
</tr>
<tr>
<td>&lt;SAMPLE_NAME&gt;_result.txt</td>
<td>A text file with key-value pairs of <code>SeqSero2</code> results</td>
</tr>
</tbody>
</table>
<h4 id="seroba">Seroba<a class="headerlink" href="#seroba" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results from <a href="https://github.com/sanger-pathogens/seroba">Seroba</a>.
More details about the outputs are available from <a href="https://sanger-pathogens.github.io/seroba/#output">Seroba - Output</a>.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>&lt;SAMPLE_NAME&gt;.tsv</td>
<td>A tab-delimited file with the predicted serotype</td>
</tr>
<tr>
<td>detailed_serogroup_info.txt</td>
<td>Detailed information about the predicted results</td>
</tr>
</tbody>
</table>
<h4 id="shigapass">ShigaPass<a class="headerlink" href="#shigapass" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results from <a href="https://github.com/imanyass/ShigaPass">ShigaPass</a>.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>&lt;SAMPLE_NAME&gt;.csv</td>
<td>A CSV file with the predicted Shigella or EIEC serotype</td>
</tr>
</tbody>
</table>
<h4 id="shigatyper">ShigaTyper<a class="headerlink" href="#shigatyper" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results from <a href="https://github.com/CFSAN-Biostatistics/shigatyper">ShigaTyyper</a>.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>&lt;SAMPLE_NAME&gt;-hits.tsv</td>
<td>Detailed statistics about each individual gene hit</td>
</tr>
<tr>
<td>&lt;SAMPLE_NAME&gt;.tsv</td>
<td>The final predicted serotype by <code>ShigaTyper</code></td>
</tr>
</tbody>
</table>
<h4 id="shigeifinder">ShigEiFinder<a class="headerlink" href="#shigeifinder" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results from <a href="https://github.com/LanLab/ShigEiFinder">ShigEiFinder</a>.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>&lt;SAMPLE_NAME&gt;.tsv</td>
<td>A tab-delimited file with the predicted Shigella or EIEC serotype</td>
</tr>
</tbody>
</table>
<h4 id="sistr">SISTR<a class="headerlink" href="#sistr" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results from <a href="https://github.com/phac-nml/sistr_cmd">SISTR</a>.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>&lt;SAMPLE_NAME&gt;-allele.fasta.gz</td>
<td>A FASTA file of the cgMLST allele search results</td>
</tr>
<tr>
<td>&lt;SAMPLE_NAME&gt;-allele.json.gz</td>
<td>JSON formated cgMLST allele search results, see  <a href="https://github.com/phac-nml/sistr_cmd#cgmlst-allele-search-results">SISTR - cgMLST search results</a> for more details</td>
</tr>
<tr>
<td>&lt;SAMPLE_NAME&gt;-cgmlst.csv</td>
<td>A comma-delimited summary of the cgMLST allele search results</td>
</tr>
<tr>
<td>&lt;SAMPLE_NAME&gt;.tsv</td>
<td>A tab-delimited file with <code>SISTR</code> results, see <a href="https://github.com/phac-nml/sistr_cmd#primary-results-output--o-sistr-results">SISTR - Primary results</a> for more details</td>
</tr>
</tbody>
</table>
<h4 id="spatyper">spaTyper<a class="headerlink" href="#spatyper" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results from <a href="https://github.com/HCGB-IGTP/spaTyper">spaTyper</a>.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>&lt;SAMPLE_NAME&gt;.tsv</td>
<td>A tab-delimited file with <code>spaTyper</code> result</td>
</tr>
</tbody>
</table>
<h4 id="ssuissero">SsuisSero<a class="headerlink" href="#ssuissero" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results from <a href="https://github.com/jimmyliu1326/SsuisSero">SsuisSero</a>.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>&lt;SAMPLE_NAME&gt;_serotyping_res.tsv</td>
<td>A tab-delimited file with <code>SsuisSero</code> results</td>
</tr>
</tbody>
</table>
<h4 id="staphopia-sccmec">staphopia-sccmec<a class="headerlink" href="#staphopia-sccmec" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results from <a href="https://github.com/staphopia/staphopia-sccmec">staphopia-sccmec</a>.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>&lt;SAMPLE_NAME&gt;.tsv</td>
<td>A tab-delimited file with <code>staphopia-sccmec</code> results</td>
</tr>
</tbody>
</table>
<h4 id="tbprofiler">TBProfiler<a class="headerlink" href="#tbprofiler" title="Permanent link">&para;</a></h4>
<p>Below is a description of the <em>per-sample</em> results from <a href="https://github.com/jodyphelan/TBProfiler">TBProfiler</a>.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>&lt;SAMPLE_NAME&gt;.results.csv</td>
<td>A CSV formated <code>TBProfiler</code> result file of resistance and strain type</td>
</tr>
<tr>
<td>&lt;SAMPLE_NAME&gt;.results.json</td>
<td>A JSON formated <code>TBProfiler</code> result file of resistance and strain type</td>
</tr>
<tr>
<td>&lt;SAMPLE_NAME&gt;.results.txt</td>
<td>A text file with <code>TBProfiler</code> results</td>
</tr>
<tr>
<td>&lt;SAMPLE_NAME&gt;.bam</td>
<td>BAM file with alignment details</td>
</tr>
<tr>
<td>&lt;SAMPLE_NAME&gt;.targets.csq.vcf.gz</td>
<td>VCF with variant info again reference genomes</td>
</tr>
</tbody>
</table>
<h3 id="_1"><a class="headerlink" href="#_1" title="Permanent link">&para;</a></h3>
<h4 id="mashdist"><i class="fa-xl fas fa-exclamation-circle"></i> mashdist<a class="headerlink" href="#mashdist" title="Permanent link">&para;</a></h4>
<table>
<thead>
<tr>
<th style="text-align: left;">Parameter</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--mash_sketch</code></td>
<td>The reference sequence as a Mash Sketch (.msh file) <br/><strong>Type:</strong> <code>string</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--mash_seed</code></td>
<td>Seed to provide to the hash function <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>42</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--mash_table</code></td>
<td>Table output (fields will be blank if they do not meet the p-value threshold) <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--mash_m</code></td>
<td>Minimum copies of each k-mer required to pass noise filter for reads <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>1</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--mash_w</code></td>
<td>Probability threshold for warning about low k-mer size. <br/><strong>Type:</strong> <code>number</code>, <strong>Default:</strong> <code>0.01</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--max_p</code></td>
<td>Maximum p-value to report. <br/><strong>Type:</strong> <code>number</code>, <strong>Default:</strong> <code>1.0</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--max_dist</code></td>
<td>Maximum distance to report. <br/><strong>Type:</strong> <code>number</code>, <strong>Default:</strong> <code>1.0</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--merlin_dist</code></td>
<td>Maximum distance to report when using Merlin . <br/><strong>Type:</strong> <code>number</code>, <strong>Default:</strong> <code>0.1</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--full_merlin</code></td>
<td>Go full Merlin and run all species-specific tools, no matter the Mash distance <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--use_fastqs</code></td>
<td>Query with FASTQs instead of the assemblies <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
</tbody>
</table>
<h4 id="agrvate_1"><i class="fa-xl fa-solid fa-toolbox"></i> AgrVATE<a class="headerlink" href="#agrvate_1" title="Permanent link">&para;</a></h4>
<table>
<thead>
<tr>
<th style="text-align: left;">Parameter</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;"><i class="fa-lg fa-solid fa-toggle-on"></i><code>--typing_only</code></td>
<td>agr typing only. Skips agr operon extraction and frameshift detection <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
</tbody>
</table>
<h4 id="ectyper_1"><i class="fa-xl fas fa-exclamation-circle"></i> ECTyper<a class="headerlink" href="#ectyper_1" title="Permanent link">&para;</a></h4>
<table>
<thead>
<tr>
<th style="text-align: left;">Parameter</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--opid</code></td>
<td>Percent identity required for an O antigen allele match <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>90</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--opcov</code></td>
<td>Minumum percent coverage required for an O antigen allele match <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>90</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--hpid</code></td>
<td>Percent identity required for an H antigen allele match <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>95</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--hpcov</code></td>
<td>Minumum percent coverage required for an H antigen allele match <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>50</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--verify</code></td>
<td>Enable E. coli species verification <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--print_alleles</code></td>
<td>Prints the allele sequences if enabled as the final column <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
</tbody>
</table>
<h4 id="emmtyper_1"><i class="fa-xl fas fa-exclamation-circle"></i> emmtyper<a class="headerlink" href="#emmtyper_1" title="Permanent link">&para;</a></h4>
<table>
<thead>
<tr>
<th style="text-align: left;">Parameter</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--emmtyper_wf</code></td>
<td>Workflow for emmtyper to use. <br/><strong>Type:</strong> <code>string</code>, <strong>Default:</strong> <code>blast</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--cluster_distance</code></td>
<td>Distance between cluster of matches to consider as different clusters <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>500</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--percid</code></td>
<td>Minimal percent identity of sequence <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>95</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--culling_limit</code></td>
<td>Total hits to return in a position <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>5</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--mismatch</code></td>
<td>Threshold for number of mismatch to allow in BLAST hit <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>5</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--align_diff</code></td>
<td>Threshold for difference between alignment length and subject length in BLAST <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>5</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--gap</code></td>
<td>Threshold gap to allow in BLAST hit <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>2</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--min_perfect</code></td>
<td>Minimum size of perfect match at 3 primer end <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>15</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--min_good</code></td>
<td>Minimum size where there must be 2 matches for each mismatch <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>15</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--max_size</code></td>
<td>Maximum size of PCR product <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>2000</code></td>
</tr>
</tbody>
</table>
<h4 id="hicap_1"><i class="fa-xl fas fa-exclamation-circle"></i> hicap<a class="headerlink" href="#hicap_1" title="Permanent link">&para;</a></h4>
<table>
<thead>
<tr>
<th style="text-align: left;">Parameter</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-boxes"></i><code>--database_dir</code></td>
<td>Directory containing locus database <br/><strong>Type:</strong> <code>string</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-boxes"></i><code>--model_fp</code></td>
<td>Path to prodigal model <br/><strong>Type:</strong> <code>string</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--full_sequence</code></td>
<td>Write the full input sequence out to the genbank file rather than just the region surrounding and including the locus <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--hicap_debug</code></td>
<td>hicap will print debug messages <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-angle-double-down"></i><code>--gene_coverage</code></td>
<td>Minimum percentage coverage to consider a single gene complete <br/><strong>Type:</strong> <code>number</code>, <strong>Default:</strong> <code>0.8</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-angle-double-down"></i><code>--gene_identity</code></td>
<td>Minimum percentage identity to consider a single gene complete <br/><strong>Type:</strong> <code>number</code>, <strong>Default:</strong> <code>0.7</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-angle-double-down"></i><code>--broken_gene_length</code></td>
<td>Minimum length to consider a broken gene <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>60</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-angle-double-down"></i><code>--broken_gene_identity</code></td>
<td>Minimum percentage identity to consider a broken gene <br/><strong>Type:</strong> <code>number</code>, <strong>Default:</strong> <code>0.8</code></td>
</tr>
</tbody>
</table>
<h4 id="genotyphi_1"><i class="fa-xl fas fa-exclamation-circle"></i> GenoTyphi<a class="headerlink" href="#genotyphi_1" title="Permanent link">&para;</a></h4>
<table>
<thead>
<tr>
<th style="text-align: left;">Parameter</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--kmer</code></td>
<td>K-mer length <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>21</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--min_depth</code></td>
<td>Minimum depth <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>1</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--model</code></td>
<td>Genotype model used. <br/><strong>Type:</strong> <code>string</code>, <strong>Default:</strong> <code>kmer_count</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--report_all_calls</code></td>
<td>Report all calls <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-italic"></i><code>--mykrobe_opts</code></td>
<td>Extra Mykrobe options in quotes <br/><strong>Type:</strong> <code>string</code></td>
</tr>
</tbody>
</table>
<h4 id="kleborate_1"><i class="fa-xl fas fa-exclamation-circle"></i> Kleborate<a class="headerlink" href="#kleborate_1" title="Permanent link">&para;</a></h4>
<table>
<thead>
<tr>
<th style="text-align: left;">Parameter</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-boxes"></i><code>--kleborate_preset</code></td>
<td>Preset module to use for Kleborate <br/><strong>Type:</strong> <code>string</code>, <strong>Default:</strong> <code>kpsc</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-italic"></i><code>--kleborate_opts</code></td>
<td>Extra options in quotes for Kleborate <br/><strong>Type:</strong> <code>string</code></td>
</tr>
</tbody>
</table>
<h4 id="legsta_1"><i class="fa-xl fas fa-exclamation-circle"></i> legsta<a class="headerlink" href="#legsta_1" title="Permanent link">&para;</a></h4>
<table>
<thead>
<tr>
<th style="text-align: left;">Parameter</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--noheader</code></td>
<td>Don't print header row <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
</tbody>
</table>
<h4 id="lissero_1"><i class="fa-xl fas fa-exclamation-circle"></i> LisSero<a class="headerlink" href="#lissero_1" title="Permanent link">&para;</a></h4>
<table>
<thead>
<tr>
<th style="text-align: left;">Parameter</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--min_id</code></td>
<td>Minimum percent identity to accept a match <br/><strong>Type:</strong> <code>number</code>, <strong>Default:</strong> <code>95.0</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--min_cov</code></td>
<td>Minimum coverage of the gene to accept a match <br/><strong>Type:</strong> <code>number</code>, <strong>Default:</strong> <code>95.0</code></td>
</tr>
</tbody>
</table>
<h4 id="meningotype_1"><i class="fa-xl fas fa-exclamation-circle"></i> meningotype<a class="headerlink" href="#meningotype_1" title="Permanent link">&para;</a></h4>
<p>You can use these parameters to fine-tune your meningotype analysis</p>
<table>
<thead>
<tr>
<th style="text-align: left;">Parameter</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--finetype</code></td>
<td>perform porA and fetA fine typing <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--porB</code></td>
<td>perform porB sequence typing (NEIS2020) <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--bast</code></td>
<td>perform Bexsero antigen sequence typing (BAST) <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--mlst</code></td>
<td>perform MLST <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--all</code></td>
<td>perform MLST, porA, fetA, porB, BAST typing <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
</tbody>
</table>
<h4 id="ngmaster_1"><i class="fa-xl fas fa-exclamation-circle"></i> ngmaster<a class="headerlink" href="#ngmaster_1" title="Permanent link">&para;</a></h4>
<table>
<thead>
<tr>
<th style="text-align: left;">Parameter</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--csv</code></td>
<td>output comma-separated format (CSV) rather than tab-separated <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
</tbody>
</table>
<h4 id="pasty_1"><i class="fa-xl fas fa-exclamation-circle"></i> pasty<a class="headerlink" href="#pasty_1" title="Permanent link">&para;</a></h4>
<table>
<thead>
<tr>
<th style="text-align: left;">Parameter</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--pasty_min_pident</code></td>
<td>Minimum percent identity to count a hit <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>95</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--pasty_min_coverage</code></td>
<td>Minimum percent coverage to count a hit <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>95</code></td>
</tr>
</tbody>
</table>
<h4 id="pbptyper_1"><i class="fa-xl fas fa-exclamation-circle"></i> pbptyper<a class="headerlink" href="#pbptyper_1" title="Permanent link">&para;</a></h4>
<table>
<thead>
<tr>
<th style="text-align: left;">Parameter</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--pbptyper_min_pident</code></td>
<td>Minimum percent identity to count a hit <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>95</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--pbptyper_min_coverage</code></td>
<td>Minimum percent coverage to count a hit <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>95</code></td>
</tr>
</tbody>
</table>
<h4 id="seqsero2_1"><i class="fa-xl fas fa-exclamation-circle"></i> SeqSero2<a class="headerlink" href="#seqsero2_1" title="Permanent link">&para;</a></h4>
<table>
<thead>
<tr>
<th style="text-align: left;">Parameter</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--run_mode</code></td>
<td>Workflow to run. 'a' allele mode, or 'k' k-mer mode <br/><strong>Type:</strong> <code>string</code>, <strong>Default:</strong> <code>k</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--input_type</code></td>
<td>Input format to analyze. 'assembly' or 'fastq' <br/><strong>Type:</strong> <code>string</code>, <strong>Default:</strong> <code>assembly</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--bwa_mode</code></td>
<td>Algorithms for bwa mapping for allele mode <br/><strong>Type:</strong> <code>string</code>, <strong>Default:</strong> <code>mem</code></td>
</tr>
</tbody>
</table>
<h4 id="sistr_1"><i class="fa-xl fas fa-exclamation-circle"></i> SISTR<a class="headerlink" href="#sistr_1" title="Permanent link">&para;</a></h4>
<table>
<thead>
<tr>
<th style="text-align: left;">Parameter</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--full_cgmlst</code></td>
<td>Use the full set of cgMLST alleles which can include highly similar alleles <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
</tbody>
</table>
<h4 id="spatyper_1"><i class="fa-xl fas fa-exclamation-circle"></i> spaTyper<a class="headerlink" href="#spatyper_1" title="Permanent link">&para;</a></h4>
<table>
<thead>
<tr>
<th style="text-align: left;">Parameter</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-file-archive"></i><code>--repeats</code></td>
<td>List of spa repeats <br/><strong>Type:</strong> <code>string</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-file-archive"></i><code>--repeat_order</code></td>
<td>List spa types and order of repeats <br/><strong>Type:</strong> <code>string</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--do_enrich</code></td>
<td>Do PCR product enrichment <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
</tbody>
</table>
<h4 id="staphopia-sccmec_1"><i class="fa-xl fas fa-exclamation-circle"></i> staphopia-sccmec<a class="headerlink" href="#staphopia-sccmec_1" title="Permanent link">&para;</a></h4>
<table>
<thead>
<tr>
<th style="text-align: left;">Parameter</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--hamming</code></td>
<td>Report the results as hamming distances <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
</tbody>
</table>
<h3 id="citations_7">Citations<a class="headerlink" href="#citations_7" title="Permanent link">&para;</a></h3>
<p>If you use Bactopia and <code>merlin</code> results in your analysis, please cite the following.</p>
<ul>
<li>
<p><a href="https://bactopia.github.io/">Bactopia</a><br />
    Petit III RA, Read TD <a href="https://doi.org/10.1128/mSystems.00190-20">Bactopia - a flexible pipeline for complete analysis of bacterial genomes.</a> <em>mSystems</em> 5 (2020)</p>
</li>
<li>
<p><a href="https://github.com/VishnuRaghuram94/AgrVATE">AgrVATE</a><br />
    Raghuram V. <a href="https://github.com/VishnuRaghuram94/AgrVATE">AgrVATE: Rapid identification of Staphylococcus aureus agr locus type and agr operon variants.</a> (GitHub)</p>
</li>
<li>
<p><a href="https://github.com/happykhan/ClermonTyping">ClermontTyping</a><br />
    Beghain J, Bridier-Nahmias A, Le Nagard H, Denamur E, Clermont O. <a href="https://doi.org/10.1099/mgen.0.000192">ClermonTyping: an easy-to-use and accurate in silico method for Escherichia genus strain phylotyping.</a> Microbial Genomics, 4(7), e000192. (2018)</p>
</li>
<li>
<p><a href="https://bioinf.shenwei.me/csvtk/">csvtk</a><br />
    Shen, W <a href="https://github.com/shenwei356/csvtk/">csvtk: A cross-platform, efficient and practical CSV/TSV toolkit in Golang.</a> (GitHub)</p>
</li>
<li>
<p><a href="https://github.com/phac-nml/ecoli_serotyping">ECTyper</a><br />
    Laing C, Bessonov K, Sung S, La Rose C <a href="https://github.com/phac-nml/ecoli_serotyping">ECTyper - In silico prediction of <em>Escherichia coli</em> serotype</a> (GitHub)  </p>
</li>
<li>
<p><a href="https://github.com/MDU-PHL/emmtyper">emmtyper</a><br />
    Tan A, Seemann T, Lacey D, Davies M, Mcintyre L, Frost H, Williamson D, Gonçalves da Silva A <a href="https://github.com/MDU-PHL/emmtyper">emmtyper - emm Automatic Isolate Labeller</a> (GitHub)</p>
</li>
<li>
<p><a href="https://github.com/katholt/genotyphi">GenoTyphi</a><br />
    Wong VK, Baker S, Connor TR, Pickard D, Page AJ, Dave J, Murphy N, Holliman R, Sefton A, Millar M, Dyson ZA, Dougan G, Holt KE, &amp; International Typhoid Consortium. <a href="https://doi.org/10.1038/ncomms12827">An extended genotyping framework for Salmonella enterica serovar Typhi, the cause of human typhoid</a> <em>Nature Communications</em> 7, 12827. (2016)</p>
</li>
<li>
<p><a href="https://github.com/scwatts/hicap">hicap</a><br />
    Watts SC, Holt KE <a href="https://doi.org/10.1128/JCM.00190-19">hicap: in silico serotyping of the Haemophilus influenzae capsule locus.</a> <em>Journal of Clinical Microbiology</em> JCM.00190-19 (2019)</p>
</li>
<li>
<p><a href="https://github.com/jimmyliu1326/HpsuisSero">HpsuisSero</a><br />
    Lui J <a href="https://github.com/jimmyliu1326/HpsuisSero">HpsuisSero: Rapid <em>Haemophilus parasuis</em> serotyping</a> (GitHub)</p>
</li>
<li>
<p><a href="https://github.com/katholt/Kleborate">Kleborate</a><br />
    Lam MMC, Wick RR, Watts, SC, Cerdeira LT, Wyres KL, Holt KE <a href="https://doi.org/10.1038/s41467-021-24448-3">A genomic surveillance framework and genotyping tool for Klebsiella pneumoniae and its related species complex.</a> <em>Nat Commun</em> 12, 4188 (2021)</p>
</li>
<li>
<p><a href="https://github.com/tseemann/legsta">legsta</a><br />
    Seemann T <a href="https://github.com/tseemann/legsta">legsta: In silico Legionella pneumophila Sequence Based Typing</a> (GitHub)</p>
</li>
<li>
<p><a href="https://github.com/MDU-PHL/LisSero">LisSero</a><br />
    Kwong J, Zhang J, Seeman T, Horan, K, Gonçalves da Silva A <a href="https://github.com/MDU-PHL/LisSero">LisSero - <em>In silico</em> serotype prediction for <em>Listeria monocytogenes</em></a> (GitHub)</p>
</li>
<li>
<p><a href="https://github.com/marbl/Mash">Mash</a><br />
    Ondov BD, Treangen TJ, Melsted P, Mallonee AB, Bergman NH, Koren S, Phillippy AM <a href="http://dx.doi.org/10.1186/s13059-016-0997-x">Mash: fast genome and metagenome distance estimation using MinHash</a>. <em>Genome Biol</em> 17, 132 (2016)</p>
</li>
<li>
<p><a href="https://github.com/MDU-PHL/meningotype">meningotype</a><br />
    Kwong JC, Gonçalves da Silva A, Stinear TP, Howden BP, &amp; Seemann T <a href="https://github.com/MDU-PHL/meningotype">meningotype: in silico typing for <em>Neisseria meningitidis</em>.</a> (GitHub)</p>
</li>
<li>
<p><a href="https://github.com/Mykrobe-tools/mykrobe">Mykrobe</a><br />
    Hunt M, Bradley P, Lapierre SG, Heys S, Thomsit M, Hall MB, Malone KM, Wintringer P, Walker TM, Cirillo DM, Comas I, Farhat MR, Fowler P, Gardy J, Ismail N, Kohl TA, Mathys V, Merker M, Niemann S, Omar SV, Sintchenko V, Smith G, Supply P, Tahseen S, Wilcox M, Arandjelovic I, Peto TEA, Crook, DW, Iqbal Z <a href="https://doi.org/10.12688/wellcomeopenres.15603.1">Antibiotic resistance prediction for Mycobacterium tuberculosis from genome sequence data with Mykrobe</a> <em>Wellcome Open Research</em> 4, 191. (2019)</p>
</li>
<li>
<p><a href="https://github.com/MDU-PHL/ngmaster">ngmaster</a><br />
    Kwong J, Gonçalves da Silva A, Schultz M, Seeman T <a href="https://github.com/MDU-PHL/ngmaster">ngmaster - <em>In silico</em> multi-antigen sequence typing for <em>Neisseria gonorrhoeae</em> (NG-MAST)</a> (GitHub)</p>
</li>
<li>
<p><a href="https://github.com/rpetit3/pasty">pasty</a><br />
    Petit III RA <a href="https://github.com/rpetit3/pasty">pasty: in silico serogrouping of <em>Pseudomonas aeruginosa</em> isolates</a> (GitHub)</p>
</li>
<li>
<p><a href="https://github.com/rpetit3/pbptyper">pbptyper</a><br />
    Petit III RA <a href="https://github.com/rpetit3/pbptyper">pbptyper: In silico Penicillin Binding Protein (PBP) typer for <em>Streptococcus pneumoniae</em> assemblies</a> (GitHub)</p>
</li>
<li>
<p><a href="https://github.com/denglab/SeqSero2">SeqSero2</a><br />
    Zhang S, Den-Bakker HC, Li S, Dinsmore BA, Lane C, Lauer AC, Fields PI, Deng X. <a href="https://doi.org/10.1128/AEM.01746-19">SeqSero2: rapid and improved Salmonella serotype determination using whole genome sequencing data.</a> <em>Appl Environ Microbiology</em> 85(23):e01746-19 (2019)</p>
</li>
<li>
<p><a href="https://github.com/imanyass/ShigaPass">shigapass</a><br />
    Yassine I, Hansen EE, Lefèvre S, Ruckly C, Carle I, Lejay-Collin M, Fabre L, Rafei R, Pardos de la Gandara M, Daboussi F, Shahin A, Weill FX <a href="https://doi.org/10.1099%2Fmgen.0.000961">ShigaPass: an in silico tool predicting Shigella serotypes from whole-genome sequencing assemblies.</a> <em>Microb Genomics</em> 9(3) (2023)</p>
</li>
<li>
<p><a href="https://github.com/CFSAN-Biostatistics/shigatyper">ShigaTyper</a><br />
    Wu Y, Lau HK, Lee T, Lau DK, Payne J <a href="https://doi.org/10.1128/AEM.00165-19">In Silico Serotyping Based on Whole-Genome Sequencing Improves the Accuracy of Shigella Identification.</a> <em>Applied and Environmental Microbiology</em>, 85(7). (2019)</p>
</li>
<li>
<p><a href="https://github.com/LanLab/ShigEiFinder">ShigEiFinder</a><br />
    Zhang X, Payne M, Nguyen T, Kaur S, Lan R <a href="https://doi.org/10.1099/mgen.0.000704">Cluster-specific gene markers enhance Shigella and enteroinvasive Escherichia coli in silico serotyping.</a> Microbial Genomics, 7(12). (2021)</p>
</li>
<li>
<p><a href="https://github.com/phac-nml/sistr_cmd">SISTR</a><br />
    Yoshida CE, Kruczkiewicz P, Laing CR, Lingohr EJ, Gannon VPJ, Nash JHE, Taboada EN <a href="https://doi.org/10.1371/journal.pone.0147101">The Salmonella In Silico Typing Resource (SISTR): An Open Web-Accessible Tool for Rapidly Typing and Subtyping Draft Salmonella Genome Assemblies.</a> <em>PloS One</em>, 11(1), e0147101. (2016)</p>
</li>
<li>
<p><a href="https://github.com/HCGB-IGTP/spaTyper">spaTyper</a><br />
    Sanchez-Herrero JF, and Sullivan M <a href="http://doi.org/10.5281/zenodo.4063625">spaTyper: Staphylococcal protein A (spa) characterization pipeline</a>. Zenodo. (2020)</p>
</li>
<li>
<p><a href="https://github.com/jimmyliu1326/SsuisSero">SsuisSero</a><br />
    Lui J <a href="https://github.com/jimmyliu1326/SsuisSero">SsuisSero: Rapid <em>Streptococcus suis</em> serotyping</a> (GitHub)</p>
</li>
<li>
<p><a href="https://github.com/staphopia/staphopia-sccmec">staphopia-sccmec</a><br />
    Petit III RA, Read TD <a href="http://dx.doi.org/10.7717/peerj.5261"><em>Staphylococcus aureus</em> viewed from the perspective of 40,000+ genomes.</a> <em>PeerJ</em> 6, e5261 (2018)</p>
</li>
<li>
<p><a href="https://github.com/jodyphelan/TBProfiler">TBProfiler</a><br />
    Phelan JE, O’Sullivan DM, Machado D, Ramos J, Oppong YEA, Campino S, O’Grady J, McNerney R, Hibberd ML, Viveiros M, Huggett JF, Clark TG <a href="https://doi.org/10.1186/s13073-019-0650-x">Integrating informatics tools and portable sequencing technology for rapid detection of resistance to anti-tuberculous drugs.</a> <em>Genome Med</em> 11, 41 (2019)</p>
</li>
</ul>
<h2 id="additional-parameters">Additional Parameters<a class="headerlink" href="#additional-parameters" title="Permanent link">&para;</a></h2>
<h3 id="optional"><i class="fa-xl fa-solid fa-gears"></i> Optional<a class="headerlink" href="#optional" title="Permanent link">&para;</a></h3>
<p>These optional parameters can be useful in certain settings.</p>
<table>
<thead>
<tr>
<th style="text-align: left;">Parameter</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-folder"></i><code>--outdir</code></td>
<td>Base directory to write results to <br/><strong>Type:</strong> <code>string</code>, <strong>Default:</strong> <code>bactopia</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-expand-arrows-alt"></i><code>--skip_compression</code></td>
<td>Ouput files will not be compressed <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-folder"></i><code>--datasets</code></td>
<td>The path to cache datasets to <br/><strong>Type:</strong> <code>string</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-trash-restore"></i><code>--keep_all_files</code></td>
<td>Keeps all analysis files created <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
</tbody>
</table>
<h3 id="max-job-request"><i class="fa-xl fa-solid fa-arrow-up-right-dots"></i> Max Job Request<a class="headerlink" href="#max-job-request" title="Permanent link">&para;</a></h3>
<p>Set the top limit for requested resources for any single job.</p>
<table>
<thead>
<tr>
<th style="text-align: left;">Parameter</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-redo"></i><code>--max_retry</code></td>
<td>Maximum times to retry a process before allowing it to fail. <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>3</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-microchip"></i><code>--max_cpus</code></td>
<td>Maximum number of CPUs that can be requested for any single job. <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>4</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-memory"></i><code>--max_memory</code></td>
<td>Maximum amount of memory that can be requested for any single job. <br/><strong>Type:</strong> <code>string</code>, <strong>Default:</strong> <code>128.GB</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg far fa-clock"></i><code>--max_time</code></td>
<td>Maximum amount of time that can be requested for any single job. <br/><strong>Type:</strong> <code>string</code>, <strong>Default:</strong> <code>240.h</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-angle-double-up"></i><code>--max_downloads</code></td>
<td>Maximum number of samples to download at a time <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>3</code></td>
</tr>
</tbody>
</table>
<h3 id="nextflow-configuration"><i class="fa-xl fa-solid fa-screwdriver-wrench"></i> Nextflow Configuration<a class="headerlink" href="#nextflow-configuration" title="Permanent link">&para;</a></h3>
<p>to fine-tune your Nextflow setup.</p>
<table>
<thead>
<tr>
<th style="text-align: left;">Parameter</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-cog"></i><code>--nfconfig</code></td>
<td>A Nextflow compatible config file for custom profiles, loaded last and will overwrite existing variables if set. <br/><strong>Type:</strong> <code>string</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-copy"></i><code>--publish_dir_mode</code></td>
<td>Method used to save pipeline results to output directory. <br/><strong>Type:</strong> <code>string</code>, <strong>Default:</strong> <code>copy</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-cogs"></i><code>--infodir</code></td>
<td>Directory to keep pipeline Nextflow logs and reports. <br/><strong>Type:</strong> <code>string</code>, <strong>Default:</strong> <code>${params.outdir}/pipeline_info</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-recycle"></i><code>--force</code></td>
<td>Nextflow will overwrite existing output files. <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-trash-alt"></i><code>--cleanup_workdir</code></td>
<td>After Bactopia is successfully executed, the <code>work</code> directory will be deleted. <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
</tbody>
</table>
<h3 id="institutional-config-options"><i class="fa-xl fas fa-university"></i> Institutional config options<a class="headerlink" href="#institutional-config-options" title="Permanent link">&para;</a></h3>
<p>used to describe centralized config profiles. These should not be edited.</p>
<table>
<thead>
<tr>
<th style="text-align: left;">Parameter</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-users-cog"></i><code>--custom_config_version</code></td>
<td>Git commit id for Institutional configs. <br/><strong>Type:</strong> <code>string</code>, <strong>Default:</strong> <code>master</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-users-cog"></i><code>--custom_config_base</code></td>
<td>Base directory for Institutional configs. <br/><strong>Type:</strong> <code>string</code>, <strong>Default:</strong> <code>https://raw.githubusercontent.com/nf-core/configs/master</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-users-cog"></i><code>--config_profile_name</code></td>
<td>Institutional config name. <br/><strong>Type:</strong> <code>string</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-users-cog"></i><code>--config_profile_description</code></td>
<td>Institutional config description. <br/><strong>Type:</strong> <code>string</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-users-cog"></i><code>--config_profile_contact</code></td>
<td>Institutional config contact information. <br/><strong>Type:</strong> <code>string</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-users-cog"></i><code>--config_profile_url</code></td>
<td>Institutional config URL link. <br/><strong>Type:</strong> <code>string</code></td>
</tr>
</tbody>
</table>
<h3 id="nextflow-profile"><i class="fa-xl fa-regular fa-address-card"></i> Nextflow Profile<a class="headerlink" href="#nextflow-profile" title="Permanent link">&para;</a></h3>
<p>to fine-tune your Nextflow setup.</p>
<table>
<thead>
<tr>
<th style="text-align: left;">Parameter</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-folder"></i><code>--condadir</code></td>
<td>Directory to Nextflow should use for Conda environments <br/><strong>Type:</strong> <code>string</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-box"></i><code>--registry</code></td>
<td>Docker registry to pull containers from. <br/><strong>Type:</strong> <code>string</code>, <strong>Default:</strong> <code>dockerhub</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-folder"></i><code>--datasets_cache</code></td>
<td>Directory where downloaded datasets should be stored. <br/><strong>Type:</strong> <code>string</code>, <strong>Default:</strong> <code>&lt;BACTOPIA_DIR&gt;/data/datasets</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-folder"></i><code>--singularity_cache_dir</code></td>
<td>Directory where remote Singularity images are stored. <br/><strong>Type:</strong> <code>string</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-toolbox"></i><code>--singularity_pull_docker_container</code></td>
<td>Instead of directly downloading Singularity images for use with Singularity, force the workflow to pull and convert Docker containers instead. <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-recycle"></i><code>--force_rebuild</code></td>
<td>Force overwrite of existing pre-built environments. <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-clipboard-list"></i><code>--queue</code></td>
<td>Comma-separated name of the queue(s) to be used by a job scheduler (e.g. AWS Batch or SLURM) <br/><strong>Type:</strong> <code>string</code>, <strong>Default:</strong> <code>general,high-memory</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-clipboard-list"></i><code>--cluster_opts</code></td>
<td>Additional options to pass to the executor. (e.g. SLURM: '--account=my_acct_name' <br/><strong>Type:</strong> <code>string</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-clipboard-list"></i><code>--container_opts</code></td>
<td>Additional options to pass to Apptainer, Docker, or Singularityu. (e.g. Singularity: '-D <code>pwd</code>' <br/><strong>Type:</strong> <code>string</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-toggle-off"></i><code>--disable_scratch</code></td>
<td>All intermediate files created on worker nodes of will be transferred to the head node. <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
</tbody>
</table>
<h3 id="helpful"><i class="fa-xl fa-solid fa-reply-all"></i> Helpful<a class="headerlink" href="#helpful" title="Permanent link">&para;</a></h3>
<p>Uncommonly used parameters that might be useful.</p>
<table>
<thead>
<tr>
<th style="text-align: left;">Parameter</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-palette"></i><code>--monochrome_logs</code></td>
<td>Do not use coloured log outputs. <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-remove-format"></i><code>--nfdir</code></td>
<td>Print directory Nextflow has pulled Bactopia to <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg far fa-clock"></i><code>--sleep_time</code></td>
<td>The amount of time (seconds) Nextflow will wait after setting up datasets before execution. <br/><strong>Type:</strong> <code>integer</code>, <strong>Default:</strong> <code>5</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-tasks"></i><code>--validate_params</code></td>
<td>Boolean whether to validate parameters against the schema at runtime <br/><strong>Type:</strong> <code>boolean</code>, <strong>Default:</strong> <code>True</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-question-circle"></i><code>--help</code></td>
<td>Display help text. <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-bacteria"></i><code>--wf</code></td>
<td>Specify which workflow or Bactopia Tool to execute <br/><strong>Type:</strong> <code>string</code>, <strong>Default:</strong> <code>bactopia</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-list"></i><code>--list_wfs</code></td>
<td>List the available workflows and Bactopia Tools to use with '--wf' <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg far fa-eye"></i><code>--show_hidden_params</code></td>
<td>Show all params when using <code>--help</code> <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-question-circle"></i><code>--help_all</code></td>
<td>An alias for --help --show_hidden_params <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
<tr>
<td style="text-align: left;"><i class="fa-lg fas fa-info"></i><code>--version</code></td>
<td>Display version text. <br/><strong>Type:</strong> <code>boolean</code></td>
</tr>
</tbody>
</table>







  
  



  




                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg>
  Back to top
</button>
        
      </main>
      
        <footer class="md-footer">
  
    
      
      <nav class="md-footer__inner md-grid" aria-label="Footer" >
        
          
          <a href="../bactopia/merlin/" class="md-footer__link md-footer__link--prev" aria-label="Previous: Merlin">
            <div class="md-footer__button md-icon">
              
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
            </div>
            <div class="md-footer__title">
              <span class="md-footer__direction">
                Previous
              </span>
              <div class="md-ellipsis">
                Merlin
              </div>
            </div>
          </a>
        
        
          
          <a href="../changelog/" class="md-footer__link md-footer__link--next" aria-label="Next: Changelog">
            <div class="md-footer__title">
              <span class="md-footer__direction">
                Next
              </span>
              <div class="md-ellipsis">
                Changelog
              </div>
            </div>
            <div class="md-footer__button md-icon">
              
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4Z"/></svg>
            </div>
          </a>
        
      </nav>
    
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright © 2019-2024 Robert A. Petit III. All rights reserved. <a href="#__consent">Change Cookie Settings</a>

    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs Insiders
    </a>
  
</div>
      
        <div class="md-social">
  
    
    
    
    
      
      
    
    <a href="https://github.com/bactopia/bactopia" target="_blank" rel="noopener" title="github.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.6.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
    </a>
  
    
    
    
    
      
      
    
    <a href="https://twitter.com/rpetit3" target="_blank" rel="noopener" title="twitter.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.6.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"/></svg>
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
      <div class="md-consent" data-md-component="consent" id="__consent" hidden>
        <div class="md-consent__overlay"></div>
        <aside class="md-consent__inner">
          <form class="md-consent__form md-grid md-typeset" name="consent">
            

  
    
  


  
    
  



  


<h4>Cookie Consent</h4>
<p>We use cookies to recognize your repeated visits and preferences, as well as to measure the effectiveness of our documentation and whether users find what they're searching for. With your consent, you're helping us to make our documentation better.</p>
<input class="md-toggle" type="checkbox" id="__settings" >
<div class="md-consent__settings">
  <ul class="task-list">
    
      
      
        
        
      
      <li class="task-list-item">
        <label class="task-list-control">
          <input type="checkbox" name="analytics" checked>
          <span class="task-list-indicator"></span>
          Google Analytics
        </label>
      </li>
    
      
      
        
        
      
      <li class="task-list-item">
        <label class="task-list-control">
          <input type="checkbox" name="github" checked>
          <span class="task-list-indicator"></span>
          GitHub
        </label>
      </li>
    
  </ul>
</div>
<div class="md-consent__controls">
  
    
      <button class="md-button md-button--primary">Accept</button>
    
    
    
  
    
    
    
      <label class="md-button" for="__settings">Manage settings</label>
    
  
</div>
          </form>
        </aside>
      </div>
      <script>var consent=__md_get("__consent");if(consent)for(var input of document.forms.consent.elements)input.name&&(input.checked=consent[input.name]||!1);else"file:"!==location.protocol&&setTimeout(function(){document.querySelector("[data-md-component=consent]").hidden=!1},250);var action,form=document.forms.consent;for(action of["submit","reset"])form.addEventListener(action,function(e){if(e.preventDefault(),"reset"===e.type)for(var n of document.forms.consent.elements)n.name&&(n.checked=!1);__md_set("__consent",Object.fromEntries(Array.from(new FormData(form).keys()).map(function(e){return[e,!0]}))),location.hash="",location.reload()})</script>
    
    <script id="__config" type="application/json">{"base": "..", "features": ["announce.dismiss", "content.action.edit", "content.code.copy", "navigation.footer", "navigation.instant", "navigation.instant.prefetch", "navigation.tabs", "navigation.tabs.sticky", "navigation.tracking", "navigation.top"], "search": "../assets/javascripts/workers/search.1e90e0fb.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": {"provider": "mike"}}</script>
    
    
      <script src="../assets/javascripts/bundle.4a1b0f67.min.js"></script>
      
        <script src="../custom/fontawesome.js"></script>
      
        <script src="../custom/jquery.min.js"></script>
      
        <script src="../custom/jquery.magnific-popup.min.js"></script>
      
        <script src="../custom/extra.js"></script>
      
    
  </body>
</html>