
Since its launch in 2019, Bactopia has seen significant academic usage, being cited in at
least 58 scholarly publications. We are excited to see Bactopia being used
by the community and hope that it continues to be a valuable tool for researchers.

!!! info "Total citations might be out of date"
    The citations below are updated regularly, but not daily. To get the most up to date list
    of citations, please visit
    [Google Scholar](https://scholar.google.com/scholar?cites=11269093140070503615) or 
    [PubMed](https://pubmed.ncbi.nlm.nih.gov/?linkname=pubmed_pubmed_citedin&from_uid=32753501)

58: Babiker HA, Al-Jardani A, Al-Azri S, Petit III RA, Saad E, Al-Mahrouqi S, Mohamed RAH, Al-Hamidhi S, Balkhair AA, Al Kharusi N, Al Balushi L, Al Zadjali S, Pérez-Pardal L, Beja-Pereira A, Babiker A. [Mycobacterium tuberculosis epidemiology in Oman: whole-genome sequencing uncovers transmission pathways](https://doi.org/10.1128/spectrum.02420-23) _Microbiology Spectrum_ (2023).

57: Almeida FMd, Campos TAd, Pappas Jr GJ. [Scalable and versatile container-based pipelines for de novo genome assembly and bacterial annotation.](https://doi.org/10.12688/f1000research.139488.1) _F1000 Research_ (2023).

56: Vanderwoude J, Azimi S, Read TD, Read TD, Diggle SP. [Evolutionary Drivers of Antimicrobial Resistance Diversity of _Pseudomonas aeruginosa_ in Cystic Fibrosis Lung Infection](https://doi.org/10.1101/2023.06.14.544983) _bioRxiv_ (2023).

55: Machado RAR, Bhat AH, Castaneda-Alvarez C, Askary TH, Půža V, Pagès S, Abolafia J. [Xenorhabdus aichiensis sp. nov., Xenorhabdus anantnagensis sp. nov., and Xenorhabdus yunnanensis sp. nov., Isolated from Steinernema Entomopathogenic Nematodes.](https://doi.org/10.1007/s00284-023-03373-2) _Curr Microbiol_ (2023).

54: Ortega-Sanz I, Barbero-Aparicio JA, Canepa-Oneto A, Rovira J, Melero B. [CamPype: an open-source workflow for automated bacterial whole-genome sequencing analysis focused on Campylobacter.](https://doi.org/10.1186/s12859-023-05414-w) _BMC Bioinformatics_ (2023).

53: Elek CKA, Brown TL, Le Viet T, Evans R, Baker DJ, Telatin A, Tiwari SK, Al-Khanaq H, Thilliez G, Kingsley RA, Hall LJ, Webber MA, Adriaenssens EM. [A hybrid and poly-polish workflow for the complete and accurate assembly of phage genomes: a case study of ten przondoviruses.](https://doi.org/10.1099/mgen.0.001065) _Microb Genom_ (2023).

52: Libuit KG, Doughty EL, Otieno JR, Ambrosio F, Kapsak CJ, Smith EA, Wright SM, Scribner MR, Petit III RA, Mendes CI, Huergo M, Legacki G, Loreth C, Park DJ, Sevinsky JR. [Accelerating bioinformatics implementation in public health.](https://doi.org/10.1099/mgen.0.001051) _Microb Genom_ (2023).

51: Hojaisa S, Jha AN [Databases and Tools for Microbial Genome and Human Microbiome Studies](https://doi.org/10.1007/978-981-99-2816-3_3) _Industrial Microbiology and Biotechnology_ (2023).

50: Raghuram V, Gunoskey JJ, Hofstetter KS, Jacko NF, Shumaker MJ, Hu YJ, Read TD, David MZ. [Comparison of genomic diversity between single and pooled _Staphylococcus aureus_ colonies isolated from human colonisation cultures.](https://doi.org/10.1101/2023.06.14.544959) _bioRxiv_ (2023).

49: Tokuda M, Yuki M, Ohkuma M, Kimbara K, Suzuki H, Shintani M. [Transconjugant range of PromA plasmids in microbial communities is predicted by sequence similarity with the bacterial host chromosome.](https://doi.org/10.1099/mgen.0.001043) _Microb Genom_ (2023).

48: Marquis B, Pillonel T, Carrara A, Bertelli C [zDB: bacterial comparative genomics made easy](https://doi.org/10.1101/2023.05.31.543076) _bioRxiv_ (2023).

47: Machado RAR, Bhat AH, Castaneda-Alvarez C, Půža V, San-Blas E [_Photorhabdus aballayi_ sp. nov. and _Photorhabdus luminescens subsp. venezuelensis_ subsp. nov., isolated from _Heterorhabditis amazonensis_ entomopathogenic nematodes.](https://doi.org/10.1099/ijsem.0.005872) _International Journal of Systematic and Evolutionary Microbiology_ (2023).

46: Pitcher NJ, Feder A, Bolden N, Zirbes CF, Pamatmat AJ, Boyken L, Hill JJ, Thurman AL, Reeb VC, Porterfield HS, Moustafa AM, Planet PJ, Fischer AJ. [Parallel Evolution of Linezolid Resistant _Staphylococcus aureus_ in Patients with Cystic Fibrosis.](https://doi.org/10.1101/2023.05.02.539145) _bioRxiv_ (2023).

45: Machado RAR, Bhat AH, Fallet P, Turlings TCJ, Kajuga J, Yan X, Toepfer S [_Xenorhabdus bovienii subsp. africana_ subsp. nov., isolated from _Steinernema africanum_ entomopathogenic nematodes.](https://doi.org/10.1099/ijsem.0.005795) _International Journal of Systematic and Evolutionary Microbiology_ (2023).

44: Martín-Rodríguez AJ, Thorell K, Joffré E, Jensie-Markopoulos S, Moore ERB, Sjöling Å [_Shewanella septentrionalis_ sp. nov. and _Shewanella holmiensis_ sp. nov., isolated from Baltic Sea water and sediments](https://doi.org/10.1099/ijsem.0.005767) _Microbiology Society_ (2023).

43: Khezri A, Avershina E, Ahmad R. [Clinical Diagnostics of Bacterial Infections and Their Resistance to Antibiotics-Current State and Whole Genome Sequencing Implementation Perspectives.](https://doi.org/10.3390/antibiotics12040781) _Antibiotics_ (2023).

42: Chomkatekaew C, Thaipadungpanit J, Hearn P, Soeng S, Pol S, Neou L, Hopkins J, Turner P, Batty EM. [Detection of maternal transmission of resistant Gram-negative bacteria in a Cambodian hospital setting.](https://doi.org/10.3389/fmicb.2023.1158056) _Front Microbiol_ (2023).

41:  [Unlocking the microbial studies through computational approaches: how far have we reached?](https://doi.org/10.1007/s11356-023-26220-0) _Environ Sci Pollut Res Int_ (2023).

40: Gompelman M, Wezendonk GTJ, Wouters Y, Beurskens-Meijerink J, Fragkos KC, Rahman FZ, Coolen JPM, van Weerdenburg IJM, Wertheim HFL, Kievit W, Akkermans RP, Serlie MJ, Bleeker-Rovers CP, Wanten GJA [Randomized clinical trial: Long-term Staphylococcus aureus decolonization in patients on home parenteral nutrition.](https://doi.org/10.1016/j.clnu.2023.03.010) _Clinical Nutrition_ (2023).

39: Legall NA [Computational Techniques for Unravelling the Genome Variations of _Mycobacterium bovis_ Across Organizational Scales](https://esploro.libs.uga.edu/esploro/outputs/9949558727702959) _Dissertation_ (2023).

38: Machado RAR, Loulou A, Bhat AH, Mastore M, Terrettaz C, Brivio MF, Kallel S [_Acinetobacter nematophilus_ sp. nov., _Alcaligenes nematophilus_ sp. nov., _Enterobacter nematophilus_ sp. nov., and _Kaistia nematophila_ sp. nov., Isolated from Soil-Borne Nematodes and Proposal for the Elevation of _Alcaligenes faecalis subsp. faecalis_, _Alcaligenes faecalis subsp. parafaecalis_, and _Alcaligenes faecalis subsp. phenolicus_ to the Species Level](https://doi.org/10.3390/taxonomy3010012) _Taxonomy_ (2023).

37: Beauchemin ET, Hunter C, Maurice CF. [Actively replicating gut bacteria identified by 5-ethynyl-2'-deoxyuridine (EdU) click chemistry and cell sorting.](https://doi.org/10.1080/19490976.2023.2180317) _Gut Microbes_ (2023).

36: Morita D, Arai H, Isobe J, Maenishi E, Kumagai T, Maruyama F, Kuroda T. [Whole-Genome and Plasmid Comparative Analysis of _Campylobacter jejuni_ from Human Patients in Toyama, Japan, from 2015 to 2019.](https://doi.org/10.1128/spectrum.02659-22) _Microbiol Spectr_ (2023).

35: Loulou A, Mastore M, Caramella S, Bhat AH, Brivio MF, Machado RAR, Kallel S. [Entomopathogenic potential of bacteria associated with soil-borne nematodes and insect immune responses to their infection.](https://doi.org/10.1371/journal.pone.0280675) _PLoS One_ (2023).

34: Jain R, Bhardwaj P, Guleria S, Pandey A, Kumar S [Polyamine metabolizing rhizobacteria _Pseudomonas sp. GBPI_506_ modulates hormone signaling to enhance lateral roots and nicotine biosynthesis in Nicotiana benthamiana.](https://doi.org/10.1016/j.plaphy.2023.01.010) _Plant Physiology and Biochemistry_ (2023).

33: Cornet L, Durieu B, Baert F, D'hooge E, Colignon D, Meunier L, Lupo V, Cleenwerck I, Daniel HM, Rigouts L, Sirjacobs D, Declerck S, Vandamme P, Wilmotte A, Baurain D, Becker P. [The GEN-ERA toolbox: unified and reproducible workflows for research in microbial genomics.](https://doi.org/10.1093/gigascience/giad022) _Gigascience_ (2022).

32: Talbot BM, Jacko NF, Petit RA, Pegues DA, Shumaker MJ, Read TD, David MZ. [Unsuspected Clonal Spread of Methicillin-Resistant _Staphylococcus aureus_ Causing Bloodstream Infections in Hospitalized Adults Detected Using Whole Genome Sequencing.](https://doi.org/10.1093/cid/ciac339) _Clin Infect Dis_ (2022).

31: Chitale P, Lemenze AD, Fogarty EC, Shah A, Grady C, Odom-Mabey AR, Johnson WE, Yang JH, Eren AM, Brosch R, Kumar P, Alland D [A comprehensive update to the _Mycobacterium tuberculosis_ H37Rv reference genome.](https://doi.org/10.1038/s41467-022-34853-x) _Nat Commun_ (2022).

30: Cavallo I, Sivori F, Truglio M, De Maio F, Lucantoni F, Cardinali G, Pontone M, Bernardi T, Sanguinetti M, Capitanio B, Cristaudo A, Ascenzioni F, Morrone A, Pimpinelli F, Di Domenico EG. [Skin dysbiosis and _Cutibacterium acnes_ biofilm in inflammatory acne lesions of adolescents.](https://doi.org/10.1038/s41598-022-25436-3) _Sci Rep_ (2022).

29: Atxaerandio-Landa A, Arrieta-Gisasola A, Laorden L, Bikandi J, Garaizar J, Martinez-Malaxetxebarria I, Martinez-Ballesteros I. [A Practical Bioinformatics Workflow for Routine Analysis of Bacterial WGS Data.](https://doi.org/10.3390/microorganisms10122364) _Microorganisms_ (2022).

28: Bhardwaj P, Jain R, Kumar S. [Draft Genome Sequence of Endophytic _Sphingomonas faeni_ Strain ALB2, Isolated from the Leaf of a Cold-Desert Medicinal Plant.](https://doi.org/10.1128/mra.00687-22) _Microbiol Resour Announc_ (2022).

27: Wang D, Fletcher GC, On SLW, Palmer JS, Gagic D, Flint SH [Biofilm formation, sodium hypochlorite susceptibility and genetic diversity of Vibrio parahaemolyticus.](https://doi.org/10.1016/j.ijfoodmicro.2022.110011) _International Journal of Food Microbiology_ (2022).

26: Taylor M, Janasky L, Vega N [Convergent structure with divergent adaptations in combinatorial microbiome communities.](https://doi.org/10.1093/femsec/fiac115) _FEMS Microbiology Ecology_ (2022).

25: Sanders JG, Yan W, Mjungu D, Lonsdorf EV, Hart JA, Sanz CM, Morgan DB, Peeters M, Hahn BH, Moeller AH. [A low-cost genomics workflow enables isolate screening and strain-level analyses within microbiomes.](https://doi.org/10.1186/s13059-022-02777-w) _Genome Biol_ (2022).

24: Babiker A, Bower C, Lutgring JD, Petit III RA, Howard-Anderson J, Ansari U, McAllister G, Adamczyk M, Breaker E, Satola SW, Jacob JT, Woodworth MH. [Clinical and Genomic Epidemiology of _mcr-9_-Carrying Carbapenem-Resistant _Enterobacterales_ Isolates in Metropolitan Atlanta, 2012 to 2017.](https://doi.org/10.1128/spectrum.02522-21) _Microbiol Spectr_ (2022).

23: Wang C, Xia Y, Liu Y, Kang C, Lu N, Tian D, Lu H, Han F, Xu J, Yomo, T [CleanSeq: A Pipeline for Contamination Detection, Cleanup, and Mutation Verifications from Microbial Genome Sequencing Data.](https://doi.org/10.3390/app12126209) _NATO Advanced Science Institutes Series E: Applied Sciences_ (2022).

22: Duré FM, Silveira MC, Rocha-de-Souza CM, Leão RS, de Oliveira Santos IC, Albano RM, Marques EA, D'Alincourt Carvalho-Assef AP, da Silva FAB. [CABGen: A Web Application for the Bioinformatic Analysis of Bacterial Genomes.](https://doi.org/10.3389/fmicb.2022.893474) _Front Microbiol_ (2022).

21: Sivertsen A, Dyrhovden R, Tellevik MG, Bruvold TS, Nybakken E, Skutlaberg DH, Skarstein I, Kommedal Ø. [_Escherichia marmotae_-a Human Pathogen Easily Misidentified as _Escherichia coli_.](https://doi.org/10.1128/spectrum.02035-21) _Microbiol Spectr_ (2022).

20: Li X, Zhu L, Wang X, Li J, Tang B. [Evaluation of IR Biotyper for _Lactiplantibacillus plantarum_ Typing and Its Application Potential in Probiotic Preliminary Screening.](https://doi.org/10.3389/fmicb.2022.823120) _Front Microbiol_ (2022).

19: Raghuram V, Alexander AM, Loo HQ, Petit III RA, Goldberg JB, Read TD. [Species-Wide Phylogenomics of the _Staphylococcus aureus_ _Agr_ Operon Revealed Convergent Evolution of Frameshift Mutations.](https://doi.org/10.1128/spectrum.01334-21) _Microbiol Spectr_ (2022).

18: Goldman SL, Sanders JG, Yan W, Denice A, Cornwall M, Ivey KN, Taylor EN, Gunderson AR, Sheehan MJ, Mjungu D, Lonsdorf EV, Pusey AE, Hahn BH, Moeller AH. [Culture-enriched community profiling improves resolution of the vertebrate gut microbiota.](https://doi.org/10.1111/1755-0998.13456) _Mol Ecol Resour_ (2022).

17:  [Application of genomics and metagenomics to improve food safety based on an enhanced characterisation of antimicrobial resistance.](https://doi.org/10.1016/j.cofs.2021.12.002) _Current Opinion in Food Science_ (2021).

16: Conrads G, Klomp T, Deng D, Wenzler JS, Braun A, Abdelbary MMH. [The Antimicrobial Susceptibility of _Porphyromonas gingivalis_: Genetic Repertoire, Global Phenotype, and Review of the Literature.](https://doi.org/10.3390/antibiotics10121438) _Antibiotics_ (2021).

15: Machado RAR, Bhat AH, Abolafia J, Muller A, Bruno P, Fallet P, Arce CCM, Turlings TCJ, Bernal JS, Kajuga J, Waweru B, Toepfer S. [Multi-locus phylogenetic analyses uncover species boundaries and reveal the occurrence of two new entomopathogenic nematode species, _Heterorhabditis ruandica_ n. sp. and _Heterorhabditis zacatecana_ n. sp.](https://doi.org/10.21307/jofnem-2021-089) _J Nematol_ (2021).

14: Bell KL, Petit III RA, Cutler A, Dobbs EK, Macpherson JM, Read TD, Burgess KS, Brosi BJ. [Comparing whole-genome shotgun sequencing and DNA metabarcoding approaches for species identification and quantification of pollen species mixtures.](https://doi.org/10.1002/ece3.8281) _Ecol Evol_ (2021).

13: Schwengers O, Jelonek L, Dieckmann MA, Beyvers S, Blom J, Goesmann A. [Bakta: rapid and standardized annotation of bacterial genomes via alignment-free sequence identification.](https://doi.org/10.1099/mgen.0.000685) _Microb Genom_ (2021).

12:  [Culture-enriched community profiling improves resolution of the vertebrate gut microbiota.](https://doi.org/10.1111/1755-0998.13456) _Molecular Ecology Resources_ (2021).

11: Murigneux V, Roberts LW, Forde BM, Phan MD, Nhu NTK, Irwin AD, Harris PNA, Paterson DL, Schembri MA, Whiley DM, Beatson SA. [MicroPIPE: validating an end- to-end workflow for high-quality complete bacterial genome construction.](https://doi.org/10.1186/s12864-021-07767-z) _BMC Genomics_ (2021).

10: Sserwadda I, Mboowa G. [rMAP: the Rapid Microbial Analysis Pipeline for ESKAPE bacterial group whole-genome sequence data.](https://doi.org/10.1099/mgen.0.000583) _Microb Genom_ (2021).

9: Pavlovikj N, Gomes-Neto JC, Deogun JS, Benson AK. [ProkEvo: an automated, reproducible, and scalable framework for high-throughput bacterial population genomics analyses.](https://doi.org/10.7717/peerj.11376) _PeerJ_ (2021).

8: Beggs GA [Biochemical and structural mechanisms of multidrug efflux pump transcription regulators, Neisseria gonorrhoeae MtrR and Escherichia coli MprA](https://hdl.handle.net/10161/22987) _Dissertation_ (2021).

7: Almeida MO, Viana MVC, Cerqueira JC, Aburjaile FF, Junior AAZ, Azevedo V, Carvalho RDO [Novel insights in bacterial vaginosis etiology through genomic approaches](https://doi.org/10.1590/0001-3765202120200945) _Anais Da Academia Brasileira de Ciencias_ (2021).

6: Korzhenkov AA [ZGA: a flexible pipeline for read processing, de novo assembly and annotation of prokaryotic genomes](https://doi.org/10.1101/2021.04.27.441618) _bioRxiv_ (2021).

5: Deneke C, Brendebach H, Uelze L, Borowiak M, Malorny B, Tausch SH. [Species-Specific Quality Control, Assembly and Contamination Detection in Microbial Isolate Sequences with AQUAMIS.](https://doi.org/10.3390/genes12050644) _Genes_ (2021).

4: Beggs GA, Ayala JC, Kavanaugh LG, Read TD, Hooks GM, Schumacher MA, Shafer WM, Brennan RG. [Structures of _Neisseria gonorrhoeae_ MtrR-operator complexes reveal molecular mechanisms of DNA recognition and antibiotic resistance-conferring clinical mutations.](https://doi.org/10.1093/nar/gkab213) _Nucleic Acids Res_ (2021).

3: Oliveira de Almeida M, Carvalho R, Figueira Aburjaile F, Malcher Miranda F, Canário Cerqueira J, Brenig B, Ghosh P, Ramos R, Kato RB, de Castro Soares S, Silva A, Azevedo V, Canário Viana MV. [Characterization of the first vaginal _Lactobacillus crispatus_ genomes isolated in Brazil.](https://doi.org/10.7717/peerj.11079) _PeerJ_ (2021).

2: Zhou X, Xuan D, Hu S, Du J, Pu J, Jin D, Zhao F, Yin F, Cui X, Huang Y, Wang G, Wu Q, Lu G, Niu L [_Pelistega ratti_ sp. nov. from _Rattus norvegicus_ of Hainan island](https://doi.org/10.1099/ijsem.0.004733) _International Journal of Systematic and Evolutionary Microbiology_ (2021).

1: Su M [Evolution of _Staphylococcus aureus_ phenotypes](https://etd.library.emory.edu/concern/etds/7m01bm754?locale=en) _Dissertation_ (2020).

