
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="A full list of the 140 datasets and software packages used by Bactopia, along with links and citaions.">
      
      
        <meta name="author" content="Robert A. Petit III">
      
      
        <link rel="canonical" href="https://bactopia.github.io/v2.1.1/acknowledgements/">
      
      
        
      
      
      <link rel="icon" href="../assets/favicon.ico">
      <meta name="generator" content="mkdocs-1.3.1, mkdocs-material-8.5.3+insiders-4.24.0">
    
    
      
        <title>Acknowledgements - Bactopia</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.38a480a7.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.cbb835fc.min.css">
        
      
      

    
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Ubuntu:300,300i,400,400i,700,700i%7CUbuntu+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Ubuntu";--md-code-font:"Ubuntu Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../custom/magnific-popup.css">
    
      <link rel="stylesheet" href="../custom/extra.css">
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      
  


  
  


  <script id="__analytics">function __md_analytics(){window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)},ga.l=+new Date,ga("create","UA-104199698-2","auto"),ga("set","anonymizeIp",!0),ga("send","pageview"),document.addEventListener("DOMContentLoaded",function(){document.forms.search&&document.forms.search.query.addEventListener("blur",function(){var e;this.value&&(e=document.location.pathname,ga("send","pageview",e+"?q="+this.value))}),document$.subscribe(function(){var a=document.forms.feedback;if(void 0!==a)for(var e of a.querySelectorAll("[type=submit]"))e.addEventListener("click",function(e){e.preventDefault();var t=document.location.pathname,e=this.getAttribute("data-md-value");ga("send","event","feedback","click",t,e),a.firstElementChild.disabled=!0;e=a.querySelector(".md-feedback__note [data-md-value='"+e+"']");e&&(e.hidden=!1)}),a.hidden=!1}),location$.subscribe(function(e){ga("send","pageview",e.pathname)})});var e=document.createElement("script");e.async=!0,e.src="https://www.google-analytics.com/analytics.js",document.getElementById("__analytics").insertAdjacentElement("afterEnd",e)}</script>

  
    <script>"undefined"!=typeof __md_analytics&&__md_analytics()</script>
  

    
    
      
        <meta  property="og:type"  content="website" >
      
        <meta  property="og:title"  content="Acknowledgements - Bactopia" >
      
        <meta  property="og:description"  content="A full list of the 140 datasets and software packages used by Bactopia, along with links and citaions." >
      
        <meta  property="og:image"  content="https://bactopia.github.io/v2.1.1/assets/images/social/acknowledgements.png" >
      
        <meta  property="og:image:type"  content="image/png" >
      
        <meta  property="og:image:width"  content="1200" >
      
        <meta  property="og:image:height"  content="630" >
      
        <meta  property="og:url"  content="https://bactopia.github.io/v2.1.1/acknowledgements/" >
      
        <meta  name="twitter:card"  content="summary_large_image" >
      
        <meta  name="twitter:title"  content="Acknowledgements - Bactopia" >
      
        <meta  name="twitter:description"  content="A full list of the 140 datasets and software packages used by Bactopia, along with links and citaions." >
      
        <meta  name="twitter:image"  content="https://bactopia.github.io/v2.1.1/assets/images/social/acknowledgements.png" >
      
    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#acknowledgements" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--lifted" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="Bactopia" class="md-header__button md-logo" aria-label="Bactopia" data-md-component="logo">
      
  <img src="../assets/bactopia-small-logo.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Bactopia
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Acknowledgements
            
          </span>
        </div>
      </div>
    </div>
    
      <form class="md-header__option" data-md-component="palette">
        
          
          
          <input class="md-option" data-md-color-media="" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_0">
          
            <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M17 6H7c-3.31 0-6 2.69-6 6s2.69 6 6 6h10c3.31 0 6-2.69 6-6s-2.69-6-6-6zm0 10H7c-2.21 0-4-1.79-4-4s1.79-4 4-4h10c2.21 0 4 1.79 4 4s-1.79 4-4 4zM7 9c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"/></svg>
            </label>
          
        
          
          
          <input class="md-option" data-md-color-media="" data-md-color-scheme="slate" data-md-color-primary="red" data-md-color-accent="red"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_1">
          
            <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_0" hidden>
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M17 7H7a5 5 0 0 0-5 5 5 5 0 0 0 5 5h10a5 5 0 0 0 5-5 5 5 0 0 0-5-5m0 8a3 3 0 0 1-3-3 3 3 0 0 1 3-3 3 3 0 0 1 3 3 3 3 0 0 1-3 3Z"/></svg>
            </label>
          
        
      </form>
    
    
      <script>var media,input,key,value,palette=__md_get("__palette");if(palette&&palette.color){"(prefers-color-scheme)"===palette.color.media&&(media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']"),palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent"));for([key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/bactopia/bactopia" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.2.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
    
      <nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        

  




  
    
    
    
      <li class="md-tabs__item">
        <a href=".." class="md-tabs__link md-tabs__link--active">
          
  
    
  
  Bactopia

        </a>
      </li>
    
  

      
        




  
    
    
    
      <li class="md-tabs__item">
        <a href="../bactopia-tools/" class="md-tabs__link">
          
  
    
  
  Bactopia Tools

        </a>
      </li>
    
  

      
    </ul>
  </div>
</nav>
    
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="Bactopia" class="md-nav__button md-logo" aria-label="Bactopia" data-md-component="logo">
      
  <img src="../assets/bactopia-small-logo.png" alt="logo">

    </a>
    Bactopia
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/bactopia/bactopia" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.2.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
    
  
  
    
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
          <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_1" type="checkbox" id="__nav_1" checked>
        
        
          <label class="md-nav__link" for="__nav_1">
            
  
  <span class="md-ellipsis">
    Bactopia
    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" aria-label="Bactopia" data-md-level="1">
          <label class="md-nav__title" for="__nav_1">
            <span class="md-nav__icon md-icon"></span>
            Bactopia
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Introduction
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../quick-start/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Quick Start
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../installation/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Installation
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../tutorial/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Tutorial
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../datasets/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Build Datasets
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../usage-basic/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Basic Usage
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../workflow-overview/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Workflow Overview
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../output-overview/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Output Overview
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../usage-complete/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Complete Usage
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../troubleshooting/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Troubleshooting
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../presentations/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Presentations
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../enhancements/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Enhancements to OSS
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  <span class="md-ellipsis">
    Acknowledgements
    
  </span>
  
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    Acknowledgements
    
  </span>
  
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#influences" class="md-nav__link">
    <span class="md-ellipsis">
      Influences
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Influences">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#nf-core" class="md-nav__link">
    <span class="md-ellipsis">
      nf-core
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#public-datasets" class="md-nav__link">
    <span class="md-ellipsis">
      Public Datasets
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Public Datasets">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#ariba-reference-datasets" class="md-nav__link">
    <span class="md-ellipsis">
      Ariba Reference Datasets
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#minmer-datasets" class="md-nav__link">
    <span class="md-ellipsis">
      Minmer Datasets
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#everything-else" class="md-nav__link">
    <span class="md-ellipsis">
      Everything Else
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#software-included-in-bactopia" class="md-nav__link">
    <span class="md-ellipsis">
      Software Included In Bactopia
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../changelog/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Changelog
    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_2" type="checkbox" id="__nav_2" >
        
        
          <label class="md-nav__link" for="__nav_2">
            
  
  <span class="md-ellipsis">
    Bactopia Tools
    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" aria-label="Bactopia Tools" data-md-level="1">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            Bactopia Tools
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Introduction
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_2_2" type="checkbox" id="__nav_2_2" >
        
        
          <label class="md-nav__link" for="__nav_2_2">
            
  
  <span class="md-ellipsis">
    Subworkflows
    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" aria-label="Subworkflows" data-md-level="2">
          <label class="md-nav__title" for="__nav_2_2">
            <span class="md-nav__icon md-icon"></span>
            Subworkflows
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/ariba/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    ariba
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/bakta/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    bakta
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/eggnog/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    eggnog
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/gtdb/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    gtdb
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/merlin/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    merlin
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/pangenome/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    pangenome
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/snippy/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    snippy
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/staphtyper/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    staphtyper
    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_2_3" type="checkbox" id="__nav_2_3" >
        
        
          <label class="md-nav__link" for="__nav_2_3">
            
  
  <span class="md-ellipsis">
    Modules
    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" aria-label="Modules" data-md-level="2">
          <label class="md-nav__title" for="__nav_2_3">
            <span class="md-nav__icon md-icon"></span>
            Modules
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/abricate/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    abricate
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/agrvate/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    agrvate
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/amrfinderplus/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    amrfinderplus
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/busco/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    busco
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/checkm/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    checkm
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/ectyper/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    ectyper
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/emmtyper/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    emmtyper
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/fastani/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    fastani
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/gamma/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    gamma
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/genotyphi/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    genotyphi
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/hicap/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    hicap
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/hpsuissero/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    hpsuissero
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/ismapper/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    ismapper
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/kleborate/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    kleborate
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/kraken2/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    kraken2
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/legsta/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    legsta
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/lissero/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    lissero
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/mashdist/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    mashdist
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/mashtree/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    mashtree
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/mcroni/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    mcroni
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/meningotype/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    meningotype
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/mlst/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    mlst
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/mobsuite/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    mobsuite
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/mykrobe/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    mykrobe
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/ngmaster/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    ngmaster
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/plasmidfinder/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    plasmidfinder
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/prokka/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    prokka
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/rgi/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    rgi
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/seqsero2/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    seqsero2
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/seroba/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    seroba
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/shigatyper/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    shigatyper
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/sistr/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    sistr
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/spatyper/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    spatyper
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/ssuissero/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    ssuissero
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/staphopiasccmec/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    staphopiasccmec
    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bactopia-tools/tbprofiler/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    tbprofiler
    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#influences" class="md-nav__link">
    <span class="md-ellipsis">
      Influences
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Influences">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#nf-core" class="md-nav__link">
    <span class="md-ellipsis">
      nf-core
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#public-datasets" class="md-nav__link">
    <span class="md-ellipsis">
      Public Datasets
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Public Datasets">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#ariba-reference-datasets" class="md-nav__link">
    <span class="md-ellipsis">
      Ariba Reference Datasets
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#minmer-datasets" class="md-nav__link">
    <span class="md-ellipsis">
      Minmer Datasets
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#everything-else" class="md-nav__link">
    <span class="md-ellipsis">
      Everything Else
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#software-included-in-bactopia" class="md-nav__link">
    <span class="md-ellipsis">
      Software Included In Bactopia
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                

<!-- Edit button -->

  <a
    href="https://github.com/bactopia/bactopia.github.io/edit/master/docs/acknowledgements.md"
    title="Edit this page"
    class="md-content__button md-icon"
  >
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25Z"/></svg>
  </a>


<!--
  Hack: check whether the content contains a h1 headline. If it
  doesn't, the page title (or respectively site name) is used
  as the main headline.
-->


<!-- Markdown content -->
<h1 id="acknowledgements">Acknowledgements<a class="headerlink" href="#acknowledgements" title="Permanent link">&para;</a></h1>
<p>Bactopia is truly a case of <em>"standing upon the shoulders of giants"</em>. Bactopia
currently includes more than 140 datasets and software packages. Nearly 
every component uses, from the workflow, datasets, software packages, even the 
framework for this site, was created by others and made freely available to the public.</p>
<p>I would like to personally extend my many thanks and gratitude to the authors
of these software packages and public datasets. If you've made it this far, I 
owe you a beer üçª (or coffee ‚òï!) if we ever encounter one another in person. 
Really, thank you very much!</p>
<div class="admonition info">
<p class="admonition-title">Please Cite Datasets and Tools</p>
<p>If you have used Bactopia in your work, please be sure to cite any datasets
or software you may have used.</p>
</div>
<h2 id="influences">Influences<a class="headerlink" href="#influences" title="Permanent link">&para;</a></h2>
<h3 id="nf-core">nf-core<a class="headerlink" href="#nf-core" title="Permanent link">&para;</a></h3>
<p><a href="https://nf-co.re/">nf-core</a> is group of great individuals volunteering 
their time to create a set of curated Nextflow analysis pipelines. The <a href="https://nf-co.re/about">nf-core Team</a>
is putting together some amazing practices that I think really strengthen the 
Nextflow community as a whole!</p>
<p>Since the beginning I have always had the idea of Bactopia one day being 
apart of nf-core. Unfortunately, I think Bactopia is a much to big at this point to fit the nf-core mold, but that's OK! 
There are still many nf-core practices I've been able to adapt into Bactopia.</p>
<ol>
<li>Arguement parsing based on nf-core library</li>
<li>All Bactopia Tools are adapted from nf-core/modules</li>
<li>Testing implemented to follow nf-core/modules</li>
</ol>
<p>By implementing these practices, Bactopia I believe is much better pipeline to use. For this I'm very 
grateful to the nf-core community! Thank you!</p>
<p>Ewels P, Peltzer A, Fillinger S, Patel H, Alneberg J, Wilm A, Garcia MU, Di Tommaso P, Nahnsen S <a href="https://dx.doi.org/10.1038/s41587-020-0439-x">The nf-core framework for community-curated bioinformatics pipelines.</a> <em>Nat Biotechnol.</em> (2020)</p>
<h2 id="public-datasets">Public Datasets<a class="headerlink" href="#public-datasets" title="Permanent link">&para;</a></h2>
<p>Below is a list of 19 public datasets that could have potentially 
been included during the <em><a href="../datasets/">Build Datasets</a></em> step.</p>
<h3 id="ariba-reference-datasets">Ariba Reference Datasets<a class="headerlink" href="#ariba-reference-datasets" title="Permanent link">&para;</a></h3>
<p>These datasets are available using Ariba's <code>getref</code> function. You can learn 
more about this function at <a href="https://github.com/sanger-pathogens/ariba/wiki/Task:-getref">Ariba's Wiki</a>.</p>
<ol>
<li>
<p><strong><a href="http://en.mediterranee-infection.com/article.php?laref=283%26titre=arg-annot">ARG-ANNOT</a></strong> <br />
Gupta SK, Padmanabhan BR, Diene SM, Lopez-Rojas R, Kempf M, Landraud L, Rolain J-M <a href="https://doi.org/10.1128/aac.01310-13">ARG-ANNOT, a new bioinformatic tool to discover antibiotic resistance genes in bacterial genomes.</a> <em>Antimicrob. Agents Chemother</em> 58, 212‚Äì220 (2014)</p>
</li>
<li>
<p><strong><a href="https://card.mcmaster.ca/">CARD</a></strong> <br />
Alcock BP, Raphenya AR, Lau TTY, Tsang KK, Bouchard M, Edalatmand A, Huynh W, Nguyen A-L V, Cheng AA, Liu S, Min SY, Miroshnichenko A, Tran H-K, Werfalli RE, Nasir JA, Oloni M, Speicher DJ, Florescu A, Singh B, Faltyn M, Hernandez-Koutoucheva A, Sharma AN, Bordeleau E, Pawlowski AC, Zubyk HL, Dooley D, Griffiths E, Maguire F, Winsor GL, Beiko RG, Brinkman FSL, Hsiao WWL, Domselaar GV, McArthur AG <a href="https://doi.org/10.1093/nar/gkz935">CARD 2020: antibiotic resistome surveillance with the comprehensive antibiotic resistance database.</a> <em>Nucleic acids research</em> 48.D1, D517-D525 (2020)</p>
</li>
<li>
<p><strong><a href="https://dx.doi.org/10.1099%2Fmgen.0.000064">EcOH</a></strong> <br />
Ingle DJ, Valcanis M, Kuzevski A, Tauschek M, Inouye M, Stinear T, Levine MM, Robins-Browne RM, Holt KE <a href="https://doi.org/10.1099/mgen.0.000064">In silico serotyping of E. coli from short read data identifies limited novel O-loci but extensive diversity of O:H serotype combinations within and between pathogenic lineages.</a> <em>Microbial Genomics</em>, 2(7), e000064. (2016)</p>
</li>
<li>
<p><strong><a href="https://megares.meglab.org/">MEGARes</a></strong> <br />
Lakin SM, Dean C, Noyes NR, Dettenwanger A, Ross AS, Doster E, Rovira P, Abdo Z, Jones KL, Ruiz J, Belk KE, Morley PS, Boucher C <a href="https://doi.org/10.1093/nar/gkw1009">MEGARes: an antimicrobial resistance database for high throughput sequencing.</a> <em>Nucleic Acids Res.</em> 45, D574‚ÄìD580 (2017)</p>
</li>
<li>
<p><strong><a href="https://megares.meglab.org/">MEGARes 2.0</a></strong> <br />
Doster E, Lakin SM, Dean CJ, Wolfe C, Young JG, Boucher C, Belk KE, Noyes NR, Morley PS <a href="https://doi.org/10.1093/nar/gkz1010">MEGARes 2.0: a database for classification of antimicrobial drug, biocide and metal resistance determinants in metagenomic sequence data.</a> <em>Nucleic Acids Research</em>, 48(D1), D561‚ÄìD569. (2020)</p>
</li>
<li>
<p><strong><a href="https://www.ncbi.nlm.nih.gov/bioproject/?term=PRJNA313047">NCBI Reference Gene Catalog</a></strong> <br />
Feldgarden M, Brover V, Haft DH, Prasad AB, Slotta DJ, Tolstoy I, Tyson GH, Zhao S, Hsu C-H, McDermott PF, Tadesse DA, Morales C, Simmons M, Tillman G, Wasilenko J, Folster JP, Klimke W <a href="https://doi.org/10.1128/AAC.00483-19">Validating the NCBI AMRFinder Tool and Resistance Gene Database Using Antimicrobial Resistance Genotype-Phenotype Correlations in a Collection of NARMS Isolates</a>. <em>Antimicrob. Agents Chemother.</em> (2019)</p>
</li>
<li>
<p><strong><a href="https://cge.cbs.dtu.dk/services/PlasmidFinder/">PlasmidFinder</a></strong> <br />
Carattoli A, Zankari E, Garc√≠a-Fern√°ndez A, Larsen MV, Lund O, Villa L, Aarestrup FM, Hasman H <a href="https://doi.org/10.1128/aac.02412-14">In silico detection and typing of plasmids using PlasmidFinder and plasmid multilocus sequence typing.</a> <em>Antimicrob. Agents Chemother.</em> 58, 3895‚Äì3903 (2014)</p>
</li>
<li>
<p><strong><a href="https://cge.cbs.dtu.dk//services/ResFinder/">ResFinder</a></strong> <br />
Zankari E, Hasman H, Cosentino S, Vestergaard M, Rasmussen S, Lund O, Aarestrup FM, Larsen MV <a href="https://doi.org/10.1093/jac/dks261">Identification of acquired antimicrobial resistance genes.</a> <em>J. Antimicrob. Chemother.</em> 67, 2640‚Äì2644 (2012)</p>
</li>
<li>
<p><strong><a href="https://github.com/katholt/srst2">SRST2</a></strong> <br />
Inouye M, Dashnow H, Raven L-A, Schultz MB, Pope BJ, Tomita T, Zobel J, Holt KE <a href="https://doi.org/10.1186/s13073-014-0090-6">SRST2: Rapid genomic surveillance for public health and hospital microbiology labs.</a> <em>Genome Med.</em> 6, 90 (2014)</p>
</li>
<li>
<p><strong><a href="http://www.mgc.ac.cn/VFs/">VFDB</a></strong> <br />
Chen L, Zheng D, Liu B, Yang J, Jin Q <a href="https://doi.org/10.1093/nar/gkv1239">VFDB 2016: hierarchical and refined dataset for big data analysis--10 years on.</a> <em>Nucleic Acids Res.</em> 44, D694‚Äì7 (2016)</p>
</li>
<li>
<p><strong><a href="https://cge.cbs.dtu.dk/services/VirulenceFinder/">VirulenceFinder</a></strong> <br />
Joensen KG, Scheutz F, Lund O, Hasman H, Kaas RS, Nielsen EM, Aarestrup FM <a href="https://doi.org/10.1128/jcm.03617-13">Real-time whole-genome sequencing for routine typing, surveillance, and outbreak detection of verotoxigenic <em>Escherichia coli</em>.</a> <em>J. Clin. Microbiol.</em> 52, 1501‚Äì1510 (2014)</p>
</li>
</ol>
<h3 id="minmer-datasets">Minmer Datasets<a class="headerlink" href="#minmer-datasets" title="Permanent link">&para;</a></h3>
<ol>
<li>
<p><strong><a href="https://mash.readthedocs.io/en/latest/data.html">Mash Refseq (release 88) Sketch</a></strong> <br />
Ondov BD, Starrett GJ, Sappington A, Kostic A, Koren S, Buck CB, Phillippy AM <a href="https://doi.org/10.1186/s13059-019-1841-x">Mash Screen: high-throughput sequence containment estimation for genome discovery</a> <em>Genome Biol</em> 20, 232 (2019)</p>
</li>
<li>
<p><strong><a href="https://sourmash.readthedocs.io/en/latest/databases.html">Sourmash Genbank LCA Signature</a></strong> <br />
Brown CT, Irber L <a href="http://dx.doi.org/10.21105/joss.00027">sourmash: a library for MinHash sketching of DNA</a>. <em>JOSS</em> 1, 27 (2016)</p>
</li>
</ol>
<h3 id="everything-else">Everything Else<a class="headerlink" href="#everything-else" title="Permanent link">&para;</a></h3>
<ol>
<li>
<p><strong><a href="http://eggnog.embl.de/">eggNOG 5.0 Database</a></strong> <br />
Huerta-Cepas J, Szklarczyk D, Heller D, Hern√°ndez-Plaza A, Forslund SK, Cook H, Mende DR, Letunic I, Rattei T, Jensen LJ, von Mering C, Bork P <a href="https://doi.org/10.1093/nar/gky1085">eggNOG 5.0: a hierarchical, functionally and phylogenetically annotated orthology resource based on 5090 organisms and 2502 viruses.</a> <em>Nucleic Acids Res.</em> 47, D309‚ÄìD314 (2019)</p>
</li>
<li>
<p><strong><a href="https://gtdb.ecogenomic.org/">Genome Taxonomy Database</a></strong> <br />
Parks DH, Chuvochina M, Rinke C, Mussig AJ, Chaumeil P-A, Hugenholtz P <a href="https://doi.org/10.1093/nar/gkab776">GTDB: an ongoing census of bacterial and archaeal diversity through a phylogenetically consistent, rank normalized and complete genome-based taxonomy</a> <em>Nucleic Acids Research</em> gkab776 (2021)</p>
</li>
<li>
<p><strong><a href="https://github.com/phac-nml/mob-suite">MOB-suite Database</a></strong> <br />
Robertson J, Bessonov K, Schonfeld J, Nash JHE. <a href="https://doi.org/10.1099/mgen.0.000435">Universal whole-sequence-based plasmid typing and its utility to prediction of host range and epidemiological surveillance.</a> <em>Microbial Genomics</em>, 6(10)(2020)</p>
</li>
<li>
<p><strong><a href="https://www.ncbi.nlm.nih.gov/refseq/">NCBI RefSeq Database</a></strong> <br />
O'Leary NA, Wright MW, Brister JR, Ciufo S, Haddad D, McVeigh R, Rajput B, Robbertse B, Smith-White B, Ako-Adjei D, Astashyn A, Badretdin A, Bao Y, Blinkova O0, Brover V, Chetvernin V, Choi J, Cox E, Ermolaeva O, Farrell CM, Goldfarb T, Gupta T, Haft D, Hatcher E, Hlavina W, Joardar VS, Kodali VK, Li W, Maglott D, Masterson P, McGarvey KM, Murphy MR, O'Neill K, Pujar S, Rangwala SH, Rausch D, Riddick LD, Schoch C, Shkeda A, Storz SS, Sun H, Thibaud-Nissen F, Tolstoy I, Tully RE, Vatsan AR, Wallin C, Webb D, Wu W, Landrum MJ, Kimchi A, Tatusova T, DiCuccio M, Kitts P, Murphy TD, Pruitt KD <a href="https://doi.org/10.1093/nar/gkv1189">Reference sequence (RefSeq) database at NCBI: current status, taxonomic expansion, and functional annotation.</a> <em>Nucleic Acids Res.</em> 44, D733‚Äì45 (2016)</p>
</li>
<li>
<p><strong><a href="https://pubmlst.org/">PubMLST.org</a></strong> <br />
Jolley KA, Bray JE, Maiden MCJ <a href="http://dx.doi.org/10.12688/wellcomeopenres.14826.1">Open-access bacterial population genomics: BIGSdb software, the PubMLST.org website and their applications.</a> <em>Wellcome Open Res</em> 3, 124 (2018)</p>
</li>
<li>
<p><strong><a href="https://www.arb-silva.de/">SILVA rRNA Database</a></strong> <br />
Quast C, Pruesse E, Yilmaz P, Gerken J, Schweer T, Yarza P, Peplies J, Gl√∂ckner FO <a href="https://doi.org/10.1093/nar/gks1219">The SILVA ribosomal RNA gene database project: improved data processing and web-based tools.</a> <em>Nucleic Acids Res.</em> 41, D590‚Äì6 (2013)</p>
</li>
</ol>
<h2 id="software-included-in-bactopia">Software Included In Bactopia<a class="headerlink" href="#software-included-in-bactopia" title="Permanent link">&para;</a></h2>
<p>Below are 121 of software packages used (directly and indirectly) by 
Bactopia. A link to the software page as well as the citation (if available) 
have been included.</p>
<ol>
<li>
<p><strong><a href="https://github.com/tseemann/abricate">Abricate</a></strong><br />
Mass screening of contigs for antimicrobial and virulence genes<br />
Seemann T <a href="https://github.com/tseemann/abricate">Abricate: mass screening of contigs for antimicrobial and virulence genes</a> (GitHub)</p>
</li>
<li>
<p><strong><a href="https://github.com/VishnuRaghuram94/AgrVATE">AgrVATE</a></strong><br />
Rapid identification of Staphylococcus aureus agr locus type and agr operon variants.<br />
Raghuram V. <a href="https://github.com/VishnuRaghuram94/AgrVATE">AgrVATE: Rapid identification of Staphylococcus aureus agr locus type and agr operon variants.</a> (GitHub)</p>
</li>
<li>
<p><strong><a href="https://github.com/ncbi/amr">AMRFinderPlus</a></strong><br />
Find acquired antimicrobial resistance genes and some point mutations in protein or assembled nucleotide sequences.<br />
Feldgarden M, Brover V, Haft DH, Prasad AB, Slotta DJ, Tolstoy I, Tyson GH, Zhao S, Hsu C-H, McDermott PF, Tadesse DA, Morales C, Simmons M, Tillman G, Wasilenko J, Folster JP, Klimke W <a href="https://doi.org/10.1128/AAC.00483-19">Validating the NCBI AMRFinder Tool and Resistance Gene Database Using Antimicrobial Resistance Genotype-Phenotype Correlations in a Collection of NARMS Isolates</a>. <em>Antimicrob. Agents Chemother.</em> (2019)</p>
</li>
<li>
<p><strong><a href="https://github.com/tseemann/any2fasta">any2fasta</a></strong><br />
Convert various sequence formats to FASTA<br />
Seemann T <a href="https://github.com/tseemann/any2fasta">any2fasta: Convert various sequence formats to FASTA</a> (GitHub)</p>
</li>
<li>
<p><strong><a href="http://130.235.244.92/ARAGORN/Downloads/">Aragorn</a></strong><br />
Finds transfer RNA features (tRNA)<br />
Laslett D, Canback B <a href="https://doi.org/10.1093/nar/gkh152">ARAGORN, a program to detect tRNA genes and tmRNA genes in nucleotide sequences.</a> <em>Nucleic Acids Res</em>. 32(1):11-6 (2004)</p>
</li>
<li>
<p><strong><a href="https://github.com/sanger-pathogens/ariba">Ariba</a></strong><br />
Antimicrobial Resistance Identification By Assembly<br />
Hunt M, Mather AE, S√°nchez-Bus√≥ L, Page AJ, Parkhill J, Keane JA, Harris SR <a href="http://dx.doi.org/10.1099/mgen.0.000131">ARIBA: rapid antimicrobial resistance genotyping directly from sequencing reads</a>. <em>Microb Genom</em> 3, e000131 (2017)</p>
</li>
<li>
<p><strong><a href="https://www.niehs.nih.gov/research/resources/software/biostatistics/art/index.cfm">ART</a></strong><br />
A set of simulation tools to generate synthetic next-generation sequencing reads<br />
Huang W, Li L, Myers JR, Marth GT <a href="http://dx.doi.org/10.1093/bioinformatics/btr708">ART: a next-generation sequencing read simulator.</a> <em>Bioinformatics</em> 28, 593‚Äì594 (2012)</p>
</li>
<li>
<p><strong><a href="https://github.com/rpetit3/assembly-scan">assembly-scan</a></strong><br />
Generate basic stats for an assembly.<br />
Petit III RA <a href="https://github.com/rpetit3/assembly-scan">assembly-scan: generate basic stats for an assembly</a> (GitHub)</p>
</li>
<li>
<p><strong><a href="https://github.com/oschwengers/bakta">Bakta</a></strong><br />
Rapid &amp; standardized annotation of bacterial genomes &amp; plasmids<br />
Schwengers O, Jelonek L, Dieckmann MA, Beyvers S, Blom J, Goesmann A <a href="https://doi.org/10.1099/mgen.0.000685">Bakta - rapid and standardized annotation of bacterial genomes via alignment-free sequence identification.</a> <em>Microbial Genomics</em> 7(11) (2021)</p>
</li>
<li>
<p><strong><a href="https://github.com/tseemann/barrnap">Barrnap</a></strong><br />
Bacterial ribosomal RNA predictor<br />
Seemann T <a href="https://github.com/tseemann/barrnap">Barrnap: Bacterial ribosomal RNA predictor</a> (GitHub)</p>
</li>
<li>
<p><strong><a href="https://jgi.doe.gov/data-and-tools/bbtools/">BBTools</a></strong><br />
BBTools is a suite of fast, multithreaded bioinformatics tools designed for analysis of DNA and RNA sequence data.<br />
Bushnell B <a href="http://sourceforge.net/projects/bbmap/">BBMap short read aligner, and other bioinformatic tools.</a> (Link)</p>
</li>
<li>
<p><strong><a href="https://github.com/samtools/bcftools">BCFtools</a></strong><br />
Utilities for variant calling and manipulating VCFs and BCFs.<br />
Danecek P, Bonfield JK, Liddle J, Marshall J, Ohan V, Pollard MO, Whitwham A, Keane T, McCarthy SA, Davies RM, Li H <a href="https://doi.org/10.1093/gigascience/giab008">Twelve years of SAMtools and BCFtools</a> <em>GigaScience</em> Volume 10, Issue 2 (2021)</p>
</li>
<li>
<p><strong><a href="https://github.com/arq5x/bedtools2">Bedtools</a></strong><br />
A powerful toolset for genome arithmetic.<br />
Quinlan AR, Hall IM <a href="http://dx.doi.org/10.1093/bioinformatics/btq033">BEDTools: a flexible suite of utilities for comparing genomic features</a>. <em>Bioinformatics</em> 26, 841‚Äì842 (2010)</p>
</li>
<li>
<p><strong><a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi">BLAST</a></strong><br />
Basic Local Alignment Search Tool<br />
Camacho C, Coulouris G, Avagyan V, Ma N, Papadopoulos J, Bealer K, Madden TL <a href="http://dx.doi.org/10.1186/1471-2105-10-421">BLAST+: architecture and applications</a>. <em>BMC Bioinformatics</em> 10, 421 (2009)</p>
</li>
<li>
<p><strong><a href="https://github.com/BenLangmead/bowtie2">Bowtie2</a></strong><br />
A fast and sensitive gapped read aligner<br />
Langmead B, Salzberg SL <a href="http://dx.doi.org/10.1038/nmeth.1923">Fast gapped-read alignment with Bowtie 2.</a> <em>Nat. Methods.</em> 9, 357‚Äì359 (2012)</p>
</li>
<li>
<p><strong><a href="https://gitlab.com/ezlab/busco">BUSCO</a></strong><br />
Assessing genome assembly and annotation completeness with Benchmarking Universal Single-Copy Orthologs (BUSCO)<br />
Manni M, Berkeley MR, Seppey M, Sim√£o FA, Zdobnov EM <a href="https://doi.org/10.1093/molbev/msab199">BUSCO Update: Novel and Streamlined Workflows along with Broader and Deeper Phylogenetic Coverage for Scoring of Eukaryotic, Prokaryotic, and Viral Genomes.</a> <em>Molecular Biology and Evolution</em> 38(10), 4647‚Äì4654. (2021)</p>
</li>
<li>
<p><strong><a href="https://github.com/lh3/bwa/">BWA</a></strong><br />
Burrow-Wheeler Aligner for short-read alignment<br />
Li H <a href="http://arxiv.org/abs/1303.3997">Aligning sequence reads, clone sequences and assembly contigs with BWA-MEM</a>. <em>arXiv</em> [q-bio.GN] (2013)</p>
</li>
<li>
<p><strong><a href="https://github.com/weizhongli/cdhit">CD-HIT</a></strong><br />
Accelerated for clustering the next-generation sequencing data<br />
Li W, Godzik A <a href="http://dx.doi.org/10.1093/bioinformatics/btl158">Cd-hit: a fast program for clustering and comparing large sets of protein or nucleotide sequences</a>. <em>Bioinformatics</em> 22, 1658‚Äì1659 (2006)</p>
</li>
<li>
<p><strong><a href="https://github.com/weizhongli/cdhit">CD-HIT-EST</a></strong><br />
Accelerated for clustering the next-generation sequencing data<br />
Fu L, Niu B, Zhu Z, Wu S, Li W <a href="http://dx.doi.org/10.1093/bioinformatics/bts565">CD-HIT: accelerated for clustering the next-generation sequencing data</a>. <em>Bioinformatics</em> 28, 3150‚Äì3152 (2012)</p>
</li>
<li>
<p><strong><a href="https://github.com/Ecogenomics/CheckM">CheckM</a></strong><br />
Assess the quality of microbial genomes recovered from isolates, single cells, and metagenomes<br />
Parks DH, Imelfort M, Skennerton CT, Hugenholtz P, Tyson GW <a href="http://dx.doi.org/10.1101/gr.186072.114">CheckM: assessing the quality of microbial genomes recovered from isolates, single cells, and metagenomes.</a> <em>Genome Res</em> 25, 1043‚Äì1055 (2015)</p>
</li>
<li>
<p><strong><a href="https://github.com/xavierdidelot/ClonalFrameML">ClonalFramML</a></strong><br />
Efficient Inference of Recombination in Whole Bacterial Genomes<br />
Didelot X, Wilson DJ <a href="https://doi.org/10.1371/journal.pcbi.1004041">ClonalFrameML: Efficient Inference of Recombination in Whole Bacterial Genomes.</a> <em>PLoS Comput Biol</em> 11(2) e1004041 (2015)</p>
</li>
<li>
<p><strong><a href="https://github.com/bbuchfink/diamond">DIAMOND</a></strong><br />
Accelerated BLAST compatible local sequence aligner.<br />
Buchfink B, Xie C, Huson DH <a href="http://dx.doi.org/10.1038/nmeth.3176">Fast and sensitive protein alignment using DIAMOND.</a> <em>Nat. Methods.</em> 12, 59‚Äì60 (2015)</p>
</li>
<li>
<p><strong><a href="https://github.com/phac-nml/ecoli_serotyping">ECTyper</a></strong><br />
In-silico prediction of <em>Escherichia coli</em> serotype<br />
Laing C, Bessonov K, Sung S, La Rose C <a href="https://github.com/phac-nml/ecoli_serotyping">ECTyper - In silico prediction of <em>Escherichia coli</em> serotype</a> (GitHub)  </p>
</li>
<li>
<p><strong><a href="https://github.com/eggnogdb/eggnog-mapper">eggNOG-mapper</a></strong><br />
Fast genome-wide functional annotation through orthology assignment<br />
Huerta-Cepas J, Forslund K, Coelho LP, Szklarczyk D, Jensen LJ, von Mering C, Bork P <a href="http://dx.doi.org/10.1093/molbev/msx148">Fast Genome-Wide Functional Annotation through Orthology Assignment by eggNOG-Mapper.</a> <em>Mol. Biol. Evol.</em> 34, 2115‚Äì2122 (2017)</p>
</li>
<li>
<p><strong><a href="https://github.com/MDU-PHL/emmtyper">emmtyper</a></strong><br />
emm Automatic Isolate Labeller<br />
Tan A, Seemann T, Lacey D, Davies M, Mcintyre L, Frost H, Williamson D, Gon√ßalves da Silva A <a href="https://github.com/MDU-PHL/emmtyper">emmtyper - emm Automatic Isolate Labeller</a> (GitHub)</p>
</li>
<li>
<p><strong><a href="https://github.com/ParBLiSS/FastANI">FastANI</a></strong><br />
Fast Whole-Genome Similarity (ANI) Estimation<br />
Jain C, Rodriguez-R LM, Phillippy AM, Konstantinidis KT, Aluru S <a href="http://dx.doi.org/10.1038/s41467-018-07641-9">High throughput ANI analysis of 90K prokaryotic genomes reveals clear species boundaries.</a> <em>Nat. Commun.</em> 9, 5114 (2018)</p>
</li>
<li>
<p><strong><a href="https://github.com/s-andrews/FastQC">FastQC</a></strong><br />
A quality control analysis tool for high throughput sequencing data.<br />
Andrews S <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc">FastQC: a quality control tool for high throughput sequence data.</a> (WebLink)</p>
</li>
<li>
<p><strong><a href="https://github.com/rpetit3/fastq-dl">fastq-dl</a></strong><br />
Download FASTQ files from SRA or ENA repositories.<br />
Petit III RA <a href="https://github.com/rpetit3/fastq-dl">fastq-dl: Download FASTQ files from SRA or ENA repositories.</a> (GitHub)</p>
</li>
<li>
<p><strong><a href="https://github.com/rpetit3/fastq-scan">fastq-scan</a></strong><br />
Output FASTQ summary statistics in JSON format<br />
Petit III RA <a href="https://github.com/rpetit3/fastq-scan">fastq-scan: generate summary statistics of input FASTQ sequences.</a> (GitHub)</p>
</li>
<li>
<p><strong><a href="http://www.microbesonline.org/fasttree">FastTree</a></strong><br />
Approximately-maximum-likelihood phylogenetic trees<br />
Price MN, Dehal PS, Arkin AP <a href="https://dx.doi.org/10.1371%2Fjournal.pone.0009490">FastTree 2 ‚Äì Approximately Maximum-Likelihood Trees for Large Alignments.</a> <em>PLoS One</em> 5, e9490 (2010)</p>
</li>
<li>
<p><strong><a href="https://ccb.jhu.edu/software/FLASH/">FLASH</a></strong><br />
A fast and accurate tool to merge paired-end reads.<br />
Magoƒç T, Salzberg SL <a href="https://doi.org/10.1093/bioinformatics/btr507">FLASH: fast length adjustment of short reads to improve genome assemblies.</a> <em>Bioinformatics</em> 27.21 2957-2963 (2011)</p>
</li>
<li>
<p><strong><a href="https://github.com/fenderglass/Flye">Flye</a></strong><br />
De novo assembler for single molecule sequencing reads using repeat graphs<br />
Kolmogorov M, Yuan J, Lin Y, Pevzner P <a href="https://doi.org/10.1038/s41587-019-0072-8">Assembly of Long Error-Prone Reads Using Repeat Graphs</a> <em>Nature Biotechnology</em> (2019)</p>
</li>
<li>
<p><strong><a href="https://github.com/ekg/freebayes">freebayes</a></strong><br />
Bayesian haplotype-based genetic polymorphism discovery and genotyping<br />
Garrison E, Marth G <a href="https://arxiv.org/abs/1207.3907">Haplotype-based variant detection from short-read sequencing.</a> arXiv preprint arXiv:1207.3907 [q-bio.GN] (2012)</p>
</li>
<li>
<p><strong><a href="https://github.com/rastanton/GAMMA">GAMMA</a></strong><br />
Gene Allele Mutation Microbial Assessment<br />
Stanton RA, Vlachos N, Halpin AL <a href="https://doi.org/10.1093/bioinformatics/btab607">GAMMA: a tool for the rapid identification, classification, and annotation of translated gene matches from sequencing data.</a> <em>Bioinformatics</em> (2021)</p>
</li>
<li>
<p><strong><a href="https://github.com/katholt/genotyphi">GenoTyphi</a></strong><br />
Assign genotypes to Salmonella Typhi genomes based on Mykrobe results<br />
Wong VK, Baker S, Connor TR, Pickard D, Page AJ, Dave J, Murphy N, Holliman R, Sefton A, Millar M, Dyson ZA, Dougan G, Holt KE, &amp; International Typhoid Consortium. <a href="https://doi.org/10.1038/ncomms12827">An extended genotyping framework for Salmonella enterica serovar Typhi, the cause of human typhoid</a> <em>Nature Communications</em> 7, 12827. (2016)</p>
</li>
<li>
<p><strong><a href="https://www.gnu.org/software/parallel/">GNU Parallel</a></strong><br />
A shell tool for executing jobs in parallel<br />
Tange O <a href="https://doi.org/10.5281/zenodo.1146014">GNU Parallel</a> (2018)</p>
</li>
<li>
<p><strong><a href="https://github.com/Ecogenomics/GTDBTk">GTDB-Tk</a></strong><br />
A toolkit for assigning objective taxonomic classifications to bacterial and archaeal genomes<br />
Chaumeil PA, Mussig AJ, Hugenholtz P, Parks DH <a href="https://doi.org/10.1093/bioinformatics/btz848">GTDB-Tk: a toolkit to classify genomes with the Genome Taxonomy Database.</a> <em>Bioinformatics</em> (2019)</p>
</li>
<li>
<p><strong><a href="https://github.com/nickjcroucher/gubbins">Gubbins</a></strong><br />
Rapid phylogenetic analysis of large samples of recombinant bacterial whole genome sequences<br />
Croucher NJ, Page AJ, Connor TR, Delaney AJ, Keane JA, Bentley SD, Parkhill J, Harris SR <a href="https://doi.org/10.1093/nar/gku1196">Rapid phylogenetic analysis of large samples of recombinant bacterial whole genome sequences using Gubbins.</a> <em>Nucleic Acids Research</em> 43(3), e15. (2015)</p>
</li>
<li>
<p><strong><a href="https://github.com/scwatts/hicap">hicap</a></strong><br />
in silico typing of the <em>H. influenzae</em> cap locus<br />
Watts SC, Holt KE <a href="https://doi.org/10.1128/JCM.00190-19">hicap: in silico serotyping of the Haemophilus influenzae capsule locus.</a> <em>Journal of Clinical Microbiology</em> JCM.00190-19 (2019)</p>
</li>
<li>
<p><strong><a href="http://hmmer.org/">HMMER</a></strong><br />
Biosequence analysis using profile hidden Markov models<br />
Eddy SR <a href="https://doi.org/10.1371/journal.pcbi.1002195">Accelerated Profile HMM Searches.</a> <em>PLoS Comput. Biol.</em> 7, e1002195 (2011)</p>
</li>
<li>
<p><strong><a href="https://github.com/jimmyliu1326/HpsuisSero">HpsuisSero</a></strong><br />
Rapid <em>Haemophilus parasuis</em> serotyping<br />
Lui J <a href="https://github.com/jimmyliu1326/HpsuisSero">HpsuisSero: Rapid <em>Haemophilus parasuis</em> serotyping</a> (GitHub)</p>
</li>
<li>
<p><strong><a href="http://eddylab.org/infernal/">Infernal</a></strong><br />
Searches DNA sequence databases for RNA structure and sequence similarities<br />
Nawrocki EP, Eddy SR <a href="https://doi.org/10.1093/bioinformatics/btt509">Infernal 1.1: 100-fold faster RNA homology searches.</a> <em>Bioinformatics</em> 29(22), 2933-2935 (2013)</p>
</li>
<li>
<p><strong><a href="https://github.com/Cibiv/IQ-TREE">IQ-TREE</a></strong><br />
Efficient phylogenomic software by maximum likelihood<br />
Nguyen L-T, Schmidt HA, von Haeseler A, Minh BQ <a href="https://doi.org/10.1093/molbev/msu300">IQ-TREE: A fast and effective stochastic algorithm for estimating maximum likelihood phylogenies.</a> <em>Mol. Biol. Evol.</em> 32:268-274 (2015)</p>
</li>
<li>
<p><strong><a href="https://github.com/Cibiv/IQ-TREE">ModelFinder</a></strong><br />
Used for automatic model selection<br />
Kalyaanamoorthy S, Minh BQ, Wong TKF, von Haeseler A, Jermiin LS <a href="https://doi.org/10.1038/nmeth.4285">ModelFinder - Fast model selection for accurate phylogenetic estimates.</a> <em>Nat. Methods</em> 14:587-589 (2017)</p>
</li>
<li>
<p><strong><a href="https://github.com/Cibiv/IQ-TREE">UFBoot2</a></strong><br />
Used to conduct ultrafast bootstrapping<br />
Hoang DT, Chernomor O, von Haeseler A, Minh BQ, Vinh LS <a href="https://doi.org/10.1093/molbev/msx281">UFBoot2: Improving the ultrafast bootstrap approximation.</a> <em>Mol. Biol. Evol.</em> 35:518‚Äì522 (2018)</p>
</li>
<li>
<p><strong><a href="https://github.com/jhawkey/IS_mapper">ISMapper</a></strong><br />
IS mapping software<br />
Hawkey J, Hamidian M, Wick RR, Edwards DJ, Billman-Jacobe H, Hall RM, Holt KE <a href="http://dx.doi.org/10.1186/s12864-015-1860-2">ISMapper: identifying transposase insertion sites in bacterial genomes from short read sequence data</a>. <em>BMC Genomics</em> 16, 667 (2015)</p>
</li>
<li>
<p><strong><a href="https://github.com/katholt/Kaptive">Kaptive</a></strong><br />
Surface polysaccharide loci for <em>Klebsiella pneumoniae</em> species complex and <em>Acinetobacter baumannii</em> genomes<br />
Wyres KL, Wick RR, Gorrie C, Jenney A, Follador R, Thomson NR, Holt KE <a href="https://doi.org/10.1099/mgen.0.000102">Identification of Klebsiella capsule synthesis loci from whole genome data.</a> <em>Microbial genomics</em> 2(12) (2016)</p>
</li>
<li>
<p><strong><a href="https://github.com/katholt/Kleborate">Kleborate</a></strong><br />
Genotyping tool for <em>Klebsiella pneumoniae</em> and its related species complex<br />
Lam MMC, Wick RR, Watts, SC, Cerdeira LT, Wyres KL, Holt KE <a href="https://doi.org/10.1038/s41467-021-24448-3">A genomic surveillance framework and genotyping tool for Klebsiella pneumoniae and its related species complex.</a> <em>Nat Commun</em> 12, 4188 (2021)</p>
</li>
<li>
<p><strong><a href="https://github.com/refresh-bio/KMC">KMC</a></strong><br />
Fast and frugal disk based k-mer counter<br />
Deorowicz S, Kokot M, Grabowski Sz, Debudaj-Grabysz A <a href="https://doi.org/10.1093/bioinformatics/btv022">KMC 2: Fast and resource-frugal k-mer counting</a> <em>Bioinformatics</em> 31(10):1569‚Äì1576 (2015)</p>
</li>
<li>
<p><strong><a href="https://github.com/DerrickWood/kraken2">Kraken2</a></strong><br />
The second version of the Kraken taxonomic sequence classification system<br />
Wood DE, Lu J, Langmead B <a href="https://doi.org/10.1186/s13059-019-1891-0">Improved metagenomic analysis with Kraken 2.</a> <em>Genome Biology</em>, 20(1), 257. (2019)</p>
</li>
<li>
<p><strong><a href="https://github.com/tseemann/legsta">legsta</a></strong><br />
In silico Legionella pneumophila Sequence Based Typing<br />
Seemann T <a href="https://github.com/tseemann/legsta">legsta: In silico Legionella pneumophila Sequence Based Typing</a> (GitHub)</p>
</li>
<li>
<p><strong><a href="https://github.com/mourisl/Lighter">Lighter</a></strong><br />
Fast and memory-efficient sequencing error corrector<br />
Song L, Florea L, Langmead B <a href="https://doi.org/10.1186/s13059-014-0509-9">Lighter: Fast and Memory-efficient Sequencing Error Correction without Counting</a>. <em>Genome Biol.</em> 15(11):509 (2014)</p>
</li>
<li>
<p><strong><a href="https://github.com/MDU-PHL/LisSero">LisSero</a></strong><br />
<em>In silico</em> serotype prediction for <em>Listeria monocytogenes</em><br />
Kwong J, Zhang J, Seeman T, Horan, K, Gon√ßalves da Silva A <a href="https://github.com/MDU-PHL/LisSero">LisSero - <em>In silico</em> serotype prediction for <em>Listeria monocytogenes</em></a> (GitHub)</p>
</li>
<li>
<p><strong><a href="https://mafft.cbrc.jp/alignment/software/">MAFFT</a></strong><br />
Multiple alignment program for amino acid or nucleotide sequences<br />
Katoh K, Standley DM <a href="https://doi.org/10.1093/molbev/mst010">MAFFT multiple sequence alignment software version 7: improvements in performance and usability.</a> <em>Mol. Biol. Evol.</em> 30, 772‚Äì780 (2013)</p>
</li>
<li>
<p><strong><a href="https://github.com/marbl/Mash">Mash</a></strong><br />
Fast genome and metagenome distance estimation using MinHash<br />
Ondov BD, Treangen TJ, Melsted P, Mallonee AB, Bergman NH, Koren S, Phillippy AM <a href="http://dx.doi.org/10.1186/s13059-016-0997-x">Mash: fast genome and metagenome distance estimation using MinHash</a>. <em>Genome Biol</em> 17, 132 (2016)</p>
</li>
<li>
<p><strong><a href="https://github.com/marbl/Mash">Mash</a></strong><br />
High-throughput sequence containment estimation<br />
Ondov BD, Starrett GJ, Sappington A, Kostic A, Koren S, Buck CB, Phillippy AM <a href="https://doi.org/10.1186/s13059-019-1841-x">Mash Screen: high-throughput sequence containment estimation for genome discovery</a> <em>Genome Biol</em> 20, 232 (2019)</p>
</li>
<li>
<p><strong><a href="https://github.com/lskatz/mashtree">Mashtree</a></strong><br />
Create a tree using Mash distances<br />
Katz LS, Griswold T, Morrison S, Caravas J, Zhang S, den Bakker HC, Deng X, Carleton HA <a href="https://doi.org/10.21105/joss.01762">Mashtree: a rapid comparison of whole genome sequence files.</a> <em>Journal of Open Source Software</em>, 4(44), 1762 (2019)</p>
</li>
<li>
<p><strong><a href="https://github.com/kwongj/maskrc-svg">maskrc-svg</a></strong><br />
Masks recombination as detected by ClonalFrameML or Gubbins<br />
Kwong J <a href="https://github.com/kwongj/maskrc-svg">maskrc-svg - Masks recombination as detected by ClonalFrameML or Gubbins and draws an SVG.</a> (GitHub)  </p>
</li>
<li>
<p><strong><a href="https://github.com/mcveanlab/mccortex">McCortex</a></strong><br />
De novo genome assembly and multisample variant calling<br />
Turner I, Garimella KV, Iqbal Z, McVean G <a href="http://dx.doi.org/10.1093/bioinformatics/bty157">Integrating long-range connectivity information into de Bruijn graphs.</a> <em>Bioinformatics</em> 34, 2556‚Äì2565 (2018)</p>
</li>
<li>
<p><strong><a href="https://github.com/liampshaw/mcroni">mcroni</a></strong><br />
Scripts for finding and processing promoter variants upstream of mcr-1<br />
Shaw L <a href="https://github.com/liampshaw/mcroni">mcroni: Scripts for finding and processing promoter variants upstream of mcr-1</a> (GitHub)</p>
</li>
<li>
<p><strong><a href="None">Medaka</a></strong><br />
Sequence correction provided by ONT Research<br />
ONT Research <a href="https://github.com/nanoporetech/medaka">Medaka: Sequence correction provided by ONT Research</a> (GitHub)</p>
</li>
<li>
<p><strong><a href="https://github.com/MDU-PHL/meningotype">meningotype</a></strong><br />
In silico serotyping, finetyping and Bexsero antigen sequence typing of <em>Neisseria meningitidis</em><br />
Kwong JC, Gon√ßalves da Silva A, Stinear TP, Howden BP, &amp; Seemann T <a href="https://github.com/MDU-PHL/meningotype">meningotype: in silico typing for <em>Neisseria meningitidis</em>.</a> (GitHub)</p>
</li>
<li>
<p><strong><a href="https://github.com/voutcn/megahit">MEGAHIT</a></strong><br />
Ultra-fast and memory-efficient (meta-)genome assembler<br />
Li D, Liu C-M, Luo R, Sadakane K, Lam T-W <a href="https://doi.org/10.1093/bioinformatics/btv033">MEGAHIT: an ultra-fast single-node solution for large and complex metagenomics assembly via succinct de Bruijn graph.</a> <em>Bioinformatics</em> 31.10 1674-1676 (2015)</p>
</li>
<li>
<p><strong><a href="https://github.com/tseemann/mlst">mlst</a></strong><br />
Scan contig files against PubMLST typing schemes<br />
Seemann T <a href="https://github.com/tseemann/mlst">mlst: scan contig files against PubMLST typing schemes</a> (GitHub)</p>
</li>
<li>
<p><strong><a href="https://github.com/ctSkennerton/minced">MinCED</a></strong><br />
Mining CRISPRs in Environmental Datasets<br />
Skennerton C <a href="https://github.com/ctSkennerton/minced">MinCED: Mining CRISPRs in Environmental Datasets</a> (GitHub)</p>
</li>
<li>
<p><strong><a href="https://github.com/lh3/miniasm">Miniasm</a></strong><br />
Ultrafast de novo assembly for long noisy reads (though having no consensus step)<br />
Li H <a href="https://github.com/lh3/miniasm">Miniasm: Ultrafast de novo assembly for long noisy reads</a> (GitHub)</p>
</li>
<li>
<p><strong><a href="https://github.com/lh3/minimap2">Minimap2</a></strong><br />
A versatile pairwise aligner for genomic and spliced nucleotide sequences<br />
Li H <a href="https://doi.org/10.1093/bioinformatics/bty191">Minimap2: pairwise alignment for nucleotide sequences.</a> <em>Bioinformatics</em> 34:3094-3100 (2018)</p>
</li>
<li>
<p><strong><a href="https://github.com/phac-nml/mob-suite">MOB-suite</a></strong><br />
Software tools for clustering, reconstruction and typing of plasmids from draft assemblies<br />
Robertson J, Nash JHE <a href="https://doi.org/10.1099/mgen.0.000206">MOB-suite: software tools for clustering, reconstruction and typing of plasmids from draft assemblies.</a> <em>Microbial Genomics</em> 4(8). (2018)</p>
</li>
<li>
<p><strong><a href="https://github.com/Mykrobe-tools/mykrobe">Mykrobe</a></strong><br />
Antibiotic resistance prediction in minutes<br />
Hunt M, Bradley P, Lapierre SG, Heys S, Thomsit M, Hall MB, Malone KM, Wintringer P, Walker TM, Cirillo DM, Comas I, Farhat MR, Fowler P, Gardy J, Ismail N, Kohl TA, Mathys V, Merker M, Niemann S, Omar SV, Sintchenko V, Smith G, Supply P, Tahseen S, Wilcox M, Arandjelovic I, Peto TEA, Crook, DW, Iqbal Z <a href="https://doi.org/10.12688/wellcomeopenres.15603.1">Antibiotic resistance prediction for Mycobacterium tuberculosis from genome sequence data with Mykrobe</a> <em>Wellcome Open Research</em> 4, 191. (2019)</p>
</li>
<li>
<p><strong><a href="https://github.com/wdecoster/NanoPlot">NanoPlot</a></strong><br />
Plotting scripts for long read sequencing data<br />
De Coster W, D‚ÄôHert S, Schultz DT, Cruts M, Van Broeckhoven C <a href="https://doi.org/10.1093/bioinformatics/bty149">NanoPack: visualizing and processing long-read sequencing data</a> <em>Bioinformatics</em> Volume 34, Issue 15 (2018) </p>
</li>
<li>
<p><strong><a href="https://github.com/esteinig/nanoq">Nanoq</a></strong><br />
Minimal but speedy quality control for nanopore reads in Rust<br />
Steinig E <a href="https://github.com/esteinig/nanoq">Nanoq: Minimal but speedy quality control for nanopore reads in Rust</a> (GitHub)</p>
</li>
<li>
<p><strong><a href="https://github.com/kblin/ncbi-genome-download">ncbi-genome-download</a></strong><br />
Scripts to download genomes from the NCBI FTP servers<br />
Blin K <a href="https://github.com/kblin/ncbi-genome-download">ncbi-genome-download: Scripts to download genomes from the NCBI FTP servers</a> (GitHub)</p>
</li>
<li>
<p><strong><a href="https://github.com/nextflow-io/nextflow">Nextflow</a></strong><br />
A DSL for data-driven computational pipelines.<br />
Di Tommaso P, Chatzou M, Floden EW, Barja PP, Palumbo E, Notredame C <a href="https://www.nature.com/articles/nbt.3820.pdf?origin=ppub">Nextflow enables reproducible computational workflows.</a> <em>Nat. Biotechnol.</em> 35, 316‚Äì319 (2017)</p>
</li>
<li>
<p><strong><a href="https://github.com/MDU-PHL/ngmaster">ngmaster</a></strong><br />
<em>In silico</em> multi-antigen sequence typing for <em>Neisseria gonorrhoeae</em> (NG-MAST)<br />
Kwong J, Gon√ßalves da Silva A, Schultz M, Seeman T <a href="https://github.com/MDU-PHL/ngmaster">ngmaster - <em>In silico</em> multi-antigen sequence typing for <em>Neisseria gonorrhoeae</em> (NG-MAST)</a> (GitHub)</p>
</li>
<li>
<p><strong><a href="http://hmmer.org/">nhmmer</a></strong><br />
DNA homology search with profile HMMs.<br />
Wheeler TJ, Eddy SR <a href="https://doi.org/10.1093/bioinformatics/btt403">nhmmer: DNA homology search with profile HMMs.</a> <em>Bioinformatics</em> 29, 2487‚Äì2489 (2013)</p>
</li>
<li>
<p><strong><a href="https://github.com/gtonkinhill/panaroo">Panaroo</a></strong><br />
An updated pipeline for pangenome investigation<br />
Tonkin-Hill G, MacAlasdair N, Ruis C, Weimann A, Horesh G, Lees JA, Gladstone RA, Lo S, Beaudoin C, Floto RA, Frost SDW, Corander J, Bentley SD, Parkhill J <a href="https://doi.org/10.1186/s13059-020-02090-4">Producing polished prokaryotic pangenomes with the Panaroo pipeline.</a> <em>Genome Biology</em> 21(1), 180. (2020)</p>
</li>
<li>
<p><strong><a href="https://github.com/HRGV/phyloFlash">phyloFlash</a></strong><br />
A pipeline to rapidly reconstruct the SSU rRNAs and explore phylogenetic composition of an illumina (meta)genomic dataset.<br />
Gruber-Vodicka HR, Seah BKB, Pruesse E <a href="https://doi.org/10.1128/mSystems.00920-20">phyloFlash: Rapid Small-Subunit rRNA Profiling and Targeted Assembly from Metagenomes</a> <em>mSystems</em> 5 (2020)</p>
</li>
<li>
<p><strong><a href="https://zlib.net/pigz/">Pigz</a></strong><br />
A parallel implementation of gzip for modern multi-processor, multi-core machines.<br />
Adler M. <a href="https://zlib.net/pigz/">pigz: A parallel implementation of gzip for modern multi-processor, multi-core machines.</a> <em>Jet Propulsion Laboratory</em> (2015)</p>
</li>
<li>
<p><strong><a href="https://github.com/broadinstitute/pilon/">Pilon</a></strong><br />
An automated genome assembly improvement and variant detection tool<br />
Walker BJ, Abeel T,  Shea T, Priest M, Abouelliel A, Sakthikumar S, Cuomo CA, Zeng Q, Wortman J, Young SK, Earl AM <a href="https://doi.org/10.1371/journal.pone.0112963">Pilon: an integrated tool for comprehensive microbial variant detection and genome assembly improvement.</a> <em>PloS one</em> 9.11 e112963 (2014)</p>
</li>
<li>
<p><strong><a href="http://github.com/SionBayliss/PIRATE">PIRATE</a></strong><br />
A toolbox for pangenome analysis and threshold evaluation.<br />
Bayliss SC, Thorpe HA, Coyle NM, Sheppard SK, Feil EJ <a href="https://doi.org/10.1093/gigascience/giz119">PIRATE: A fast and scalable pangenomics toolbox for clustering diverged orthologues in bacteria.</a> <em>Gigascience</em> 8 (2019)</p>
</li>
<li>
<p><strong><a href="https://bitbucket.org/genomicepidemiology/plasmidfinder">PlasmidFinder</a></strong><br />
Identifies plasmids in total or partial sequenced isolates of bacteria<br />
Carattoli A, Zankari E, Garc√≠a-Fern√°ndez A, Voldby Larsen M, Lund O, Villa L, M√∏ller Aarestrup F, Hasman H <a href="https://doi.org/10.1128/AAC.02412-14">In silico detection and typing of plasmids using PlasmidFinder and plasmid multilocus sequence typing.</a> <em>Antimicrobial Agents and Chemotherapy</em> 58(7), 3895‚Äì3903. (2014)</p>
</li>
<li>
<p><strong><a href="https://github.com/rrwick/Porechop">Porechop</a></strong><br />
adapter trimmer for Oxford Nanopore reads<br />
Wick RR, Judd LM, Gorrie CL, Holt KE. <a href="https://doi.org/10.1099/mgen.0.000132">Completing bacterial genome assemblies with multiplex MinION sequencing.</a> <em>Microb Genom.</em> 3(10):e000132 (2017)</p>
</li>
<li>
<p><strong><a href="https://github.com/matsen/pplacer">pplacer</a></strong><br />
Phylogenetic placement and downstream analysis<br />
Matsen FA, Kodner RB, Armbrust EV <a href="https://doi.org/10.1186/1471-2105-11-538">pplacer: linear time maximum-likelihood and Bayesian phylogenetic placement of sequences onto a fixed reference tree.</a> <em>BMC Bioinformatics</em> 11, 538 (2010)</p>
</li>
<li>
<p><strong><a href="https://github.com/hyattpd/Prodigal">Prodigal</a></strong><br />
Fast, reliable protein-coding gene prediction for prokaryotic genomes.<br />
Hyatt D, Chen G-L, LoCascio PF, Land ML, Larimer FW, Hauser LJ <a href="https://doi.org/10.1186/1471-2105-11-119">Prodigal: prokaryotic gene recognition and translation initiation site identification.</a> <em>BMC Bioinformatics</em> 11.1 119 (2010)</p>
</li>
<li>
<p><strong><a href="https://github.com/tseemann/prokka">Prokka</a></strong><br />
Rapid prokaryotic genome annotation<br />
Seemann T <a href="http://dx.doi.org/10.1093/bioinformatics/btu153">Prokka: rapid prokaryotic genome annotation</a> <em>Bioinformatics</em> 30, 2068‚Äì2069 (2014)</p>
</li>
<li>
<p><strong><a href="http://quast.sourceforge.net/">QUAST</a></strong><br />
Quality Assessment Tool for Genome<br />
Gurevich A, Saveliev V, Vyahhi N, Tesler G <a href="http://dx.doi.org/10.1093/bioinformatics/btt086">QUAST: quality assessment tool for genome assemblies.</a> <em>Bioinformatics</em> 29, 1072‚Äì1075 (2013)</p>
</li>
<li>
<p><strong><a href="https://github.com/lbcb-sci/racon">Racon</a></strong><br />
Ultrafast consensus module for raw de novo genome assembly of long uncorrected reads<br />
Vaser R, Soviƒá I, Nagarajan N, ≈†ikiƒá M <a href="http://dx.doi.org/10.1101/gr.214270.116">Fast and accurate de novo genome assembly from long uncorrected reads.</a> <em>Genome Res</em> 27, 737‚Äì746 (2017)</p>
</li>
<li>
<p><strong><a href="https://github.com/mbhall88/rasusa">Rasusa</a></strong><br />
Randomly subsample sequencing reads to a specified coverage<br />
Hall MB <a href="https://doi.org/10.5281/zenodo.3731394">Rasusa: Randomly subsample sequencing reads to a specified coverage.</a> (2019).</p>
</li>
<li>
<p><strong><a href="https://github.com/lbcb-sci/raven">Raven</a></strong><br />
De novo genome assembler for long uncorrected reads<br />
Vaser R, ≈†ikiƒá M <a href="https://doi.org/10.1038/s43588-021-00073-4">Time- and memory-efficient genome assembly with Raven.</a> <em>Nat Comput Sci</em> 1, 332‚Äì336 (2021)</p>
</li>
<li>
<p><strong><a href="https://github.com/arpcard/rgi">Resistance Gene Identifier (RGI)</a></strong><br />
Software to predict resistomes from protein or nucleotide data, based on homology and SNP models.<br />
Alcock BP, Raphenya AR, Lau TTY, Tsang KK, Bouchard M, Edalatmand A, Huynh W, Nguyen A-L V, Cheng AA, Liu S, Min SY, Miroshnichenko A, Tran H-K, Werfalli RE, Nasir JA, Oloni M, Speicher DJ, Florescu A, Singh B, Faltyn M, Hernandez-Koutoucheva A, Sharma AN, Bordeleau E, Pawlowski AC, Zubyk HL, Dooley D, Griffiths E, Maguire F, Winsor GL, Beiko RG, Brinkman FSL, Hsiao WWL, Domselaar GV, McArthur AG <a href="https://doi.org/10.1093/nar/gkz935">CARD 2020: antibiotic resistome surveillance with the comprehensive antibiotic resistance database.</a> <em>Nucleic acids research</em> 48.D1, D517-D525 (2020)</p>
</li>
<li>
<p><strong><a href="http://www.cbs.dtu.dk/services/RNAmmer/">RNAmmer</a></strong><br />
Consistent and rapid annotation of ribosomal RNA genes<br />
Lagesen K, Hallin P, R√∏dland EA, St√¶rfeldt H-H, Rognes T, Ussery DW <a href="https://dx.doi.org/10.1093%2Fnar%2Fgkm160">RNAmmer: consistent annotation of rRNA genes in genomic sequences.</a> <em>Nucleic Acids Res</em> 35.9: 3100-3108 (2007)</p>
</li>
<li>
<p><strong><a href="https://github.com/sanger-pathogens/Roary">Roary</a></strong><br />
Rapid large-scale prokaryote pan genome analysis<br />
Page AJ, Cummins CA, Hunt M, Wong VK, Reuter S, Holden MTG, Fookes M, Falush D, Keane JA, Parkhill J <a href="https://doi.org/10.1093/bioinformatics/btv421">Roary: rapid large-scale prokaryote pan genome analysis.</a> <em>Bioinformatics</em> 31, 3691‚Äì3693 (2015)</p>
</li>
<li>
<p><strong><a href="https://github.com/tseemann/samclip">samclip</a></strong><br />
Filter SAM file for soft and hard clipped alignments<br />
Seemann T <a href="https://github.com/tseemann/samclip">Samclip: Filter SAM file for soft and hard clipped alignments</a> (GitHub)</p>
</li>
<li>
<p><strong><a href="https://github.com/samtools/samtools">Samtools</a></strong><br />
Tools for manipulating next-generation sequencing data<br />
Li H, Handsaker B, Wysoker A, Fennell T, Ruan J, Homer N, Marth G, Abecasis G, Durbin R <a href="http://dx.doi.org/10.1093/bioinformatics/btp352">The Sequence Alignment/Map format and SAMtools</a>. <em>Bioinformatics</em> 25, 2078‚Äì2079 (2009)</p>
</li>
<li>
<p><strong><a href="https://github.com/AdmiralenOla/Scoary">Scoary</a></strong><br />
Pan-genome wide association studies<br />
Brynildsrud O, Bohlin J, Scheffer L, Eldholm V <a href="https://doi.org/10.1186/s13059-016-1108-8">Rapid scoring of genes in microbial pan-genome-wide association studies with Scoary.</a> <em>Genome Biol.</em> 17:238 (2016)</p>
</li>
<li>
<p><strong><a href="https://github.com/denglab/SeqSero2">SeqSero2</a></strong><br />
Salmonella serotype prediction from genome sequencing data<br />
Zhang S, Den-Bakker HC, Li S, Dinsmore BA, Lane C, Lauer AC, Fields PI, Deng X. <a href="https://doi.org/10.1128/AEM.01746-19">SeqSero2: rapid and improved Salmonella serotype determination using whole genome sequencing data.</a> <em>Appl Environ Microbiology</em> 85(23):e01746-19 (2019)</p>
</li>
<li>
<p><strong><a href="https://github.com/lh3/seqtk">Seqtk</a></strong><br />
A fast and lightweight tool for processing sequences in the FASTA or FASTQ format.<br />
Li H <a href="https://github.com/lh3/seqtk">Toolkit for processing sequences in FASTA/Q formats</a> (GitHub)</p>
</li>
<li>
<p><strong><a href="https://github.com/sanger-pathogens/seroba">Seroba</a></strong><br />
k-mer based pipeline to identify the serotype of <em>Streptococcus pneumoniae</em> from Illumina NGS reads<br />
Epping L, van Tonder AJ, Gladstone RA, The Global Pneumococcal Sequencing Consortium, Bentley SD, Page AJ, Keane JA <a href="https://doi.org/10.1099/mgen.0.000186">SeroBA: rapid high-throughput serotyping of Streptococcus pneumoniae from whole genome sequence data.</a> <em>Microbial Genomics</em>, 4(7) (2018)</p>
</li>
<li>
<p><strong><a href="https://github.com/CFSAN-Biostatistics/shigatyper">ShigaTyper</a></strong><br />
Shigella serotype from Illumina or Oxford Nanopore reads<br />
Wu Y, Lau HK, Lee T, Lau DK, Payne J <a href="https://doi.org/10.1128/AEM.00165-19">In Silico Serotyping Based on Whole-Genome Sequencing Improves the Accuracy of Shigella Identification.</a> <em>Applied and Environmental Microbiology</em>, 85(7). (2019)</p>
</li>
<li>
<p><strong><a href="https://github.com/tseemann/shovill">Shovill</a></strong><br />
Faster assembly of Illumina reads<br />
Seemann T <a href="https://github.com/tseemann/shovill">Shovill: De novo assembly pipeline for Illumina paired reads</a> (GitHub)</p>
</li>
<li>
<p><strong><a href="http://www.cbs.dtu.dk/services/SignalP-4.0/">SignalP</a></strong><br />
SISTR (Salmonella In Silico Typing Resource) command-line tool<br />
Petersen TN, Brunak S, von Heijne G, Nielsen H <a href="https://doi.org/10.1038/nmeth.1701">SignalP 4.0: discriminating signal peptides from transmembrane regions.</a> <em>Nature methods</em> 8.10: 785 (2011)</p>
</li>
<li>
<p><strong><a href="https://github.com/phac-nml/sistr_cmd">SISTR</a></strong><br />
Finds signal peptide features in CDS<br />
Yoshida CE, Kruczkiewicz P, Laing CR, Lingohr EJ, Gannon VPJ, Nash JHE, Taboada EN <a href="https://doi.org/10.1371/journal.pone.0147101">The Salmonella In Silico Typing Resource (SISTR): An Open Web-Accessible Tool for Rapidly Typing and Subtyping Draft Salmonella Genome Assemblies.</a> <em>PloS One</em>, 11(1), e0147101. (2016)</p>
</li>
<li>
<p><strong><a href="https://github.com/ncbi/SKESA">SKESA</a></strong><br />
Strategic Kmer Extension for Scrupulous Assemblies<br />
Souvorov A, Agarwala R, Lipman DJ <a href="https://doi.org/10.1186/s13059-018-1540-z">SKESA: strategic k-mer extension for scrupulous assemblies.</a> <em>Genome Biology</em> 19:153 (2018)</p>
</li>
<li>
<p><strong><a href="https://github.com/tseemann/snippy">Snippy</a></strong><br />
Rapid haploid variant calling and core genome alignment<br />
Seemann T <a href="https://github.com/tseemann/snippy">Snippy: fast bacterial variant calling from NGS reads</a> (GitHub)</p>
</li>
<li>
<p><strong><a href="http://snpeff.sourceforge.net/">SnpEff</a></strong><br />
Genomic variant annotations and functional effect prediction toolbox.<br />
Cingolani P, Platts A, Wang LL, Coon M, Nguyen T, Wang L, Land SJ, Lu X, Douglas M <a href="https://doi.org/10.4161/fly.19695">A program for annotating and predicting the effects of single nucleotide polymorphisms, SnpEff: SNPs in the genome of Drosophila melanogaster strain w1118; iso-2; iso-3.</a> <em>Fly</em> 6(2), 80-92 (2012)</p>
</li>
<li>
<p><strong><a href="https://github.com/tseemann/snp-dists">snp-dists</a></strong><br />
Pairwise SNP distance matrix from a FASTA sequence alignment<br />
Seemann T <a href="https://github.com/tseemann/snp-dists">snp-dists - Pairwise SNP distance matrix from a FASTA sequence alignment.</a> (GitHub)</p>
</li>
<li>
<p><strong><a href="https://github.com/sanger-pathogens/snp-sites">SNP-sites</a></strong><br />
Rapidly extracts SNPs from a multi-FASTA alignment.<br />
Page AJ, Taylor B, Delaney AJ, Soares J, Seemann T, Keane JA, Harris SR <a href="https://dx.doi.org/10.1099%2Fmgen.0.000056">SNP-sites: rapid efficient extraction of SNPs from multi-FASTA alignments.</a> <em>Microbial Genomics</em> 2.4‚ÄÖ(2016)</p>
</li>
<li>
<p><strong><a href="https://github.com/dib-lab/sourmash">Sourmash</a></strong><br />
Compute and compare MinHash signatures for DNA data sets.<br />
Brown CT, Irber L <a href="http://dx.doi.org/10.21105/joss.00027">sourmash: a library for MinHash sketching of DNA</a>. <em>JOSS</em> 1, 27 (2016)</p>
</li>
<li>
<p><strong><a href="https://github.com/ablab/spades">SPAdes</a></strong><br />
An assembly toolkit containing various assembly pipelines.<br />
Bankevich A, Nurk S, Antipov D, Gurevich AA, Dvorkin M, Kulikov AS, Lesin VM, Nikolenko SI, Pham S, Prjibelski AD, Pyshkin AV, Sirotkin AV, Vyahhi N, Tesler G, Alekseyev MA, Pevzner PA <a href="https://doi.org/10.1089/cmb.2012.0021">SPAdes: a new genome assembly algorithm and its applications to single-cell sequencing.</a> <em>Journal of computational biology</em> 19.5 455-477 (2012)</p>
</li>
<li>
<p><strong><a href="https://github.com/HCGB-IGTP/spaTyper">spaTyper</a></strong><br />
Computational method for finding spa types.<br />
Sanchez-Herrero JF, and Sullivan M <a href="http://doi.org/10.5281/zenodo.4063625">spaTyper: Staphylococcal protein A (spa) characterization pipeline</a>. Zenodo. (2020)</p>
</li>
<li>
<p><strong><a href="https://cge.cbs.dtu.dk/services/spatyper/">spaTyper Database</a></strong><br />
Database used by spaTyper<br />
Harmsen D, Claus H, Witte W, Rothg√§nger J, Claus H, Turnwald D, and Vogel U <a href="https://doi.org/10.1128/jcm.41.12.5442-5448.2003">Typing of methicillin-resistant <em>Staphylococcus aureus</em> in a university hospital setting using a novel software for spa-repeat determination and database management.</a> <em>J. Clin. Microbiol.</em> 41:5442-5448 (2003)</p>
</li>
<li>
<p><strong><a href="https://github.com/jimmyliu1326/SsuisSero">SsuisSero</a></strong><br />
Rapid <em>Streptococcus suis</em> serotyping<br />
Lui J <a href="https://github.com/jimmyliu1326/SsuisSero">SsuisSero: Rapid <em>Streptococcus suis</em> serotyping</a> (GitHub)</p>
</li>
<li>
<p><strong><a href="https://github.com/staphopia/staphopia-sccmec">staphopia-sccmec</a></strong><br />
A standalone version of Staphopia's SCCmec typing method.<br />
Petit III RA, Read TD <a href="http://dx.doi.org/10.7717/peerj.5261"><em>Staphylococcus aureus</em> viewed from the perspective of 40,000+ genomes.</a> <em>PeerJ</em> 6, e5261 (2018)</p>
</li>
<li>
<p><strong><a href="https://github.com/jodyphelan/TBProfiler">TBProfiler</a></strong><br />
Profiling tool for <em>Mycobacterium tuberculosis</em> to detect resistance and strain type<br />
Phelan JE, O‚ÄôSullivan DM, Machado D, Ramos J, Oppong YEA, Campino S, O‚ÄôGrady J, McNerney R, Hibberd ML, Viveiros M, Huggett JF, Clark TG <a href="https://doi.org/10.1186/s13073-019-0650-x">Integrating informatics tools and portable sequencing technology for rapid detection of resistance to anti-tuberculous drugs.</a> <em>Genome Med</em> 11, 41 (2019)</p>
</li>
<li>
<p><strong><a href="http://www.usadellab.org/cms/index.php?page=trimmomatic">Trimmomatic</a></strong><br />
A flexible read trimming tool for Illumina NGS data<br />
Bolger AM, Lohse M, Usadel B <a href="https://doi.org/10.1093/bioinformatics/btu170">Trimmomatic: a flexible trimmer for Illumina sequence data.</a> <em>Bioinformatics</em> 30.15 2114-2120 (2014)</p>
</li>
<li>
<p><strong><a href="https://github.com/rrwick/Unicycler">Unicycler</a></strong><br />
Hybrid assembly pipeline for bacterial genomes<br />
Wick RR, Judd LM, Gorrie CL, Holt KE <a href="http://dx.doi.org/10.1371/journal.pcbi.1005595">Unicycler: Resolving bacterial genome assemblies from short and long sequencing reads.</a> <em>PLoS Comput. Biol.</em> 13, e1005595 (2017)</p>
</li>
<li>
<p><strong><a href="https://github.com/rpetit3/vcf-annotator">VCF-Annotator</a></strong><br />
Add biological annotations to variants in a VCF file.<br />
Petit III RA <a href="https://github.com/rpetit3/vcf-annotator">VCF-Annotator: Add biological annotations to variants in a VCF file.</a> (GitHub)</p>
</li>
<li>
<p><strong><a href="https://github.com/vcflib/vcflib">Vcflib</a></strong><br />
a simple C++ library for parsing and manipulating VCF files<br />
Garrison E <a href="https://github.com/vcflib/vcflib">Vcflib: A C++ library for parsing and manipulating VCF files</a> (GitHub)</p>
</li>
<li>
<p><strong><a href="https://github.com/dzerbino/velvet">Velvet</a></strong><br />
Short read de novo assembler using de Bruijn graphs<br />
Zerbino DR, Birney E <a href="http://www.genome.org/cgi/doi/10.1101/gr.074492.107">Velvet: algorithms for de novo short read assembly using de Bruijn graphs.</a> <em>Genome research</em> 18.5 821-829 (2008)</p>
</li>
<li>
<p><strong><a href="https://github.com/torognes/vsearch">VSEARCH</a></strong><br />
Versatile open-source tool for metagenomics<br />
Rognes T, Flouri T, Nichols B, Quince C, Mah√© F <a href="https://doi.org/10.7717/peerj.2584">VSEARCH: a versatile open source tool for metagenomics.</a> <em>PeerJ</em> 4, e2584 (2016)</p>
</li>
<li>
<p><strong><a href="https://github.com/atks/vt">vt</a></strong><br />
A tool set for short variant discovery in genetic sequence data.<br />
Tan A, Abecasis GR, Kang HM <a href="https://doi.org/10.1093/bioinformatics/btv112">Unified representation of genetic variants.</a> <em>Bioinformatics</em> 31(13), 2202-2204 (2015)</p>
</li>
</ol>

<!-- Last update of source file -->

  


              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
    
    <nav class="md-footer__inner md-grid" aria-label="Footer" >
      
        
        <a href="../enhancements/" class="md-footer__link md-footer__link--prev" aria-label="Previous: Enhancements to OSS" rel="prev">
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
          </div>
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Previous
              </span>
              Enhancements to OSS
            </div>
          </div>
        </a>
      
      
        
        <a href="../changelog/" class="md-footer__link md-footer__link--next" aria-label="Next: Changelog" rel="next">
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Next
              </span>
              Changelog
            </div>
          </div>
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4Z"/></svg>
          </div>
        </a>
      
    </nav>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright ¬© 2019-2021 Robert A. Petit III. All rights reserved.
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs Insiders
    </a>
  
</div>
      
        <div class="md-social">
  
    
    
      
      
    
    <a href="https://github.com/bactopia/bactopia" target="_blank" rel="noopener" title="github.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.2.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
    </a>
  
    
    
      
      
    
    <a href="https://gitter.im/bactopia/Lobby" target="_blank" rel="noopener" title="gitter.im" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 384 512"><!--! Font Awesome Free 6.2.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M66.4 322.5H16V0h50.4v322.5zM166.9 76.1h-50.4V512h50.4V76.1zm100.6 0h-50.4V512h50.4V76.1zM368 76h-50.4v247H368V76z"/></svg>
    </a>
  
    
    
      
      
    
    <a href="https://hub.docker.com/r/bactopia/bactopia" target="_blank" rel="noopener" title="hub.docker.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 640 512"><!--! Font Awesome Free 6.2.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M349.9 236.3h-66.1v-59.4h66.1v59.4zm0-204.3h-66.1v60.7h66.1V32zm78.2 144.8H362v59.4h66.1v-59.4zm-156.3-72.1h-66.1v60.1h66.1v-60.1zm78.1 0h-66.1v60.1h66.1v-60.1zm276.8 100c-14.4-9.7-47.6-13.2-73.1-8.4-3.3-24-16.7-44.9-41.1-63.7l-14-9.3-9.3 14c-18.4 27.8-23.4 73.6-3.7 103.8-8.7 4.7-25.8 11.1-48.4 10.7H2.4c-8.7 50.8 5.8 116.8 44 162.1 37.1 43.9 92.7 66.2 165.4 66.2 157.4 0 273.9-72.5 328.4-204.2 21.4.4 67.6.1 91.3-45.2 1.5-2.5 6.6-13.2 8.5-17.1l-13.3-8.9zm-511.1-27.9h-66v59.4h66.1v-59.4zm78.1 0h-66.1v59.4h66.1v-59.4zm78.1 0h-66.1v59.4h66.1v-59.4zm-78.1-72.1h-66.1v60.1h66.1v-60.1z"/></svg>
    </a>
  
    
    
      
      
    
    <a href="https://twitter.com/rpetit3" target="_blank" rel="noopener" title="twitter.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.2.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"/></svg>
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    <script id="__config" type="application/json">{"base": "..", "features": ["navigation.tabs", "navigation.tabs.sticky"], "search": "../assets/javascripts/workers/search.720157f5.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.config.lang": "en", "search.config.pipeline": "stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version.title": "Select version"}}</script>
    
    
      <script src="../assets/javascripts/bundle.f4c6f307.min.js"></script>
      
        <script src="../custom/jquery.min.js"></script>
      
        <script src="../custom/jquery.magnific-popup.min.js"></script>
      
        <script src="../custom/extra.js"></script>
      
    
  </body>
</html>