---
title: qc
description: A Bactopia Tool which uses a variety of tools to perform quality control on Illumina and Oxford Nanopore reads.

---

The `qc` module uses  a variety of tools to perform quality control on Illumina and
Oxford Nanopore reads. The tools used are:

| Tool | Technology | Description |
|------|------------|-------------|
| [bbtools](https://jgi.doe.gov/data-and-tools/bbtools/) | Illumina | A suite of tools for manipulating reads |
| [fastp](https://github.com/OpenGene/fastp) | Illumina | A tool designed to provide fast all-in-one preprocessing for FastQ files |
| [fastqc](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/) | Illumina | A quality control tool for high throughput sequence data |
| [fastq_scan](https://github.com/rpetit3/fastq-scan) | Nanopore | A tool for quickly scanning FASTQ files |
| [lighter](https://github.com/mourisl/Lighter) | Illumina | A tool for correcting sequencing errors in Illumina reads |
| [NanoPlot](https://github.com/wdecoster/NanoPlot) | Nanopore | A tool for plotting long read sequencing data |
| [nanoq](https://github.com/esteinig/nanoq) | Nanopore | A tool for calculating quality metrics for Oxford Nanopore reads |
| [porechop](https://github.com/rrwick/Porechop) | Nanopore | A tool for removing adapters from Oxford Nanopore reads |
| [rasusa](https://github.com/mbhall88/rasusa) | Nanopore | Randomly subsample sequencing reads to a specified coverage |

Similar to the `gather` step, the `qc` step will also stop samples that fail to meet
basic QC checks from continuing downstream.


## Output Overview

Below is the default output structure for the `qc` step in Bactopia. Where
possible the file descriptions below were modified from a tools description.

```{bash}
<BACTOPIA_DIR>
├── <SAMPLE_NAME>
│   └── main
│       └── qc
│           ├── extra
│           ├── logs
│           │   ├── nf-qc.{begin,err,log,out,run,sh,trace}
│           │   ├── output-fastp.log
│           │   └── versions.yml
│           ├── <SAMPLE_NAME>-low-read-count-error.txt
│           ├── <SAMPLE_NAME>-low-sequence-coverage-error.txt
│           ├── <SAMPLE_NAME>-low-sequence-depth-error.txt
│           ├── <SAMPLE_NAME>{|_R1|_R2}.error-fastq.gz
│           ├── <SAMPLE_NAME>{|_R1|_R2}.fastq.gz
│           └── summary
│               ├── <SAMPLE_NAME>.fastp.{html|json}
│               ├── <SAMPLE_NAME>{|_R1|_R2}-{final|original}.json
│               └── <SAMPLE_NAME>{|_R1|_R2}-{final|original}_fastqc.{html|zip}
│               └── <SAMPLE_NAME>{|_R1|_R2}-{final|original}_NanoPlot-report.html
│               └── <SAMPLE_NAME>{|_R1|_R2}-{final|original}_NanoPlot.tar.gz
└── bactopia-runs
    └── bactopia-<TIMESTAMP>
        └── nf-reports
            ├── bactopia-dag.dot
            ├── bactopia-report.html
            ├── bactopia-timeline.html
            └── bactopia-trace.txt

```



### Results

#### Quality Control

Below is a description of the _per-sample_ results from `qc` subworkflow.


| Filename                      | Description |
|-------------------------------|-------------|
| &lt;SAMPLE_NAME&gt;.fastq.gz | A gzipped FASTQ file containing the cleaned Illumina single-end, or Oxford Nanopore reads |
| &lt;SAMPLE_NAME&gt;_R{1\|2}.fastq.gz | A gzipped FASTQ file containing the cleaned Illumina paired-end reads |
| &lt;SAMPLE_NAME&gt;-{final\|original}.json | A JSON file containing the QC results generated by [fastq-scan](https://github.com/rpetit3/fastq-scan) |
| &lt;SAMPLE_NAME&gt;-{final\|original}_fastqc.html | (Illumina Only) A HTML report of the QC results generated by [fastqc](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/) |
| &lt;SAMPLE_NAME&gt;-{final\|original}_fastqc.zip | (Illumina Only) A zip file containing the complete set of [fastqc](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/) results |
| &lt;SAMPLE_NAME&gt;-{final\|original}_fastp.json | (Illumina Only) A JSON file containing the QC results generated by [fastp](https://github.com/OpenGene/fastp) |
| &lt;SAMPLE_NAME&gt;-{final\|original}_fastp.html | (Illumina Only) A HTML report of the QC results generated by [fastp](https://github.com/OpenGene/fastp) |
| &lt;SAMPLE_NAME&gt;-{final\|original}_NanoPlot-report.html | (ONT Only) A HTML report of the QC results generated by [NanoPlot](https://github.com/wdecoster/NanoPlot) |
| &lt;SAMPLE_NAME&gt;-{final\|original}_NanoPlot.tar.gz | (ONT Only) A tarball containing the complete set of [NanoPlot](https://github.com/wdecoster/NanoPlot) results |






#### Failed Quality Checks

Built into Bactopia are few basic quality checks to help prevent downstream failures.
If a sample fails one of these checks, it will be excluded from further analysis. By
excluding these samples, complete pipeline failures are prevented.


| Extension                     | Description |
|-------------------------------|-------------|
| .error-fastq.gz | A gzipped FASTQ file of Illumina Single-End or Oxford Nanopore reads that failed QC |
| _R{1\|2}.error-fastq.gz | A gzipped FASTQ file of Illumina Single-End or Oxford Nanopore reads that failed QC |
| -low-read-count-error.txt | Sample failed read count checks and excluded from further analysis |
| -low-sequence-coverage-error.txt | Sample failed sequenced coverage checks and excluded from further analysis |
| -low-sequence-depth-error.txt | Sample failed sequenced basepair checks and excluded from further analysis |



!!! info "Poor samples are excluded to prevent downstream failures"
    Samples that fail any of the QC checks will be excluded from further analysis.
    Those samples will generate a `*-error.txt` file with the error message. Excluding
    these samples prevents downstream failures that cause the whole workflow to fail.



??? warning "Example Error: After QC, too few reads remain"
    If after cleaning reads, a sample has less than the minimum required reads, the
    sample will be excluded from further analysis. You can adjust this minimum read
    count using the `--min_reads` parameter.

    __Example Text from &lt;SAMPLE_NAME&gt;-low-read-count-error.txt__  
    _&lt;SAMPLE_NAME&gt; FASTQ(s) contain `X` total reads. This does not exceed the required
    minimum `Y` read count. Further analysis is discontinued._

??? warning "Example Error: After QC, too little sequence coverage remains"
    If after cleaning reads, a sample has failed to meet the minimum sequence 
    coverage required, the sample will be excluded from further analysis. You can
    adjust this minimum read count using the `--min_coverage` parameter.

    __Note:__ This check is only performed when a genome size is available.

    __Example Text from &lt;SAMPLE_NAME&gt;-low-sequence-coverage-error.txt__  
    _After QC, &lt;SAMPLE_NAME&gt; FASTQ(s) contain `X` total basepairs. This does not
    exceed the required minimum `Y` bp (`Z`x coverage). Further analysis is
    discontinued._

??? warning "Example Error: After QC, too little sequenced basepairs remain"
    If after cleaning reads, a sample has failed to meet the minimum number of
    sequenced basepairs, the sample will be excluded from further analysis. You can
    adjust this minimum read count using the `--min_basepairs` parameter.

    __Example Text from &lt;SAMPLE_NAME&gt;-low-sequence-depth-error.txt__  
    _&lt;SAMPLE_NAME&gt; FASTQ(s) contain `X` total basepairs. This does not exceed the
    required minimum `Y` bp. Further analysis is discontinued._





### Audit Trail

Below are files that can assist you in understanding which parameters and program versions
were used.

#### Logs 

Each process that is executed will have a folder named `logs`. In this folder are helpful
files for you to review if the need ever arises.

| Extension    | Description |
|--------------|-------------|
| .begin       | An empty file used to designate the process started |
| .err         | Contains STDERR outputs from the process |
| .log         | Contains both STDERR and STDOUT outputs from the process |
| .out         | Contains STDOUT outputs from the process |
| .run         | The script Nextflow uses to stage/unstage files and queue processes based on given profile |
| .sh          | The script executed by bash for the process  |
| .trace       | The Nextflow [Trace](https://www.nextflow.io/docs/latest/tracing.html#trace-report) report for the process |
| versions.yml | A YAML formatted file with program versions |

## Parameters


### <i class="fa-xl fas fa-exclamation-circle"></i> QC 


| Parameter | Description |
|:---|---|
| <i class="fa-lg fas fa-fast-forward"></i>` --use_bbmap` | Illumina reads will be QC'd using BBMap <br/>**Type:** `boolean` |
| <i class="fa-lg fas fa-fast-forward"></i>` --use_porechop` | Use Porechop to remove adapters from ONT reads <br/>**Type:** `boolean`, **Default:** `turned off` |
| <i class="fa-lg fas fa-fast-forward"></i>` --skip_qc` | The QC step will be skipped and it will be assumed the inputs sequences have already been QCed. <br/>**Type:** `boolean` |
| <i class="fa-lg fas fa-fast-forward"></i>` --skip_qc_plots` | QC Plot creation by FastQC or Nanoplot will be skipped <br/>**Type:** `boolean` |
| <i class="fa-lg fas fa-fast-forward"></i>` --skip_error_correction` | FLASH error correction of reads will be skipped. <br/>**Type:** `boolean` |
| <i class="fa-lg fas fa-file-alt"></i>` --adapters` | A FASTA file containing adapters to remove <br/>**Type:** `string` |
| <i class="fa-lg fas fa-hashtag"></i>` --adapter_k` | Kmer length used for finding adapters. <br/>**Type:** `integer`, **Default:** `23` |
| <i class="fa-lg fas fa-file-alt"></i>` --phix` | phiX174 reference genome to remove <br/>**Type:** `string` |
| <i class="fa-lg fas fa-hashtag"></i>` --phix_k` | Kmer length used for finding phiX174. <br/>**Type:** `integer`, **Default:** `31` |
| <i class="fa-lg fas fa-boxes"></i>` --ktrim` | Trim reads to remove bases matching reference kmers <br/>**Type:** `string`, **Default:** `r` |
| <i class="fa-lg fas fa-angle-double-down"></i>` --mink` | Look for shorter kmers at read tips down to this length, when k-trimming or masking. <br/>**Type:** `integer`, **Default:** `11` |
| <i class="fa-lg fas fa-hashtag"></i>` --hdist` | Maximum Hamming distance for ref kmers (subs only) <br/>**Type:** `integer`, **Default:** `1` |
| <i class="fa-lg fas fa-boxes"></i>` --tpe` | When kmer right-trimming, trim both reads to the minimum length of either <br/>**Type:** `string`, **Default:** `t` |
| <i class="fa-lg fas fa-boxes"></i>` --tbo` | Trim adapters based on where paired reads overlap <br/>**Type:** `string`, **Default:** `t` |
| <i class="fa-lg fas fa-boxes"></i>` --qtrim` | Trim read ends to remove bases with quality below trimq. <br/>**Type:** `string`, **Default:** `rl` |
| <i class="fa-lg fas fa-hashtag"></i>` --trimq` | Regions with average quality BELOW this will be trimmed if qtrim is set to something other than f <br/>**Type:** `integer`, **Default:** `6` |
| <i class="fa-lg fas fa-angle-double-down"></i>` --maq` | Reads with average quality (after trimming) below this will be discarded <br/>**Type:** `integer`, **Default:** `10` |
| <i class="fa-lg fas fa-angle-double-down"></i>` --minlength` | Reads shorter than this after trimming will be discarded <br/>**Type:** `integer`, **Default:** `35` |
| <i class="fa-lg fas fa-hashtag"></i>` --ftm` | If positive, right-trim length to be equal to zero, modulo this number <br/>**Type:** `integer`, **Default:** `5` |
| <i class="fa-lg fas fa-boxes"></i>` --tossjunk` | Discard reads with invalid characters as bases <br/>**Type:** `string`, **Default:** `t` |
| <i class="fa-lg fas fa-boxes"></i>` --ain` | When detecting pair names, allow identical names <br/>**Type:** `string`, **Default:** `f` |
| <i class="fa-lg fas fa-boxes"></i>` --qout` | PHRED offset to use for output FASTQs <br/>**Type:** `string`, **Default:** `33` |
| <i class="fa-lg fas fa-hashtag"></i>` --maxcor` | Max number of corrections within a 20bp window <br/>**Type:** `integer`, **Default:** `1` |
| <i class="fa-lg fas fa-hashtag"></i>` --sampleseed` | Set to a positive number to use as the random number generator seed for sampling <br/>**Type:** `integer`, **Default:** `42` |
| <i class="fa-lg fas fa-angle-double-down"></i>` --ont_minlength` | ONT Reads shorter than this will be discarded <br/>**Type:** `integer`, **Default:** `1000` |
| <i class="fa-lg fas fa-angle-double-down"></i>` --ont_minqual` | Minimum average read quality filter of ONT reads <br/>**Type:** `integer` |
| <i class="fa-lg fas fa-italic"></i>` --porechop_opts` | Extra Porechop options in quotes <br/>**Type:** `integer` |
| <i class="fa-lg fas fa-italic"></i>` --nanoplot_opts` | Extra NanoPlot options in quotes <br/>**Type:** `integer` |
| <i class="fa-lg fas fa-italic"></i>` --bbduk_opts` | Extra BBDuk options in quotes <br/>**Type:** `integer` |
| <i class="fa-lg fas fa-italic"></i>` --fastp_opts` | Extra fastp options in quotes <br/>**Type:** `integer` |

## Citations
If you use Bactopia and `qc` in your analysis, please cite the following.

- [Bactopia](https://bactopia.github.io/)  
    Petit III RA, Read TD [Bactopia - a flexible pipeline for complete analysis of bacterial genomes.](https://doi.org/10.1128/mSystems.00190-20) _mSystems_ 5 (2020)
  

- [BBTools](https://jgi.doe.gov/data-and-tools/bbtools/)  
    Bushnell B [BBMap short read aligner, and other bioinformatic tools.](http://sourceforge.net/projects/bbmap/) (Link)
  
- [fastp](https://github.com/OpenGene/fastp)  
    Chen S, Zhou Y, Chen Y, and Gu J [fastp: an ultra-fast all-in-one FASTQ preprocessor.](https://doi.org/10.1093/bioinformatics/bty560) _Bioinformatics_, 34(17), i884–i890. (2018)
  
- [fastq-scan](https://github.com/rpetit3/fastq-scan)  
    Petit III RA [fastq-scan: generate summary statistics of input FASTQ sequences.](https://github.com/rpetit3/fastq-scan) (GitHub)
  
- [FastQC](https://github.com/s-andrews/FastQC)  
    Andrews S [FastQC: a quality control tool for high throughput sequence data.](http://www.bioinformatics.babraham.ac.uk/projects/fastqc) (WebLink)
  
- [Lighter](https://github.com/mourisl/Lighter)  
    Song L, Florea L, Langmead B [Lighter: Fast and Memory-efficient Sequencing Error Correction without Counting](https://doi.org/10.1186/s13059-014-0509-9). _Genome Biol._ 15(11):509 (2014)
  
- [NanoPlot](https://github.com/wdecoster/NanoPlot)  
    De Coster W, D’Hert S, Schultz DT, Cruts M, Van Broeckhoven C [NanoPack: visualizing and processing long-read sequencing data](https://doi.org/10.1093/bioinformatics/bty149) _Bioinformatics_ Volume 34, Issue 15 (2018) 
  
- [Nanoq](https://github.com/esteinig/nanoq)  
    Steinig E [Nanoq: Minimal but speedy quality control for nanopore reads in Rust](https://github.com/esteinig/nanoq) (GitHub)
  
- [Pigz](https://zlib.net/pigz/)  
    Adler M. [pigz: A parallel implementation of gzip for modern multi-processor, multi-core machines.](https://zlib.net/pigz/) _Jet Propulsion Laboratory_ (2015)
  
- [Porechop](https://github.com/rrwick/Porechop)  
    Wick RR, Judd LM, Gorrie CL, Holt KE. [Completing bacterial genome assemblies with multiplex MinION sequencing.](https://doi.org/10.1099/mgen.0.000132) _Microb Genom._ 3(10):e000132 (2017)
  
- [Rasusa](https://github.com/mbhall88/rasusa)  
    Hall MB [Rasusa: Randomly subsample sequencing reads to a specified coverage.](https://doi.org/10.5281/zenodo.3731394) (2019).
  
